{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (0.8.8) is out-of-date. Latest is 0.9.0. Please run 'pip install --upgrade cptac' to update it. (<ipython-input-1-a52d0cab607c>, line 1)\n"
     ]
    }
   ],
   "source": [
    "import cptac\n",
    "import pandas as pd\n",
    "import scipy.stats as stats\n",
    "import cptac.utils as ut\n",
    "import statsmodels.stats.multitest as ssm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    }
   ],
   "source": [
    "ccrcc = cptac.Ccrcc()\n",
    "#en = cptac.Endometrial()\n",
    "#luad = cptac.Luad()\n",
    "#hnscc  = cptac.Hnscc()\n",
    "#lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [ccrcc]#, en, luad, hnscc, lscc]\n",
    "cancer_names = ['CCRCC', 'Endometrial', 'LUAD', 'HNSCC', 'LSCC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-4-3c2e46c775cb>, line 6)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-4-3c2e46c775cb>, line 16)\n"
     ]
    }
   ],
   "source": [
    "prot_df = []\n",
    "trans_df = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_normal_df = cancer.get_proteomics('normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID')\n",
    "    trans_normal_df = cancer.get_transcriptomics('normal')\n",
    "    if isinstance(trans_normal_df.columns, pd.MultiIndex):\n",
    "        trans_normal_df = ut.reduce_multiindex(df = trans_normal_df, levels_to_drop='Database_ID')\n",
    "    prot_normal_df['Patient_ID'] = prot_normal_df.index\n",
    "    trans_normal_df['Patient_ID'] = trans_normal_df.index\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    trans_normal_df = trans_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    prot_tumor_df = cancer.get_proteomics('tumor')\n",
    "    if isinstance(prot_tumor_df.columns, pd.MultiIndex):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID')\n",
    "    trans_tumor_df = cancer.get_transcriptomics('tumor')\n",
    "    if isinstance(trans_tumor_df.columns, pd.MultiIndex):\n",
    "        trans_tumor_df = ut.reduce_multiindex(df = trans_tumor_df, levels_to_drop='Database_ID')\n",
    "    prot_tumor_df['Patient_ID'] = prot_tumor_df.index\n",
    "    trans_tumor_df['Patient_ID'] = trans_tumor_df.index\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    trans_tumor_df = trans_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    prot_tumor_df['Tissue'] = ['tumor'] * len(prot_tumor_df)\n",
    "    prot_normal_df['Tissue'] = ['normal'] * len(prot_normal_df)\n",
    "    trans_tumor_df['Tissue'] = ['tumor'] * len(trans_tumor_df)\n",
    "    trans_normal_df['Tissue'] = ['normal'] * len(trans_normal_df)\n",
    "    prot_tumor_df['Cancer'] = [cancer_name] * len(prot_tumor_df)\n",
    "    prot_normal_df['Cancer'] = [cancer_name] * len(prot_normal_df)\n",
    "    trans_tumor_df['Cancer'] = [cancer_name] * len(trans_tumor_df)\n",
    "    trans_normal_df['Cancer'] = [cancer_name] * len(trans_normal_df)\n",
    "    prot_df.extend([prot_tumor_df, prot_normal_df])\n",
    "    trans_df.extend([trans_tumor_df, trans_normal_df])\n",
    "prot_df = pd.concat(prot_df)\n",
    "trans_df = pd.concat(trans_df)\n",
    "\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Proteomic</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.304302</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>1.195915</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.286155</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.135730</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.123959</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983633</th>\n",
       "      <td>C3N-01522.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.089533</td>\n",
       "      <td>normal</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983636</th>\n",
       "      <td>C3N-01648.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.200438</td>\n",
       "      <td>normal</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983637</th>\n",
       "      <td>C3N-01649.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.111064</td>\n",
       "      <td>normal</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983638</th>\n",
       "      <td>C3N-01651.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.185857</td>\n",
       "      <td>normal</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983639</th>\n",
       "      <td>C3N-01808.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.052599</td>\n",
       "      <td>normal</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1774834 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Patient_ID  Gene  Proteomic  Tissue Cancer\n",
       "0         C3L-00004  A1BG  -0.304302   tumor  CCRCC\n",
       "1         C3L-00010  A1BG   1.195915   tumor  CCRCC\n",
       "2         C3L-00011  A1BG  -0.286155   tumor  CCRCC\n",
       "3         C3L-00026  A1BG   0.135730   tumor  CCRCC\n",
       "4         C3L-00079  A1BG  -0.123959   tumor  CCRCC\n",
       "...             ...   ...        ...     ...    ...\n",
       "983633  C3N-01522.N  ZZZ3   0.089533  normal  CCRCC\n",
       "983636  C3N-01648.N  ZZZ3   0.200438  normal  CCRCC\n",
       "983637  C3N-01649.N  ZZZ3   0.111064  normal  CCRCC\n",
       "983638  C3N-01651.N  ZZZ3   0.185857  normal  CCRCC\n",
       "983639  C3N-01808.N  ZZZ3  -0.052599  normal  CCRCC\n",
       "\n",
       "[1774834 rows x 5 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_df = prot_df.dropna()\n",
    "prot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.995336</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.679400</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.354549</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>2.543775</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>4.355205</td>\n",
       "      <td>tumor</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2048443</th>\n",
       "      <td>C3N-03880.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>17.705300</td>\n",
       "      <td>normal</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2048444</th>\n",
       "      <td>C3N-03882.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>17.762500</td>\n",
       "      <td>normal</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2048445</th>\n",
       "      <td>C3N-04124.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>17.979700</td>\n",
       "      <td>normal</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2048446</th>\n",
       "      <td>C3N-04127.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>17.735200</td>\n",
       "      <td>normal</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2048447</th>\n",
       "      <td>C3N-04162.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>17.543900</td>\n",
       "      <td>normal</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>20694143 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID  Gene  Transcriptomics  Tissue Cancer\n",
       "0          C3L-00004  A1BG         0.995336   tumor  CCRCC\n",
       "1          C3L-00010  A1BG         0.679400   tumor  CCRCC\n",
       "2          C3L-00011  A1BG         0.354549   tumor  CCRCC\n",
       "3          C3L-00026  A1BG         2.543775   tumor  CCRCC\n",
       "4          C3L-00079  A1BG         4.355205   tumor  CCRCC\n",
       "...              ...   ...              ...     ...    ...\n",
       "2048443  C3N-03880.N  ZZZ3        17.705300  normal   LSCC\n",
       "2048444  C3N-03882.N  ZZZ3        17.762500  normal   LSCC\n",
       "2048445  C3N-04124.N  ZZZ3        17.979700  normal   LSCC\n",
       "2048446  C3N-04127.N  ZZZ3        17.735200  normal   LSCC\n",
       "2048447  C3N-04162.N  ZZZ3        17.543900  normal   LSCC\n",
       "\n",
       "[20694143 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trans_df = trans_df.dropna()\n",
    "trans_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "unequal length arrays",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-7-f467b6ae1bbf>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[0mnormal\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgroup\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mgroup\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mTissue\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'normal'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      6\u001b[0m     \u001b[0mtumor\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgroup\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mgroup\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mTissue\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;34m'tumor'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 7\u001b[1;33m     \u001b[0mt\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpval\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mstats\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mttest_rel\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnormal\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mProteomic\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtumor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mProteomic\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      8\u001b[0m     \u001b[0mrow\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      9\u001b[0m     \u001b[0mrow\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'Cancer'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcancer\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py\u001b[0m in \u001b[0;36mttest_rel\u001b[1;34m(a, b, axis, nan_policy)\u001b[0m\n\u001b[0;32m   5760\u001b[0m     \u001b[0mnb\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0m_get_len\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mb\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"second argument\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5761\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mna\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mnb\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5762\u001b[1;33m         \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'unequal length arrays'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   5763\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5764\u001b[0m     \u001b[1;32mif\u001b[0m \u001b[0mna\u001b[0m \u001b[1;33m==\u001b[0m \u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: unequal length arrays"
     ]
    }
   ],
   "source": [
    "prot_group = prot_df.groupby(['Cancer', 'Gene'])\n",
    "rows = []\n",
    "for name, group in prot_group:\n",
    "    cancer, gene = name\n",
    "    normal = group[group.Tissue == 'normal']\n",
    "    tumor = group[group.Tissue == 'tumor']\n",
    "    t, pval = stats.ttest_rel(normal.Proteomic, tumor.Proteomic)\n",
    "    row = dict()\n",
    "    row['Cancer'] = cancer\n",
    "    row['Gene'] = gene\n",
    "    row['t_score'] = t\n",
    "    row['pval'] = pval\n",
    "    rows.append(row)\n",
    "prot_t_test_df = pd.DataFrame(rows)\n",
    "prot_t_test_df = prot_t_test_df.dropna()\n",
    "prot_t_test_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-8-d4baa8e72c50>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['qval'] = ssm.fdrcorrection(df.pval)[1]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "      <th>t_score</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-4.357646</td>\n",
       "      <td>2.139405e-05</td>\n",
       "      <td>3.499398e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>7.793847</td>\n",
       "      <td>3.999889e-13</td>\n",
       "      <td>9.480498e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-7.416771</td>\n",
       "      <td>3.757940e-12</td>\n",
       "      <td>8.575264e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A4GALT</td>\n",
       "      <td>-1.538266</td>\n",
       "      <td>1.678723e-01</td>\n",
       "      <td>1.965757e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>-11.785243</td>\n",
       "      <td>1.787427e-24</td>\n",
       "      <td>6.537957e-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45101</th>\n",
       "      <td>LSCC</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>-11.840228</td>\n",
       "      <td>1.588201e-24</td>\n",
       "      <td>6.318733e-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45102</th>\n",
       "      <td>LSCC</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>11.295875</td>\n",
       "      <td>2.561137e-23</td>\n",
       "      <td>9.684108e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45103</th>\n",
       "      <td>LSCC</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>22.169011</td>\n",
       "      <td>2.387573e-56</td>\n",
       "      <td>5.532851e-55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45104</th>\n",
       "      <td>LSCC</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>14.374062</td>\n",
       "      <td>7.289689e-33</td>\n",
       "      <td>4.344920e-32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45105</th>\n",
       "      <td>LSCC</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>2.888575</td>\n",
       "      <td>4.285567e-03</td>\n",
       "      <td>5.669931e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55337 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Cancer    Gene    t_score          pval          qval\n",
       "0      CCRCC    A1BG  -4.357646  2.139405e-05  3.499398e-05\n",
       "1      CCRCC    A1CF   7.793847  3.999889e-13  9.480498e-13\n",
       "2      CCRCC     A2M  -7.416771  3.757940e-12  8.575264e-12\n",
       "3      CCRCC  A4GALT  -1.538266  1.678723e-01  1.965757e-01\n",
       "4      CCRCC    AAAS -11.785243  1.787427e-24  6.537957e-24\n",
       "...      ...     ...        ...           ...           ...\n",
       "45101   LSCC    ZXDC -11.840228  1.588201e-24  6.318733e-24\n",
       "45102   LSCC  ZYG11B  11.295875  2.561137e-23  9.684108e-23\n",
       "45103   LSCC     ZYX  22.169011  2.387573e-56  5.532851e-55\n",
       "45104   LSCC   ZZEF1  14.374062  7.289689e-33  4.344920e-32\n",
       "45105   LSCC    ZZZ3   2.888575  4.285567e-03  5.669931e-03\n",
       "\n",
       "[55337 rows x 5 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# correct the p-values\n",
    "corrected_dfs = []\n",
    "for cancer in pd.unique(prot_df.Cancer):\n",
    "    df = prot_t_test_df[prot_t_test_df.Cancer == cancer]\n",
    "    df['qval'] = ssm.fdrcorrection(df.pval)[1]    \n",
    "    corrected_dfs.append(df)\n",
    "prot_t_test_df = pd.concat(corrected_dfs)\n",
    "prot_t_test_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3583: RuntimeWarning: Degrees of freedom <= 0 for slice\n",
      "  return _methods._var(a, axis=axis, dtype=dtype, out=out, ddof=ddof,\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\numpy\\core\\_methods.py:209: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  ret = ret.dtype.type(ret / rcount)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "      <th>t_score</th>\n",
       "      <th>pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-5.468802</td>\n",
       "      <td>1.471386e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>5.126843</td>\n",
       "      <td>7.454574e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-6.159303</td>\n",
       "      <td>4.526458e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>2.163946</td>\n",
       "      <td>3.176444e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>-1.573925</td>\n",
       "      <td>1.172320e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125674</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>-16.499280</td>\n",
       "      <td>1.031040e-39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125675</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>6.577774</td>\n",
       "      <td>3.763714e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125676</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>9.108707</td>\n",
       "      <td>6.873646e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125677</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>11.350264</td>\n",
       "      <td>1.425407e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125678</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-2.847916</td>\n",
       "      <td>4.840047e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120955 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Cancer     Gene    t_score          pval\n",
       "0       CCRCC     A1BG  -5.468802  1.471386e-07\n",
       "1       CCRCC     A1CF   5.126843  7.454574e-07\n",
       "2       CCRCC      A2M  -6.159303  4.526458e-09\n",
       "3       CCRCC    A2ML1   2.163946  3.176444e-02\n",
       "4       CCRCC  A3GALT2  -1.573925  1.172320e-01\n",
       "...       ...      ...        ...           ...\n",
       "125674   LUAD   ZYG11A -16.499280  1.031040e-39\n",
       "125675   LUAD   ZYG11B   6.577774  3.763714e-10\n",
       "125676   LUAD      ZYX   9.108707  6.873646e-17\n",
       "125677   LUAD    ZZEF1  11.350264  1.425407e-23\n",
       "125678   LUAD     ZZZ3  -2.847916  4.840047e-03\n",
       "\n",
       "[120955 rows x 4 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trans_group = trans_df.groupby(['Cancer', 'Gene'])\n",
    "rows = []\n",
    "for name, group in trans_group:\n",
    "    cancer, gene = name\n",
    "    normal = group[group.Tissue == 'normal']\n",
    "    tumor = group[group.Tissue == 'tumor']\n",
    "    t, pval = stats.ttest_ind(normal.Transcriptomics, tumor.Transcriptomics)\n",
    "    row = dict()\n",
    "    row['Cancer'] = cancer\n",
    "    row['Gene'] = gene\n",
    "    row['t_score'] = t\n",
    "    row['pval'] = pval\n",
    "    rows.append(row)\n",
    "trans_t_test_df = pd.DataFrame(rows)\n",
    "trans_t_test_df = trans_t_test_df.dropna()\n",
    "trans_t_test_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-10-a4c8df15721b>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df['qval'] = ssm.fdrcorrection(df.pval)[1]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "      <th>t_score</th>\n",
       "      <th>pval</th>\n",
       "      <th>qval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-5.468802</td>\n",
       "      <td>1.471386e-07</td>\n",
       "      <td>3.321956e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>5.126843</td>\n",
       "      <td>7.454574e-07</td>\n",
       "      <td>1.592595e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-6.159303</td>\n",
       "      <td>4.526458e-09</td>\n",
       "      <td>1.136298e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>2.163946</td>\n",
       "      <td>3.176444e-02</td>\n",
       "      <td>4.337747e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>-1.573925</td>\n",
       "      <td>1.172320e-01</td>\n",
       "      <td>1.467886e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125674</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>-16.499280</td>\n",
       "      <td>1.031040e-39</td>\n",
       "      <td>1.258244e-38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125675</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>6.577774</td>\n",
       "      <td>3.763714e-10</td>\n",
       "      <td>8.354679e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125676</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>9.108707</td>\n",
       "      <td>6.873646e-17</td>\n",
       "      <td>2.216665e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125677</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>11.350264</td>\n",
       "      <td>1.425407e-23</td>\n",
       "      <td>6.662452e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125678</th>\n",
       "      <td>LUAD</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-2.847916</td>\n",
       "      <td>4.840047e-03</td>\n",
       "      <td>6.605971e-03</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>120955 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Cancer     Gene    t_score          pval          qval\n",
       "0       CCRCC     A1BG  -5.468802  1.471386e-07  3.321956e-07\n",
       "1       CCRCC     A1CF   5.126843  7.454574e-07  1.592595e-06\n",
       "2       CCRCC      A2M  -6.159303  4.526458e-09  1.136298e-08\n",
       "3       CCRCC    A2ML1   2.163946  3.176444e-02  4.337747e-02\n",
       "4       CCRCC  A3GALT2  -1.573925  1.172320e-01  1.467886e-01\n",
       "...       ...      ...        ...           ...           ...\n",
       "125674   LUAD   ZYG11A -16.499280  1.031040e-39  1.258244e-38\n",
       "125675   LUAD   ZYG11B   6.577774  3.763714e-10  8.354679e-10\n",
       "125676   LUAD      ZYX   9.108707  6.873646e-17  2.216665e-16\n",
       "125677   LUAD    ZZEF1  11.350264  1.425407e-23  6.662452e-23\n",
       "125678   LUAD     ZZZ3  -2.847916  4.840047e-03  6.605971e-03\n",
       "\n",
       "[120955 rows x 5 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# correct the p-values\n",
    "corrected_dfs = []\n",
    "for cancer in pd.unique(trans_t_test_df.Cancer):\n",
    "    df = trans_t_test_df[trans_t_test_df.Cancer == cancer]\n",
    "    df['qval'] = ssm.fdrcorrection(df.pval)[1]    \n",
    "    corrected_dfs.append(df)\n",
    "trans_t_test_df = pd.concat(corrected_dfs)\n",
    "trans_t_test_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_t_test_df.to_csv('data/proteomics_t_tests.csv', index = False)\n",
    "trans_t_test_df.to_csv('data/transcriptomics_t_tests.csv', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
