{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from gprofiler import GProfiler\n",
    "gp = GProfiler(return_dataframe=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Delta_Correlation</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.268533</td>\n",
       "      <td>5.703182e-02</td>\n",
       "      <td>1.320375e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.192038</td>\n",
       "      <td>1.063340e-04</td>\n",
       "      <td>6.401858e-04</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.191619</td>\n",
       "      <td>1.277644e-01</td>\n",
       "      <td>2.439276e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.019654</td>\n",
       "      <td>8.963138e-01</td>\n",
       "      <td>9.409267e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS</td>\n",
       "      <td>-0.169937</td>\n",
       "      <td>6.007042e-02</td>\n",
       "      <td>1.375402e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>-0.263372</td>\n",
       "      <td>1.531939e-02</td>\n",
       "      <td>4.572896e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AAED1</td>\n",
       "      <td>0.190506</td>\n",
       "      <td>2.882440e-01</td>\n",
       "      <td>4.385777e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.364999</td>\n",
       "      <td>7.633656e-04</td>\n",
       "      <td>3.646252e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.356932</td>\n",
       "      <td>6.347464e-14</td>\n",
       "      <td>1.385608e-12</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>0.281272</td>\n",
       "      <td>2.536223e-02</td>\n",
       "      <td>6.920503e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AAR2</td>\n",
       "      <td>0.168046</td>\n",
       "      <td>1.250698e-01</td>\n",
       "      <td>2.403575e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.415815</td>\n",
       "      <td>6.031667e-03</td>\n",
       "      <td>2.119475e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>-0.022583</td>\n",
       "      <td>8.611434e-01</td>\n",
       "      <td>9.167396e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>0.464768</td>\n",
       "      <td>4.388407e-04</td>\n",
       "      <td>2.263637e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AASDH</td>\n",
       "      <td>-0.543619</td>\n",
       "      <td>4.759680e-02</td>\n",
       "      <td>1.146645e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AASDHPPT</td>\n",
       "      <td>0.067474</td>\n",
       "      <td>5.707809e-01</td>\n",
       "      <td>7.043680e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AASS</td>\n",
       "      <td>-0.278696</td>\n",
       "      <td>1.150360e-06</td>\n",
       "      <td>1.030519e-05</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AATF</td>\n",
       "      <td>-0.067020</td>\n",
       "      <td>2.215056e-01</td>\n",
       "      <td>3.651859e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ABAT</td>\n",
       "      <td>-0.612928</td>\n",
       "      <td>7.461635e-19</td>\n",
       "      <td>2.593633e-17</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ABCA1</td>\n",
       "      <td>0.243467</td>\n",
       "      <td>6.276414e-02</td>\n",
       "      <td>1.422388e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ABCA12</td>\n",
       "      <td>-0.089958</td>\n",
       "      <td>7.175205e-01</td>\n",
       "      <td>8.188977e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ABCA2</td>\n",
       "      <td>0.056825</td>\n",
       "      <td>6.997512e-01</td>\n",
       "      <td>8.064600e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ABCA6</td>\n",
       "      <td>0.370021</td>\n",
       "      <td>7.012739e-04</td>\n",
       "      <td>3.389490e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ABCA8</td>\n",
       "      <td>-0.084953</td>\n",
       "      <td>1.692837e-01</td>\n",
       "      <td>3.012979e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ABCB1</td>\n",
       "      <td>0.072682</td>\n",
       "      <td>2.975122e-03</td>\n",
       "      <td>1.171726e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ABCB10</td>\n",
       "      <td>-0.256389</td>\n",
       "      <td>3.868174e-02</td>\n",
       "      <td>9.715232e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ABCB6</td>\n",
       "      <td>0.125776</td>\n",
       "      <td>5.996023e-03</td>\n",
       "      <td>2.109179e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ABCB7</td>\n",
       "      <td>-0.227777</td>\n",
       "      <td>2.300171e-01</td>\n",
       "      <td>3.751268e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ABCB8</td>\n",
       "      <td>-0.095291</td>\n",
       "      <td>1.518858e-02</td>\n",
       "      <td>4.539284e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ABCC1</td>\n",
       "      <td>0.505836</td>\n",
       "      <td>6.018861e-15</td>\n",
       "      <td>1.439908e-13</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50278</th>\n",
       "      <td>ZNF804B</td>\n",
       "      <td>0.259350</td>\n",
       "      <td>2.735740e-01</td>\n",
       "      <td>3.399482e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50279</th>\n",
       "      <td>ZNF827</td>\n",
       "      <td>0.503660</td>\n",
       "      <td>2.645396e-02</td>\n",
       "      <td>4.292721e-02</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50280</th>\n",
       "      <td>ZNF830</td>\n",
       "      <td>0.605859</td>\n",
       "      <td>2.272676e-06</td>\n",
       "      <td>9.573220e-06</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50281</th>\n",
       "      <td>ZNF831</td>\n",
       "      <td>-0.255886</td>\n",
       "      <td>1.202825e-01</td>\n",
       "      <td>1.656746e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50282</th>\n",
       "      <td>ZNF837</td>\n",
       "      <td>0.161310</td>\n",
       "      <td>3.262350e-01</td>\n",
       "      <td>3.962325e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50283</th>\n",
       "      <td>ZNF860</td>\n",
       "      <td>-0.072901</td>\n",
       "      <td>6.550842e-01</td>\n",
       "      <td>7.100679e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50284</th>\n",
       "      <td>ZNF92</td>\n",
       "      <td>0.009830</td>\n",
       "      <td>9.482255e-01</td>\n",
       "      <td>9.570740e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50285</th>\n",
       "      <td>ZNF98</td>\n",
       "      <td>0.139564</td>\n",
       "      <td>4.447458e-01</td>\n",
       "      <td>5.140409e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50286</th>\n",
       "      <td>ZNHIT1</td>\n",
       "      <td>-0.222195</td>\n",
       "      <td>1.777904e-01</td>\n",
       "      <td>2.324400e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50287</th>\n",
       "      <td>ZNHIT2</td>\n",
       "      <td>0.734775</td>\n",
       "      <td>1.808708e-07</td>\n",
       "      <td>9.480871e-07</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50288</th>\n",
       "      <td>ZNHIT3</td>\n",
       "      <td>0.196080</td>\n",
       "      <td>1.355067e-01</td>\n",
       "      <td>1.835773e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50289</th>\n",
       "      <td>ZNHIT6</td>\n",
       "      <td>0.585166</td>\n",
       "      <td>2.861270e-06</td>\n",
       "      <td>1.182478e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50290</th>\n",
       "      <td>ZNRD1</td>\n",
       "      <td>0.130238</td>\n",
       "      <td>2.494071e-01</td>\n",
       "      <td>3.137159e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50291</th>\n",
       "      <td>ZNRF2</td>\n",
       "      <td>0.730968</td>\n",
       "      <td>1.473860e-14</td>\n",
       "      <td>2.442431e-13</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50292</th>\n",
       "      <td>ZPR1</td>\n",
       "      <td>0.712566</td>\n",
       "      <td>4.114687e-08</td>\n",
       "      <td>2.477347e-07</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50293</th>\n",
       "      <td>ZRANB2</td>\n",
       "      <td>0.094088</td>\n",
       "      <td>4.713707e-01</td>\n",
       "      <td>5.398518e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50294</th>\n",
       "      <td>ZRSR2</td>\n",
       "      <td>0.239083</td>\n",
       "      <td>8.855146e-02</td>\n",
       "      <td>1.263193e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50295</th>\n",
       "      <td>ZSCAN16</td>\n",
       "      <td>0.155307</td>\n",
       "      <td>3.793274e-01</td>\n",
       "      <td>4.507566e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50296</th>\n",
       "      <td>ZSCAN18</td>\n",
       "      <td>0.391928</td>\n",
       "      <td>1.055189e-05</td>\n",
       "      <td>3.835977e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50297</th>\n",
       "      <td>ZSCAN23</td>\n",
       "      <td>0.245749</td>\n",
       "      <td>3.112602e-01</td>\n",
       "      <td>3.803871e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50298</th>\n",
       "      <td>ZSCAN26</td>\n",
       "      <td>0.381625</td>\n",
       "      <td>1.278330e-02</td>\n",
       "      <td>2.242804e-02</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50299</th>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>0.510219</td>\n",
       "      <td>1.242018e-03</td>\n",
       "      <td>2.804028e-03</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50300</th>\n",
       "      <td>ZW10</td>\n",
       "      <td>0.571723</td>\n",
       "      <td>3.114751e-09</td>\n",
       "      <td>2.320835e-08</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50301</th>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>0.832168</td>\n",
       "      <td>4.549808e-12</td>\n",
       "      <td>5.461398e-11</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50302</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>0.837582</td>\n",
       "      <td>2.770355e-08</td>\n",
       "      <td>1.722312e-07</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50303</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.073854</td>\n",
       "      <td>6.395306e-01</td>\n",
       "      <td>6.962162e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50304</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.734569</td>\n",
       "      <td>1.583743e-10</td>\n",
       "      <td>1.504331e-09</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50305</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.440861</td>\n",
       "      <td>2.253307e-06</td>\n",
       "      <td>9.499585e-06</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50306</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.422001</td>\n",
       "      <td>3.731185e-11</td>\n",
       "      <td>3.889196e-10</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50307</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.717501</td>\n",
       "      <td>2.955080e-06</td>\n",
       "      <td>1.215796e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50308 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Gene  Delta_Correlation       P_Value           FDR Cancer\n",
       "0          A1BG          -0.268533  5.703182e-02  1.320375e-01  CCRCC\n",
       "1          A1CF           0.192038  1.063340e-04  6.401858e-04  CCRCC\n",
       "2           A2M          -0.191619  1.277644e-01  2.439276e-01  CCRCC\n",
       "3          AAAS           0.019654  8.963138e-01  9.409267e-01  CCRCC\n",
       "4          AACS          -0.169937  6.007042e-02  1.375402e-01  CCRCC\n",
       "5         AADAT          -0.263372  1.531939e-02  4.572896e-02  CCRCC\n",
       "6         AAED1           0.190506  2.882440e-01  4.385777e-01  CCRCC\n",
       "7         AAGAB           0.364999  7.633656e-04  3.646252e-03  CCRCC\n",
       "8          AAK1           0.356932  6.347464e-14  1.385608e-12  CCRCC\n",
       "9          AAMP           0.281272  2.536223e-02  6.920503e-02  CCRCC\n",
       "10         AAR2           0.168046  1.250698e-01  2.403575e-01  CCRCC\n",
       "11         AARS           0.415815  6.031667e-03  2.119475e-02  CCRCC\n",
       "12        AARS2          -0.022583  8.611434e-01  9.167396e-01  CCRCC\n",
       "13       AARSD1           0.464768  4.388407e-04  2.263637e-03  CCRCC\n",
       "14        AASDH          -0.543619  4.759680e-02  1.146645e-01  CCRCC\n",
       "15     AASDHPPT           0.067474  5.707809e-01  7.043680e-01  CCRCC\n",
       "16         AASS          -0.278696  1.150360e-06  1.030519e-05  CCRCC\n",
       "17         AATF          -0.067020  2.215056e-01  3.651859e-01  CCRCC\n",
       "18         ABAT          -0.612928  7.461635e-19  2.593633e-17  CCRCC\n",
       "19        ABCA1           0.243467  6.276414e-02  1.422388e-01  CCRCC\n",
       "20       ABCA12          -0.089958  7.175205e-01  8.188977e-01  CCRCC\n",
       "21        ABCA2           0.056825  6.997512e-01  8.064600e-01  CCRCC\n",
       "22        ABCA6           0.370021  7.012739e-04  3.389490e-03  CCRCC\n",
       "23        ABCA8          -0.084953  1.692837e-01  3.012979e-01  CCRCC\n",
       "24        ABCB1           0.072682  2.975122e-03  1.171726e-02  CCRCC\n",
       "25       ABCB10          -0.256389  3.868174e-02  9.715232e-02  CCRCC\n",
       "26        ABCB6           0.125776  5.996023e-03  2.109179e-02  CCRCC\n",
       "27        ABCB7          -0.227777  2.300171e-01  3.751268e-01  CCRCC\n",
       "28        ABCB8          -0.095291  1.518858e-02  4.539284e-02  CCRCC\n",
       "29        ABCC1           0.505836  6.018861e-15  1.439908e-13  CCRCC\n",
       "...         ...                ...           ...           ...    ...\n",
       "50278   ZNF804B           0.259350  2.735740e-01  3.399482e-01   LUAD\n",
       "50279    ZNF827           0.503660  2.645396e-02  4.292721e-02   LUAD\n",
       "50280    ZNF830           0.605859  2.272676e-06  9.573220e-06   LUAD\n",
       "50281    ZNF831          -0.255886  1.202825e-01  1.656746e-01   LUAD\n",
       "50282    ZNF837           0.161310  3.262350e-01  3.962325e-01   LUAD\n",
       "50283    ZNF860          -0.072901  6.550842e-01  7.100679e-01   LUAD\n",
       "50284     ZNF92           0.009830  9.482255e-01  9.570740e-01   LUAD\n",
       "50285     ZNF98           0.139564  4.447458e-01  5.140409e-01   LUAD\n",
       "50286    ZNHIT1          -0.222195  1.777904e-01  2.324400e-01   LUAD\n",
       "50287    ZNHIT2           0.734775  1.808708e-07  9.480871e-07   LUAD\n",
       "50288    ZNHIT3           0.196080  1.355067e-01  1.835773e-01   LUAD\n",
       "50289    ZNHIT6           0.585166  2.861270e-06  1.182478e-05   LUAD\n",
       "50290     ZNRD1           0.130238  2.494071e-01  3.137159e-01   LUAD\n",
       "50291     ZNRF2           0.730968  1.473860e-14  2.442431e-13   LUAD\n",
       "50292      ZPR1           0.712566  4.114687e-08  2.477347e-07   LUAD\n",
       "50293    ZRANB2           0.094088  4.713707e-01  5.398518e-01   LUAD\n",
       "50294     ZRSR2           0.239083  8.855146e-02  1.263193e-01   LUAD\n",
       "50295   ZSCAN16           0.155307  3.793274e-01  4.507566e-01   LUAD\n",
       "50296   ZSCAN18           0.391928  1.055189e-05  3.835977e-05   LUAD\n",
       "50297   ZSCAN23           0.245749  3.112602e-01  3.803871e-01   LUAD\n",
       "50298   ZSCAN26           0.381625  1.278330e-02  2.242804e-02   LUAD\n",
       "50299   ZSCAN31           0.510219  1.242018e-03  2.804028e-03   LUAD\n",
       "50300      ZW10           0.571723  3.114751e-09  2.320835e-08   LUAD\n",
       "50301    ZWILCH           0.832168  4.549808e-12  5.461398e-11   LUAD\n",
       "50302     ZWINT           0.837582  2.770355e-08  1.722312e-07   LUAD\n",
       "50303      ZXDC           0.073854  6.395306e-01  6.962162e-01   LUAD\n",
       "50304    ZYG11B           0.734569  1.583743e-10  1.504331e-09   LUAD\n",
       "50305       ZYX           0.440861  2.253307e-06  9.499585e-06   LUAD\n",
       "50306     ZZEF1           0.422001  3.731185e-11  3.889196e-10   LUAD\n",
       "50307      ZZZ3           0.717501  2.955080e-06  1.215796e-05   LUAD\n",
       "\n",
       "[50308 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_correlation_df = pd.read_csv('delta_correlation_df.csv')\n",
    "delta_correlation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_types = pd.unique(delta_correlation_df.Cancer) # ['CCRCC', 'Endometrial', 'HNSCC', 'LSCC', 'LUAD']\n",
    "all_profiles = []\n",
    "for cancer in cancer_types:\n",
    "    cancer_df = delta_correlation_df[delta_correlation_df.Cancer == cancer]\n",
    "#     background_genes  = list(pd.unique(cancer_df.Gene))\n",
    "    cancer_df = cancer_df[cancer_df.FDR < 0.05]\n",
    "    background_genes  = list(pd.unique(cancer_df.Gene))\n",
    "    cancer_profile = gp.profile(organism='hsapiens', query = background_genes, no_iea=True, sources = [\"KEGG\"],\n",
    "                       ordered=True, no_evidences=False, background= background_genes)\n",
    "    all_profiles.append(cancer_profile)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['CCRCC', 'Endometrial', 'HNSCC', 'LSCC', 'LUAD'], dtype=object)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_types"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:02010</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>7.545223e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>8</td>\n",
       "      <td>3365</td>\n",
       "      <td>0.571429</td>\n",
       "      <td>0.727273</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABCA6, ABCB1, ABCB6, ABCB8, ABCC1, ABCC3, ABC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00071</td>\n",
       "      <td>Fatty acid degradation</td>\n",
       "      <td>7.183519e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid degradation</td>\n",
       "      <td>19</td>\n",
       "      <td>122</td>\n",
       "      <td>13</td>\n",
       "      <td>3365</td>\n",
       "      <td>0.106557</td>\n",
       "      <td>0.684211</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA1, ACADL, ACADM, ACADSB, ACAT1, ACOX1, AC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00640</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>2.367187e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>23</td>\n",
       "      <td>51</td>\n",
       "      <td>7</td>\n",
       "      <td>3365</td>\n",
       "      <td>0.137255</td>\n",
       "      <td>0.304348</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABAT, ACAT1, ACOX1, ACOX3, ACSS1, ACSS2, ACSS3]</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01212</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>2.714408e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>21</td>\n",
       "      <td>45</td>\n",
       "      <td>8</td>\n",
       "      <td>3365</td>\n",
       "      <td>0.177778</td>\n",
       "      <td>0.380952</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA1, ACADL, ACADM, ACADSB, ACAT1, ACOX1, AC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01040</td>\n",
       "      <td>Biosynthesis of unsaturated fatty acids</td>\n",
       "      <td>3.185833e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of unsaturated fatty acids</td>\n",
       "      <td>11</td>\n",
       "      <td>42</td>\n",
       "      <td>6</td>\n",
       "      <td>3365</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA1, ACOT1, ACOT2, ACOT7, ACOX1, ACOX3]</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source      native                                     name       p_value  \\\n",
       "0   KEGG  KEGG:02010                         ABC transporters  7.545223e-08   \n",
       "1   KEGG  KEGG:00071                   Fatty acid degradation  7.183519e-05   \n",
       "2   KEGG  KEGG:00640                    Propanoate metabolism  2.367187e-02   \n",
       "3   KEGG  KEGG:01212                    Fatty acid metabolism  2.714408e-02   \n",
       "4   KEGG  KEGG:01040  Biosynthesis of unsaturated fatty acids  3.185833e-02   \n",
       "\n",
       "   significant                              description  term_size  \\\n",
       "0         True                         ABC transporters         11   \n",
       "1         True                   Fatty acid degradation         19   \n",
       "2         True                    Propanoate metabolism         23   \n",
       "3         True                    Fatty acid metabolism         21   \n",
       "4         True  Biosynthesis of unsaturated fatty acids         11   \n",
       "\n",
       "   query_size  intersection_size  effective_domain_size  precision    recall  \\\n",
       "0          14                  8                   3365   0.571429  0.727273   \n",
       "1         122                 13                   3365   0.106557  0.684211   \n",
       "2          51                  7                   3365   0.137255  0.304348   \n",
       "3          45                  8                   3365   0.177778  0.380952   \n",
       "4          42                  6                   3365   0.142857  0.545455   \n",
       "\n",
       "     query       parents                                      intersections  \\\n",
       "0  query_1  [KEGG:00000]  [ABCA6, ABCB1, ABCB6, ABCB8, ABCC1, ABCC3, ABC...   \n",
       "1  query_1  [KEGG:00000]  [ACAA1, ACADL, ACADM, ACADSB, ACAT1, ACOX1, AC...   \n",
       "2  query_1  [KEGG:00000]   [ABAT, ACAT1, ACOX1, ACOX3, ACSS1, ACSS2, ACSS3]   \n",
       "3  query_1  [KEGG:00000]  [ACAA1, ACADL, ACADM, ACADSB, ACAT1, ACOX1, AC...   \n",
       "4  query_1  [KEGG:00000]         [ACAA1, ACOT1, ACOT2, ACOT7, ACOX1, ACOX3]   \n",
       "\n",
       "                                           evidences  \n",
       "0  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "1  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "2  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "3  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "4   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG]]  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#CCRCC\n",
    "(all_profiles[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01212</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>0.016862</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>18</td>\n",
       "      <td>34</td>\n",
       "      <td>9</td>\n",
       "      <td>1876</td>\n",
       "      <td>0.264706</td>\n",
       "      <td>0.5</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA2, ACACA, ACADM, ACADS, ACAT2, ACOX3, ACS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source      native                   name   p_value  significant  \\\n",
       "0   KEGG  KEGG:01212  Fatty acid metabolism  0.016862         True   \n",
       "\n",
       "             description  term_size  query_size  intersection_size  \\\n",
       "0  Fatty acid metabolism         18          34                  9   \n",
       "\n",
       "   effective_domain_size  precision  recall    query       parents  \\\n",
       "0                   1876   0.264706     0.5  query_1  [KEGG:00000]   \n",
       "\n",
       "                                       intersections  \\\n",
       "0  [ACAA2, ACACA, ACADM, ACADS, ACAT2, ACOX3, ACS...   \n",
       "\n",
       "                                           evidences  \n",
       "0  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Endometrial\n",
    "(all_profiles[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [source, native, name, p_value, significant, description, term_size, query_size, intersection_size, effective_domain_size, precision, recall, query, parents, intersections, evidences]\n",
       "Index: []"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HNSCC\n",
    "(all_profiles[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:02010</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>9.095193e-24</td>\n",
       "      <td>True</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>18</td>\n",
       "      <td>32</td>\n",
       "      <td>16</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>0.888889</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABCA1, ABCA12, ABCA3, ABCA6, ABCA8, ABCB1, AB...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00071</td>\n",
       "      <td>Fatty acid degradation</td>\n",
       "      <td>1.585756e-18</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid degradation</td>\n",
       "      <td>32</td>\n",
       "      <td>242</td>\n",
       "      <td>23</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.095041</td>\n",
       "      <td>0.718750</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA1, ACAA2, ACADL, ACADM, ACADS, ACADSB, AC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01212</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>5.159682e-13</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>38</td>\n",
       "      <td>95</td>\n",
       "      <td>16</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.168421</td>\n",
       "      <td>0.421053</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA1, ACAA2, ACACA, ACADL, ACADM, ACADS, ACA...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00280</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>8.439474e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>38</td>\n",
       "      <td>242</td>\n",
       "      <td>16</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.066116</td>\n",
       "      <td>0.421053</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[AACS, ABAT, ACAA1, ACAA2, ACADM, ACADS, ACADS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00620</td>\n",
       "      <td>Pyruvate metabolism</td>\n",
       "      <td>1.207363e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>Pyruvate metabolism</td>\n",
       "      <td>31</td>\n",
       "      <td>242</td>\n",
       "      <td>14</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.057851</td>\n",
       "      <td>0.451613</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACACA, ACAT1, ACAT2, ACSS1, ACYP1, ADH1B, ADH...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00410</td>\n",
       "      <td>beta-Alanine metabolism</td>\n",
       "      <td>8.723237e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>beta-Alanine metabolism</td>\n",
       "      <td>18</td>\n",
       "      <td>242</td>\n",
       "      <td>11</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.045455</td>\n",
       "      <td>0.611111</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABAT, ACADS, ACOX3, ALDH1B1, ALDH2, ALDH3A1, ...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00010</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "      <td>2.463394e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "      <td>40</td>\n",
       "      <td>244</td>\n",
       "      <td>15</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.061475</td>\n",
       "      <td>0.375000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACSS1, ADH1B, ADH1C, ADH5, ADH7, ADPGK, ALDH1...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00640</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>6.071100e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>21</td>\n",
       "      <td>97</td>\n",
       "      <td>8</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.082474</td>\n",
       "      <td>0.380952</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABAT, ACACA, ACADS, ACAT1, ACAT2, ACOX3, ACSS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00061</td>\n",
       "      <td>Fatty acid biosynthesis</td>\n",
       "      <td>3.955072e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid biosynthesis</td>\n",
       "      <td>9</td>\n",
       "      <td>95</td>\n",
       "      <td>6</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.063158</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACACA, ACSF3, ACSL1, ACSL3, ACSL4, ACSL5]</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:04146</td>\n",
       "      <td>Peroxisome</td>\n",
       "      <td>4.816363e-03</td>\n",
       "      <td>True</td>\n",
       "      <td>Peroxisome</td>\n",
       "      <td>46</td>\n",
       "      <td>95</td>\n",
       "      <td>10</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.217391</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABCD3, ABCD4, ACAA1, ACOT8, ACOX2, ACOX3, ACS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03320</td>\n",
       "      <td>PPAR signaling pathway</td>\n",
       "      <td>2.807692e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>PPAR signaling pathway</td>\n",
       "      <td>34</td>\n",
       "      <td>95</td>\n",
       "      <td>9</td>\n",
       "      <td>7111</td>\n",
       "      <td>0.094737</td>\n",
       "      <td>0.264706</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA1, ACADL, ACADM, ACOX2, ACOX3, ACSL1, ACS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   source      native                                        name  \\\n",
       "0    KEGG  KEGG:02010                            ABC transporters   \n",
       "1    KEGG  KEGG:00071                      Fatty acid degradation   \n",
       "2    KEGG  KEGG:01212                       Fatty acid metabolism   \n",
       "3    KEGG  KEGG:00280  Valine, leucine and isoleucine degradation   \n",
       "4    KEGG  KEGG:00620                         Pyruvate metabolism   \n",
       "5    KEGG  KEGG:00410                     beta-Alanine metabolism   \n",
       "6    KEGG  KEGG:00010                Glycolysis / Gluconeogenesis   \n",
       "7    KEGG  KEGG:00640                       Propanoate metabolism   \n",
       "8    KEGG  KEGG:00061                     Fatty acid biosynthesis   \n",
       "9    KEGG  KEGG:04146                                  Peroxisome   \n",
       "10   KEGG  KEGG:03320                      PPAR signaling pathway   \n",
       "\n",
       "         p_value  significant                                 description  \\\n",
       "0   9.095193e-24         True                            ABC transporters   \n",
       "1   1.585756e-18         True                      Fatty acid degradation   \n",
       "2   5.159682e-13         True                       Fatty acid metabolism   \n",
       "3   8.439474e-08         True  Valine, leucine and isoleucine degradation   \n",
       "4   1.207363e-05         True                         Pyruvate metabolism   \n",
       "5   8.723237e-05         True                     beta-Alanine metabolism   \n",
       "6   2.463394e-04         True                Glycolysis / Gluconeogenesis   \n",
       "7   6.071100e-04         True                       Propanoate metabolism   \n",
       "8   3.955072e-03         True                     Fatty acid biosynthesis   \n",
       "9   4.816363e-03         True                                  Peroxisome   \n",
       "10  2.807692e-02         True                      PPAR signaling pathway   \n",
       "\n",
       "    term_size  query_size  intersection_size  effective_domain_size  \\\n",
       "0          18          32                 16                   7111   \n",
       "1          32         242                 23                   7111   \n",
       "2          38          95                 16                   7111   \n",
       "3          38         242                 16                   7111   \n",
       "4          31         242                 14                   7111   \n",
       "5          18         242                 11                   7111   \n",
       "6          40         244                 15                   7111   \n",
       "7          21          97                  8                   7111   \n",
       "8           9          95                  6                   7111   \n",
       "9          46          95                 10                   7111   \n",
       "10         34          95                  9                   7111   \n",
       "\n",
       "    precision    recall    query       parents  \\\n",
       "0    0.500000  0.888889  query_1  [KEGG:00000]   \n",
       "1    0.095041  0.718750  query_1  [KEGG:00000]   \n",
       "2    0.168421  0.421053  query_1  [KEGG:00000]   \n",
       "3    0.066116  0.421053  query_1  [KEGG:00000]   \n",
       "4    0.057851  0.451613  query_1  [KEGG:00000]   \n",
       "5    0.045455  0.611111  query_1  [KEGG:00000]   \n",
       "6    0.061475  0.375000  query_1  [KEGG:00000]   \n",
       "7    0.082474  0.380952  query_1  [KEGG:00000]   \n",
       "8    0.063158  0.666667  query_1  [KEGG:00000]   \n",
       "9    0.105263  0.217391  query_1  [KEGG:00000]   \n",
       "10   0.094737  0.264706  query_1  [KEGG:00000]   \n",
       "\n",
       "                                        intersections  \\\n",
       "0   [ABCA1, ABCA12, ABCA3, ABCA6, ABCA8, ABCB1, AB...   \n",
       "1   [ACAA1, ACAA2, ACADL, ACADM, ACADS, ACADSB, AC...   \n",
       "2   [ACAA1, ACAA2, ACACA, ACADL, ACADM, ACADS, ACA...   \n",
       "3   [AACS, ABAT, ACAA1, ACAA2, ACADM, ACADS, ACADS...   \n",
       "4   [ACACA, ACAT1, ACAT2, ACSS1, ACYP1, ADH1B, ADH...   \n",
       "5   [ABAT, ACADS, ACOX3, ALDH1B1, ALDH2, ALDH3A1, ...   \n",
       "6   [ACSS1, ADH1B, ADH1C, ADH5, ADH7, ADPGK, ALDH1...   \n",
       "7   [ABAT, ACACA, ACADS, ACAT1, ACAT2, ACOX3, ACSS...   \n",
       "8          [ACACA, ACSF3, ACSL1, ACSL3, ACSL4, ACSL5]   \n",
       "9   [ABCD3, ABCD4, ACAA1, ACOT8, ACOX2, ACOX3, ACS...   \n",
       "10  [ACAA1, ACADL, ACADM, ACOX2, ACOX3, ACSL1, ACS...   \n",
       "\n",
       "                                            evidences  \n",
       "0   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "1   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "2   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "3   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "4   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "5   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "6   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "7   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "8    [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG]]  \n",
       "9   [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "10  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#LSCC\n",
    "(all_profiles[3])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:02010</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>4.090863e-22</td>\n",
       "      <td>True</td>\n",
       "      <td>ABC transporters</td>\n",
       "      <td>17</td>\n",
       "      <td>27</td>\n",
       "      <td>15</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.555556</td>\n",
       "      <td>0.882353</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABCA1, ABCA12, ABCA3, ABCB10, ABCB6, ABCB7, A...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00071</td>\n",
       "      <td>Fatty acid degradation</td>\n",
       "      <td>2.333943e-12</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid degradation</td>\n",
       "      <td>28</td>\n",
       "      <td>222</td>\n",
       "      <td>19</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.085586</td>\n",
       "      <td>0.678571</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA2, ACADL, ACADM, ACADS, ACADSB, ACADVL, A...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01212</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>1.515898e-08</td>\n",
       "      <td>True</td>\n",
       "      <td>Fatty acid metabolism</td>\n",
       "      <td>34</td>\n",
       "      <td>85</td>\n",
       "      <td>13</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.152941</td>\n",
       "      <td>0.382353</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACAA2, ACACA, ACADL, ACADM, ACADS, ACADSB, AC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00280</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>2.649866e-05</td>\n",
       "      <td>True</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>34</td>\n",
       "      <td>222</td>\n",
       "      <td>14</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.063063</td>\n",
       "      <td>0.411765</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[AACS, ABAT, ACAA2, ACAD8, ACADM, ACADS, ACADS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00620</td>\n",
       "      <td>Pyruvate metabolism</td>\n",
       "      <td>2.318216e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>Pyruvate metabolism</td>\n",
       "      <td>26</td>\n",
       "      <td>222</td>\n",
       "      <td>13</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.058559</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACACA, ACAT1, ACSS1, ACSS2, ACYP2, ADH1B, ADH...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00640</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>7.989158e-04</td>\n",
       "      <td>True</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>20</td>\n",
       "      <td>89</td>\n",
       "      <td>8</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.089888</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ABAT, ACACA, ACADS, ACAT1, ACOX3, ACSS1, ACSS...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00010</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "      <td>2.602875e-02</td>\n",
       "      <td>True</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "      <td>39</td>\n",
       "      <td>225</td>\n",
       "      <td>13</td>\n",
       "      <td>6276</td>\n",
       "      <td>0.057778</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ACSS1, ACSS2, ADH1B, ADH1C, ADH5, ADH7, ALDH1...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  source      native                                        name  \\\n",
       "0   KEGG  KEGG:02010                            ABC transporters   \n",
       "1   KEGG  KEGG:00071                      Fatty acid degradation   \n",
       "2   KEGG  KEGG:01212                       Fatty acid metabolism   \n",
       "3   KEGG  KEGG:00280  Valine, leucine and isoleucine degradation   \n",
       "4   KEGG  KEGG:00620                         Pyruvate metabolism   \n",
       "5   KEGG  KEGG:00640                       Propanoate metabolism   \n",
       "6   KEGG  KEGG:00010                Glycolysis / Gluconeogenesis   \n",
       "\n",
       "        p_value  significant                                 description  \\\n",
       "0  4.090863e-22         True                            ABC transporters   \n",
       "1  2.333943e-12         True                      Fatty acid degradation   \n",
       "2  1.515898e-08         True                       Fatty acid metabolism   \n",
       "3  2.649866e-05         True  Valine, leucine and isoleucine degradation   \n",
       "4  2.318216e-04         True                         Pyruvate metabolism   \n",
       "5  7.989158e-04         True                       Propanoate metabolism   \n",
       "6  2.602875e-02         True                Glycolysis / Gluconeogenesis   \n",
       "\n",
       "   term_size  query_size  intersection_size  effective_domain_size  precision  \\\n",
       "0         17          27                 15                   6276   0.555556   \n",
       "1         28         222                 19                   6276   0.085586   \n",
       "2         34          85                 13                   6276   0.152941   \n",
       "3         34         222                 14                   6276   0.063063   \n",
       "4         26         222                 13                   6276   0.058559   \n",
       "5         20          89                  8                   6276   0.089888   \n",
       "6         39         225                 13                   6276   0.057778   \n",
       "\n",
       "     recall    query       parents  \\\n",
       "0  0.882353  query_1  [KEGG:00000]   \n",
       "1  0.678571  query_1  [KEGG:00000]   \n",
       "2  0.382353  query_1  [KEGG:00000]   \n",
       "3  0.411765  query_1  [KEGG:00000]   \n",
       "4  0.500000  query_1  [KEGG:00000]   \n",
       "5  0.400000  query_1  [KEGG:00000]   \n",
       "6  0.333333  query_1  [KEGG:00000]   \n",
       "\n",
       "                                       intersections  \\\n",
       "0  [ABCA1, ABCA12, ABCA3, ABCB10, ABCB6, ABCB7, A...   \n",
       "1  [ACAA2, ACADL, ACADM, ACADS, ACADSB, ACADVL, A...   \n",
       "2  [ACAA2, ACACA, ACADL, ACADM, ACADS, ACADSB, AC...   \n",
       "3  [AACS, ABAT, ACAA2, ACAD8, ACADM, ACADS, ACADS...   \n",
       "4  [ACACA, ACAT1, ACSS1, ACSS2, ACYP2, ADH1B, ADH...   \n",
       "5  [ABAT, ACACA, ACADS, ACAT1, ACOX3, ACSS1, ACSS...   \n",
       "6  [ACSS1, ACSS2, ADH1B, ADH1C, ADH5, ADH7, ALDH1...   \n",
       "\n",
       "                                           evidences  \n",
       "0  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "1  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "2  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "3  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "4  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "5  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "6  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#LUAD\n",
    "(all_profiles[4])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exploration: Sustaining and Proliferating Hallmark Pathways"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These are the Sustaining and Proliferating Hallmark Pathways\n",
    "sus_prof_pathways = [\n",
    "    'ABC transporters',\n",
    "    'Fatty acid degradation',\n",
    "    'Metabolic pathways',\n",
    "    'Fatty acid metabolism',\n",
    "    'Pyruvate metabolism',\n",
    "    'Valine, leucine and isoleucine degradation',\n",
    "    'Butanoate metabolism',\n",
    "    'Glycolysis / Gluconeogenesis',\n",
    "    'Lysosome',\n",
    "    'Peroxisome',\n",
    "    'Biosynthesis of unsaturated fatty acids',\n",
    "    'Thiamine metabolism',\n",
    "    'Purine metabolism',\n",
    "    'Glyoxylate and dicarboxylate metabolism',\n",
    "    'Tyrosine metabolism',\n",
    "    'Carbon metabolism',\n",
    "    'Various types of N-glycan biosynthesis',\n",
    "    'Lysine degradation',\n",
    "    'Chemical carcinogenesis',\n",
    "    'Arginine biosynthesis',\n",
    "    'Cysteine and methionine metabolism',\n",
    "    'Proteasome',\n",
    "    'Drug metabolism - cytochrome P450',\n",
    "    'Arginine and proline metabolism',\n",
    "    'Synthesis and degradation of ketone bodies',\n",
    "    'Citrate cycle (TCA cycle)',\n",
    "    'Pantothenate and CoA biosynthesis',\n",
    "    'Oxidative phosphorylation',\n",
    "    'Fructose and mannose metabolism',\n",
    "    'Biosynthesis of amino acids',\n",
    "    'DNA replication',\n",
    "    'D-Glutamine and D-glutamate metabolism',\n",
    "    'Phenylalanine metabolism',\n",
    "    'Translation Factors',\n",
    "    'Mitochondrial LC-Fatty Acid Beta-Oxidation',\n",
    "    'Fatty Acid Beta Oxidation',\n",
    "    'Fatty Acid Biosynthesis',\n",
    "    'Nuclear Receptors in Lipid Metabolism and Toxicity',\n",
    "    'Fatty Acid Omega Oxidation',\n",
    "    'Fluoroacetic acid toxicity',\n",
    "    'Statin Pathway',\n",
    "    'Pregnane X Receptor pathway',\n",
    "    'Integrated Cancer Pathway',\n",
    "    'G Protein Signaling Pathways',\n",
    "    'Cytoplasmic Ribosomal Proteins',\n",
    "    'Mitochondrial complex I assembly model OXPHOS system',\n",
    "    'Nanoparticle triggered autophagic cell death',\n",
    "    'Nanomaterial induced apoptosis',\n",
    "    'Composition of Lipid Particles'\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Pathway searching\n",
    "\n",
    "First we'll make one dataframe that has all the enriched pathways for all cancer types, adding a column that identifies which cancer type each pathway came from. Note that if one pathway is enriched in multiple cancer types, it will be in the table multiple times, once for each cancer type."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "all_profiles_df = pd.DataFrame()\n",
    "for profile, cancer_type in zip(all_profiles, cancer_types):\n",
    "    profile = profile.assign(cancer_type=cancer_type)\n",
    "    all_profiles_df = all_profiles_df.append(profile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we'll write a function that will take a list of pathways we're interested in, and return a table showing which cancer types those pathways are enriched in."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cancers</th>\n",
       "      <th>pathways</th>\n",
       "      <th>pathways_ct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[CCRCC, Endometrial, LSCC, LUAD]</td>\n",
       "      <td>[Fatty acid metabolism]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>[CCRCC, LSCC, LUAD]</td>\n",
       "      <td>[ABC transporters, Fatty acid degradation]</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>[LSCC, LUAD]</td>\n",
       "      <td>[Glycolysis / Gluconeogenesis, Pyruvate metabo...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>[CCRCC]</td>\n",
       "      <td>[Biosynthesis of unsaturated fatty acids]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>[LSCC]</td>\n",
       "      <td>[Peroxisome]</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                            cancers  \\\n",
       "0  [CCRCC, Endometrial, LSCC, LUAD]   \n",
       "1               [CCRCC, LSCC, LUAD]   \n",
       "2                      [LSCC, LUAD]   \n",
       "3                           [CCRCC]   \n",
       "4                            [LSCC]   \n",
       "\n",
       "                                            pathways  pathways_ct  \n",
       "0                            [Fatty acid metabolism]            1  \n",
       "1         [ABC transporters, Fatty acid degradation]            2  \n",
       "2  [Glycolysis / Gluconeogenesis, Pyruvate metabo...            3  \n",
       "3          [Biosynthesis of unsaturated fatty acids]            1  \n",
       "4                                       [Peroxisome]            1  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def summarize_pathways(all_profiles, selected_pathways):\n",
    "    \n",
    "    # Pull out the profiles for our selected pathways\n",
    "    # Note that if a pathway isn't found in the profiles table, this will omit it silently\n",
    "    selected_profiles = all_profiles[all_profiles[\"name\"].isin(selected_pathways)]\n",
    "    \n",
    "    # Group by pathway, to have an entry for which cancer types each pathway is enriched in\n",
    "    # Then group by which cancer types each pathway is enriched in, so for each cancer type\n",
    "    # group we have a list of all pathways that are enriched in all members of that group of\n",
    "    # cancer types\n",
    "    summ = selected_profiles.\\\n",
    "    groupby(\"name\").\\\n",
    "    agg(cancers_str=(\"cancer_type\", lambda x: \"_\".join(sorted(set(x))))).\\\n",
    "    reset_index(drop=False).\\\n",
    "    groupby(\"cancers_str\").\\\n",
    "    agg(pathways=(\"name\", lambda x: sorted(set(x)))).\\\n",
    "    reset_index(drop=False)\n",
    "\n",
    "    # Sort everything nicely\n",
    "    summ.insert(0, \"cancers\", summ[\"cancers_str\"].str.split(\"_\"))\n",
    "    summ = summ.\\\n",
    "    assign(cancers_ct=summ[\"cancers\"].apply(len)).\\\n",
    "    sort_values(by=[\"cancers_ct\", \"cancers_str\"], ascending=[False, True]).\\\n",
    "    drop(columns=[\"cancers_str\", \"cancers_ct\"]).\\\n",
    "    reset_index(drop=True)\n",
    "\n",
    "    summ = summ.\\\n",
    "    assign(pathways_ct=summ[\"pathways\"].apply(len))\n",
    "\n",
    "    return summ\n",
    "\n",
    "sus_prof_sum = summarize_pathways(all_profiles_df, sus_prof_pathways)\n",
    "sus_prof_sum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Fatty acid metabolism']"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sus_prof_sum.loc[0, \"pathways\"]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This list copied from output of cell above. Descriptions from KEGG and other searches.\n",
    "\n",
    "- ABC transporters\n",
    "    - Use ATP hydrolysis for active transport of many substrates, including ions, sugars, lipids, sterols, peptides, proteins, and drugs.\n",
    "- Arginine and proline metabolism\n",
    "    - These reactions are bidirectional, so adjusts to the current needs of the cell\n",
    "    - Metabolic needs shift to support metastasis or other activities [[28492237]](https://pubmed.ncbi.nlm.nih.gov/28492237/)\n",
    "- Biosynthesis of unsaturated fatty acids\n",
    "    - Important for cancer to change growth patterns [[31819192]](https://pubmed.ncbi.nlm.nih.gov/31819192/)\n",
    "- Butanoate metabolism\n",
    "    - Type of fatty acid metabolism\n",
    "    - Involved in gut microbiome homeostasis?\n",
    "- Carbon metabolism\n",
    "- Citrate cycle (TCA cycle)\n",
    "- Fatty acid degradation\n",
    "- Fatty acid metabolism\n",
    "- Glycolysis / Gluconeogenesis\n",
    "- Glyoxylate and dicarboxylate metabolism\n",
    "- Lysine degradation\n",
    "- Lysosome\n",
    "- Metabolic pathways\n",
    "- Oxidative phosphorylation\n",
    "- Pantothenate and CoA biosynthesis\n",
    "- Peroxisome\n",
    "- Purine metabolism\n",
    "- Pyruvate metabolism\n",
    "- Synthesis and degradation of ketone bodies\n",
    "- Thiamine metabolism\n",
    "- Tyrosine metabolism\n",
    "- Valine, leucine and isoleucine degradation\n",
    "- Various types of N-glycan biosynthesis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Let's explore which pathways are found in which cancer types\n",
    "\n",
    "Why are some pathways not showing up in any cancers in our table?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "not_found = []\n",
    "found = {}\n",
    "cancers_map = {cancer: [] for cancer in cancer_types}\n",
    "\n",
    "for pw in sus_prof_pathways:\n",
    "    cancers = all_profiles_df[all_profiles_df[\"name\"] == pw][\"cancer_type\"]\n",
    "    if len(cancers) == 0:\n",
    "        not_found.append(pw)\n",
    "    else:\n",
    "        found[pw] = []\n",
    "        for cancer in cancers:\n",
    "            found[pw].append(cancer)\n",
    "            cancers_map[cancer].append(pw)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Metabolic pathways',\n",
       " 'Butanoate metabolism',\n",
       " 'Lysosome',\n",
       " 'Thiamine metabolism',\n",
       " 'Purine metabolism',\n",
       " 'Glyoxylate and dicarboxylate metabolism',\n",
       " 'Tyrosine metabolism',\n",
       " 'Carbon metabolism',\n",
       " 'Various types of N-glycan biosynthesis',\n",
       " 'Lysine degradation',\n",
       " 'Chemical carcinogenesis',\n",
       " 'Arginine biosynthesis',\n",
       " 'Cysteine and methionine metabolism',\n",
       " 'Proteasome',\n",
       " 'Drug metabolism - cytochrome P450',\n",
       " 'Arginine and proline metabolism',\n",
       " 'Synthesis and degradation of ketone bodies',\n",
       " 'Citrate cycle (TCA cycle)',\n",
       " 'Pantothenate and CoA biosynthesis',\n",
       " 'Oxidative phosphorylation',\n",
       " 'Fructose and mannose metabolism',\n",
       " 'Biosynthesis of amino acids',\n",
       " 'DNA replication',\n",
       " 'D-Glutamine and D-glutamate metabolism',\n",
       " 'Phenylalanine metabolism',\n",
       " 'Translation Factors',\n",
       " 'Mitochondrial LC-Fatty Acid Beta-Oxidation',\n",
       " 'Fatty Acid Beta Oxidation',\n",
       " 'Fatty Acid Biosynthesis',\n",
       " 'Nuclear Receptors in Lipid Metabolism and Toxicity',\n",
       " 'Fatty Acid Omega Oxidation',\n",
       " 'Fluoroacetic acid toxicity',\n",
       " 'Statin Pathway',\n",
       " 'Pregnane X Receptor pathway',\n",
       " 'Integrated Cancer Pathway',\n",
       " 'G Protein Signaling Pathways',\n",
       " 'Cytoplasmic Ribosomal Proteins',\n",
       " 'Mitochondrial complex I assembly model OXPHOS system',\n",
       " 'Nanoparticle triggered autophagic cell death',\n",
       " 'Nanomaterial induced apoptosis',\n",
       " 'Composition of Lipid Particles']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "not_found"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "def print_map(dmap):\n",
    "    for k in sorted(dmap.keys()):\n",
    "        print(k)\n",
    "        for v in sorted(dmap[k]):\n",
    "            print(f\"\\t{v}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ABC transporters\n",
      "\tCCRCC\n",
      "\tLSCC\n",
      "\tLUAD\n",
      "Biosynthesis of unsaturated fatty acids\n",
      "\tCCRCC\n",
      "Fatty acid degradation\n",
      "\tCCRCC\n",
      "\tLSCC\n",
      "\tLUAD\n",
      "Fatty acid metabolism\n",
      "\tCCRCC\n",
      "\tEndometrial\n",
      "\tLSCC\n",
      "\tLUAD\n",
      "Glycolysis / Gluconeogenesis\n",
      "\tLSCC\n",
      "\tLUAD\n",
      "Peroxisome\n",
      "\tLSCC\n",
      "Pyruvate metabolism\n",
      "\tLSCC\n",
      "\tLUAD\n",
      "Valine, leucine and isoleucine degradation\n",
      "\tLSCC\n",
      "\tLUAD\n"
     ]
    }
   ],
   "source": [
    "print_map(found)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CCRCC\n",
      "\tABC transporters\n",
      "\tBiosynthesis of unsaturated fatty acids\n",
      "\tFatty acid degradation\n",
      "\tFatty acid metabolism\n",
      "Endometrial\n",
      "\tFatty acid metabolism\n",
      "HNSCC\n",
      "LSCC\n",
      "\tABC transporters\n",
      "\tFatty acid degradation\n",
      "\tFatty acid metabolism\n",
      "\tGlycolysis / Gluconeogenesis\n",
      "\tPeroxisome\n",
      "\tPyruvate metabolism\n",
      "\tValine, leucine and isoleucine degradation\n",
      "LUAD\n",
      "\tABC transporters\n",
      "\tFatty acid degradation\n",
      "\tFatty acid metabolism\n",
      "\tGlycolysis / Gluconeogenesis\n",
      "\tPyruvate metabolism\n",
      "\tValine, leucine and isoleucine degradation\n"
     ]
    }
   ],
   "source": [
    "print_map(cancers_map)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
