{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import scipy\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import math\n",
    "import pandas as pd\n",
    "import statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that luad index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (<ipython-input-2-00c37755498a>, line 15)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that lscc index is up-to-date...   \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-00c37755498a>, line 18)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-00c37755498a>, line 19)\n"
     ]
    }
   ],
   "source": [
    "cptac.download(\"brca\")\n",
    "cptac.download(\"ccrcc\")\n",
    "cptac.download(\"colon\")\n",
    "cptac.download(\"endometrial\")\n",
    "cptac.download(\"gbm\")\n",
    "cptac.download(\"luad\")\n",
    "cptac.download(\"ovarian\")\n",
    "cptac.download(\"hnscc\")\n",
    "cptac.download(\"lscc\")\n",
    "\n",
    "brca = cptac.Brca()\n",
    "ccrcc = cptac.Ccrcc()\n",
    "colon = cptac.Colon()\n",
    "en = cptac.Endometrial()\n",
    "gbm = cptac.Gbm()\n",
    "luad = cptac.Luad()\n",
    "ovarian = cptac.Ovarian()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_list(tissue):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    gene_list = []\n",
    "    prot = tissue.get_proteomics()\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = prot.columns.get_level_values(0)\n",
    "    trans = tissue.get_transcriptomics()\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = trans.columns.get_level_values(0)\n",
    "    for i in prot:\n",
    "        if i in trans and i not in gene_list:\n",
    "            gene_list.append(i)\n",
    "    return gene_list\n",
    "\n",
    "\n",
    "def correlation_list(tissue, gene_list, valid_sample_number = 30, tissue_type = \"both\"):\n",
    "    #Returns a list of lists containing the name of the gene and the correlation\n",
    "    corr_list = []\n",
    "    pval_list = []\n",
    "    \n",
    "    prot = tissue.get_proteomics(tissue_type)\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = cptac.utils.reduce_multiindex(prot, levels_to_drop = \"Database_ID\")\n",
    "    \n",
    "    trans = tissue.get_transcriptomics(tissue_type)\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = cptac.utils.reduce_multiindex(trans, levels_to_drop = \"Database_ID\")\n",
    "        \n",
    "    prot_index_values = list(prot.index.values)\n",
    "    trans_index_values = list(trans.index.values)\n",
    "    \n",
    "\n",
    "    for found_index in prot_index_values:\n",
    "        if found_index not in trans_index_values:\n",
    "            prot = prot.drop(index=found_index)\n",
    "    for found_index in trans_index_values:\n",
    "        if found_index not in prot_index_values:\n",
    "            trans = trans.drop(index=found_index)\n",
    "  \n",
    "    for gene in gene_list:\n",
    "        prot_measurements = prot[gene]\n",
    "        prot_count = len(prot_measurements.dropna())\n",
    "        \n",
    "        trans_measurements = trans[gene]\n",
    "        trans_count = len(trans_measurements.dropna())\n",
    "        \n",
    "        if prot_count < valid_sample_number or trans_count < valid_sample_number:\n",
    "            continue\n",
    "            \n",
    "        nan_indices = set()\n",
    "        prot_indices = list(prot_measurements.index.values)\n",
    "        trans_indices = list(trans_measurements.index.values)\n",
    "        \n",
    "        #Here we are only currently taking the first column of multi_indices\n",
    "        if isinstance(prot_measurements, pd.core.frame.DataFrame): \n",
    "            prot_measurements = prot_measurements.iloc[:,0]\n",
    "        if isinstance(trans_measurements, pd.core.frame.DataFrame):\n",
    "            trans_measurements = trans_measurements.iloc[:,0]\n",
    "        for i in range(len(prot_measurements)):\n",
    "            if math.isnan(prot_measurements[i]):\n",
    "                nan_indices.add(prot_indices[i])\n",
    "                \n",
    "        for i in range(len(trans_measurements)):\n",
    "            if math.isnan(trans_measurements[i]):\n",
    "                nan_indices.add(trans_indices[i])\n",
    "        \n",
    "        nan_indices = list(nan_indices)\n",
    "        prot_measurements = prot_measurements.drop(nan_indices)\n",
    "        trans_measurements = trans_measurements.drop(nan_indices)\n",
    "        \n",
    "        correlation,pval = scipy.stats.pearsonr(prot_measurements, trans_measurements)\n",
    "        if math.isnan(correlation):\n",
    "            continue\n",
    "\n",
    "        corr_list.append([gene,correlation])\n",
    "        pval_list.append([gene, pval])\n",
    "    return corr_list, pval_list\n",
    "def ret_list(li):\n",
    "    #Returns a list of correlations from all genes\n",
    "    ret_li = []\n",
    "    for i in li:\n",
    "        ret_li.append(i[1])   \n",
    "    return ret_li\n",
    "def minmax(val_list):\n",
    "    if len(val_list) == 0:\n",
    "        return np.nan, np.nan\n",
    "    else:\n",
    "        min_val = min(val_list)\n",
    "        max_val = max(val_list)\n",
    "\n",
    "    return (min_val, max_val)\n",
    "def fill_genes_nan(all_genes, tissue_corr):\n",
    "    tissue_genes = []\n",
    "    for i, j  in tissue_corr:\n",
    "        tissue_genes.append(i)\n",
    "        \n",
    "    for i in all_genes:\n",
    "        if i not in tissue_genes:\n",
    "            tissue_corr.append([i, np.nan])\n",
    "    end_list = sorted(tissue_corr)\n",
    "           \n",
    "    return end_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2495 duplicated column headers. (<ipython-input-11-864b366a296f>, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2495 duplicated column headers. (<ipython-input-11-864b366a296f>, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-11-864b366a296f>, line 23)\n"
     ]
    }
   ],
   "source": [
    "brca_gene_list = get_gene_list(brca)\n",
    "ccrcc_gene_list = get_gene_list(ccrcc)\n",
    "colon_gene_list = get_gene_list(colon)\n",
    "gbm_gene_list = get_gene_list(gbm)\n",
    "luad_gene_list = get_gene_list(luad)\n",
    "ovarian_gene_list = get_gene_list(ovarian)\n",
    "en_gene_list = get_gene_list(en)\n",
    "hnscc_gene_list = get_gene_list(hnscc)\n",
    "lscc_gene_list = get_gene_list(lscc)\n",
    "\n",
    "brca_corr, brca_pval = correlation_list(brca, brca_gene_list, tissue_type = \"tumor\")\n",
    "ccrcc_corr, ccrcc_pval = correlation_list(ccrcc, ccrcc_gene_list, tissue_type = \"tumor\")\n",
    "colon_corr, colon_pval = correlation_list(colon, colon_gene_list, tissue_type = \"tumor\")\n",
    "en_corr, en_pval = correlation_list(en, en_gene_list, tissue_type = \"tumor\")\n",
    "gbm_corr, gbm_pval = correlation_list(gbm, gbm_gene_list, tissue_type = \"tumor\")\n",
    "luad_corr, luad_pval = correlation_list(luad,luad_gene_list, tissue_type = \"tumor\")\n",
    "ovarian_corr, ovarian_pval = correlation_list(ovarian, ovarian_gene_list, tissue_type = \"tumor\")\n",
    "hnscc_corr, hnscc_pval = correlation_list(hnscc, hnscc_gene_list, tissue_type = \"tumor\")\n",
    "lscc_corr, lscc_pval = correlation_list(lscc, lscc_gene_list, tissue_type = \"tumor\")\n",
    "\n",
    "normal_brca_corr, normal_brca_pval = correlation_list(brca, brca_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_ccrcc_corr, normal_ccrcc_pval = correlation_list(ccrcc, ccrcc_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_colon_corr, normal_colon_pval = correlation_list(colon, colon_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_en_corr, normal_en_pval = correlation_list(en, en_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_gbm_corr, normal_gbm_pval = correlation_list(gbm, gbm_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_luad_corr, normal_luad_pval = correlation_list(luad,luad_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_ovarian_corr, normal_ovarian_pval = correlation_list(ovarian, ovarian_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_hnscc_corr, normal_hnscc_pval = correlation_list(hnscc, hnscc_gene_list,valid_sample_number = 7, tissue_type=\"normal\")\n",
    "normal_lscc_corr, normal_lscc_pval = correlation_list(lscc, lscc_gene_list,valid_sample_number = 7, tissue_type=\"normal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_gene_list =[]\n",
    "df_correlation_list = []\n",
    "df_index = []\n",
    "tissue_type_list = []\n",
    "for gene , correlation in brca_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"BRCA\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in ccrcc_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"CCRCC\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "for gene , correlation in normal_ccrcc_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"CCRCC\")    \n",
    "    tissue_type_list.append(\"normal\")\n",
    "    \n",
    "for gene , correlation in colon_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"colon\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in en_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"Endometrial\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "for gene , correlation in normal_en_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"Endometrial\")    \n",
    "    tissue_type_list.append(\"normal\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in gbm_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"GBM\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "for gene , correlation in normal_gbm_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"GBM\")    \n",
    "    tissue_type_list.append(\"normal\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in luad_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"LUAD\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "for gene , correlation in normal_luad_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"LUAD\")    \n",
    "    tissue_type_list.append(\"normal\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in ovarian_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"Ovarian\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in hnscc_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"HNSCC\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "for gene , correlation in normal_hnscc_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"HNSCC\")    \n",
    "    tissue_type_list.append(\"normal\")\n",
    "    \n",
    "    \n",
    "for gene , correlation in lscc_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"LSCC\")    \n",
    "    tissue_type_list.append(\"tumor\")\n",
    "for gene , correlation in normal_lscc_corr:\n",
    "    df_gene_list.append(gene)\n",
    "    df_correlation_list.append(correlation)\n",
    "    df_index.append(\"LSCC\")    \n",
    "    tissue_type_list.append(\"normal\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_pval_list =[]\n",
    "for gene , pval in brca_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in ccrcc_pval:\n",
    "    df_pval_list.append(pval)\n",
    "for gene , pval in normal_ccrcc_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in colon_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in en_pval:\n",
    "    df_pval_list.append(pval)\n",
    "for gene , pval in normal_en_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in gbm_pval:\n",
    "    df_pval_list.append(pval)\n",
    "for gene , pval in normal_gbm_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in luad_pval:\n",
    "    df_pval_list.append(pval)\n",
    "for gene , pval in normal_luad_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in ovarian_pval:\n",
    "    df_pval_list.append(pval)\n",
    "\n",
    "for gene , pval in hnscc_pval:\n",
    "    df_pval_list.append(pval)\n",
    "for gene , pval in normal_hnscc_pval:\n",
    "    df_pval_list.append(pval)\n",
    "    \n",
    "for gene , pval in lscc_pval:\n",
    "    df_pval_list.append(pval)\n",
    "for gene , pval in normal_lscc_pval:\n",
    "    df_pval_list.append(pval)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "df_columns = {'Tissue Type': (tissue_type_list),'Gene': (df_gene_list), \n",
    "       'Correlation': (df_correlation_list), 'P-value': (df_pval_list)}\n",
    "       #'R-squared': (gbm_list_corr)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(df_pval_list) - len (tissue_type_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.DataFrame(df_columns , columns = [\"Tissue Type\", \"Gene\", \"Correlation\", \"P-value\"], index=df_index)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tissue Type</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Correlation</th>\n",
       "      <th>P-value</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>tumor</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.131286</td>\n",
       "      <td>1.657241e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>tumor</td>\n",
       "      <td>A2M</td>\n",
       "      <td>0.422497</td>\n",
       "      <td>1.249853e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>tumor</td>\n",
       "      <td>A2ML1</td>\n",
       "      <td>0.713289</td>\n",
       "      <td>1.373971e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>tumor</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.086532</td>\n",
       "      <td>3.432747e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BRCA</th>\n",
       "      <td>tumor</td>\n",
       "      <td>AACS</td>\n",
       "      <td>0.747434</td>\n",
       "      <td>4.695304e-23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LSCC</th>\n",
       "      <td>normal</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.268095</td>\n",
       "      <td>1.257086e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LSCC</th>\n",
       "      <td>normal</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.132199</td>\n",
       "      <td>2.040310e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LSCC</th>\n",
       "      <td>normal</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.225703</td>\n",
       "      <td>2.872053e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LSCC</th>\n",
       "      <td>normal</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.356916</td>\n",
       "      <td>4.137038e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LSCC</th>\n",
       "      <td>normal</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.046046</td>\n",
       "      <td>6.594335e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>147275 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Tissue Type    Gene  Correlation       P-value\n",
       "BRCA       tumor    A1BG     0.131286  1.657241e-01\n",
       "BRCA       tumor     A2M     0.422497  1.249853e-06\n",
       "BRCA       tumor   A2ML1     0.713289  1.373971e-17\n",
       "BRCA       tumor    AAAS     0.086532  3.432747e-01\n",
       "BRCA       tumor    AACS     0.747434  4.695304e-23\n",
       "...          ...     ...          ...           ...\n",
       "LSCC      normal    ZXDC     0.268095  1.257086e-02\n",
       "LSCC      normal  ZYG11B     0.132199  2.040310e-01\n",
       "LSCC      normal     ZYX     0.225703  2.872053e-02\n",
       "LSCC      normal   ZZEF1     0.356916  4.137038e-04\n",
       "LSCC      normal    ZZZ3     0.046046  6.594335e-01\n",
       "\n",
       "[147275 rows x 4 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
