{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import scipy\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import math\n",
    "import pandas as pd\n",
    "import statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_list(tissue):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    gene_list = []\n",
    "    prot = tissue.get_proteomics()\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = prot.columns.get_level_values(0)\n",
    "    trans = tissue.get_transcriptomics()\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = trans.columns.get_level_values(0)\n",
    "    for i in prot:\n",
    "        if i in trans and i not in gene_list:\n",
    "            gene_list.append(i)\n",
    "    return gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def correlation_list(tissue, valid_sample_number = 30, tissue_type = \"both\"):\n",
    "    #Returns a list of lists containing the name of the gene and the correlation\n",
    "    corr_list = []\n",
    "    pval_list = []\n",
    "    prot = tissue.get_proteomics(tissue_type)\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = cptac.utils.reduce_multiindex(prot, levels_to_drop = \"Database_ID\")\n",
    "    \n",
    "    trans = tissue.get_transcriptomics(tissue_type)\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = cptac.utils.reduce_multiindex(trans, levels_to_drop = \"Database_ID\")\n",
    "        \n",
    "    prot_index_values = list(prot.index.values)\n",
    "    trans_index_values = list(trans.index.values)\n",
    "    \n",
    "\n",
    "    for found_index in prot_index_values:\n",
    "        if found_index not in trans_index_values:\n",
    "            prot = prot.drop(index=found_index)\n",
    "    for found_index in trans_index_values:\n",
    "        if found_index not in prot_index_values:\n",
    "            trans = trans.drop(index=found_index)\n",
    "        \n",
    "    for gene in gene_list:\n",
    "        prot_measurements = prot[gene]\n",
    "        prot_count = len(prot_measurements.dropna())\n",
    "        \n",
    "        trans_measurements = trans[gene]\n",
    "        trans_count = len(trans_measurements.dropna())\n",
    "        \n",
    "        if prot_count < valid_sample_number or trans_count < valid_sample_number:\n",
    "            continue\n",
    "            \n",
    "        nan_indices = set()\n",
    "        prot_indices = list(prot_measurements.index.values)\n",
    "        trans_indices = list(trans_measurements.index.values)\n",
    "        \n",
    "        #Here we are only currently taking the first column of multi_indices\n",
    "        if isinstance(prot_measurements, pd.core.frame.DataFrame): \n",
    "            prot_measurements = prot_measurements.iloc[:,0]\n",
    "        if isinstance(trans_measurements, pd.core.frame.DataFrame):\n",
    "            trans_measurements = trans_measurements.iloc[:,0]\n",
    "        for i in range(len(prot_measurements)):\n",
    "            if math.isnan(prot_measurements[i]):\n",
    "                nan_indices.add(prot_indices[i])\n",
    "                \n",
    "        for i in range(len(trans_measurements)):\n",
    "            if math.isnan(trans_measurements[i]):\n",
    "                nan_indices.add(trans_indices[i])\n",
    "        \n",
    "        nan_indices = list(nan_indices)\n",
    "        prot_measurements = prot_measurements.drop(nan_indices)\n",
    "        trans_measurements = trans_measurements.drop(nan_indices)\n",
    "        \n",
    "        correlation,pval = scipy.stats.pearsonr(prot_measurements, trans_measurements)\n",
    "        if math.isnan(correlation):\n",
    "            continue\n",
    "        corr_list.append([gene,correlation])\n",
    "        pval_list.append([gene, pval])\n",
    "    return corr_list, pval_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def ret_list(li):\n",
    "    #Returns a list of correlations from all genes\n",
    "    ret_li = []\n",
    "    for i in li:\n",
    "        ret_li.append(i[1])   \n",
    "    return ret_li"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1BG-AS1</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A2MP1</th>\n",
       "      <th>A3GALT2</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>A4GNT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>...</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDB</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11A</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CPT000814</th>\n",
       "      <td>-0.2020</td>\n",
       "      <td>0.4851</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.0264</td>\n",
       "      <td>5.3754</td>\n",
       "      <td>-1.7357</td>\n",
       "      <td>-2.6428</td>\n",
       "      <td>-1.0939</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2623</td>\n",
       "      <td>...</td>\n",
       "      <td>0.2799</td>\n",
       "      <td>0.5755</td>\n",
       "      <td>-0.5886</td>\n",
       "      <td>-0.7428</td>\n",
       "      <td>1.5763</td>\n",
       "      <td>2.3063</td>\n",
       "      <td>-0.9448</td>\n",
       "      <td>0.6603</td>\n",
       "      <td>0.2159</td>\n",
       "      <td>0.1554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CPT001846</th>\n",
       "      <td>1.5602</td>\n",
       "      <td>0.8676</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.1659</td>\n",
       "      <td>-2.3353</td>\n",
       "      <td>1.7850</td>\n",
       "      <td>7.8635</td>\n",
       "      <td>1.3619</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0019</td>\n",
       "      <td>...</td>\n",
       "      <td>0.4305</td>\n",
       "      <td>-0.4707</td>\n",
       "      <td>-2.0699</td>\n",
       "      <td>-2.0325</td>\n",
       "      <td>-0.0221</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0745</td>\n",
       "      <td>1.9567</td>\n",
       "      <td>-0.4364</td>\n",
       "      <td>-0.1601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR001</th>\n",
       "      <td>-0.4547</td>\n",
       "      <td>1.7415</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3769</td>\n",
       "      <td>2.1803</td>\n",
       "      <td>1.3121</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3131</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.2010</td>\n",
       "      <td>-1.6472</td>\n",
       "      <td>0.3005</td>\n",
       "      <td>0.3112</td>\n",
       "      <td>0.3747</td>\n",
       "      <td>1.0260</td>\n",
       "      <td>0.2738</td>\n",
       "      <td>-0.0211</td>\n",
       "      <td>0.3720</td>\n",
       "      <td>-0.6570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR008</th>\n",
       "      <td>-1.4653</td>\n",
       "      <td>0.4251</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5979</td>\n",
       "      <td>5.8009</td>\n",
       "      <td>0.5635</td>\n",
       "      <td>-2.6931</td>\n",
       "      <td>-1.0861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0043</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3331</td>\n",
       "      <td>1.3433</td>\n",
       "      <td>-1.8711</td>\n",
       "      <td>-1.3578</td>\n",
       "      <td>0.0305</td>\n",
       "      <td>1.3676</td>\n",
       "      <td>-1.6430</td>\n",
       "      <td>0.3679</td>\n",
       "      <td>0.6431</td>\n",
       "      <td>-0.1793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X01BR009</th>\n",
       "      <td>1.0341</td>\n",
       "      <td>2.0925</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6195</td>\n",
       "      <td>7.0649</td>\n",
       "      <td>-0.7143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.2492</td>\n",
       "      <td>-2.2484</td>\n",
       "      <td>0.4772</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0054</td>\n",
       "      <td>-1.0478</td>\n",
       "      <td>-1.3751</td>\n",
       "      <td>-1.8079</td>\n",
       "      <td>-0.1200</td>\n",
       "      <td>-4.6279</td>\n",
       "      <td>-0.8619</td>\n",
       "      <td>-0.0338</td>\n",
       "      <td>1.5769</td>\n",
       "      <td>-0.6337</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X21BR001</th>\n",
       "      <td>-0.0196</td>\n",
       "      <td>0.2171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3940</td>\n",
       "      <td>-2.0997</td>\n",
       "      <td>1.0183</td>\n",
       "      <td>6.4045</td>\n",
       "      <td>-0.2501</td>\n",
       "      <td>-1.7260</td>\n",
       "      <td>0.6178</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7115</td>\n",
       "      <td>0.1819</td>\n",
       "      <td>-1.2171</td>\n",
       "      <td>-1.3631</td>\n",
       "      <td>1.3455</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7515</td>\n",
       "      <td>0.3044</td>\n",
       "      <td>1.3806</td>\n",
       "      <td>-1.1197</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X21BR002</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.1073</td>\n",
       "      <td>-4.0915</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.8259</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6250</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.7193</td>\n",
       "      <td>-0.3678</td>\n",
       "      <td>-1.8014</td>\n",
       "      <td>-1.4538</td>\n",
       "      <td>1.5637</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.5786</td>\n",
       "      <td>0.8320</td>\n",
       "      <td>1.0435</td>\n",
       "      <td>-1.1151</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X21BR010</th>\n",
       "      <td>-3.2675</td>\n",
       "      <td>-1.0806</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.4900</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-5.5735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.2094</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.6299</td>\n",
       "      <td>-3.1189</td>\n",
       "      <td>0.7028</td>\n",
       "      <td>0.7773</td>\n",
       "      <td>-0.1069</td>\n",
       "      <td>-4.5936</td>\n",
       "      <td>0.9043</td>\n",
       "      <td>-1.5439</td>\n",
       "      <td>-1.4296</td>\n",
       "      <td>0.9932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X22BR005</th>\n",
       "      <td>-4.2705</td>\n",
       "      <td>-1.4557</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.3112</td>\n",
       "      <td>1.8311</td>\n",
       "      <td>1.1163</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0104</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0333</td>\n",
       "      <td>1.2037</td>\n",
       "      <td>-0.0319</td>\n",
       "      <td>0.2372</td>\n",
       "      <td>-0.9557</td>\n",
       "      <td>0.5448</td>\n",
       "      <td>-0.4072</td>\n",
       "      <td>-1.2626</td>\n",
       "      <td>-0.9846</td>\n",
       "      <td>1.2978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X22BR006</th>\n",
       "      <td>-2.4837</td>\n",
       "      <td>-0.2383</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.4869</td>\n",
       "      <td>-2.5440</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.1171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.8315</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.9753</td>\n",
       "      <td>-2.5921</td>\n",
       "      <td>1.1066</td>\n",
       "      <td>0.8053</td>\n",
       "      <td>-1.1356</td>\n",
       "      <td>-0.2454</td>\n",
       "      <td>2.0016</td>\n",
       "      <td>2.2719</td>\n",
       "      <td>1.3618</td>\n",
       "      <td>-0.2480</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>122 rows × 23121 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name          A1BG  A1BG-AS1  A1CF     A2M   A2ML1   A2MP1  A3GALT2  A4GALT  \\\n",
       "Patient_ID                                                                    \n",
       "CPT000814  -0.2020    0.4851   NaN -1.0264  5.3754 -1.7357  -2.6428 -1.0939   \n",
       "CPT001846   1.5602    0.8676   NaN -1.1659 -2.3353  1.7850   7.8635  1.3619   \n",
       "X01BR001   -0.4547    1.7415   NaN -0.3769  2.1803  1.3121      NaN  0.8310   \n",
       "X01BR008   -1.4653    0.4251   NaN -0.5979  5.8009  0.5635  -2.6931 -1.0861   \n",
       "X01BR009    1.0341    2.0925   NaN  0.6195  7.0649 -0.7143      NaN  0.2492   \n",
       "...            ...       ...   ...     ...     ...     ...      ...     ...   \n",
       "X21BR001   -0.0196    0.2171   NaN -0.3940 -2.0997  1.0183   6.4045 -0.2501   \n",
       "X21BR002       NaN       NaN   NaN  2.1073 -4.0915     NaN      NaN  3.8259   \n",
       "X21BR010   -3.2675   -1.0806   NaN  0.4900     NaN -0.1154      NaN -5.5735   \n",
       "X22BR005   -4.2705   -1.4557   NaN  0.3112  1.8311  1.1163      NaN  0.6396   \n",
       "X22BR006   -2.4837   -0.2383   NaN  1.4869 -2.5440     NaN      NaN -0.1171   \n",
       "\n",
       "Name         A4GNT    AAAS  ...  ZWILCH   ZWINT    ZXDA    ZXDB    ZXDC  \\\n",
       "Patient_ID                  ...                                           \n",
       "CPT000814      NaN  0.2623  ...  0.2799  0.5755 -0.5886 -0.7428  1.5763   \n",
       "CPT001846      NaN -0.0019  ...  0.4305 -0.4707 -2.0699 -2.0325 -0.0221   \n",
       "X01BR001       NaN -0.3131  ... -0.2010 -1.6472  0.3005  0.3112  0.3747   \n",
       "X01BR008       NaN  0.0043  ...  0.3331  1.3433 -1.8711 -1.3578  0.0305   \n",
       "X01BR009   -2.2484  0.4772  ... -0.0054 -1.0478 -1.3751 -1.8079 -0.1200   \n",
       "...            ...     ...  ...     ...     ...     ...     ...     ...   \n",
       "X21BR001   -1.7260  0.6178  ... -0.7115  0.1819 -1.2171 -1.3631  1.3455   \n",
       "X21BR002       NaN  0.6250  ... -0.7193 -0.3678 -1.8014 -1.4538  1.5637   \n",
       "X21BR010       NaN -1.2094  ... -0.6299 -3.1189  0.7028  0.7773 -0.1069   \n",
       "X22BR005       NaN  0.0104  ... -0.0333  1.2037 -0.0319  0.2372 -0.9557   \n",
       "X22BR006       NaN  0.8315  ... -0.9753 -2.5921  1.1066  0.8053 -1.1356   \n",
       "\n",
       "Name        ZYG11A  ZYG11B     ZYX   ZZEF1    ZZZ3  \n",
       "Patient_ID                                          \n",
       "CPT000814   2.3063 -0.9448  0.6603  0.2159  0.1554  \n",
       "CPT001846      NaN  0.0745  1.9567 -0.4364 -0.1601  \n",
       "X01BR001    1.0260  0.2738 -0.0211  0.3720 -0.6570  \n",
       "X01BR008    1.3676 -1.6430  0.3679  0.6431 -0.1793  \n",
       "X01BR009   -4.6279 -0.8619 -0.0338  1.5769 -0.6337  \n",
       "...            ...     ...     ...     ...     ...  \n",
       "X21BR001       NaN -0.7515  0.3044  1.3806 -1.1197  \n",
       "X21BR002       NaN -0.5786  0.8320  1.0435 -1.1151  \n",
       "X21BR010   -4.5936  0.9043 -1.5439 -1.4296  0.9932  \n",
       "X22BR005    0.5448 -0.4072 -1.2626 -0.9846  1.2978  \n",
       "X22BR006   -0.2454  2.0016  2.2719  1.3618 -0.2480  \n",
       "\n",
       "[122 rows x 23121 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca.get_transcriptomics()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that luad index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 14)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that ovarian index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LUAD dataset is under publication embargo until July 01, 2020. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 15)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that lscc index is up-to-date...   \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 17)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 18)\n"
     ]
    }
   ],
   "source": [
    "# cptac.download(\"brca\")\n",
    "# cptac.download(\"ccrcc\")\n",
    "# cptac.download(\"colon\")\n",
    "# cptac.download(\"endometrial\")\n",
    "# cptac.download(\"gbm\")\n",
    "# cptac.download(\"luad\")\n",
    "# cptac.download(\"ovarian\")\n",
    "# cptac.download(\"hnscc\")\n",
    "# cptac.download(\"lscc\")\n",
    "brca = cptac.Brca()\n",
    "ccrcc = cptac.Ccrcc()\n",
    "colon = cptac.Colon()\n",
    "en = cptac.Endometrial()\n",
    "gbm = cptac.Gbm()\n",
    "luad = cptac.Luad()\n",
    "ovarian = cptac.Ovarian()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_gene_list = get_gene_list(brca)\n",
    "ccrcc_gene_list = get_gene_list(ccrcc)\n",
    "colon_gene_list = get_gene_list(colon)\n",
    "gbm_gene_list = get_gene_list(gbm)\n",
    "luad_gene_list = get_gene_list(luad)\n",
    "ovarian_gene_list = get_gene_list(ovarian)\n",
    "en_gene_list = get_gene_list(en)\n",
    "hnscc_gene_list = get_gene_list(hnscc)\n",
    "lscc_gene_list = get_gene_list(lscc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 7)\n",
      "/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/scipy/stats/stats.py:3508: PearsonRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(PearsonRConstantInputWarning())\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2495 duplicated column headers. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 830 duplicated column headers. (/Users/benkk/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/ipykernel_launcher.py, line 7)\n"
     ]
    }
   ],
   "source": [
    "brca_corr, brca_pval = correlation_list(brca, brca_gene_list, tissue_type = \"tumor\")\n",
    "ccrcc_corr, ccrcc_pval = correlation_list(ccrcc, ccrcc_gene_list, tissue_type = \"tumor\")\n",
    "colon_corr, colon_pval = correlation_list(colon, colon_gene_list, tissue_type = \"tumor\")\n",
    "en_corr, en_pval = correlation_list(en, en_gene_list, tissue_type = \"tumor\")\n",
    "gbm_corr, gbm_pval = correlation_list(gbm, gbm_gene_list, tissue_type = \"tumor\")\n",
    "luad_corr, luad_pval = correlation_list(luad,luad_gene_list, tissue_type = \"tumor\")\n",
    "ovarian_corr, ovarian_pval = correlation_list(ovarian, ovarian_gene_list, tissue_type = \"tumor\")\n",
    "hnscc_corr, hnscc_pval = correlation_list(hnscc, hnscc_gene_list, tissue_type = \"tumor\")\n",
    "lscc_corr, lscc_pval = correlation_list(lscc, lscc_gene_list, tissue_type = \"tumor\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = brca_gene_list + ccrcc_gene_list + colon_gene_list + en_gene_list + gbm_gene_list+ luad_gene_list + ovarian_gene_list + hnscc_gene_list + lscc_gene_list\n",
    "common_genes = []\n",
    "for i in genes:\n",
    "    if i not in common_genes:\n",
    "        common_genes.append(i)\n",
    "common_genes = sorted(common_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for item in brca_corr:\n",
    "#     if item[0] == \"RGMA\":\n",
    "#         print(\"Here:\")\n",
    "#         print(item[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for item in brca_corr:\n",
    "#     print(item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_genes_nan(all_genes, tissue_corr):\n",
    "    tissue_genes = []\n",
    "    for i, j  in tissue_corr:\n",
    "        tissue_genes.append(i)\n",
    "        \n",
    "    for i in all_genes:\n",
    "        if i not in tissue_genes:\n",
    "            tissue_corr.append([i, np.nan])\n",
    "    end_list = sorted(tissue_corr)\n",
    "           \n",
    "    return end_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# brca_list_1 = fill_genes_nan(common_genes, brca_corr)\n",
    "# ccrcc_list_1 = fill_genes_nan(common_genes, ccrcc_corr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# print(brca_list_1[13478])\n",
    "# print(ccrcc_list_1[13478])\n",
    "# print(brca_list_1[10379])\n",
    "# print(ccrcc_list_1[10379])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for i in range(len(brca_list_1)):\n",
    "#     if brca_list_1[i][0] == \"RGMA\":\n",
    "#         print(\"i = \",i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# # print(brca_list_corr)\n",
    "# # print(len(lscc_list_corr))\n",
    "# # print(len(common_genes))\n",
    "# common_genes_2 = sorted(common_genes)\n",
    "# print(common_genes[13478])\n",
    "# print(common_genes_2[13478])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BRCA</th>\n",
       "      <th>CCRCC</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Endometrial</th>\n",
       "      <th>GBM</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>LSCC</th>\n",
       "      <th>Median</th>\n",
       "      <th>Stnd Deviation</th>\n",
       "      <th>+/- 3 sigma</th>\n",
       "      <th>Range</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.131286</td>\n",
       "      <td>-0.059781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.386703</td>\n",
       "      <td>-0.200579</td>\n",
       "      <td>-0.067920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.112133</td>\n",
       "      <td>0.165873</td>\n",
       "      <td>0.112133</td>\n",
       "      <td>0.193232</td>\n",
       "      <td>False</td>\n",
       "      <td>0.587282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.892757</td>\n",
       "      <td>0.698025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.795391</td>\n",
       "      <td>0.137696</td>\n",
       "      <td>False</td>\n",
       "      <td>0.194732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.422497</td>\n",
       "      <td>0.010289</td>\n",
       "      <td>-0.122255</td>\n",
       "      <td>0.024022</td>\n",
       "      <td>0.312922</td>\n",
       "      <td>0.381025</td>\n",
       "      <td>0.140341</td>\n",
       "      <td>0.201083</td>\n",
       "      <td>0.355647</td>\n",
       "      <td>0.201083</td>\n",
       "      <td>0.191378</td>\n",
       "      <td>False</td>\n",
       "      <td>0.544752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.713289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.835080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.531368</td>\n",
       "      <td>0.862612</td>\n",
       "      <td>0.817462</td>\n",
       "      <td>0.817462</td>\n",
       "      <td>0.135646</td>\n",
       "      <td>False</td>\n",
       "      <td>0.331245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.486499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.286010</td>\n",
       "      <td>0.386255</td>\n",
       "      <td>0.141767</td>\n",
       "      <td>False</td>\n",
       "      <td>0.200489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>0.200364</td>\n",
       "      <td>0.124010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.146719</td>\n",
       "      <td>0.185779</td>\n",
       "      <td>0.212842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.488809</td>\n",
       "      <td>0.583187</td>\n",
       "      <td>0.200364</td>\n",
       "      <td>0.181339</td>\n",
       "      <td>False</td>\n",
       "      <td>0.459178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>-0.029503</td>\n",
       "      <td>0.241315</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.231203</td>\n",
       "      <td>0.458107</td>\n",
       "      <td>0.556451</td>\n",
       "      <td>0.371712</td>\n",
       "      <td>0.324304</td>\n",
       "      <td>0.387166</td>\n",
       "      <td>0.348008</td>\n",
       "      <td>0.176623</td>\n",
       "      <td>False</td>\n",
       "      <td>0.585954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0.546670</td>\n",
       "      <td>0.217698</td>\n",
       "      <td>0.144728</td>\n",
       "      <td>0.532447</td>\n",
       "      <td>0.628549</td>\n",
       "      <td>0.414920</td>\n",
       "      <td>0.300631</td>\n",
       "      <td>0.544006</td>\n",
       "      <td>0.522374</td>\n",
       "      <td>0.522374</td>\n",
       "      <td>0.168982</td>\n",
       "      <td>False</td>\n",
       "      <td>0.483821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.280712</td>\n",
       "      <td>0.197289</td>\n",
       "      <td>0.087799</td>\n",
       "      <td>0.299816</td>\n",
       "      <td>0.424073</td>\n",
       "      <td>0.679603</td>\n",
       "      <td>0.218811</td>\n",
       "      <td>0.555744</td>\n",
       "      <td>0.630379</td>\n",
       "      <td>0.299816</td>\n",
       "      <td>0.207991</td>\n",
       "      <td>False</td>\n",
       "      <td>0.591803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.374921</td>\n",
       "      <td>0.101076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011190</td>\n",
       "      <td>0.263872</td>\n",
       "      <td>0.490226</td>\n",
       "      <td>0.332541</td>\n",
       "      <td>0.110028</td>\n",
       "      <td>0.515773</td>\n",
       "      <td>0.298207</td>\n",
       "      <td>0.191616</td>\n",
       "      <td>False</td>\n",
       "      <td>0.526962</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14636 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            BRCA     CCRCC     Colon  Endometrial       GBM      LUAD  \\\n",
       "A1BG    0.131286 -0.059781       NaN     0.386703 -0.200579 -0.067920   \n",
       "A1CF         NaN  0.892757  0.698025          NaN       NaN       NaN   \n",
       "A2M     0.422497  0.010289 -0.122255     0.024022  0.312922  0.381025   \n",
       "A2ML1   0.713289       NaN       NaN     0.835080       NaN       NaN   \n",
       "A4GALT       NaN       NaN       NaN     0.486499       NaN       NaN   \n",
       "...          ...       ...       ...          ...       ...       ...   \n",
       "ZXDC    0.200364  0.124010       NaN     0.146719  0.185779  0.212842   \n",
       "ZYG11B -0.029503  0.241315       NaN     0.231203  0.458107  0.556451   \n",
       "ZYX     0.546670  0.217698  0.144728     0.532447  0.628549  0.414920   \n",
       "ZZEF1   0.280712  0.197289  0.087799     0.299816  0.424073  0.679603   \n",
       "ZZZ3    0.374921  0.101076       NaN    -0.011190  0.263872  0.490226   \n",
       "\n",
       "         Ovarian     HNSCC      LSCC    Median  Stnd Deviation  +/- 3 sigma  \\\n",
       "A1BG         NaN  0.112133  0.165873  0.112133        0.193232        False   \n",
       "A1CF         NaN       NaN       NaN  0.795391        0.137696        False   \n",
       "A2M     0.140341  0.201083  0.355647  0.201083        0.191378        False   \n",
       "A2ML1   0.531368  0.862612  0.817462  0.817462        0.135646        False   \n",
       "A4GALT       NaN       NaN  0.286010  0.386255        0.141767        False   \n",
       "...          ...       ...       ...       ...             ...          ...   \n",
       "ZXDC         NaN  0.488809  0.583187  0.200364        0.181339        False   \n",
       "ZYG11B  0.371712  0.324304  0.387166  0.348008        0.176623        False   \n",
       "ZYX     0.300631  0.544006  0.522374  0.522374        0.168982        False   \n",
       "ZZEF1   0.218811  0.555744  0.630379  0.299816        0.207991        False   \n",
       "ZZZ3    0.332541  0.110028  0.515773  0.298207        0.191616        False   \n",
       "\n",
       "           Range  \n",
       "A1BG    0.587282  \n",
       "A1CF    0.194732  \n",
       "A2M     0.544752  \n",
       "A2ML1   0.331245  \n",
       "A4GALT  0.200489  \n",
       "...          ...  \n",
       "ZXDC    0.459178  \n",
       "ZYG11B  0.585954  \n",
       "ZYX     0.483821  \n",
       "ZZEF1   0.591803  \n",
       "ZZZ3    0.526962  \n",
       "\n",
       "[14636 rows x 13 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_list_corr = ret_list(fill_genes_nan(common_genes, brca_corr))\n",
    "ccrcc_list_corr = ret_list(fill_genes_nan(common_genes, ccrcc_corr))\n",
    "colon_list_corr = ret_list(fill_genes_nan(common_genes, colon_corr))\n",
    "en_list_corr = ret_list(fill_genes_nan(common_genes, en_corr))\n",
    "gbm_list_corr = ret_list(fill_genes_nan(common_genes, gbm_corr))\n",
    "luad_list_corr = ret_list(fill_genes_nan(common_genes, luad_corr))\n",
    "ovarian_list_corr = ret_list(fill_genes_nan(common_genes, ovarian_corr))\n",
    "hnscc_list_corr = ret_list(fill_genes_nan(common_genes, hnscc_corr))\n",
    "lscc_list_corr = ret_list(fill_genes_nan(common_genes, lscc_corr))\n",
    "\n",
    "median_list = []\n",
    "std_list = []\n",
    "sigma_list = []\n",
    "range_list = []\n",
    "def minmax(val_list):\n",
    "    if len(val_list) == 0:\n",
    "        return np.nan, np.nan\n",
    "    else:\n",
    "        min_val = min(val_list)\n",
    "        max_val = max(val_list)\n",
    "\n",
    "    return (min_val, max_val)\n",
    "# curr_gene = False\n",
    "# print(len(common_genes))\n",
    "for i in range(len(common_genes)):\n",
    "#     if common_genes[i] == \"RGMA\":\n",
    "#         curr_gene = True\n",
    "#         print(\"Found RGMA\")\n",
    "#         print(brca_list_corr[i])\n",
    "#         print(\"i = \",i)\n",
    "    row = list(filter(lambda x:not math.isnan(x),[brca_list_corr[i], ccrcc_list_corr[i], colon_list_corr[i], \n",
    "             en_list_corr[i],gbm_list_corr[i],luad_list_corr[i],ovarian_list_corr[i],\n",
    "             hnscc_list_corr[i],lscc_list_corr[i]]))#Creates a list without Nan on each row\n",
    "\n",
    "#     counter = len(row)\n",
    "#     if counter == 0:\n",
    "#         counter = 1\n",
    "        \n",
    "    ###Median calculation\n",
    "#     print(len(row))\n",
    "    if len(row) == 0:\n",
    "        list_median = np.nan\n",
    "    else:\n",
    "        list_median = statistics.median(row)\n",
    "    median_list.append(list_median)\n",
    "    \n",
    "    ###Standard Deviation calculation\n",
    "    std = 0\n",
    "    if len(row) < 2:\n",
    "        std = np.nan    \n",
    "    else:\n",
    "        std = statistics.stdev(row)\n",
    "    std_list.append(std)\n",
    "    \n",
    "    ### 3 sigma calculation\n",
    "    neg_sigma = list_median - 2.8 * std\n",
    "    pos_sigma = list_median + 2.8 * std\n",
    "    \n",
    "    for i in row:\n",
    "        sigma_bool = False\n",
    "        if i > pos_sigma or i < neg_sigma:\n",
    "            sigma_bool = True\n",
    "            break\n",
    "    sigma_list.append(sigma_bool)\n",
    "    \n",
    "    ###Calculating range\n",
    "    min_value, max_value = minmax(row)\n",
    "    range_list.append(max_value - min_value)\n",
    "    \n",
    "    \n",
    "correlation_columns = {'BRCA': (brca_list_corr),'CCRCC': (ccrcc_list_corr), \n",
    "       'Colon': (colon_list_corr), 'Endometrial': (en_list_corr),\n",
    "       'GBM': (gbm_list_corr), 'LUAD': (luad_list_corr), 'Ovarian': (ovarian_list_corr),\n",
    "        'HNSCC' : (hnscc_list_corr), 'LSCC' : (lscc_list_corr), 'Median' : median_list, 'Stnd Deviation' :std_list, '+/- 3 sigma' : sigma_list, 'Range' : range_list }\n",
    "\n",
    "correlation_df = pd.DataFrame(correlation_columns , columns = ['BRCA', 'CCRCC', 'Colon', 'Endometrial', 'GBM', 'LUAD', 'Ovarian', 'HNSCC', 'LSCC', 'Median', 'Stnd Deviation', '+/- 3 sigma', 'Range'], index=common_genes)\n",
    "correlation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BRCA</th>\n",
       "      <th>CCRCC</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Endometrial</th>\n",
       "      <th>GBM</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>LSCC</th>\n",
       "      <th>Median</th>\n",
       "      <th>Stnd Deviation</th>\n",
       "      <th>+/- 3 sigma</th>\n",
       "      <th>Range</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ABCC4</th>\n",
       "      <td>0.511142</td>\n",
       "      <td>0.604304</td>\n",
       "      <td>0.509307</td>\n",
       "      <td>0.691190</td>\n",
       "      <td>0.666117</td>\n",
       "      <td>0.778407</td>\n",
       "      <td>0.725755</td>\n",
       "      <td>-0.042563</td>\n",
       "      <td>0.645570</td>\n",
       "      <td>0.645570</td>\n",
       "      <td>0.245128</td>\n",
       "      <td>True</td>\n",
       "      <td>0.820970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKR1B10</th>\n",
       "      <td>0.826170</td>\n",
       "      <td>0.416940</td>\n",
       "      <td>0.594124</td>\n",
       "      <td>0.840090</td>\n",
       "      <td>0.489908</td>\n",
       "      <td>0.825854</td>\n",
       "      <td>-0.094944</td>\n",
       "      <td>0.908394</td>\n",
       "      <td>0.836027</td>\n",
       "      <td>0.825854</td>\n",
       "      <td>0.322626</td>\n",
       "      <td>True</td>\n",
       "      <td>1.003339</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARHGAP4</th>\n",
       "      <td>0.395950</td>\n",
       "      <td>0.730204</td>\n",
       "      <td>0.638369</td>\n",
       "      <td>0.852650</td>\n",
       "      <td>0.815684</td>\n",
       "      <td>0.836753</td>\n",
       "      <td>0.670715</td>\n",
       "      <td>0.849480</td>\n",
       "      <td>0.817124</td>\n",
       "      <td>0.815684</td>\n",
       "      <td>0.149581</td>\n",
       "      <td>True</td>\n",
       "      <td>0.456700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASP1</th>\n",
       "      <td>0.727167</td>\n",
       "      <td>0.787449</td>\n",
       "      <td>0.856944</td>\n",
       "      <td>0.835613</td>\n",
       "      <td>0.804726</td>\n",
       "      <td>0.847280</td>\n",
       "      <td>0.482290</td>\n",
       "      <td>0.889323</td>\n",
       "      <td>0.846390</td>\n",
       "      <td>0.835613</td>\n",
       "      <td>0.123301</td>\n",
       "      <td>True</td>\n",
       "      <td>0.407032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFL2</th>\n",
       "      <td>0.698358</td>\n",
       "      <td>0.739900</td>\n",
       "      <td>0.644461</td>\n",
       "      <td>0.751337</td>\n",
       "      <td>0.623642</td>\n",
       "      <td>0.674934</td>\n",
       "      <td>-0.012246</td>\n",
       "      <td>0.816952</td>\n",
       "      <td>0.577462</td>\n",
       "      <td>0.674934</td>\n",
       "      <td>0.245270</td>\n",
       "      <td>True</td>\n",
       "      <td>0.829198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CRISPLD2</th>\n",
       "      <td>0.675279</td>\n",
       "      <td>0.557736</td>\n",
       "      <td>0.295992</td>\n",
       "      <td>0.639656</td>\n",
       "      <td>0.605271</td>\n",
       "      <td>0.636899</td>\n",
       "      <td>0.668486</td>\n",
       "      <td>0.621525</td>\n",
       "      <td>0.662528</td>\n",
       "      <td>0.636899</td>\n",
       "      <td>0.118190</td>\n",
       "      <td>True</td>\n",
       "      <td>0.379287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EGFR</th>\n",
       "      <td>0.784136</td>\n",
       "      <td>0.615396</td>\n",
       "      <td>-0.020567</td>\n",
       "      <td>0.705223</td>\n",
       "      <td>0.816821</td>\n",
       "      <td>0.848057</td>\n",
       "      <td>0.657607</td>\n",
       "      <td>0.913202</td>\n",
       "      <td>0.900464</td>\n",
       "      <td>0.784136</td>\n",
       "      <td>0.286309</td>\n",
       "      <td>True</td>\n",
       "      <td>0.933769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EIF4ENIF1</th>\n",
       "      <td>0.627583</td>\n",
       "      <td>0.609696</td>\n",
       "      <td>-0.120662</td>\n",
       "      <td>0.656449</td>\n",
       "      <td>0.712156</td>\n",
       "      <td>0.635596</td>\n",
       "      <td>0.591006</td>\n",
       "      <td>0.720241</td>\n",
       "      <td>0.651353</td>\n",
       "      <td>0.635596</td>\n",
       "      <td>0.260596</td>\n",
       "      <td>True</td>\n",
       "      <td>0.840903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>EML2</th>\n",
       "      <td>0.699754</td>\n",
       "      <td>0.106465</td>\n",
       "      <td>0.569057</td>\n",
       "      <td>0.728429</td>\n",
       "      <td>0.615791</td>\n",
       "      <td>0.766045</td>\n",
       "      <td>0.582013</td>\n",
       "      <td>0.746030</td>\n",
       "      <td>0.695806</td>\n",
       "      <td>0.695806</td>\n",
       "      <td>0.202576</td>\n",
       "      <td>True</td>\n",
       "      <td>0.659580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FUCA1</th>\n",
       "      <td>0.696446</td>\n",
       "      <td>0.681138</td>\n",
       "      <td>0.382992</td>\n",
       "      <td>0.622044</td>\n",
       "      <td>0.646362</td>\n",
       "      <td>0.707473</td>\n",
       "      <td>0.717193</td>\n",
       "      <td>0.729365</td>\n",
       "      <td>0.704841</td>\n",
       "      <td>0.696446</td>\n",
       "      <td>0.107343</td>\n",
       "      <td>True</td>\n",
       "      <td>0.346373</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GBE1</th>\n",
       "      <td>0.739463</td>\n",
       "      <td>0.623390</td>\n",
       "      <td>0.321040</td>\n",
       "      <td>0.755462</td>\n",
       "      <td>0.769420</td>\n",
       "      <td>0.784586</td>\n",
       "      <td>0.638961</td>\n",
       "      <td>0.754238</td>\n",
       "      <td>0.775540</td>\n",
       "      <td>0.754238</td>\n",
       "      <td>0.148470</td>\n",
       "      <td>True</td>\n",
       "      <td>0.463546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GULP1</th>\n",
       "      <td>0.690514</td>\n",
       "      <td>0.734908</td>\n",
       "      <td>0.096438</td>\n",
       "      <td>0.784630</td>\n",
       "      <td>0.631638</td>\n",
       "      <td>0.754185</td>\n",
       "      <td>0.519199</td>\n",
       "      <td>0.709247</td>\n",
       "      <td>0.705767</td>\n",
       "      <td>0.705767</td>\n",
       "      <td>0.212949</td>\n",
       "      <td>True</td>\n",
       "      <td>0.688191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HLA-DPA1</th>\n",
       "      <td>0.584553</td>\n",
       "      <td>0.722609</td>\n",
       "      <td>0.558308</td>\n",
       "      <td>0.720700</td>\n",
       "      <td>0.792768</td>\n",
       "      <td>0.718981</td>\n",
       "      <td>0.154412</td>\n",
       "      <td>0.736682</td>\n",
       "      <td>0.655453</td>\n",
       "      <td>0.718981</td>\n",
       "      <td>0.192536</td>\n",
       "      <td>True</td>\n",
       "      <td>0.638356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IMPDH1</th>\n",
       "      <td>0.567074</td>\n",
       "      <td>0.712594</td>\n",
       "      <td>0.510780</td>\n",
       "      <td>0.786561</td>\n",
       "      <td>0.664013</td>\n",
       "      <td>0.722927</td>\n",
       "      <td>0.038880</td>\n",
       "      <td>0.684296</td>\n",
       "      <td>0.672341</td>\n",
       "      <td>0.672341</td>\n",
       "      <td>0.224373</td>\n",
       "      <td>True</td>\n",
       "      <td>0.747681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IREB2</th>\n",
       "      <td>0.437539</td>\n",
       "      <td>-0.007255</td>\n",
       "      <td>0.361175</td>\n",
       "      <td>0.462670</td>\n",
       "      <td>0.299769</td>\n",
       "      <td>0.432525</td>\n",
       "      <td>0.368629</td>\n",
       "      <td>0.496991</td>\n",
       "      <td>0.471606</td>\n",
       "      <td>0.432525</td>\n",
       "      <td>0.154450</td>\n",
       "      <td>True</td>\n",
       "      <td>0.504245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IVD</th>\n",
       "      <td>0.771037</td>\n",
       "      <td>-0.103327</td>\n",
       "      <td>0.428964</td>\n",
       "      <td>0.794557</td>\n",
       "      <td>0.735900</td>\n",
       "      <td>0.653613</td>\n",
       "      <td>0.673239</td>\n",
       "      <td>0.800636</td>\n",
       "      <td>0.713852</td>\n",
       "      <td>0.713852</td>\n",
       "      <td>0.289489</td>\n",
       "      <td>True</td>\n",
       "      <td>0.903963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KARS</th>\n",
       "      <td>0.773275</td>\n",
       "      <td>0.683755</td>\n",
       "      <td>0.316380</td>\n",
       "      <td>0.725110</td>\n",
       "      <td>0.681289</td>\n",
       "      <td>0.783006</td>\n",
       "      <td>0.748120</td>\n",
       "      <td>0.706938</td>\n",
       "      <td>0.781861</td>\n",
       "      <td>0.725110</td>\n",
       "      <td>0.145179</td>\n",
       "      <td>True</td>\n",
       "      <td>0.466626</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KHDRBS3</th>\n",
       "      <td>0.837233</td>\n",
       "      <td>0.809998</td>\n",
       "      <td>0.835391</td>\n",
       "      <td>0.906788</td>\n",
       "      <td>0.825993</td>\n",
       "      <td>0.857303</td>\n",
       "      <td>0.417757</td>\n",
       "      <td>0.781615</td>\n",
       "      <td>0.892706</td>\n",
       "      <td>0.835391</td>\n",
       "      <td>0.147057</td>\n",
       "      <td>True</td>\n",
       "      <td>0.489031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LPCAT2</th>\n",
       "      <td>0.271617</td>\n",
       "      <td>0.689473</td>\n",
       "      <td>0.744612</td>\n",
       "      <td>0.702410</td>\n",
       "      <td>0.750686</td>\n",
       "      <td>0.738311</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.762852</td>\n",
       "      <td>0.762306</td>\n",
       "      <td>0.741462</td>\n",
       "      <td>0.166279</td>\n",
       "      <td>True</td>\n",
       "      <td>0.491236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>METTL1</th>\n",
       "      <td>0.250558</td>\n",
       "      <td>0.284784</td>\n",
       "      <td>0.247096</td>\n",
       "      <td>0.427342</td>\n",
       "      <td>0.566503</td>\n",
       "      <td>0.421835</td>\n",
       "      <td>-0.165263</td>\n",
       "      <td>0.425828</td>\n",
       "      <td>0.424595</td>\n",
       "      <td>0.421835</td>\n",
       "      <td>0.209658</td>\n",
       "      <td>True</td>\n",
       "      <td>0.731765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NOL3</th>\n",
       "      <td>-0.097384</td>\n",
       "      <td>0.770391</td>\n",
       "      <td>0.493258</td>\n",
       "      <td>0.760641</td>\n",
       "      <td>0.764850</td>\n",
       "      <td>0.764999</td>\n",
       "      <td>0.628435</td>\n",
       "      <td>0.611956</td>\n",
       "      <td>0.778121</td>\n",
       "      <td>0.760641</td>\n",
       "      <td>0.282628</td>\n",
       "      <td>True</td>\n",
       "      <td>0.875506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NTM</th>\n",
       "      <td>0.734932</td>\n",
       "      <td>0.144675</td>\n",
       "      <td>0.617261</td>\n",
       "      <td>0.606503</td>\n",
       "      <td>0.731177</td>\n",
       "      <td>0.753389</td>\n",
       "      <td>0.731595</td>\n",
       "      <td>0.637519</td>\n",
       "      <td>0.718153</td>\n",
       "      <td>0.718153</td>\n",
       "      <td>0.190681</td>\n",
       "      <td>True</td>\n",
       "      <td>0.608714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PMM2</th>\n",
       "      <td>0.552862</td>\n",
       "      <td>0.635562</td>\n",
       "      <td>0.146906</td>\n",
       "      <td>0.727357</td>\n",
       "      <td>0.620492</td>\n",
       "      <td>0.661793</td>\n",
       "      <td>0.655104</td>\n",
       "      <td>0.689441</td>\n",
       "      <td>0.544095</td>\n",
       "      <td>0.635562</td>\n",
       "      <td>0.173322</td>\n",
       "      <td>True</td>\n",
       "      <td>0.580451</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PPIC</th>\n",
       "      <td>0.767638</td>\n",
       "      <td>0.579829</td>\n",
       "      <td>0.610209</td>\n",
       "      <td>0.755198</td>\n",
       "      <td>0.737467</td>\n",
       "      <td>0.723840</td>\n",
       "      <td>0.285926</td>\n",
       "      <td>0.611377</td>\n",
       "      <td>0.725904</td>\n",
       "      <td>0.723840</td>\n",
       "      <td>0.151662</td>\n",
       "      <td>True</td>\n",
       "      <td>0.481712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PSTPIP1</th>\n",
       "      <td>0.690316</td>\n",
       "      <td>0.705296</td>\n",
       "      <td>0.631509</td>\n",
       "      <td>0.724814</td>\n",
       "      <td>0.793440</td>\n",
       "      <td>0.700288</td>\n",
       "      <td>0.226931</td>\n",
       "      <td>0.701017</td>\n",
       "      <td>0.722457</td>\n",
       "      <td>0.701017</td>\n",
       "      <td>0.165956</td>\n",
       "      <td>True</td>\n",
       "      <td>0.566510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PXK</th>\n",
       "      <td>0.461879</td>\n",
       "      <td>0.483921</td>\n",
       "      <td>-0.149073</td>\n",
       "      <td>0.663593</td>\n",
       "      <td>0.531911</td>\n",
       "      <td>0.669128</td>\n",
       "      <td>0.640607</td>\n",
       "      <td>0.699162</td>\n",
       "      <td>0.764322</td>\n",
       "      <td>0.640607</td>\n",
       "      <td>0.274059</td>\n",
       "      <td>True</td>\n",
       "      <td>0.913395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RCN3</th>\n",
       "      <td>0.103712</td>\n",
       "      <td>0.802036</td>\n",
       "      <td>0.665820</td>\n",
       "      <td>0.801990</td>\n",
       "      <td>0.702966</td>\n",
       "      <td>0.852907</td>\n",
       "      <td>0.456371</td>\n",
       "      <td>0.801420</td>\n",
       "      <td>0.874600</td>\n",
       "      <td>0.801420</td>\n",
       "      <td>0.248685</td>\n",
       "      <td>True</td>\n",
       "      <td>0.770887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SASH3</th>\n",
       "      <td>0.706611</td>\n",
       "      <td>0.746389</td>\n",
       "      <td>0.264880</td>\n",
       "      <td>0.791185</td>\n",
       "      <td>0.808296</td>\n",
       "      <td>0.809943</td>\n",
       "      <td>0.692803</td>\n",
       "      <td>0.873148</td>\n",
       "      <td>0.814355</td>\n",
       "      <td>0.791185</td>\n",
       "      <td>0.180961</td>\n",
       "      <td>True</td>\n",
       "      <td>0.608268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SERPINF1</th>\n",
       "      <td>0.619753</td>\n",
       "      <td>0.199213</td>\n",
       "      <td>0.545733</td>\n",
       "      <td>0.627718</td>\n",
       "      <td>0.516837</td>\n",
       "      <td>0.612136</td>\n",
       "      <td>0.644906</td>\n",
       "      <td>0.596940</td>\n",
       "      <td>0.507713</td>\n",
       "      <td>0.596940</td>\n",
       "      <td>0.137496</td>\n",
       "      <td>True</td>\n",
       "      <td>0.445693</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SH3PXD2B</th>\n",
       "      <td>0.839330</td>\n",
       "      <td>0.624821</td>\n",
       "      <td>0.237788</td>\n",
       "      <td>0.798657</td>\n",
       "      <td>0.734055</td>\n",
       "      <td>0.844389</td>\n",
       "      <td>0.664304</td>\n",
       "      <td>0.823594</td>\n",
       "      <td>0.799638</td>\n",
       "      <td>0.798657</td>\n",
       "      <td>0.192494</td>\n",
       "      <td>True</td>\n",
       "      <td>0.606601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC4A7</th>\n",
       "      <td>0.575583</td>\n",
       "      <td>0.653060</td>\n",
       "      <td>0.392839</td>\n",
       "      <td>0.521605</td>\n",
       "      <td>0.608348</td>\n",
       "      <td>0.627902</td>\n",
       "      <td>0.043326</td>\n",
       "      <td>0.572303</td>\n",
       "      <td>0.513104</td>\n",
       "      <td>0.572303</td>\n",
       "      <td>0.188126</td>\n",
       "      <td>True</td>\n",
       "      <td>0.609733</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>STARD13</th>\n",
       "      <td>0.737320</td>\n",
       "      <td>0.539458</td>\n",
       "      <td>-0.125408</td>\n",
       "      <td>0.677195</td>\n",
       "      <td>0.679426</td>\n",
       "      <td>0.528829</td>\n",
       "      <td>0.371099</td>\n",
       "      <td>0.701527</td>\n",
       "      <td>0.661878</td>\n",
       "      <td>0.661878</td>\n",
       "      <td>0.271268</td>\n",
       "      <td>True</td>\n",
       "      <td>0.862728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TACC2</th>\n",
       "      <td>0.580223</td>\n",
       "      <td>0.746140</td>\n",
       "      <td>0.015732</td>\n",
       "      <td>0.778093</td>\n",
       "      <td>0.657541</td>\n",
       "      <td>0.861281</td>\n",
       "      <td>0.444940</td>\n",
       "      <td>0.765510</td>\n",
       "      <td>0.772458</td>\n",
       "      <td>0.746140</td>\n",
       "      <td>0.260317</td>\n",
       "      <td>True</td>\n",
       "      <td>0.845550</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TAP1</th>\n",
       "      <td>0.678376</td>\n",
       "      <td>0.795205</td>\n",
       "      <td>0.850012</td>\n",
       "      <td>0.736942</td>\n",
       "      <td>0.675080</td>\n",
       "      <td>0.849838</td>\n",
       "      <td>0.246675</td>\n",
       "      <td>0.848217</td>\n",
       "      <td>0.839963</td>\n",
       "      <td>0.795205</td>\n",
       "      <td>0.193048</td>\n",
       "      <td>True</td>\n",
       "      <td>0.603338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBL5</th>\n",
       "      <td>0.171003</td>\n",
       "      <td>0.177282</td>\n",
       "      <td>0.256073</td>\n",
       "      <td>0.200977</td>\n",
       "      <td>0.210857</td>\n",
       "      <td>0.216585</td>\n",
       "      <td>0.645589</td>\n",
       "      <td>0.395009</td>\n",
       "      <td>0.240097</td>\n",
       "      <td>0.216585</td>\n",
       "      <td>0.152693</td>\n",
       "      <td>True</td>\n",
       "      <td>0.474587</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UQCC1</th>\n",
       "      <td>0.440982</td>\n",
       "      <td>0.524675</td>\n",
       "      <td>0.528753</td>\n",
       "      <td>0.522274</td>\n",
       "      <td>0.156899</td>\n",
       "      <td>0.443219</td>\n",
       "      <td>0.543341</td>\n",
       "      <td>0.428459</td>\n",
       "      <td>0.583740</td>\n",
       "      <td>0.522274</td>\n",
       "      <td>0.126707</td>\n",
       "      <td>True</td>\n",
       "      <td>0.426841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VCL</th>\n",
       "      <td>0.722270</td>\n",
       "      <td>0.673013</td>\n",
       "      <td>0.136985</td>\n",
       "      <td>0.664586</td>\n",
       "      <td>0.712123</td>\n",
       "      <td>0.641212</td>\n",
       "      <td>0.688833</td>\n",
       "      <td>0.683509</td>\n",
       "      <td>0.624079</td>\n",
       "      <td>0.673013</td>\n",
       "      <td>0.182401</td>\n",
       "      <td>True</td>\n",
       "      <td>0.585285</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               BRCA     CCRCC     Colon  Endometrial       GBM      LUAD  \\\n",
       "ABCC4      0.511142  0.604304  0.509307     0.691190  0.666117  0.778407   \n",
       "AKR1B10    0.826170  0.416940  0.594124     0.840090  0.489908  0.825854   \n",
       "ARHGAP4    0.395950  0.730204  0.638369     0.852650  0.815684  0.836753   \n",
       "CASP1      0.727167  0.787449  0.856944     0.835613  0.804726  0.847280   \n",
       "CFL2       0.698358  0.739900  0.644461     0.751337  0.623642  0.674934   \n",
       "CRISPLD2   0.675279  0.557736  0.295992     0.639656  0.605271  0.636899   \n",
       "EGFR       0.784136  0.615396 -0.020567     0.705223  0.816821  0.848057   \n",
       "EIF4ENIF1  0.627583  0.609696 -0.120662     0.656449  0.712156  0.635596   \n",
       "EML2       0.699754  0.106465  0.569057     0.728429  0.615791  0.766045   \n",
       "FUCA1      0.696446  0.681138  0.382992     0.622044  0.646362  0.707473   \n",
       "GBE1       0.739463  0.623390  0.321040     0.755462  0.769420  0.784586   \n",
       "GULP1      0.690514  0.734908  0.096438     0.784630  0.631638  0.754185   \n",
       "HLA-DPA1   0.584553  0.722609  0.558308     0.720700  0.792768  0.718981   \n",
       "IMPDH1     0.567074  0.712594  0.510780     0.786561  0.664013  0.722927   \n",
       "IREB2      0.437539 -0.007255  0.361175     0.462670  0.299769  0.432525   \n",
       "IVD        0.771037 -0.103327  0.428964     0.794557  0.735900  0.653613   \n",
       "KARS       0.773275  0.683755  0.316380     0.725110  0.681289  0.783006   \n",
       "KHDRBS3    0.837233  0.809998  0.835391     0.906788  0.825993  0.857303   \n",
       "LPCAT2     0.271617  0.689473  0.744612     0.702410  0.750686  0.738311   \n",
       "METTL1     0.250558  0.284784  0.247096     0.427342  0.566503  0.421835   \n",
       "NOL3      -0.097384  0.770391  0.493258     0.760641  0.764850  0.764999   \n",
       "NTM        0.734932  0.144675  0.617261     0.606503  0.731177  0.753389   \n",
       "PMM2       0.552862  0.635562  0.146906     0.727357  0.620492  0.661793   \n",
       "PPIC       0.767638  0.579829  0.610209     0.755198  0.737467  0.723840   \n",
       "PSTPIP1    0.690316  0.705296  0.631509     0.724814  0.793440  0.700288   \n",
       "PXK        0.461879  0.483921 -0.149073     0.663593  0.531911  0.669128   \n",
       "RCN3       0.103712  0.802036  0.665820     0.801990  0.702966  0.852907   \n",
       "SASH3      0.706611  0.746389  0.264880     0.791185  0.808296  0.809943   \n",
       "SERPINF1   0.619753  0.199213  0.545733     0.627718  0.516837  0.612136   \n",
       "SH3PXD2B   0.839330  0.624821  0.237788     0.798657  0.734055  0.844389   \n",
       "SLC4A7     0.575583  0.653060  0.392839     0.521605  0.608348  0.627902   \n",
       "STARD13    0.737320  0.539458 -0.125408     0.677195  0.679426  0.528829   \n",
       "TACC2      0.580223  0.746140  0.015732     0.778093  0.657541  0.861281   \n",
       "TAP1       0.678376  0.795205  0.850012     0.736942  0.675080  0.849838   \n",
       "UBL5       0.171003  0.177282  0.256073     0.200977  0.210857  0.216585   \n",
       "UQCC1      0.440982  0.524675  0.528753     0.522274  0.156899  0.443219   \n",
       "VCL        0.722270  0.673013  0.136985     0.664586  0.712123  0.641212   \n",
       "\n",
       "            Ovarian     HNSCC      LSCC    Median  Stnd Deviation  \\\n",
       "ABCC4      0.725755 -0.042563  0.645570  0.645570        0.245128   \n",
       "AKR1B10   -0.094944  0.908394  0.836027  0.825854        0.322626   \n",
       "ARHGAP4    0.670715  0.849480  0.817124  0.815684        0.149581   \n",
       "CASP1      0.482290  0.889323  0.846390  0.835613        0.123301   \n",
       "CFL2      -0.012246  0.816952  0.577462  0.674934        0.245270   \n",
       "CRISPLD2   0.668486  0.621525  0.662528  0.636899        0.118190   \n",
       "EGFR       0.657607  0.913202  0.900464  0.784136        0.286309   \n",
       "EIF4ENIF1  0.591006  0.720241  0.651353  0.635596        0.260596   \n",
       "EML2       0.582013  0.746030  0.695806  0.695806        0.202576   \n",
       "FUCA1      0.717193  0.729365  0.704841  0.696446        0.107343   \n",
       "GBE1       0.638961  0.754238  0.775540  0.754238        0.148470   \n",
       "GULP1      0.519199  0.709247  0.705767  0.705767        0.212949   \n",
       "HLA-DPA1   0.154412  0.736682  0.655453  0.718981        0.192536   \n",
       "IMPDH1     0.038880  0.684296  0.672341  0.672341        0.224373   \n",
       "IREB2      0.368629  0.496991  0.471606  0.432525        0.154450   \n",
       "IVD        0.673239  0.800636  0.713852  0.713852        0.289489   \n",
       "KARS       0.748120  0.706938  0.781861  0.725110        0.145179   \n",
       "KHDRBS3    0.417757  0.781615  0.892706  0.835391        0.147057   \n",
       "LPCAT2          NaN  0.762852  0.762306  0.741462        0.166279   \n",
       "METTL1    -0.165263  0.425828  0.424595  0.421835        0.209658   \n",
       "NOL3       0.628435  0.611956  0.778121  0.760641        0.282628   \n",
       "NTM        0.731595  0.637519  0.718153  0.718153        0.190681   \n",
       "PMM2       0.655104  0.689441  0.544095  0.635562        0.173322   \n",
       "PPIC       0.285926  0.611377  0.725904  0.723840        0.151662   \n",
       "PSTPIP1    0.226931  0.701017  0.722457  0.701017        0.165956   \n",
       "PXK        0.640607  0.699162  0.764322  0.640607        0.274059   \n",
       "RCN3       0.456371  0.801420  0.874600  0.801420        0.248685   \n",
       "SASH3      0.692803  0.873148  0.814355  0.791185        0.180961   \n",
       "SERPINF1   0.644906  0.596940  0.507713  0.596940        0.137496   \n",
       "SH3PXD2B   0.664304  0.823594  0.799638  0.798657        0.192494   \n",
       "SLC4A7     0.043326  0.572303  0.513104  0.572303        0.188126   \n",
       "STARD13    0.371099  0.701527  0.661878  0.661878        0.271268   \n",
       "TACC2      0.444940  0.765510  0.772458  0.746140        0.260317   \n",
       "TAP1       0.246675  0.848217  0.839963  0.795205        0.193048   \n",
       "UBL5       0.645589  0.395009  0.240097  0.216585        0.152693   \n",
       "UQCC1      0.543341  0.428459  0.583740  0.522274        0.126707   \n",
       "VCL        0.688833  0.683509  0.624079  0.673013        0.182401   \n",
       "\n",
       "           +/- 3 sigma     Range  \n",
       "ABCC4             True  0.820970  \n",
       "AKR1B10           True  1.003339  \n",
       "ARHGAP4           True  0.456700  \n",
       "CASP1             True  0.407032  \n",
       "CFL2              True  0.829198  \n",
       "CRISPLD2          True  0.379287  \n",
       "EGFR              True  0.933769  \n",
       "EIF4ENIF1         True  0.840903  \n",
       "EML2              True  0.659580  \n",
       "FUCA1             True  0.346373  \n",
       "GBE1              True  0.463546  \n",
       "GULP1             True  0.688191  \n",
       "HLA-DPA1          True  0.638356  \n",
       "IMPDH1            True  0.747681  \n",
       "IREB2             True  0.504245  \n",
       "IVD               True  0.903963  \n",
       "KARS              True  0.466626  \n",
       "KHDRBS3           True  0.489031  \n",
       "LPCAT2            True  0.491236  \n",
       "METTL1            True  0.731765  \n",
       "NOL3              True  0.875506  \n",
       "NTM               True  0.608714  \n",
       "PMM2              True  0.580451  \n",
       "PPIC              True  0.481712  \n",
       "PSTPIP1           True  0.566510  \n",
       "PXK               True  0.913395  \n",
       "RCN3              True  0.770887  \n",
       "SASH3             True  0.608268  \n",
       "SERPINF1          True  0.445693  \n",
       "SH3PXD2B          True  0.606601  \n",
       "SLC4A7            True  0.609733  \n",
       "STARD13           True  0.862728  \n",
       "TACC2             True  0.845550  \n",
       "TAP1              True  0.603338  \n",
       "UBL5              True  0.474587  \n",
       "UQCC1             True  0.426841  \n",
       "VCL               True  0.585285  "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_df.loc[correlation_df['+/- 3 sigma'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BRCA</th>\n",
       "      <th>CCRCC</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Endometrial</th>\n",
       "      <th>GBM</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>LSCC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.657241e-01</td>\n",
       "      <td>5.350095e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.084161e-04</td>\n",
       "      <td>4.651711e-02</td>\n",
       "      <td>4.807936e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.456948e-01</td>\n",
       "      <td>9.082049e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3.432584e-39</td>\n",
       "      <td>5.325574e-15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>1.249853e-06</td>\n",
       "      <td>9.150454e-01</td>\n",
       "      <td>2.353794e-01</td>\n",
       "      <td>8.172572e-01</td>\n",
       "      <td>1.612770e-03</td>\n",
       "      <td>4.015329e-05</td>\n",
       "      <td>0.208548</td>\n",
       "      <td>3.602656e-02</td>\n",
       "      <td>1.703835e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.373971e-17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.999470e-23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>1.908976e-33</td>\n",
       "      <td>6.589307e-27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.594752e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.805955e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>4.909360e-02</td>\n",
       "      <td>2.958874e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.559574e-01</td>\n",
       "      <td>7.958310e-02</td>\n",
       "      <td>5.049816e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.776369e-07</td>\n",
       "      <td>3.654355e-10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>7.470119e-01</td>\n",
       "      <td>1.109573e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.417794e-02</td>\n",
       "      <td>1.861457e-06</td>\n",
       "      <td>2.757689e-10</td>\n",
       "      <td>0.000585</td>\n",
       "      <td>5.811263e-04</td>\n",
       "      <td>3.793685e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>7.319264e-11</td>\n",
       "      <td>2.569032e-02</td>\n",
       "      <td>1.594617e-01</td>\n",
       "      <td>2.798595e-08</td>\n",
       "      <td>3.274769e-12</td>\n",
       "      <td>6.578244e-06</td>\n",
       "      <td>0.006064</td>\n",
       "      <td>9.742303e-10</td>\n",
       "      <td>7.881491e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>1.736969e-03</td>\n",
       "      <td>3.883803e-02</td>\n",
       "      <td>3.949742e-01</td>\n",
       "      <td>3.159430e-03</td>\n",
       "      <td>1.218953e-05</td>\n",
       "      <td>3.279695e-16</td>\n",
       "      <td>0.048267</td>\n",
       "      <td>3.547520e-10</td>\n",
       "      <td>3.444445e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>2.092650e-05</td>\n",
       "      <td>3.754370e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.142955e-01</td>\n",
       "      <td>8.312799e-03</td>\n",
       "      <td>3.453729e-07</td>\n",
       "      <td>0.003557</td>\n",
       "      <td>3.473585e-01</td>\n",
       "      <td>1.305147e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14636 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                BRCA         CCRCC         Colon   Endometrial           GBM  \\\n",
       "A1BG    1.657241e-01  5.350095e-01           NaN  1.084161e-04  4.651711e-02   \n",
       "A1CF             NaN  3.432584e-39  5.325574e-15           NaN           NaN   \n",
       "A2M     1.249853e-06  9.150454e-01  2.353794e-01  8.172572e-01  1.612770e-03   \n",
       "A2ML1   1.373971e-17           NaN           NaN  9.999470e-23           NaN   \n",
       "A4GALT           NaN           NaN           NaN  2.594752e-05           NaN   \n",
       "...              ...           ...           ...           ...           ...   \n",
       "ZXDC    4.909360e-02  2.958874e-01           NaN  1.559574e-01  7.958310e-02   \n",
       "ZYG11B  7.470119e-01  1.109573e-02           NaN  2.417794e-02  1.861457e-06   \n",
       "ZYX     7.319264e-11  2.569032e-02  1.594617e-01  2.798595e-08  3.274769e-12   \n",
       "ZZEF1   1.736969e-03  3.883803e-02  3.949742e-01  3.159430e-03  1.218953e-05   \n",
       "ZZZ3    2.092650e-05  3.754370e-01           NaN  9.142955e-01  8.312799e-03   \n",
       "\n",
       "                LUAD   Ovarian         HNSCC          LSCC  \n",
       "A1BG    4.807936e-01       NaN  2.456948e-01  9.082049e-02  \n",
       "A1CF             NaN       NaN           NaN           NaN  \n",
       "A2M     4.015329e-05  0.208548  3.602656e-02  1.703835e-04  \n",
       "A2ML1            NaN  0.000013  1.908976e-33  6.589307e-27  \n",
       "A4GALT           NaN       NaN           NaN  1.805955e-02  \n",
       "...              ...       ...           ...           ...  \n",
       "ZXDC    5.049816e-02       NaN  3.776369e-07  3.654355e-10  \n",
       "ZYG11B  2.757689e-10  0.000585  5.811263e-04  3.793685e-05  \n",
       "ZYX     6.578244e-06  0.006064  9.742303e-10  7.881491e-09  \n",
       "ZZEF1   3.279695e-16  0.048267  3.547520e-10  3.444445e-13  \n",
       "ZZZ3    3.453729e-07  0.003557  3.473585e-01  1.305147e-08  \n",
       "\n",
       "[14636 rows x 9 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_list_pval = (ret_list(fill_genes_nan(common_genes, brca_pval)))\n",
    "ccrcc_list_pval = (ret_list(fill_genes_nan(common_genes, ccrcc_pval)))\n",
    "colon_list_pval = (ret_list(fill_genes_nan(common_genes, colon_pval)))\n",
    "en_list_pval = (ret_list(fill_genes_nan(common_genes, en_pval)))\n",
    "gbm_list_pval = (ret_list(fill_genes_nan(common_genes, gbm_pval)))\n",
    "luad_list_pval = (ret_list(fill_genes_nan(common_genes, luad_pval)))\n",
    "ovarian_list_pval = (ret_list(fill_genes_nan(common_genes, ovarian_pval)))\n",
    "hnscc_list_pval = (ret_list(fill_genes_nan(common_genes, hnscc_pval)))\n",
    "lscc_list_pval = (ret_list(fill_genes_nan(common_genes, lscc_pval)))\n",
    "\n",
    "\n",
    "pval_columns = {'BRCA': (brca_list_pval),\n",
    "        'CCRCC': (ccrcc_list_pval), 'Colon': (colon_list_pval), 'Endometrial': (en_list_pval),\n",
    "       'GBM': (gbm_list_pval), 'LUAD': (luad_list_pval), 'Ovarian': (ovarian_list_pval), 'HNSCC' : (hnscc_list_pval), 'LSCC' : (lscc_list_pval)}\n",
    "\n",
    "pval_df= pd.DataFrame(pval_columns , columns = ['BRCA', 'CCRCC', 'Colon', 'Endometrial', 'GBM', 'LUAD', 'Ovarian', 'HNSCC', 'LSCC'], index=common_genes)\n",
    "pval_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
