{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import cptac.utils as ut\n",
    "import scipy\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import math\n",
    "import pandas as pd\n",
    "import statistics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_list(tissue):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    gene_list = []\n",
    "    prot = tissue.get_proteomics()\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = prot.columns.get_level_values(0)\n",
    "    trans = tissue.get_transcriptomics()\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = trans.columns.get_level_values(0)\n",
    "    for i in prot:\n",
    "        if i in trans and i not in gene_list:\n",
    "            gene_list.append(i)\n",
    "    return gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def correlation_list(tissue, gene_list, valid_sample_number = 30, tissue_type = \"both\"):\n",
    "    #Returns a list of lists containing the name of the gene and the correlation\n",
    "    corr_list = []\n",
    "    pval_list = []\n",
    "    prot = tissue.get_proteomics(tissue_type)\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = ut.reduce_multiindex(df = prot, levels_to_drop=\"Database_ID\")\n",
    "    \n",
    "    trans = tissue.get_transcriptomics(tissue_type)\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = ut.reduce_multiindex(df = trans, levels_to_drop=\"Database_ID\")\n",
    "        \n",
    "    prot_index_values = list(prot.index.values)\n",
    "    trans_index_values = list(trans.index.values)\n",
    "    \n",
    "\n",
    "    for found_index in prot_index_values:\n",
    "        if found_index not in trans_index_values:\n",
    "            prot = prot.drop(index=found_index)\n",
    "    for found_index in trans_index_values:\n",
    "        if found_index not in prot_index_values:\n",
    "            trans = trans.drop(index=found_index)\n",
    "        \n",
    "    for gene in gene_list:\n",
    "        prot_measurements = prot[gene]\n",
    "        prot_count = len(prot_measurements.dropna())\n",
    "        \n",
    "        trans_measurements = trans[gene]\n",
    "        trans_count = len(trans_measurements.dropna())\n",
    "        \n",
    "        if prot_count < valid_sample_number or trans_count < valid_sample_number:\n",
    "            continue\n",
    "            \n",
    "        nan_indices = set()\n",
    "        prot_indices = list(prot_measurements.index.values)\n",
    "        trans_indices = list(trans_measurements.index.values)\n",
    "        \n",
    "        #Here we are only currently taking the first column of multi_indices\n",
    "        if isinstance(prot_measurements, pd.core.frame.DataFrame): \n",
    "            prot_measurements = prot_measurements.iloc[:,0]\n",
    "        if isinstance(trans_measurements, pd.core.frame.DataFrame):\n",
    "            trans_measurements = trans_measurements.iloc[:,0]\n",
    "        for i in range(len(prot_measurements)):\n",
    "            if math.isnan(prot_measurements[i]):\n",
    "                nan_indices.add(prot_indices[i])\n",
    "                \n",
    "        for i in range(len(trans_measurements)):\n",
    "            if math.isnan(trans_measurements[i]):\n",
    "                nan_indices.add(trans_indices[i])\n",
    "        \n",
    "        nan_indices = list(nan_indices)\n",
    "        prot_measurements = prot_measurements.drop(nan_indices)\n",
    "        trans_measurements = trans_measurements.drop(nan_indices)\n",
    "        \n",
    "        correlation,pval = scipy.stats.pearsonr(prot_measurements, trans_measurements)\n",
    "        if math.isnan(correlation):\n",
    "            continue\n",
    "        corr_list.append([gene,correlation])\n",
    "        pval_list.append([gene, pval])\n",
    "    return corr_list, pval_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def ret_list(li):\n",
    "    #Returns a list of correlations from all genes\n",
    "    ret_li = []\n",
    "    for i in li:\n",
    "        ret_li.append(i[1])   \n",
    "    return ret_li"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that luad index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The GBM dataset is under publication embargo until March 01, 2021. CPTAC is a community resource project and data are made available rapidly after generation for community research use. The embargo allows exploring and utilizing the data, but analysis may not be published until after the embargo date. Please see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 14)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that lscc index is up-to-date...   \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The HNSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 17)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "version 3scc v3.2.......                 \n",
      "                            \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 18)\n"
     ]
    }
   ],
   "source": [
    "# cptac.download(\"brca\")\n",
    "# cptac.download(\"ccrcc\")\n",
    "# cptac.download(\"colon\")\n",
    "# cptac.download(\"endometrial\")\n",
    "# cptac.download(\"gbm\")\n",
    "# cptac.download(\"luad\")\n",
    "# cptac.download(\"ovarian\")\n",
    "# cptac.download(\"hnscc\")\n",
    "# cptac.download(\"lscc\")\n",
    "brca = cptac.Brca()\n",
    "ccrcc = cptac.Ccrcc()\n",
    "colon = cptac.Colon()\n",
    "en = cptac.Endometrial()\n",
    "gbm = cptac.Gbm()\n",
    "luad = cptac.Luad()\n",
    "ovarian = cptac.Ovarian()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "brca_gene_list = get_gene_list(brca)\n",
    "ccrcc_gene_list = get_gene_list(ccrcc)\n",
    "colon_gene_list = get_gene_list(colon)\n",
    "gbm_gene_list = get_gene_list(gbm)\n",
    "luad_gene_list = get_gene_list(luad)\n",
    "ovarian_gene_list = get_gene_list(ovarian)\n",
    "en_gene_list = get_gene_list(en)\n",
    "hnscc_gene_list = get_gene_list(hnscc)\n",
    "lscc_gene_list = get_gene_list(lscc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 655 duplicated column headers. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\scipy\\stats\\stats.py:3508: PearsonRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(PearsonRConstantInputWarning())\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 2495 duplicated column headers. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 11)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 1162 duplicated column headers. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\humbe\\miniconda3\\lib\\site-packages\\ipykernel_launcher.py, line 7)\n"
     ]
    }
   ],
   "source": [
    "brca_corr, brca_pval = correlation_list(brca, brca_gene_list, tissue_type = \"tumor\")\n",
    "ccrcc_corr, ccrcc_pval = correlation_list(ccrcc, ccrcc_gene_list, tissue_type = \"tumor\")\n",
    "colon_corr, colon_pval = correlation_list(colon, colon_gene_list, tissue_type = \"tumor\")\n",
    "en_corr, en_pval = correlation_list(en, en_gene_list, tissue_type = \"tumor\")\n",
    "gbm_corr, gbm_pval = correlation_list(gbm, gbm_gene_list, tissue_type = \"tumor\")\n",
    "luad_corr, luad_pval = correlation_list(luad,luad_gene_list, tissue_type = \"tumor\")\n",
    "ovarian_corr, ovarian_pval = correlation_list(ovarian, ovarian_gene_list, tissue_type = \"tumor\")\n",
    "hnscc_corr, hnscc_pval = correlation_list(hnscc, hnscc_gene_list, tissue_type = \"tumor\")\n",
    "lscc_corr, lscc_pval = correlation_list(lscc, lscc_gene_list, tissue_type = \"tumor\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "genes = brca_gene_list + ccrcc_gene_list + colon_gene_list + en_gene_list + gbm_gene_list+ luad_gene_list + ovarian_gene_list + hnscc_gene_list + lscc_gene_list\n",
    "common_genes = []\n",
    "for i in genes:\n",
    "    if i not in common_genes:\n",
    "        common_genes.append(i)\n",
    "common_genes = sorted(common_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def fill_genes_nan(all_genes, tissue_corr):\n",
    "    tissue_genes = []\n",
    "    for i, j  in tissue_corr:\n",
    "        tissue_genes.append(i)\n",
    "        \n",
    "    for i in all_genes:\n",
    "        if i not in tissue_genes:\n",
    "            tissue_corr.append([i, np.nan])\n",
    "    end_list = sorted(tissue_corr)\n",
    "           \n",
    "    return end_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BRCA</th>\n",
       "      <th>CCRCC</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Endometrial</th>\n",
       "      <th>GBM</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>LSCC</th>\n",
       "      <th>Mean</th>\n",
       "      <th>Stnd Deviation</th>\n",
       "      <th>+/- 3 sigma</th>\n",
       "      <th>Range</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.131286</td>\n",
       "      <td>-0.059781</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.386703</td>\n",
       "      <td>-0.200579</td>\n",
       "      <td>-0.067920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.112133</td>\n",
       "      <td>0.122072</td>\n",
       "      <td>0.060559</td>\n",
       "      <td>0.190174</td>\n",
       "      <td>False</td>\n",
       "      <td>0.587282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.892757</td>\n",
       "      <td>0.698025</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.795391</td>\n",
       "      <td>0.137696</td>\n",
       "      <td>False</td>\n",
       "      <td>0.194732</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.422497</td>\n",
       "      <td>0.010289</td>\n",
       "      <td>-0.122255</td>\n",
       "      <td>0.024022</td>\n",
       "      <td>0.312922</td>\n",
       "      <td>0.381025</td>\n",
       "      <td>0.140341</td>\n",
       "      <td>0.201083</td>\n",
       "      <td>0.345069</td>\n",
       "      <td>0.190555</td>\n",
       "      <td>0.190275</td>\n",
       "      <td>False</td>\n",
       "      <td>0.544752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>0.713289</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.835080</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.531368</td>\n",
       "      <td>0.862612</td>\n",
       "      <td>0.832191</td>\n",
       "      <td>0.754908</td>\n",
       "      <td>0.137571</td>\n",
       "      <td>False</td>\n",
       "      <td>0.331245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.486499</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.312484</td>\n",
       "      <td>0.399491</td>\n",
       "      <td>0.123047</td>\n",
       "      <td>False</td>\n",
       "      <td>0.174015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>0.200364</td>\n",
       "      <td>0.124010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.146719</td>\n",
       "      <td>0.185779</td>\n",
       "      <td>0.212842</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.488809</td>\n",
       "      <td>0.600445</td>\n",
       "      <td>0.279853</td>\n",
       "      <td>0.186240</td>\n",
       "      <td>False</td>\n",
       "      <td>0.476435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>-0.029503</td>\n",
       "      <td>0.241315</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.231203</td>\n",
       "      <td>0.458107</td>\n",
       "      <td>0.556451</td>\n",
       "      <td>0.371712</td>\n",
       "      <td>0.324304</td>\n",
       "      <td>0.363366</td>\n",
       "      <td>0.314619</td>\n",
       "      <td>0.175480</td>\n",
       "      <td>False</td>\n",
       "      <td>0.585954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>0.546670</td>\n",
       "      <td>0.217698</td>\n",
       "      <td>0.144728</td>\n",
       "      <td>0.532447</td>\n",
       "      <td>0.628549</td>\n",
       "      <td>0.414920</td>\n",
       "      <td>0.300631</td>\n",
       "      <td>0.544006</td>\n",
       "      <td>0.541382</td>\n",
       "      <td>0.430114</td>\n",
       "      <td>0.170422</td>\n",
       "      <td>False</td>\n",
       "      <td>0.483821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>0.280712</td>\n",
       "      <td>0.197289</td>\n",
       "      <td>0.087799</td>\n",
       "      <td>0.299816</td>\n",
       "      <td>0.424073</td>\n",
       "      <td>0.679603</td>\n",
       "      <td>0.218811</td>\n",
       "      <td>0.555744</td>\n",
       "      <td>0.625155</td>\n",
       "      <td>0.374334</td>\n",
       "      <td>0.207194</td>\n",
       "      <td>False</td>\n",
       "      <td>0.591803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>0.374921</td>\n",
       "      <td>0.101076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.011190</td>\n",
       "      <td>0.263872</td>\n",
       "      <td>0.490226</td>\n",
       "      <td>0.332541</td>\n",
       "      <td>0.110028</td>\n",
       "      <td>0.492600</td>\n",
       "      <td>0.269259</td>\n",
       "      <td>0.187539</td>\n",
       "      <td>False</td>\n",
       "      <td>0.503789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14636 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            BRCA     CCRCC     Colon  Endometrial       GBM      LUAD  \\\n",
       "A1BG    0.131286 -0.059781       NaN     0.386703 -0.200579 -0.067920   \n",
       "A1CF         NaN  0.892757  0.698025          NaN       NaN       NaN   \n",
       "A2M     0.422497  0.010289 -0.122255     0.024022  0.312922  0.381025   \n",
       "A2ML1   0.713289       NaN       NaN     0.835080       NaN       NaN   \n",
       "A4GALT       NaN       NaN       NaN     0.486499       NaN       NaN   \n",
       "...          ...       ...       ...          ...       ...       ...   \n",
       "ZXDC    0.200364  0.124010       NaN     0.146719  0.185779  0.212842   \n",
       "ZYG11B -0.029503  0.241315       NaN     0.231203  0.458107  0.556451   \n",
       "ZYX     0.546670  0.217698  0.144728     0.532447  0.628549  0.414920   \n",
       "ZZEF1   0.280712  0.197289  0.087799     0.299816  0.424073  0.679603   \n",
       "ZZZ3    0.374921  0.101076       NaN    -0.011190  0.263872  0.490226   \n",
       "\n",
       "         Ovarian     HNSCC      LSCC      Mean  Stnd Deviation  +/- 3 sigma  \\\n",
       "A1BG         NaN  0.112133  0.122072  0.060559        0.190174        False   \n",
       "A1CF         NaN       NaN       NaN  0.795391        0.137696        False   \n",
       "A2M     0.140341  0.201083  0.345069  0.190555        0.190275        False   \n",
       "A2ML1   0.531368  0.862612  0.832191  0.754908        0.137571        False   \n",
       "A4GALT       NaN       NaN  0.312484  0.399491        0.123047        False   \n",
       "...          ...       ...       ...       ...             ...          ...   \n",
       "ZXDC         NaN  0.488809  0.600445  0.279853        0.186240        False   \n",
       "ZYG11B  0.371712  0.324304  0.363366  0.314619        0.175480        False   \n",
       "ZYX     0.300631  0.544006  0.541382  0.430114        0.170422        False   \n",
       "ZZEF1   0.218811  0.555744  0.625155  0.374334        0.207194        False   \n",
       "ZZZ3    0.332541  0.110028  0.492600  0.269259        0.187539        False   \n",
       "\n",
       "           Range  \n",
       "A1BG    0.587282  \n",
       "A1CF    0.194732  \n",
       "A2M     0.544752  \n",
       "A2ML1   0.331245  \n",
       "A4GALT  0.174015  \n",
       "...          ...  \n",
       "ZXDC    0.476435  \n",
       "ZYG11B  0.585954  \n",
       "ZYX     0.483821  \n",
       "ZZEF1   0.591803  \n",
       "ZZZ3    0.503789  \n",
       "\n",
       "[14636 rows x 13 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_list_corr = ret_list(fill_genes_nan(common_genes, brca_corr))\n",
    "ccrcc_list_corr = ret_list(fill_genes_nan(common_genes, ccrcc_corr))\n",
    "colon_list_corr = ret_list(fill_genes_nan(common_genes, colon_corr))\n",
    "en_list_corr = ret_list(fill_genes_nan(common_genes, en_corr))\n",
    "gbm_list_corr = ret_list(fill_genes_nan(common_genes, gbm_corr))\n",
    "luad_list_corr = ret_list(fill_genes_nan(common_genes, luad_corr))\n",
    "ovarian_list_corr = ret_list(fill_genes_nan(common_genes, ovarian_corr))\n",
    "hnscc_list_corr = ret_list(fill_genes_nan(common_genes, hnscc_corr))\n",
    "lscc_list_corr = ret_list(fill_genes_nan(common_genes, lscc_corr))\n",
    "\n",
    "mean_list = []\n",
    "std_list = []\n",
    "sigma_list = []\n",
    "range_list = []\n",
    "def minmax(val_list):\n",
    "    if len(val_list) == 0:\n",
    "        return np.nan, np.nan\n",
    "    else:\n",
    "        min_val = min(val_list)\n",
    "        max_val = max(val_list)\n",
    "\n",
    "    return (min_val, max_val)\n",
    "for i in range(len(common_genes)):\n",
    "    \n",
    "    row = list(filter(lambda x:not math.isnan(x),[brca_list_corr[i], ccrcc_list_corr[i], colon_list_corr[i], \n",
    "             en_list_corr[i],gbm_list_corr[i],luad_list_corr[i],ovarian_list_corr[i],\n",
    "             hnscc_list_corr[i],lscc_list_corr[i]]))#Creates a list without Nan on each row\n",
    "\n",
    "    counter = len(row)\n",
    "    if counter == 0:\n",
    "        counter = 1\n",
    "        \n",
    "    ###Mean calculation\n",
    "    mean = sum(row)/counter\n",
    "    mean_list.append(mean)\n",
    "    \n",
    "    ###Standard Deviation calculation\n",
    "    std = 0\n",
    "    if len(row) < 2:\n",
    "        std = np.nan    \n",
    "    else:\n",
    "        std = statistics.stdev(row)\n",
    "    std_list.append(std)\n",
    "    \n",
    "    ### 3 sigma calculation\n",
    "    neg_sigma = mean - 3 * std\n",
    "    pos_sigma = mean + 3 * std\n",
    "    \n",
    "    for i in row:\n",
    "        sigma_bool = False\n",
    "        if i > pos_sigma or i < neg_sigma:\n",
    "            sigma_bool = True\n",
    "            break\n",
    "    sigma_list.append(sigma_bool)\n",
    "    \n",
    "    ###Calculating range\n",
    "    min_value, max_value = minmax(row)\n",
    "    range_list.append(max_value - min_value)\n",
    "    \n",
    "    \n",
    "correlation_columns = {'BRCA': (brca_list_corr),'CCRCC': (ccrcc_list_corr), \n",
    "       'Colon': (colon_list_corr), 'Endometrial': (en_list_corr),\n",
    "       'GBM': (gbm_list_corr), 'LUAD': (luad_list_corr), 'Ovarian': (ovarian_list_corr),\n",
    "        'HNSCC' : (hnscc_list_corr), 'LSCC' : (lscc_list_corr), 'Mean' : mean_list, 'Stnd Deviation' :std_list, '+/- 3 sigma' : sigma_list, 'Range' : range_list }\n",
    "\n",
    "correlation_df = pd.DataFrame(correlation_columns , columns = ['BRCA', 'CCRCC', 'Colon', 'Endometrial', 'GBM', 'LUAD', 'Ovarian', 'HNSCC', 'LSCC', 'Mean', 'Stnd Deviation', '+/- 3 sigma', 'Range'], index=common_genes)\n",
    "correlation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BRCA</th>\n",
       "      <th>CCRCC</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Endometrial</th>\n",
       "      <th>GBM</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>LSCC</th>\n",
       "      <th>Mean</th>\n",
       "      <th>Stnd Deviation</th>\n",
       "      <th>+/- 3 sigma</th>\n",
       "      <th>Range</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [BRCA, CCRCC, Colon, Endometrial, GBM, LUAD, Ovarian, HNSCC, LSCC, Mean, Stnd Deviation, +/- 3 sigma, Range]\n",
       "Index: []"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_df.loc[correlation_df['+/- 3 sigma'] == True]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BRCA</th>\n",
       "      <th>CCRCC</th>\n",
       "      <th>Colon</th>\n",
       "      <th>Endometrial</th>\n",
       "      <th>GBM</th>\n",
       "      <th>LUAD</th>\n",
       "      <th>Ovarian</th>\n",
       "      <th>HNSCC</th>\n",
       "      <th>LSCC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>1.657241e-01</td>\n",
       "      <td>5.350095e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.084161e-04</td>\n",
       "      <td>4.651711e-02</td>\n",
       "      <td>4.807936e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.456948e-01</td>\n",
       "      <td>2.125541e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3.432584e-39</td>\n",
       "      <td>5.325574e-15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>1.249853e-06</td>\n",
       "      <td>9.150454e-01</td>\n",
       "      <td>2.353794e-01</td>\n",
       "      <td>8.172572e-01</td>\n",
       "      <td>1.612770e-03</td>\n",
       "      <td>4.015329e-05</td>\n",
       "      <td>0.208548</td>\n",
       "      <td>3.602656e-02</td>\n",
       "      <td>2.548479e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.373971e-17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.999470e-23</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>1.908976e-33</td>\n",
       "      <td>6.590706e-29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A4GALT</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.594752e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.447014e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZXDC</th>\n",
       "      <td>4.909360e-02</td>\n",
       "      <td>2.958874e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.559574e-01</td>\n",
       "      <td>7.958310e-02</td>\n",
       "      <td>5.049816e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.776369e-07</td>\n",
       "      <td>5.053928e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYG11B</th>\n",
       "      <td>7.470119e-01</td>\n",
       "      <td>1.109573e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.417794e-02</td>\n",
       "      <td>1.861457e-06</td>\n",
       "      <td>2.757689e-10</td>\n",
       "      <td>0.000585</td>\n",
       "      <td>5.811263e-04</td>\n",
       "      <td>1.109731e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZYX</th>\n",
       "      <td>7.319264e-11</td>\n",
       "      <td>2.569032e-02</td>\n",
       "      <td>1.594617e-01</td>\n",
       "      <td>2.798595e-08</td>\n",
       "      <td>3.274769e-12</td>\n",
       "      <td>6.578244e-06</td>\n",
       "      <td>0.006064</td>\n",
       "      <td>9.742303e-10</td>\n",
       "      <td>1.451108e-09</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZEF1</th>\n",
       "      <td>1.736969e-03</td>\n",
       "      <td>3.883803e-02</td>\n",
       "      <td>3.949742e-01</td>\n",
       "      <td>3.159430e-03</td>\n",
       "      <td>1.218953e-05</td>\n",
       "      <td>3.279695e-16</td>\n",
       "      <td>0.048267</td>\n",
       "      <td>3.547520e-10</td>\n",
       "      <td>4.761787e-13</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>2.092650e-05</td>\n",
       "      <td>3.754370e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.142955e-01</td>\n",
       "      <td>8.312799e-03</td>\n",
       "      <td>3.453729e-07</td>\n",
       "      <td>0.003557</td>\n",
       "      <td>3.473585e-01</td>\n",
       "      <td>6.117741e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14636 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                BRCA         CCRCC         Colon   Endometrial           GBM  \\\n",
       "A1BG    1.657241e-01  5.350095e-01           NaN  1.084161e-04  4.651711e-02   \n",
       "A1CF             NaN  3.432584e-39  5.325574e-15           NaN           NaN   \n",
       "A2M     1.249853e-06  9.150454e-01  2.353794e-01  8.172572e-01  1.612770e-03   \n",
       "A2ML1   1.373971e-17           NaN           NaN  9.999470e-23           NaN   \n",
       "A4GALT           NaN           NaN           NaN  2.594752e-05           NaN   \n",
       "...              ...           ...           ...           ...           ...   \n",
       "ZXDC    4.909360e-02  2.958874e-01           NaN  1.559574e-01  7.958310e-02   \n",
       "ZYG11B  7.470119e-01  1.109573e-02           NaN  2.417794e-02  1.861457e-06   \n",
       "ZYX     7.319264e-11  2.569032e-02  1.594617e-01  2.798595e-08  3.274769e-12   \n",
       "ZZEF1   1.736969e-03  3.883803e-02  3.949742e-01  3.159430e-03  1.218953e-05   \n",
       "ZZZ3    2.092650e-05  3.754370e-01           NaN  9.142955e-01  8.312799e-03   \n",
       "\n",
       "                LUAD   Ovarian         HNSCC          LSCC  \n",
       "A1BG    4.807936e-01       NaN  2.456948e-01  2.125541e-01  \n",
       "A1CF             NaN       NaN           NaN           NaN  \n",
       "A2M     4.015329e-05  0.208548  3.602656e-02  2.548479e-04  \n",
       "A2ML1            NaN  0.000013  1.908976e-33  6.590706e-29  \n",
       "A4GALT           NaN       NaN           NaN  8.447014e-03  \n",
       "...              ...       ...           ...           ...  \n",
       "ZXDC    5.049816e-02       NaN  3.776369e-07  5.053928e-11  \n",
       "ZYG11B  2.757689e-10  0.000585  5.811263e-04  1.109731e-04  \n",
       "ZYX     6.578244e-06  0.006064  9.742303e-10  1.451108e-09  \n",
       "ZZEF1   3.279695e-16  0.048267  3.547520e-10  4.761787e-13  \n",
       "ZZZ3    3.453729e-07  0.003557  3.473585e-01  6.117741e-08  \n",
       "\n",
       "[14636 rows x 9 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "brca_list_pval = (ret_list(fill_genes_nan(common_genes, brca_pval)))\n",
    "ccrcc_list_pval = (ret_list(fill_genes_nan(common_genes, ccrcc_pval)))\n",
    "colon_list_pval = (ret_list(fill_genes_nan(common_genes, colon_pval)))\n",
    "en_list_pval = (ret_list(fill_genes_nan(common_genes, en_pval)))\n",
    "gbm_list_pval = (ret_list(fill_genes_nan(common_genes, gbm_pval)))\n",
    "luad_list_pval = (ret_list(fill_genes_nan(common_genes, luad_pval)))\n",
    "ovarian_list_pval = (ret_list(fill_genes_nan(common_genes, ovarian_pval)))\n",
    "hnscc_list_pval = (ret_list(fill_genes_nan(common_genes, hnscc_pval)))\n",
    "lscc_list_pval = (ret_list(fill_genes_nan(common_genes, lscc_pval)))\n",
    "\n",
    "\n",
    "pval_columns = {'BRCA': (brca_list_pval),\n",
    "        'CCRCC': (ccrcc_list_pval), 'Colon': (colon_list_pval), 'Endometrial': (en_list_pval),\n",
    "       'GBM': (gbm_list_pval), 'LUAD': (luad_list_pval), 'Ovarian': (ovarian_list_pval), 'HNSCC' : (hnscc_list_pval), 'LSCC' : (lscc_list_pval)}\n",
    "\n",
    "pval_df= pd.DataFrame(pval_columns , columns = ['BRCA', 'CCRCC', 'Colon', 'Endometrial', 'GBM', 'LUAD', 'Ovarian', 'HNSCC', 'LSCC'], index=common_genes)\n",
    "pval_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
