{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (0.9.4) is out-of-date. Latest is 0.9.6. Please run 'pip install --upgrade cptac' to update it. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 1)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import sys, os\n",
    "currentdir = os.path.dirname(os.path.realpath('make_regression_df.py'))\n",
    "parentdir = os.path.dirname(currentdir)\n",
    "sys.path.append(parentdir)\n",
    "import Delta_Correlation as dc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 14)\n"
     ]
    }
   ],
   "source": [
    "cancers, cancer_names = dc.load_cancers()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 36)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.304302</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>0.995336</td>\n",
       "      <td>Grade III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.641447</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>16.677828</td>\n",
       "      <td>Grade III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.000025</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>353.263362</td>\n",
       "      <td>Grade III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.207831</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>15.831130</td>\n",
       "      <td>Grade III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>AACS</td>\n",
       "      <td>-0.364128</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>2.938550</td>\n",
       "      <td>Grade III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454341</th>\n",
       "      <td>C3N-03933.N</td>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>22.136796</td>\n",
       "      <td>Normal</td>\n",
       "      <td>8.900000</td>\n",
       "      <td>Normal</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454342</th>\n",
       "      <td>C3N-03933.N</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>19.411112</td>\n",
       "      <td>Normal</td>\n",
       "      <td>11.160000</td>\n",
       "      <td>Normal</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454343</th>\n",
       "      <td>C3N-03933.N</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>23.439007</td>\n",
       "      <td>Normal</td>\n",
       "      <td>11.960000</td>\n",
       "      <td>Normal</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454344</th>\n",
       "      <td>C3N-03933.N</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>27.619972</td>\n",
       "      <td>Normal</td>\n",
       "      <td>11.330000</td>\n",
       "      <td>Normal</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1454345</th>\n",
       "      <td>C3N-03933.N</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>24.746321</td>\n",
       "      <td>Normal</td>\n",
       "      <td>12.730000</td>\n",
       "      <td>Normal</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4182198 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID    Gene  Proteomics  Tissue  Transcriptomics      Grade  \\\n",
       "0          C3L-00004    A1BG   -0.304302   Tumor         0.995336  Grade III   \n",
       "1          C3L-00004    A1CF    0.641447   Tumor        16.677828  Grade III   \n",
       "2          C3L-00004     A2M   -0.000025   Tumor       353.263362  Grade III   \n",
       "3          C3L-00004    AAAS    0.207831   Tumor        15.831130  Grade III   \n",
       "4          C3L-00004    AACS   -0.364128   Tumor         2.938550  Grade III   \n",
       "...              ...     ...         ...     ...              ...        ...   \n",
       "1454341  C3N-03933.N  ZWILCH   22.136796  Normal         8.900000     Normal   \n",
       "1454342  C3N-03933.N    ZXDC   19.411112  Normal        11.160000     Normal   \n",
       "1454343  C3N-03933.N  ZYG11B   23.439007  Normal        11.960000     Normal   \n",
       "1454344  C3N-03933.N     ZYX   27.619972  Normal        11.330000     Normal   \n",
       "1454345  C3N-03933.N   ZZEF1   24.746321  Normal        12.730000     Normal   \n",
       "\n",
       "        Cancer  \n",
       "0        CCRCC  \n",
       "1        CCRCC  \n",
       "2        CCRCC  \n",
       "3        CCRCC  \n",
       "4        CCRCC  \n",
       "...        ...  \n",
       "1454341  HNSCC  \n",
       "1454342  HNSCC  \n",
       "1454343  HNSCC  \n",
       "1454344  HNSCC  \n",
       "1454345  HNSCC  \n",
       "\n",
       "[4182198 rows x 7 columns]"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):    \n",
    "    if cancer_name == 'CCRCC':\n",
    "        grade = cancer.get_clinical('tumor')[['histologic_grade']]\n",
    "        grade['Patient_ID'] = grade.index\n",
    "        grade = grade.rename(columns = {'histologic_grade': 'Grade'})\n",
    "        grade = grade.reset_index(level = 0, drop = True)\n",
    "        grades = []\n",
    "        for g in grade['Grade']:\n",
    "            if g == 'G1':\n",
    "                grades.append('Grade I')\n",
    "            if g == 'G2':\n",
    "                grades.append('Grade II')\n",
    "            if g == 'G3':\n",
    "                grades.append('Grade III')\n",
    "            if g == 'G4':\n",
    "                grades.append('Grade IV')\n",
    "        grade['Grade'] = grades\n",
    "    elif cancer_name == 'Endometrial':\n",
    "        grade = cancer.get_clinical('tumor')[['Histologic_Grade_FIGO']]\n",
    "        grade['Patient_ID'] = grade.index\n",
    "        grade = grade.rename(columns = {'Histologic_Grade_FIGO': 'Grade'})\n",
    "        grade = grade.reset_index(level = 0, drop = True)\n",
    "        grades = []\n",
    "        for g in grade['Grade']:\n",
    "            if g in ['FIGO grade 1']:\n",
    "                grades.append('Grade I')\n",
    "            elif g in ['FIGO grade 2']:\n",
    "                grades.append('Grade II')\n",
    "            elif g in ['FIGO grade 3']:\n",
    "                grades.append('Grade III')\n",
    "            else:\n",
    "                grades.append(float('NaN'))\n",
    "        grade['Grade'] = grades\n",
    "    elif cancer_name == 'HNSCC':\n",
    "        grade = cancer.get_clinical('tumor')[['histologic_grade']]\n",
    "        grade['Patient_ID'] = grade.index\n",
    "        grade = grade.rename(columns = {'histologic_grade': 'Grade'})\n",
    "        grade = grade.reset_index(level = 0, drop = True)\n",
    "        grades = []\n",
    "        for g in grade['Grade']:\n",
    "            if g in ['G1 Well differentiated']:\n",
    "                grades.append('Grade I')\n",
    "            elif g in ['G2 Moderately differentiated']:\n",
    "                grades.append('Grade II')\n",
    "            elif g in ['G3 Poorly differentiated']:\n",
    "                grades.append('Grade III')\n",
    "            else:\n",
    "                grades.append(float('NaN'))\n",
    "        grade['Grade'] = grades\n",
    "    grade = grade.dropna()\n",
    "    cancer_df = dc.get_prot_trans_df(cancer)\n",
    "    cancer_df = cancer_df.dropna()\n",
    "    cancer_df = pd.merge(cancer_df, grade, how = 'outer')\n",
    "    cancer_df[['Grade']] = cancer_df[['Grade']].fillna(value='Normal')\n",
    "    cancer_df = cancer_df.dropna()\n",
    "    cancer_df['Cancer'] = [cancer_name] * len(cancer_df)\n",
    "    cancer_dfs.append(cancer_df)\n",
    "prot_trans_stages_df = pd.concat(cancer_dfs)\n",
    "prot_trans_stages_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Grade</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Spearman_Rho</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Grade I</td>\n",
       "      <td>ABLIM1</td>\n",
       "      <td>0.442981</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Grade I</td>\n",
       "      <td>ABLIM3</td>\n",
       "      <td>0.691161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Grade I</td>\n",
       "      <td>ACLY</td>\n",
       "      <td>0.310087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Grade I</td>\n",
       "      <td>ACOX1</td>\n",
       "      <td>-0.014110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Grade I</td>\n",
       "      <td>ACP1</td>\n",
       "      <td>0.623290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143911</th>\n",
       "      <td>HNSCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>-0.068118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143912</th>\n",
       "      <td>HNSCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.809273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143913</th>\n",
       "      <td>HNSCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.511977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143914</th>\n",
       "      <td>HNSCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.129024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143915</th>\n",
       "      <td>HNSCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.033533</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100864 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Cancer    Grade    Gene  Spearman_Rho\n",
       "67      CCRCC  Grade I  ABLIM1      0.442981\n",
       "69      CCRCC  Grade I  ABLIM3      0.691161\n",
       "103     CCRCC  Grade I    ACLY      0.310087\n",
       "115     CCRCC  Grade I   ACOX1     -0.014110\n",
       "118     CCRCC  Grade I    ACP1      0.623290\n",
       "...       ...      ...     ...           ...\n",
       "143911  HNSCC   Normal    ZXDC     -0.068118\n",
       "143912  HNSCC   Normal  ZYG11B      0.809273\n",
       "143913  HNSCC   Normal     ZYX      0.511977\n",
       "143914  HNSCC   Normal   ZZEF1      0.129024\n",
       "143915  HNSCC   Normal    ZZZ3      0.033533\n",
       "\n",
       "[100864 rows x 4 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = prot_trans_stages_df\n",
    "df = df.drop(columns = 'Patient_ID')\n",
    "grouped_stages = df.groupby(['Cancer', 'Grade', 'Gene'])\n",
    "corr_df = grouped_stages.apply(lambda x: x.corr(method = 'spearman', min_periods = 10).iloc[0][1])\n",
    "corr_df = pd.DataFrame(corr_df, columns = ['Spearman_Rho'])\n",
    "corr_df.reset_index(inplace = True)\n",
    "corr_df = corr_df.dropna()\n",
    "corr_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df.to_csv('Cancer_grades_correlations.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = (cancers[0:2] + [cancers[-2]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CCRCC', 'Endometrial', 'HNSCC']"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_names = cancer_names[0:2] + [cancer_names[-2]]\n",
    "cancer_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 36)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 36)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (/Users/humbertogiraldez/Documents/GitHub/pancancerProteinMRNA/notebook_steps_Spearman/Delta_Correlation.py, line 36)\n"
     ]
    }
   ],
   "source": [
    "cancer_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):    \n",
    "    if cancer_name == 'CCRCC':\n",
    "        grade = cancer.get_clinical('tumor')[['histologic_grade']]\n",
    "        grade['Patient_ID'] = grade.index\n",
    "        grade = grade.rename(columns = {'histologic_grade': 'Grade'})\n",
    "        grade = grade.reset_index(level = 0, drop = True)\n",
    "        grades = []\n",
    "        for g in grade['Grade']:\n",
    "            if g == 'G1':\n",
    "                grades.append('Grade I')\n",
    "            if g == 'G2':\n",
    "                grades.append('Grade II')\n",
    "            if g == 'G3':\n",
    "                grades.append('Grade III')\n",
    "            if g == 'G4':\n",
    "                grades.append('Grade IV')\n",
    "        grade['Grade'] = grades\n",
    "    elif cancer_name == 'Endometrial':\n",
    "        grade = cancer.get_clinical('tumor')[['Histologic_Grade_FIGO']]\n",
    "        grade['Patient_ID'] = grade.index\n",
    "        grade = grade.rename(columns = {'Histologic_Grade_FIGO': 'Grade'})\n",
    "        grade = grade.reset_index(level = 0, drop = True)\n",
    "        grades = []\n",
    "        for g in grade['Grade']:\n",
    "            if g in ['FIGO grade 1']:\n",
    "                grades.append('Grade I')\n",
    "            elif g in ['FIGO grade 2']:\n",
    "                grades.append('Grade II')\n",
    "            elif g in ['FIGO grade 3']:\n",
    "                grades.append('Grade III')\n",
    "            else:\n",
    "                grades.append(float('NaN'))\n",
    "        grade['Grade'] = grades\n",
    "    elif cancer_name == 'HNSCC':\n",
    "        grade = cancer.get_clinical('tumor')[['histologic_grade']]\n",
    "        grade['Patient_ID'] = grade.index\n",
    "        grade = grade.rename(columns = {'histologic_grade': 'Grade'})\n",
    "        grade = grade.reset_index(level = 0, drop = True)\n",
    "        grades = []\n",
    "        for g in grade['Grade']:\n",
    "            if g in ['G1 Well differentiated']:\n",
    "                grades.append('Grade I')\n",
    "            elif g in ['G2 Moderately differentiated']:\n",
    "                grades.append('Grade II')\n",
    "            elif g in ['G3 Poorly differentiated']:\n",
    "                grades.append('Grade III')\n",
    "            else:\n",
    "                grades.append(float('NaN'))\n",
    "        grade['Grade'] = grades\n",
    "    grade = grade.dropna()\n",
    "    cancer_df = dc.get_prot_trans_df(cancer)\n",
    "    cancer_df = cancer_df.dropna()\n",
    "    cancer_df = pd.merge(cancer_df, grade, how = 'outer')\n",
    "    cancer_df[['Grade']] = cancer_df[['Grade']].fillna(value='Normal')\n",
    "    cancer_df = cancer_df.dropna()\n",
    "    cancer_df['Cancer'] = [cancer_name] * len(cancer_df)\n",
    "    cancer_dfs.append(cancer_df)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
