{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import cptac\n",
    "import statistics as st\n",
    "import cptac.utils as ut\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "import transform_pval \n",
    "from scipy import stats\n",
    "import scipy"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-5c25f03e6d4a>, line 5)\n"
     ]
    }
   ],
   "source": [
    "ccrcc = cptac.Ccrcc()\n",
    "en = cptac.Endometrial()\n",
    "luad = cptac.Luad()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_dict = {ccrcc:\"ccrcc\",en:\"endometrial\",luad:\"luad\", hnscc:\"hnscc\",lscc:\"lscc\"}\n",
    "cancer_list = [ccrcc,en,luad,hnscc,lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "mutated_gene = 'TP53'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_omics_df(cancer):\n",
    "    transcriptomics_df = cancer.get_transcriptomics(tissue_type='tumor')\n",
    "    proteomics_df = cancer.get_proteomics(tissue_type='tumor')\n",
    "    if isinstance(proteomics_df.columns, pd.MultiIndex):\n",
    "        proteomics_df = proteomics_df.droplevel('Database_ID', axis = 1)\n",
    "    if isinstance(transcriptomics_df.columns, pd.MultiIndex):\n",
    "        transcriptomics_df = transcriptomics_df.droplevel('Database_ID', axis = 1)\n",
    "    proteomics_df['patient_ID'] = proteomics_df.index\n",
    "    transcriptomics_df['patient_ID'] = transcriptomics_df.index\n",
    "    transcriptomics_df = transcriptomics_df.melt(id_vars='patient_ID', var_name = 'gene', value_name='transcriptomics')\n",
    "    proteomics_df = proteomics_df.melt(id_vars='patient_ID', var_name = 'gene', value_name='proteomics')\n",
    "    mutation_df = cancer.get_somatic_mutation()\n",
    "    mutation_df = mutation_df[mutation_df.Gene == mutated_gene]\n",
    "    omics_df = pd.merge(transcriptomics_df, proteomics_df, how = 'inner')\n",
    "    omics_df['mutation_status'] = omics_df.patient_ID.isin(mutation_df.index)\n",
    "    omics_df = omics_df.dropna()\n",
    "    return omics_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_corr_df(omics_df):\n",
    "    mut_corrs = []\n",
    "    mut_p_vals = []\n",
    "    non_mut_corrs = []\n",
    "    non_mut_p_vals = []\n",
    "    corr_diffs = []\n",
    "    corr_diff_pvals = []\n",
    "    genes = []\n",
    "    for gene in pd.unique(omics_df.gene):\n",
    "        df = omics_df[omics_df.gene == gene]\n",
    "        mut_df = df[df.mutation_status == True]\n",
    "        non_mut_df = df[df.mutation_status == False]\n",
    "        if len(mut_df) < 4 or len(non_mut_df) < 4:\n",
    "            continue\n",
    "        mut_r, mut_p = stats.pearsonr(mut_df.transcriptomics, mut_df.proteomics)\n",
    "        non_mut_r, non_mut_p = stats.pearsonr(non_mut_df.transcriptomics, non_mut_df.proteomics)\n",
    "        mut_corrs.append(mut_r)\n",
    "        mut_p_vals.append(mut_p)\n",
    "        non_mut_corrs.append(non_mut_r)\n",
    "        non_mut_p_vals.append(non_mut_p)\n",
    "        corr_diff = mut_r - non_mut_r\n",
    "        corr_diffs.append(corr_diff)\n",
    "        diff_p_val = transform_pval.compare_correlations(mut_r, non_mut_r, len(mut_df), len(non_mut_df))\n",
    "        corr_diff_pvals.append(diff_p_val)\n",
    "        genes.append(gene)\n",
    "    correlation_df = pd.DataFrame({'gene': genes, 'mutated_correlation': mut_corrs, 'non_mutated_correlation': non_mut_corrs,\n",
    "                                   'non_mutated_p_vals': non_mut_p_vals, 'mutated_p_vals': mut_p_vals,\n",
    "                                   'delta_correlation': corr_diffs, 'delta_correlation_pval': corr_diff_pvals})\n",
    "    return correlation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ccrcc\n",
      "endometrial\n",
      "luad\n",
      "hnscc\n",
      "lscc\n"
     ]
    }
   ],
   "source": [
    "omics_dfs = []\n",
    "for cancer in cancer_list:\n",
    "    print(cancer_dict[cancer])\n",
    "    df = get_omics_df(cancer)\n",
    "    df['cancer'] = [cancer_dict[cancer]] * len(df)\n",
    "    omics_dfs.append(df)\n",
    "omics_df = pd.concat(omics_dfs)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['ccrcc']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:3845: PearsonRConstantInputWarning: An input array is constant; the correlation coefficent is not defined.\n",
      "  warnings.warn(PearsonRConstantInputWarning())\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['endometrial']\n",
      "['luad']\n",
      "['hnscc']\n",
      "['lscc']\n"
     ]
    }
   ],
   "source": [
    "correlation_dfs = []\n",
    "for df in omics_dfs:\n",
    "    cancer = list(pd.unique(df['cancer']))\n",
    "    print(cancer)\n",
    "    corr_df = get_corr_df(df)\n",
    "    corr_df['cancer'] =  cancer * len(corr_df)\n",
    "    correlation_dfs.append(corr_df)\n",
    "correlation_df = pd.concat(correlation_dfs)\n",
    "correlation_df.to_csv('TP53_trans_effects.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>gene</th>\n",
       "      <th>mutated_correlation</th>\n",
       "      <th>non_mutated_correlation</th>\n",
       "      <th>non_mutated_p_vals</th>\n",
       "      <th>mutated_p_vals</th>\n",
       "      <th>delta_correlation</th>\n",
       "      <th>delta_correlation_pval</th>\n",
       "      <th>cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.104701</td>\n",
       "      <td>-0.092503</td>\n",
       "      <td>3.503288e-01</td>\n",
       "      <td>8.435229e-01</td>\n",
       "      <td>0.197204</td>\n",
       "      <td>0.735579</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.949266</td>\n",
       "      <td>0.889170</td>\n",
       "      <td>2.059497e-36</td>\n",
       "      <td>3.795552e-03</td>\n",
       "      <td>0.060096</td>\n",
       "      <td>0.487926</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.491574</td>\n",
       "      <td>-0.017783</td>\n",
       "      <td>8.578013e-01</td>\n",
       "      <td>3.220318e-01</td>\n",
       "      <td>-0.473791</td>\n",
       "      <td>0.374445</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>-0.848880</td>\n",
       "      <td>0.162907</td>\n",
       "      <td>9.847157e-02</td>\n",
       "      <td>3.253011e-02</td>\n",
       "      <td>-1.011787</td>\n",
       "      <td>0.015614</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>AACS</td>\n",
       "      <td>0.738195</td>\n",
       "      <td>0.580735</td>\n",
       "      <td>1.031490e-10</td>\n",
       "      <td>9.384053e-02</td>\n",
       "      <td>0.157460</td>\n",
       "      <td>0.629146</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49121</th>\n",
       "      <td>9673</td>\n",
       "      <td>ZWINT</td>\n",
       "      <td>0.794975</td>\n",
       "      <td>0.859211</td>\n",
       "      <td>1.407891e-01</td>\n",
       "      <td>2.749241e-17</td>\n",
       "      <td>-0.064236</td>\n",
       "      <td>0.838289</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49122</th>\n",
       "      <td>9674</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.364882</td>\n",
       "      <td>0.136414</td>\n",
       "      <td>8.268526e-01</td>\n",
       "      <td>1.509645e-04</td>\n",
       "      <td>0.228468</td>\n",
       "      <td>0.731297</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49123</th>\n",
       "      <td>9675</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.508387</td>\n",
       "      <td>0.923179</td>\n",
       "      <td>2.526303e-02</td>\n",
       "      <td>4.197810e-08</td>\n",
       "      <td>-0.414792</td>\n",
       "      <td>0.141643</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49124</th>\n",
       "      <td>9676</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.591263</td>\n",
       "      <td>0.874924</td>\n",
       "      <td>5.209261e-02</td>\n",
       "      <td>4.844711e-11</td>\n",
       "      <td>-0.283660</td>\n",
       "      <td>0.345211</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49125</th>\n",
       "      <td>9677</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.502935</td>\n",
       "      <td>-0.373936</td>\n",
       "      <td>5.352303e-01</td>\n",
       "      <td>6.164005e-08</td>\n",
       "      <td>0.876871</td>\n",
       "      <td>0.185181</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49126 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0    gene  mutated_correlation  non_mutated_correlation  \\\n",
       "0               0    A1BG             0.104701                -0.092503   \n",
       "1               1    A1CF             0.949266                 0.889170   \n",
       "2               2     A2M            -0.491574                -0.017783   \n",
       "3               3    AAAS            -0.848880                 0.162907   \n",
       "4               4    AACS             0.738195                 0.580735   \n",
       "...           ...     ...                  ...                      ...   \n",
       "49121        9673   ZWINT             0.794975                 0.859211   \n",
       "49122        9674  ZYG11B             0.364882                 0.136414   \n",
       "49123        9675     ZYX             0.508387                 0.923179   \n",
       "49124        9676   ZZEF1             0.591263                 0.874924   \n",
       "49125        9677    ZZZ3             0.502935                -0.373936   \n",
       "\n",
       "       non_mutated_p_vals  mutated_p_vals  delta_correlation  \\\n",
       "0            3.503288e-01    8.435229e-01           0.197204   \n",
       "1            2.059497e-36    3.795552e-03           0.060096   \n",
       "2            8.578013e-01    3.220318e-01          -0.473791   \n",
       "3            9.847157e-02    3.253011e-02          -1.011787   \n",
       "4            1.031490e-10    9.384053e-02           0.157460   \n",
       "...                   ...             ...                ...   \n",
       "49121        1.407891e-01    2.749241e-17          -0.064236   \n",
       "49122        8.268526e-01    1.509645e-04           0.228468   \n",
       "49123        2.526303e-02    4.197810e-08          -0.414792   \n",
       "49124        5.209261e-02    4.844711e-11          -0.283660   \n",
       "49125        5.352303e-01    6.164005e-08           0.876871   \n",
       "\n",
       "       delta_correlation_pval cancer  \n",
       "0                    0.735579  ccrcc  \n",
       "1                    0.487926  ccrcc  \n",
       "2                    0.374445  ccrcc  \n",
       "3                    0.015614  ccrcc  \n",
       "4                    0.629146  ccrcc  \n",
       "...                       ...    ...  \n",
       "49121                0.838289   lscc  \n",
       "49122                0.731297   lscc  \n",
       "49123                0.141643   lscc  \n",
       "49124                0.345211   lscc  \n",
       "49125                0.185181   lscc  \n",
       "\n",
       "[49126 rows x 9 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "correlation_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
