{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import pandas as pd\n",
    "from scipy import stats\n",
    "import numpy as np\n",
    "import statsmodels.stats.multitest as ssm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-5c25f03e6d4a>, line 5)\n"
     ]
    }
   ],
   "source": [
    "ccrcc = cptac.Ccrcc()\n",
    "en = cptac.Endometrial()\n",
    "luad = cptac.Luad()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Make docstrings for what is required\n",
    "def get_single_gene_correlations(gene, cancer_type_list,type_dict,input_tissue_type = \"tumor\"):\n",
    "    correlation_list = []\n",
    "    type_list = []\n",
    "    pval_list = []\n",
    "    for cancer in cancer_type_list:\n",
    "        correlation,pval,num_samples = get_correlations.correlation_list(cancer, [gene],tissue_type = input_tissue_type)\n",
    "        if len(correlation) == 0:\n",
    "            continue\n",
    "        else:\n",
    "            correlation_list.append(correlation[0])\n",
    "            pval_list.append(pval[0][1])\n",
    "            type_list.append(type_dict[cancer])\n",
    "    return correlation_list, type_list, num_samples, pval_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_cancer_string(cancer):\n",
    "    if cancer == ccrcc:\n",
    "        return \"ccrcc\"\n",
    "    if cancer == en:\n",
    "        return 'endometrial'\n",
    "    if cancer == luad:\n",
    "        return 'luad'\n",
    "    if cancer == hnscc:\n",
    "        return 'hnscc'\n",
    "    if cancer == lscc:\n",
    "        return 'lscc'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See https://statisticsbyjim.com/regression/comparing-regression-lines/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "from rpy2.robjects import r, pandas2ri\n",
    "from rpy2.robjects.packages import importr\n",
    "import rpy2.robjects as robj\n",
    "import pandas as pd\n",
    "def linear_model(data, Input, Output, Condition):\n",
    "    try:\n",
    "        stats = importr('stats')\n",
    "        base = importr('base')\n",
    "        pandas2ri.activate()\n",
    "        r_df = pandas2ri.py2rpy(data)\n",
    "        pandas2ri.deactivate()\n",
    "        formula = '{y}~{x}*{condition}'.format(y = Output, x = Input, condition = Condition)\n",
    "        lm = stats.lm(formula, r_df)\n",
    "        summary = (base.summary(lm))\n",
    "        results = summary.rx2('coefficients')\n",
    "        results_df = base.as_data_frame_matrix(results)\n",
    "        py_results_df = pd.DataFrame(results_df).transpose()\n",
    "        py_results_df.columns = results_df.colnames\n",
    "        py_results_df.index = results_df.rownames\n",
    "        return(py_results_df)\n",
    "    except:\n",
    "        print(data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_norm_tumor_corr_effect(cancer_type, gene):\n",
    "    tumor = cancer_type.multi_join({'proteomics': gene, 'transcriptomics': gene}, tissue_type= 'tumor', flatten = True)\n",
    "    normal = cancer_type.multi_join({'proteomics': gene, 'transcriptomics': gene}, tissue_type= 'normal', flatten = True)\n",
    "    tumor = tumor.dropna()\n",
    "    normal = normal.dropna()\n",
    "    if len(normal) < 2 or len(tumor) < 2:\n",
    "        return float(\"NaN\"), float(\"NaN\"), float(\"NaN\")\n",
    "    tumor.columns = ['proteomics', 'transcriptomics']\n",
    "    normal.columns = ['proteomics', 'transcriptomics']\n",
    "    groups = ['tumor'] * len(tumor)\n",
    "    groups.extend(['normal']*len(normal))\n",
    "    prot_list = list(tumor['proteomics'])\n",
    "    prot_list.extend(list(normal['proteomics']))\n",
    "    trans_list = list(tumor['transcriptomics'])\n",
    "    trans_list.extend(list(normal['transcriptomics']))\n",
    "    gene_df = pd.DataFrame({'Type': groups, 'Proteomics': prot_list, 'Transcriptomics': trans_list})\n",
    "    is_tum = gene_df['Type'] == 'tumor'\n",
    "    is_norm = gene_df['Type'] == 'normal'\n",
    "    tum_cor, tum_pval = stats.pearsonr(gene_df[is_tum]['Proteomics'], gene_df[is_tum]['Transcriptomics'])\n",
    "    norm_cor, norm_pval = stats.pearsonr(gene_df[is_norm]['Proteomics'], gene_df[is_norm]['Transcriptomics'])\n",
    "    return tum_cor, tum_pval, norm_cor, norm_pval, gene_df "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_genes = ['BRAF','FAT1','IDH1','TP53','KMT2D','KRAS','PIK3CA','PTEN','SPTA1']\n",
    "cancer_list = [ccrcc,en,luad,hnscc,lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    }
   ],
   "source": [
    "# Make df for grid plot\n",
    "dfs = []\n",
    "rows = []\n",
    "for cancer in cancer_list:\n",
    "    for gene in cancer_genes:\n",
    "        if gene == \"KRAS\" and cancer == ccrcc:\n",
    "            continue\n",
    "        tum_cor, tum_pval, norm_cor, norm_pval, gene_df = find_norm_tumor_corr_effect(cancer, gene)\n",
    "        if not np.isnan(tum_cor):\n",
    "            d = {}\n",
    "            gene_df['Cancer'] = [get_cancer_string(cancer)] * len(gene_df)\n",
    "            gene_df['Gene'] = [gene] * len(gene_df)\n",
    "            dfs.append(gene_df)\n",
    "            d['cancer'] = get_cancer_string(cancer)\n",
    "            d['gene'] = gene\n",
    "            d['tum_pval'] = tum_pval\n",
    "            d['tum_corr'] = tum_cor\n",
    "            d['norm_pval'] = norm_pval\n",
    "            d['norm_corr'] = norm_cor\n",
    "            \n",
    "            rows.append(d)\n",
    "full_df = pd.concat(dfs)\n",
    "full_df = full_df.rename(columns ={'Type': 'Tissue'})\n",
    "corr_df = pd.DataFrame(rows)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>tumor</td>\n",
       "      <td>0.235538</td>\n",
       "      <td>21.616591</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>tumor</td>\n",
       "      <td>0.232718</td>\n",
       "      <td>19.209432</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>tumor</td>\n",
       "      <td>0.034752</td>\n",
       "      <td>14.278585</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>tumor</td>\n",
       "      <td>0.328557</td>\n",
       "      <td>22.972120</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>tumor</td>\n",
       "      <td>0.136153</td>\n",
       "      <td>15.809168</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>BRAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>195</th>\n",
       "      <td>normal</td>\n",
       "      <td>-0.788000</td>\n",
       "      <td>10.293500</td>\n",
       "      <td>lscc</td>\n",
       "      <td>SPTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>normal</td>\n",
       "      <td>-0.322600</td>\n",
       "      <td>9.841800</td>\n",
       "      <td>lscc</td>\n",
       "      <td>SPTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>normal</td>\n",
       "      <td>0.660700</td>\n",
       "      <td>10.935900</td>\n",
       "      <td>lscc</td>\n",
       "      <td>SPTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>normal</td>\n",
       "      <td>1.128500</td>\n",
       "      <td>10.572600</td>\n",
       "      <td>lscc</td>\n",
       "      <td>SPTA1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>normal</td>\n",
       "      <td>-1.959600</td>\n",
       "      <td>10.151200</td>\n",
       "      <td>lscc</td>\n",
       "      <td>SPTA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7313 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Tissue  Proteomics  Transcriptomics Cancer   Gene\n",
       "0     tumor    0.235538        21.616591  ccrcc   BRAF\n",
       "1     tumor    0.232718        19.209432  ccrcc   BRAF\n",
       "2     tumor    0.034752        14.278585  ccrcc   BRAF\n",
       "3     tumor    0.328557        22.972120  ccrcc   BRAF\n",
       "4     tumor    0.136153        15.809168  ccrcc   BRAF\n",
       "..      ...         ...              ...    ...    ...\n",
       "195  normal   -0.788000        10.293500   lscc  SPTA1\n",
       "196  normal   -0.322600         9.841800   lscc  SPTA1\n",
       "197  normal    0.660700        10.935900   lscc  SPTA1\n",
       "198  normal    1.128500        10.572600   lscc  SPTA1\n",
       "199  normal   -1.959600        10.151200   lscc  SPTA1\n",
       "\n",
       "[7313 rows x 5 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cancer</th>\n",
       "      <th>gene</th>\n",
       "      <th>tum_pval</th>\n",
       "      <th>tum_corr</th>\n",
       "      <th>norm_pval</th>\n",
       "      <th>norm_corr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>6.404490e-10</td>\n",
       "      <td>0.546805</td>\n",
       "      <td>4.861585e-01</td>\n",
       "      <td>0.081654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>4.138834e-12</td>\n",
       "      <td>0.600361</td>\n",
       "      <td>2.655076e-04</td>\n",
       "      <td>0.409362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>1.885896e-17</td>\n",
       "      <td>0.699513</td>\n",
       "      <td>2.697707e-14</td>\n",
       "      <td>0.741719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>TP53</td>\n",
       "      <td>4.480253e-01</td>\n",
       "      <td>0.146573</td>\n",
       "      <td>1.872371e-01</td>\n",
       "      <td>-0.278706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>4.143698e-01</td>\n",
       "      <td>0.078602</td>\n",
       "      <td>5.426068e-01</td>\n",
       "      <td>-0.071417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>5.065631e-09</td>\n",
       "      <td>0.521751</td>\n",
       "      <td>8.552509e-02</td>\n",
       "      <td>0.199898</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>3.267175e-08</td>\n",
       "      <td>0.497284</td>\n",
       "      <td>2.995446e-10</td>\n",
       "      <td>0.649130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>SPTA1</td>\n",
       "      <td>5.929323e-03</td>\n",
       "      <td>0.260784</td>\n",
       "      <td>1.837206e-01</td>\n",
       "      <td>-0.155179</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>1.740190e-12</td>\n",
       "      <td>0.644945</td>\n",
       "      <td>8.626160e-01</td>\n",
       "      <td>-0.050971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>7.399839e-29</td>\n",
       "      <td>0.859540</td>\n",
       "      <td>2.661306e-01</td>\n",
       "      <td>0.319095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>2.675118e-40</td>\n",
       "      <td>0.922802</td>\n",
       "      <td>1.028107e-01</td>\n",
       "      <td>0.454173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>TP53</td>\n",
       "      <td>3.832454e-01</td>\n",
       "      <td>0.096952</td>\n",
       "      <td>8.953207e-01</td>\n",
       "      <td>-0.045068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>4.446940e-02</td>\n",
       "      <td>0.206689</td>\n",
       "      <td>3.452136e-01</td>\n",
       "      <td>-0.272880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>5.752369e-05</td>\n",
       "      <td>0.400604</td>\n",
       "      <td>5.919370e-01</td>\n",
       "      <td>0.157005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3.422092e-12</td>\n",
       "      <td>0.638375</td>\n",
       "      <td>2.528128e-01</td>\n",
       "      <td>0.327641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>3.406507e-10</td>\n",
       "      <td>0.589048</td>\n",
       "      <td>4.292413e-02</td>\n",
       "      <td>0.547029</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>SPTA1</td>\n",
       "      <td>2.998187e-01</td>\n",
       "      <td>-0.107490</td>\n",
       "      <td>8.650617e-01</td>\n",
       "      <td>-0.050056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>luad</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>6.461032e-10</td>\n",
       "      <td>0.546703</td>\n",
       "      <td>1.397435e-01</td>\n",
       "      <td>0.147973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>luad</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>2.311694e-21</td>\n",
       "      <td>0.753126</td>\n",
       "      <td>1.219206e-04</td>\n",
       "      <td>0.373062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>luad</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>4.446926e-20</td>\n",
       "      <td>0.736898</td>\n",
       "      <td>1.001981e-15</td>\n",
       "      <td>0.692806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>luad</td>\n",
       "      <td>TP53</td>\n",
       "      <td>1.523660e-06</td>\n",
       "      <td>0.496929</td>\n",
       "      <td>9.447010e-02</td>\n",
       "      <td>-0.193213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>luad</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>2.433462e-04</td>\n",
       "      <td>0.343080</td>\n",
       "      <td>2.862975e-01</td>\n",
       "      <td>-0.107127</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>luad</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>2.368358e-02</td>\n",
       "      <td>0.219640</td>\n",
       "      <td>1.127935e-01</td>\n",
       "      <td>-0.162044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>luad</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>1.085663e-05</td>\n",
       "      <td>0.405897</td>\n",
       "      <td>6.000407e-02</td>\n",
       "      <td>0.187812</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>luad</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>4.114600e-12</td>\n",
       "      <td>0.600418</td>\n",
       "      <td>3.584929e-02</td>\n",
       "      <td>0.209110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>luad</td>\n",
       "      <td>SPTA1</td>\n",
       "      <td>3.893842e-01</td>\n",
       "      <td>-0.082873</td>\n",
       "      <td>2.778005e-01</td>\n",
       "      <td>0.109024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>7.578852e-13</td>\n",
       "      <td>0.618639</td>\n",
       "      <td>2.713018e-01</td>\n",
       "      <td>0.173692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>5.556750e-27</td>\n",
       "      <td>0.813865</td>\n",
       "      <td>1.282240e-01</td>\n",
       "      <td>0.238513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>9.646958e-31</td>\n",
       "      <td>0.844177</td>\n",
       "      <td>2.019960e-08</td>\n",
       "      <td>0.740893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>TP53</td>\n",
       "      <td>2.981543e-18</td>\n",
       "      <td>0.742948</td>\n",
       "      <td>3.699540e-01</td>\n",
       "      <td>0.149618</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>1.051409e-02</td>\n",
       "      <td>0.244149</td>\n",
       "      <td>4.539083e-01</td>\n",
       "      <td>0.118734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>8.859633e-09</td>\n",
       "      <td>0.516669</td>\n",
       "      <td>1.346871e-01</td>\n",
       "      <td>0.234663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>4.821807e-27</td>\n",
       "      <td>0.814412</td>\n",
       "      <td>8.598514e-01</td>\n",
       "      <td>0.028087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>9.640723e-14</td>\n",
       "      <td>0.636979</td>\n",
       "      <td>6.444848e-01</td>\n",
       "      <td>-0.073318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>SPTA1</td>\n",
       "      <td>1.080085e-01</td>\n",
       "      <td>0.154798</td>\n",
       "      <td>1.032935e-02</td>\n",
       "      <td>0.391600</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>lscc</td>\n",
       "      <td>BRAF</td>\n",
       "      <td>5.434862e-01</td>\n",
       "      <td>-0.059098</td>\n",
       "      <td>7.176714e-01</td>\n",
       "      <td>-0.037786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>lscc</td>\n",
       "      <td>FAT1</td>\n",
       "      <td>8.985484e-30</td>\n",
       "      <td>0.838953</td>\n",
       "      <td>7.860562e-07</td>\n",
       "      <td>0.483709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>lscc</td>\n",
       "      <td>IDH1</td>\n",
       "      <td>1.594770e-24</td>\n",
       "      <td>0.792698</td>\n",
       "      <td>6.768440e-12</td>\n",
       "      <td>0.634218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>lscc</td>\n",
       "      <td>TP53</td>\n",
       "      <td>9.640307e-19</td>\n",
       "      <td>0.723227</td>\n",
       "      <td>6.760100e-04</td>\n",
       "      <td>-0.344411</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>lscc</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>1.211582e-02</td>\n",
       "      <td>0.240646</td>\n",
       "      <td>7.444075e-01</td>\n",
       "      <td>-0.034073</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>lscc</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>1.395545e-18</td>\n",
       "      <td>0.720933</td>\n",
       "      <td>5.499014e-02</td>\n",
       "      <td>0.198606</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>lscc</td>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>3.444115e-27</td>\n",
       "      <td>0.817814</td>\n",
       "      <td>1.278941e-02</td>\n",
       "      <td>0.255913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>lscc</td>\n",
       "      <td>PTEN</td>\n",
       "      <td>6.855659e-17</td>\n",
       "      <td>0.695318</td>\n",
       "      <td>6.016573e-01</td>\n",
       "      <td>0.054532</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>lscc</td>\n",
       "      <td>SPTA1</td>\n",
       "      <td>4.306072e-03</td>\n",
       "      <td>-0.275174</td>\n",
       "      <td>6.508849e-01</td>\n",
       "      <td>0.047282</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         cancer    gene      tum_pval  tum_corr     norm_pval  norm_corr\n",
       "0         ccrcc    BRAF  6.404490e-10  0.546805  4.861585e-01   0.081654\n",
       "1         ccrcc    FAT1  4.138834e-12  0.600361  2.655076e-04   0.409362\n",
       "2         ccrcc    IDH1  1.885896e-17  0.699513  2.697707e-14   0.741719\n",
       "3         ccrcc    TP53  4.480253e-01  0.146573  1.872371e-01  -0.278706\n",
       "4         ccrcc   KMT2D  4.143698e-01  0.078602  5.426068e-01  -0.071417\n",
       "5         ccrcc  PIK3CA  5.065631e-09  0.521751  8.552509e-02   0.199898\n",
       "6         ccrcc    PTEN  3.267175e-08  0.497284  2.995446e-10   0.649130\n",
       "7         ccrcc   SPTA1  5.929323e-03  0.260784  1.837206e-01  -0.155179\n",
       "8   endometrial    BRAF  1.740190e-12  0.644945  8.626160e-01  -0.050971\n",
       "9   endometrial    FAT1  7.399839e-29  0.859540  2.661306e-01   0.319095\n",
       "10  endometrial    IDH1  2.675118e-40  0.922802  1.028107e-01   0.454173\n",
       "11  endometrial    TP53  3.832454e-01  0.096952  8.953207e-01  -0.045068\n",
       "12  endometrial   KMT2D  4.446940e-02  0.206689  3.452136e-01  -0.272880\n",
       "13  endometrial    KRAS  5.752369e-05  0.400604  5.919370e-01   0.157005\n",
       "14  endometrial  PIK3CA  3.422092e-12  0.638375  2.528128e-01   0.327641\n",
       "15  endometrial    PTEN  3.406507e-10  0.589048  4.292413e-02   0.547029\n",
       "16  endometrial   SPTA1  2.998187e-01 -0.107490  8.650617e-01  -0.050056\n",
       "17         luad    BRAF  6.461032e-10  0.546703  1.397435e-01   0.147973\n",
       "18         luad    FAT1  2.311694e-21  0.753126  1.219206e-04   0.373062\n",
       "19         luad    IDH1  4.446926e-20  0.736898  1.001981e-15   0.692806\n",
       "20         luad    TP53  1.523660e-06  0.496929  9.447010e-02  -0.193213\n",
       "21         luad   KMT2D  2.433462e-04  0.343080  2.862975e-01  -0.107127\n",
       "22         luad    KRAS  2.368358e-02  0.219640  1.127935e-01  -0.162044\n",
       "23         luad  PIK3CA  1.085663e-05  0.405897  6.000407e-02   0.187812\n",
       "24         luad    PTEN  4.114600e-12  0.600418  3.584929e-02   0.209110\n",
       "25         luad   SPTA1  3.893842e-01 -0.082873  2.778005e-01   0.109024\n",
       "26        hnscc    BRAF  7.578852e-13  0.618639  2.713018e-01   0.173692\n",
       "27        hnscc    FAT1  5.556750e-27  0.813865  1.282240e-01   0.238513\n",
       "28        hnscc    IDH1  9.646958e-31  0.844177  2.019960e-08   0.740893\n",
       "29        hnscc    TP53  2.981543e-18  0.742948  3.699540e-01   0.149618\n",
       "30        hnscc   KMT2D  1.051409e-02  0.244149  4.539083e-01   0.118734\n",
       "31        hnscc    KRAS  8.859633e-09  0.516669  1.346871e-01   0.234663\n",
       "32        hnscc  PIK3CA  4.821807e-27  0.814412  8.598514e-01   0.028087\n",
       "33        hnscc    PTEN  9.640723e-14  0.636979  6.444848e-01  -0.073318\n",
       "34        hnscc   SPTA1  1.080085e-01  0.154798  1.032935e-02   0.391600\n",
       "35         lscc    BRAF  5.434862e-01 -0.059098  7.176714e-01  -0.037786\n",
       "36         lscc    FAT1  8.985484e-30  0.838953  7.860562e-07   0.483709\n",
       "37         lscc    IDH1  1.594770e-24  0.792698  6.768440e-12   0.634218\n",
       "38         lscc    TP53  9.640307e-19  0.723227  6.760100e-04  -0.344411\n",
       "39         lscc   KMT2D  1.211582e-02  0.240646  7.444075e-01  -0.034073\n",
       "40         lscc    KRAS  1.395545e-18  0.720933  5.499014e-02   0.198606\n",
       "41         lscc  PIK3CA  3.444115e-27  0.817814  1.278941e-02   0.255913\n",
       "42         lscc    PTEN  6.855659e-17  0.695318  6.016573e-01   0.054532\n",
       "43         lscc   SPTA1  4.306072e-03 -0.275174  6.508849e-01   0.047282"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "corr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ccrcc\n",
      "BRAF\n",
      "FAT1\n",
      "IDH1\n",
      "TP53\n",
      "KMT2D\n",
      "PIK3CA\n",
      "PTEN\n",
      "SPTA1\n",
      "KRAS\n",
      "endometrial\n",
      "BRAF\n",
      "FAT1\n",
      "IDH1\n",
      "TP53\n",
      "KMT2D\n",
      "PIK3CA\n",
      "PTEN\n",
      "SPTA1\n",
      "KRAS\n",
      "luad\n",
      "BRAF\n",
      "FAT1\n",
      "IDH1\n",
      "TP53\n",
      "KMT2D\n",
      "PIK3CA\n",
      "PTEN\n",
      "SPTA1\n",
      "KRAS\n",
      "hnscc\n",
      "BRAF\n",
      "FAT1\n",
      "IDH1\n",
      "TP53\n",
      "KMT2D\n",
      "PIK3CA\n",
      "PTEN\n",
      "SPTA1\n",
      "KRAS\n",
      "lscc\n",
      "BRAF\n",
      "FAT1\n",
      "IDH1\n",
      "TP53\n",
      "KMT2D\n",
      "PIK3CA\n",
      "PTEN\n",
      "SPTA1\n",
      "KRAS\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>cancer</th>\n",
       "      <th>interaction_coeff</th>\n",
       "      <th>condition_coeff</th>\n",
       "      <th>transcript_coeff</th>\n",
       "      <th>intercept</th>\n",
       "      <th>interaction_pval</th>\n",
       "      <th>condition_pval</th>\n",
       "      <th>transcript_pval</th>\n",
       "      <th>intercept_pval</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.010412</td>\n",
       "      <td>-0.081675</td>\n",
       "      <td>0.004956</td>\n",
       "      <td>-0.113339</td>\n",
       "      <td>3.414814e-01</td>\n",
       "      <td>6.409939e-01</td>\n",
       "      <td>5.627178e-01</td>\n",
       "      <td>5.742203e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>-0.002134</td>\n",
       "      <td>0.743207</td>\n",
       "      <td>0.006258</td>\n",
       "      <td>-0.815901</td>\n",
       "      <td>4.017072e-01</td>\n",
       "      <td>2.601158e-04</td>\n",
       "      <td>1.696685e-02</td>\n",
       "      <td>3.223774e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>-0.002057</td>\n",
       "      <td>-0.414354</td>\n",
       "      <td>0.012565</td>\n",
       "      <td>-0.300409</td>\n",
       "      <td>3.414814e-01</td>\n",
       "      <td>1.174697e-05</td>\n",
       "      <td>3.436337e-21</td>\n",
       "      <td>1.087158e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.146506</td>\n",
       "      <td>-0.414724</td>\n",
       "      <td>-0.124654</td>\n",
       "      <td>0.441984</td>\n",
       "      <td>3.414814e-01</td>\n",
       "      <td>4.509170e-01</td>\n",
       "      <td>3.095905e-01</td>\n",
       "      <td>4.057605e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.008612</td>\n",
       "      <td>0.075793</td>\n",
       "      <td>-0.005563</td>\n",
       "      <td>0.083785</td>\n",
       "      <td>4.017072e-01</td>\n",
       "      <td>2.834865e-01</td>\n",
       "      <td>5.627178e-01</td>\n",
       "      <td>2.290060e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.019525</td>\n",
       "      <td>-0.230643</td>\n",
       "      <td>0.025151</td>\n",
       "      <td>-0.061899</td>\n",
       "      <td>4.017072e-01</td>\n",
       "      <td>2.049129e-01</td>\n",
       "      <td>3.095905e-01</td>\n",
       "      <td>6.624260e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>-0.044845</td>\n",
       "      <td>0.542478</td>\n",
       "      <td>0.054366</td>\n",
       "      <td>-0.890484</td>\n",
       "      <td>1.694057e-07</td>\n",
       "      <td>2.616658e-03</td>\n",
       "      <td>4.241792e-11</td>\n",
       "      <td>4.169225e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SPTA1</td>\n",
       "      <td>ccrcc</td>\n",
       "      <td>7.095579</td>\n",
       "      <td>1.207372</td>\n",
       "      <td>-5.901764</td>\n",
       "      <td>-1.172718</td>\n",
       "      <td>3.414814e-01</td>\n",
       "      <td>2.598936e-09</td>\n",
       "      <td>3.407768e-01</td>\n",
       "      <td>1.223506e-11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.560156</td>\n",
       "      <td>-6.634918</td>\n",
       "      <td>-0.028917</td>\n",
       "      <td>0.377898</td>\n",
       "      <td>1.273634e-01</td>\n",
       "      <td>1.334145e-01</td>\n",
       "      <td>9.358392e-01</td>\n",
       "      <td>9.337822e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.259647</td>\n",
       "      <td>-3.545581</td>\n",
       "      <td>0.110960</td>\n",
       "      <td>-1.627843</td>\n",
       "      <td>1.273634e-01</td>\n",
       "      <td>1.334145e-01</td>\n",
       "      <td>8.137769e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.696324</td>\n",
       "      <td>-8.376503</td>\n",
       "      <td>0.166922</td>\n",
       "      <td>-2.307084</td>\n",
       "      <td>9.313318e-07</td>\n",
       "      <td>1.794343e-07</td>\n",
       "      <td>8.137769e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.216767</td>\n",
       "      <td>-2.142691</td>\n",
       "      <td>-0.066085</td>\n",
       "      <td>0.693141</td>\n",
       "      <td>8.146880e-01</td>\n",
       "      <td>8.020032e-01</td>\n",
       "      <td>9.358392e-01</td>\n",
       "      <td>9.337822e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.175178</td>\n",
       "      <td>-2.140750</td>\n",
       "      <td>-0.103726</td>\n",
       "      <td>1.215829</td>\n",
       "      <td>5.889034e-01</td>\n",
       "      <td>5.274874e-01</td>\n",
       "      <td>8.137769e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.261513</td>\n",
       "      <td>-3.319234</td>\n",
       "      <td>0.250617</td>\n",
       "      <td>-2.509353</td>\n",
       "      <td>5.889034e-01</td>\n",
       "      <td>5.274874e-01</td>\n",
       "      <td>8.137769e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.254787</td>\n",
       "      <td>-3.880342</td>\n",
       "      <td>0.430141</td>\n",
       "      <td>-5.317317</td>\n",
       "      <td>8.001482e-01</td>\n",
       "      <td>6.160422e-01</td>\n",
       "      <td>8.137769e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SPTA1</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>-0.058152</td>\n",
       "      <td>-0.683825</td>\n",
       "      <td>-0.020472</td>\n",
       "      <td>0.787907</td>\n",
       "      <td>8.146880e-01</td>\n",
       "      <td>6.160422e-01</td>\n",
       "      <td>9.358392e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.164433</td>\n",
       "      <td>-1.799088</td>\n",
       "      <td>0.220494</td>\n",
       "      <td>-2.493825</td>\n",
       "      <td>8.146880e-01</td>\n",
       "      <td>7.417380e-01</td>\n",
       "      <td>8.137769e-01</td>\n",
       "      <td>6.780257e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>luad</td>\n",
       "      <td>0.685593</td>\n",
       "      <td>-3.528648</td>\n",
       "      <td>0.568099</td>\n",
       "      <td>-2.232003</td>\n",
       "      <td>2.026499e-01</td>\n",
       "      <td>1.137364e-01</td>\n",
       "      <td>2.726685e-01</td>\n",
       "      <td>3.180543e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>luad</td>\n",
       "      <td>0.534283</td>\n",
       "      <td>-3.346308</td>\n",
       "      <td>0.568181</td>\n",
       "      <td>-2.540912</td>\n",
       "      <td>2.995525e-02</td>\n",
       "      <td>2.808152e-03</td>\n",
       "      <td>2.198705e-02</td>\n",
       "      <td>1.135370e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>luad</td>\n",
       "      <td>0.076171</td>\n",
       "      <td>-0.542342</td>\n",
       "      <td>1.204356</td>\n",
       "      <td>-6.546108</td>\n",
       "      <td>7.428237e-01</td>\n",
       "      <td>6.319126e-01</td>\n",
       "      <td>5.001409e-15</td>\n",
       "      <td>8.765103e-17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>luad</td>\n",
       "      <td>3.764936</td>\n",
       "      <td>-9.532491</td>\n",
       "      <td>-1.428287</td>\n",
       "      <td>1.843437</td>\n",
       "      <td>1.862003e-02</td>\n",
       "      <td>1.223848e-02</td>\n",
       "      <td>2.849488e-01</td>\n",
       "      <td>6.065517e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>luad</td>\n",
       "      <td>0.490484</td>\n",
       "      <td>-1.219498</td>\n",
       "      <td>-0.099277</td>\n",
       "      <td>0.155863</td>\n",
       "      <td>9.132771e-03</td>\n",
       "      <td>7.419616e-03</td>\n",
       "      <td>3.981141e-01</td>\n",
       "      <td>6.065517e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>luad</td>\n",
       "      <td>-0.128660</td>\n",
       "      <td>-0.617585</td>\n",
       "      <td>0.809636</td>\n",
       "      <td>-1.681925</td>\n",
       "      <td>7.578645e-01</td>\n",
       "      <td>6.390444e-01</td>\n",
       "      <td>1.053326e-01</td>\n",
       "      <td>2.897705e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>luad</td>\n",
       "      <td>0.699612</td>\n",
       "      <td>-3.264382</td>\n",
       "      <td>0.426196</td>\n",
       "      <td>-1.487936</td>\n",
       "      <td>2.995525e-02</td>\n",
       "      <td>1.495423e-02</td>\n",
       "      <td>1.546128e-01</td>\n",
       "      <td>2.897705e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SPTA1</td>\n",
       "      <td>luad</td>\n",
       "      <td>-0.364259</td>\n",
       "      <td>-7.266318</td>\n",
       "      <td>0.247496</td>\n",
       "      <td>3.561006</td>\n",
       "      <td>2.060564e-01</td>\n",
       "      <td>7.444146e-10</td>\n",
       "      <td>2.849488e-01</td>\n",
       "      <td>1.214984e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>luad</td>\n",
       "      <td>1.507012</td>\n",
       "      <td>-5.700630</td>\n",
       "      <td>-1.058874</td>\n",
       "      <td>3.969742</td>\n",
       "      <td>3.821068e-02</td>\n",
       "      <td>1.183908e-02</td>\n",
       "      <td>1.546128e-01</td>\n",
       "      <td>8.194769e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.223519</td>\n",
       "      <td>-2.610744</td>\n",
       "      <td>0.220150</td>\n",
       "      <td>19.743516</td>\n",
       "      <td>3.158205e-01</td>\n",
       "      <td>3.820806e-01</td>\n",
       "      <td>5.215037e-01</td>\n",
       "      <td>1.708986e-16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.338782</td>\n",
       "      <td>-4.698292</td>\n",
       "      <td>0.120531</td>\n",
       "      <td>23.387686</td>\n",
       "      <td>4.034440e-03</td>\n",
       "      <td>4.580811e-03</td>\n",
       "      <td>5.215037e-01</td>\n",
       "      <td>1.590684e-37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.074228</td>\n",
       "      <td>-1.096204</td>\n",
       "      <td>0.536851</td>\n",
       "      <td>22.328785</td>\n",
       "      <td>4.015635e-01</td>\n",
       "      <td>3.820806e-01</td>\n",
       "      <td>2.243340e-11</td>\n",
       "      <td>1.108988e-55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.480006</td>\n",
       "      <td>-3.908124</td>\n",
       "      <td>0.303470</td>\n",
       "      <td>18.022086</td>\n",
       "      <td>2.926086e-01</td>\n",
       "      <td>3.820806e-01</td>\n",
       "      <td>5.814570e-01</td>\n",
       "      <td>2.599421e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.035466</td>\n",
       "      <td>-0.502427</td>\n",
       "      <td>0.049169</td>\n",
       "      <td>22.667491</td>\n",
       "      <td>6.271805e-01</td>\n",
       "      <td>6.198790e-01</td>\n",
       "      <td>5.814570e-01</td>\n",
       "      <td>1.955292e-64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.419034</td>\n",
       "      <td>-4.716726</td>\n",
       "      <td>0.014433</td>\n",
       "      <td>22.554205</td>\n",
       "      <td>1.368097e-04</td>\n",
       "      <td>1.373303e-04</td>\n",
       "      <td>8.714605e-01</td>\n",
       "      <td>2.052131e-49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.553132</td>\n",
       "      <td>-6.874431</td>\n",
       "      <td>-0.065336</td>\n",
       "      <td>24.444256</td>\n",
       "      <td>4.034440e-03</td>\n",
       "      <td>4.580811e-03</td>\n",
       "      <td>7.692865e-01</td>\n",
       "      <td>7.935639e-24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SPTA1</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>-0.375500</td>\n",
       "      <td>0.041738</td>\n",
       "      <td>0.464321</td>\n",
       "      <td>24.819005</td>\n",
       "      <td>3.239966e-02</td>\n",
       "      <td>9.378032e-01</td>\n",
       "      <td>4.620949e-03</td>\n",
       "      <td>3.478749e-94</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.248712</td>\n",
       "      <td>-3.048619</td>\n",
       "      <td>0.096946</td>\n",
       "      <td>24.851266</td>\n",
       "      <td>4.067777e-02</td>\n",
       "      <td>2.749830e-02</td>\n",
       "      <td>5.606188e-01</td>\n",
       "      <td>3.557979e-51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>lscc</td>\n",
       "      <td>-0.037316</td>\n",
       "      <td>0.354570</td>\n",
       "      <td>-0.079324</td>\n",
       "      <td>1.597180</td>\n",
       "      <td>9.007401e-01</td>\n",
       "      <td>9.467863e-01</td>\n",
       "      <td>8.267101e-01</td>\n",
       "      <td>8.213378e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>lscc</td>\n",
       "      <td>0.948409</td>\n",
       "      <td>-18.470329</td>\n",
       "      <td>0.661108</td>\n",
       "      <td>-12.911891</td>\n",
       "      <td>5.454881e-05</td>\n",
       "      <td>2.810194e-05</td>\n",
       "      <td>6.148023e-03</td>\n",
       "      <td>3.614429e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>IDH1</td>\n",
       "      <td>lscc</td>\n",
       "      <td>0.204197</td>\n",
       "      <td>-4.454225</td>\n",
       "      <td>1.079801</td>\n",
       "      <td>-20.289023</td>\n",
       "      <td>2.523383e-01</td>\n",
       "      <td>2.090197e-01</td>\n",
       "      <td>2.196804e-12</td>\n",
       "      <td>2.068574e-12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TP53</td>\n",
       "      <td>lscc</td>\n",
       "      <td>4.997480</td>\n",
       "      <td>-83.964923</td>\n",
       "      <td>-1.919483</td>\n",
       "      <td>29.552870</td>\n",
       "      <td>7.006034e-06</td>\n",
       "      <td>9.084214e-06</td>\n",
       "      <td>1.185113e-01</td>\n",
       "      <td>1.239996e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>lscc</td>\n",
       "      <td>0.289215</td>\n",
       "      <td>-5.000330</td>\n",
       "      <td>-0.033060</td>\n",
       "      <td>0.582962</td>\n",
       "      <td>1.209917e-01</td>\n",
       "      <td>1.365588e-01</td>\n",
       "      <td>8.267101e-01</td>\n",
       "      <td>8.213378e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>lscc</td>\n",
       "      <td>0.955611</td>\n",
       "      <td>-17.397305</td>\n",
       "      <td>0.365969</td>\n",
       "      <td>-6.346826</td>\n",
       "      <td>3.869694e-05</td>\n",
       "      <td>2.329491e-05</td>\n",
       "      <td>1.185113e-01</td>\n",
       "      <td>1.239996e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>lscc</td>\n",
       "      <td>1.120741</td>\n",
       "      <td>-20.944437</td>\n",
       "      <td>0.090312</td>\n",
       "      <td>-0.951894</td>\n",
       "      <td>3.869694e-05</td>\n",
       "      <td>2.713795e-05</td>\n",
       "      <td>8.267101e-01</td>\n",
       "      <td>8.213378e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SPTA1</td>\n",
       "      <td>lscc</td>\n",
       "      <td>-0.575340</td>\n",
       "      <td>-0.411296</td>\n",
       "      <td>0.127011</td>\n",
       "      <td>1.051816</td>\n",
       "      <td>1.095711e-01</td>\n",
       "      <td>9.467863e-01</td>\n",
       "      <td>8.267101e-01</td>\n",
       "      <td>8.213378e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>lscc</td>\n",
       "      <td>0.517899</td>\n",
       "      <td>-10.238431</td>\n",
       "      <td>0.374548</td>\n",
       "      <td>-6.427021</td>\n",
       "      <td>2.968179e-02</td>\n",
       "      <td>1.075511e-02</td>\n",
       "      <td>1.185113e-01</td>\n",
       "      <td>1.239996e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     gene       cancer  interaction_coeff  condition_coeff  transcript_coeff  \\\n",
       "0    BRAF        ccrcc           0.010412        -0.081675          0.004956   \n",
       "1    FAT1        ccrcc          -0.002134         0.743207          0.006258   \n",
       "2    IDH1        ccrcc          -0.002057        -0.414354          0.012565   \n",
       "3    TP53        ccrcc           0.146506        -0.414724         -0.124654   \n",
       "4   KMT2D        ccrcc           0.008612         0.075793         -0.005563   \n",
       "5  PIK3CA        ccrcc           0.019525        -0.230643          0.025151   \n",
       "6    PTEN        ccrcc          -0.044845         0.542478          0.054366   \n",
       "7   SPTA1        ccrcc           7.095579         1.207372         -5.901764   \n",
       "0    BRAF  endometrial           0.560156        -6.634918         -0.028917   \n",
       "1    FAT1  endometrial           0.259647        -3.545581          0.110960   \n",
       "2    IDH1  endometrial           0.696324        -8.376503          0.166922   \n",
       "3    TP53  endometrial           0.216767        -2.142691         -0.066085   \n",
       "4   KMT2D  endometrial           0.175178        -2.140750         -0.103726   \n",
       "5  PIK3CA  endometrial           0.261513        -3.319234          0.250617   \n",
       "6    PTEN  endometrial           0.254787        -3.880342          0.430141   \n",
       "7   SPTA1  endometrial          -0.058152        -0.683825         -0.020472   \n",
       "8    KRAS  endometrial           0.164433        -1.799088          0.220494   \n",
       "0    BRAF         luad           0.685593        -3.528648          0.568099   \n",
       "1    FAT1         luad           0.534283        -3.346308          0.568181   \n",
       "2    IDH1         luad           0.076171        -0.542342          1.204356   \n",
       "3    TP53         luad           3.764936        -9.532491         -1.428287   \n",
       "4   KMT2D         luad           0.490484        -1.219498         -0.099277   \n",
       "5  PIK3CA         luad          -0.128660        -0.617585          0.809636   \n",
       "6    PTEN         luad           0.699612        -3.264382          0.426196   \n",
       "7   SPTA1         luad          -0.364259        -7.266318          0.247496   \n",
       "8    KRAS         luad           1.507012        -5.700630         -1.058874   \n",
       "0    BRAF        hnscc           0.223519        -2.610744          0.220150   \n",
       "1    FAT1        hnscc           0.338782        -4.698292          0.120531   \n",
       "2    IDH1        hnscc           0.074228        -1.096204          0.536851   \n",
       "3    TP53        hnscc           0.480006        -3.908124          0.303470   \n",
       "4   KMT2D        hnscc           0.035466        -0.502427          0.049169   \n",
       "5  PIK3CA        hnscc           0.419034        -4.716726          0.014433   \n",
       "6    PTEN        hnscc           0.553132        -6.874431         -0.065336   \n",
       "7   SPTA1        hnscc          -0.375500         0.041738          0.464321   \n",
       "8    KRAS        hnscc           0.248712        -3.048619          0.096946   \n",
       "0    BRAF         lscc          -0.037316         0.354570         -0.079324   \n",
       "1    FAT1         lscc           0.948409       -18.470329          0.661108   \n",
       "2    IDH1         lscc           0.204197        -4.454225          1.079801   \n",
       "3    TP53         lscc           4.997480       -83.964923         -1.919483   \n",
       "4   KMT2D         lscc           0.289215        -5.000330         -0.033060   \n",
       "5  PIK3CA         lscc           0.955611       -17.397305          0.365969   \n",
       "6    PTEN         lscc           1.120741       -20.944437          0.090312   \n",
       "7   SPTA1         lscc          -0.575340        -0.411296          0.127011   \n",
       "8    KRAS         lscc           0.517899       -10.238431          0.374548   \n",
       "\n",
       "   intercept  interaction_pval  condition_pval  transcript_pval  \\\n",
       "0  -0.113339      3.414814e-01    6.409939e-01     5.627178e-01   \n",
       "1  -0.815901      4.017072e-01    2.601158e-04     1.696685e-02   \n",
       "2  -0.300409      3.414814e-01    1.174697e-05     3.436337e-21   \n",
       "3   0.441984      3.414814e-01    4.509170e-01     3.095905e-01   \n",
       "4   0.083785      4.017072e-01    2.834865e-01     5.627178e-01   \n",
       "5  -0.061899      4.017072e-01    2.049129e-01     3.095905e-01   \n",
       "6  -0.890484      1.694057e-07    2.616658e-03     4.241792e-11   \n",
       "7  -1.172718      3.414814e-01    2.598936e-09     3.407768e-01   \n",
       "0   0.377898      1.273634e-01    1.334145e-01     9.358392e-01   \n",
       "1  -1.627843      1.273634e-01    1.334145e-01     8.137769e-01   \n",
       "2  -2.307084      9.313318e-07    1.794343e-07     8.137769e-01   \n",
       "3   0.693141      8.146880e-01    8.020032e-01     9.358392e-01   \n",
       "4   1.215829      5.889034e-01    5.274874e-01     8.137769e-01   \n",
       "5  -2.509353      5.889034e-01    5.274874e-01     8.137769e-01   \n",
       "6  -5.317317      8.001482e-01    6.160422e-01     8.137769e-01   \n",
       "7   0.787907      8.146880e-01    6.160422e-01     9.358392e-01   \n",
       "8  -2.493825      8.146880e-01    7.417380e-01     8.137769e-01   \n",
       "0  -2.232003      2.026499e-01    1.137364e-01     2.726685e-01   \n",
       "1  -2.540912      2.995525e-02    2.808152e-03     2.198705e-02   \n",
       "2  -6.546108      7.428237e-01    6.319126e-01     5.001409e-15   \n",
       "3   1.843437      1.862003e-02    1.223848e-02     2.849488e-01   \n",
       "4   0.155863      9.132771e-03    7.419616e-03     3.981141e-01   \n",
       "5  -1.681925      7.578645e-01    6.390444e-01     1.053326e-01   \n",
       "6  -1.487936      2.995525e-02    1.495423e-02     1.546128e-01   \n",
       "7   3.561006      2.060564e-01    7.444146e-10     2.849488e-01   \n",
       "8   3.969742      3.821068e-02    1.183908e-02     1.546128e-01   \n",
       "0  19.743516      3.158205e-01    3.820806e-01     5.215037e-01   \n",
       "1  23.387686      4.034440e-03    4.580811e-03     5.215037e-01   \n",
       "2  22.328785      4.015635e-01    3.820806e-01     2.243340e-11   \n",
       "3  18.022086      2.926086e-01    3.820806e-01     5.814570e-01   \n",
       "4  22.667491      6.271805e-01    6.198790e-01     5.814570e-01   \n",
       "5  22.554205      1.368097e-04    1.373303e-04     8.714605e-01   \n",
       "6  24.444256      4.034440e-03    4.580811e-03     7.692865e-01   \n",
       "7  24.819005      3.239966e-02    9.378032e-01     4.620949e-03   \n",
       "8  24.851266      4.067777e-02    2.749830e-02     5.606188e-01   \n",
       "0   1.597180      9.007401e-01    9.467863e-01     8.267101e-01   \n",
       "1 -12.911891      5.454881e-05    2.810194e-05     6.148023e-03   \n",
       "2 -20.289023      2.523383e-01    2.090197e-01     2.196804e-12   \n",
       "3  29.552870      7.006034e-06    9.084214e-06     1.185113e-01   \n",
       "4   0.582962      1.209917e-01    1.365588e-01     8.267101e-01   \n",
       "5  -6.346826      3.869694e-05    2.329491e-05     1.185113e-01   \n",
       "6  -0.951894      3.869694e-05    2.713795e-05     8.267101e-01   \n",
       "7   1.051816      1.095711e-01    9.467863e-01     8.267101e-01   \n",
       "8  -6.427021      2.968179e-02    1.075511e-02     1.185113e-01   \n",
       "\n",
       "   intercept_pval  \n",
       "0    5.742203e-01  \n",
       "1    3.223774e-05  \n",
       "2    1.087158e-04  \n",
       "3    4.057605e-01  \n",
       "4    2.290060e-01  \n",
       "5    6.624260e-01  \n",
       "6    4.169225e-07  \n",
       "7    1.223506e-11  \n",
       "0    9.337822e-01  \n",
       "1    6.780257e-01  \n",
       "2    6.780257e-01  \n",
       "3    9.337822e-01  \n",
       "4    6.780257e-01  \n",
       "5    6.780257e-01  \n",
       "6    6.780257e-01  \n",
       "7    6.780257e-01  \n",
       "8    6.780257e-01  \n",
       "0    3.180543e-01  \n",
       "1    1.135370e-02  \n",
       "2    8.765103e-17  \n",
       "3    6.065517e-01  \n",
       "4    6.065517e-01  \n",
       "5    2.897705e-01  \n",
       "6    2.897705e-01  \n",
       "7    1.214984e-04  \n",
       "8    8.194769e-02  \n",
       "0    1.708986e-16  \n",
       "1    1.590684e-37  \n",
       "2    1.108988e-55  \n",
       "3    2.599421e-06  \n",
       "4    1.955292e-64  \n",
       "5    2.052131e-49  \n",
       "6    7.935639e-24  \n",
       "7    3.478749e-94  \n",
       "8    3.557979e-51  \n",
       "0    8.213378e-01  \n",
       "1    3.614429e-03  \n",
       "2    2.068574e-12  \n",
       "3    1.239996e-01  \n",
       "4    8.213378e-01  \n",
       "5    1.239996e-01  \n",
       "6    8.213378e-01  \n",
       "7    8.213378e-01  \n",
       "8    1.239996e-01  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_dfs = []\n",
    "for cancer in pd.unique(full_df.Cancer):\n",
    "    print(cancer)\n",
    "    rows = []\n",
    "    for gene in list(pd.unique(full_df.Gene)):\n",
    "        print(gene)\n",
    "        d = {}\n",
    "        df = full_df[full_df.Gene == gene]\n",
    "        df = df[df.Cancer == cancer]\n",
    "        if len(df) < 4:\n",
    "            continue\n",
    "        df = df[['Tissue', 'Proteomics', 'Transcriptomics']]\n",
    "        lm_df= linear_model(df, 'Transcriptomics', 'Proteomics', 'Tissue')\n",
    "        d['gene'] = gene\n",
    "        d['cancer'] = cancer\n",
    "        d['interaction_coeff'] = lm_df['Estimate'][3]\n",
    "        d['condition_coeff'] = lm_df['Estimate'][2]\n",
    "        d['transcript_coeff'] = lm_df['Estimate'][1]\n",
    "        d['intercept'] = lm_df['Estimate'][0]\n",
    "        d['interaction_pval'] = lm_df['Pr(>|t|)'][3]\n",
    "        d['condition_pval'] = lm_df['Pr(>|t|)'][2]\n",
    "        d['transcript_pval'] = lm_df['Pr(>|t|)'][1]\n",
    "        d['intercept_pval'] = lm_df['Pr(>|t|)'][0]\n",
    "        rows.append(d)\n",
    "    cancer_df = pd.DataFrame(rows)\n",
    "    for column in cancer_df:\n",
    "        if 'pval' in column:\n",
    "            old_pvals = list(cancer_df[column])\n",
    "            adj_pvals = list(ssm.fdrcorrection(old_pvals)[1])\n",
    "            cancer_df[column] = adj_pvals    \n",
    "    cancer_dfs.append(cancer_df)\n",
    "            \n",
    "lm_df = pd.concat(cancer_dfs)\n",
    "lm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'get_correlations'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-16-26e699618ea6>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m#CODE IMPORTED FROM FIGURE 10 GENE CORRELATION CHANGES\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      2\u001b[0m \u001b[1;31m#Make docstrings for what is required\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 3\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mget_correlations\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      4\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0mget_single_gene_correlations\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mgene\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcancer_type_list\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mtype_dict\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0minput_tissue_type\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;34m\"tumor\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m     \u001b[0mcorrelation_list\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mModuleNotFoundError\u001b[0m: No module named 'get_correlations'"
     ]
    }
   ],
   "source": [
    "#CODE IMPORTED FROM FIGURE 10 GENE CORRELATION CHANGES\n",
    "#Make docstrings for what is required\n",
    "import get_correlations\n",
    "def get_single_gene_correlations(gene, cancer_type_list,type_dict,input_tissue_type = \"tumor\"):\n",
    "    correlation_list = []\n",
    "    type_list = []\n",
    "    pval_list = []\n",
    "    for cancer in cancer_type_list:\n",
    "        correlation,pval,num_samples = get_correlations.correlation_list(cancer, [gene],tissue_type = input_tissue_type)\n",
    "        if len(correlation) == 0:\n",
    "            continue\n",
    "        else:\n",
    "            correlation_list.append(correlation[0])\n",
    "            pval_list.append(pval[0][1])\n",
    "            type_list.append(type_dict[cancer])\n",
    "    return correlation_list, type_list, num_samples, pval_list\n",
    "\n",
    "def make_df_with_genes(gene_list,cancer_list,name_dict,tissue_type):\n",
    "    all_genes = []\n",
    "    all_correlations = []\n",
    "    all_cancer_types = []\n",
    "    all_pvals = []\n",
    "    for gene in gene_list:\n",
    "        gene_corr_list,type_list,num_samples, pval_list = get_single_gene_correlations(gene,cancer_list,name_dict,tissue_type)\n",
    "        gene_x,gene_y = get_x_y(gene_corr_list)\n",
    "        all_genes.extend(gene_x)\n",
    "        all_correlations.extend(gene_y)\n",
    "        all_cancer_types.extend(type_list)\n",
    "        all_pvals.extend(pval_list)\n",
    "    df = pd.DataFrame(list(zip(all_genes,all_correlations,all_cancer_types, all_pvals)), columns =['Gene', 'Correlation','Cancer', 'p-val'])\n",
    "    return(df)\n",
    "\n",
    "def get_x_y(correlation_list):\n",
    "    x = []\n",
    "    y = []\n",
    "    for item in correlation_list:\n",
    "        x.append(item[0])\n",
    "        y.append(item[1])\n",
    "    return x,y\n",
    "\n",
    "type_dict = {ccrcc:\"ccrcc\",en:\"endometrial\",luad:\"luad\", hnscc:\"hnscc\",lscc:\"lscc\"}\n",
    "\n",
    "# get delta correlation p-values\n",
    "\n",
    "tumor_df = make_df_with_genes(cancer_genes,cancer_list,type_dict,tissue_type = \"tumor\")\n",
    "tumor_df['Tissue'] = ['Tumor'] * len(tumor_df)\n",
    "normal_df = make_df_with_genes(cancer_genes,cancer_list,type_dict,tissue_type = \"normal\")\n",
    "normal_df['Tissue'] = ['Normal'] * len(normal_df)\n",
    "\n",
    "\n",
    "delta_corr_pvals_df = pd.read_csv('data/p_val.csv')\n",
    "combined_df = pd.merge(normal_df, tumor_df, how = 'outer')\n",
    "combined_df['Significant Correlation'] = combined_df['p-val'] <= 0.05\n",
    "delta_corr_pvals = []\n",
    "for index, row in combined_df.iterrows():\n",
    "    cancer = row['Cancer']\n",
    "    gene = row['Gene']\n",
    "    pval = delta_corr_pvals_df[delta_corr_pvals_df.Cancer == cancer].reset_index()[gene][0]\n",
    "    delta_corr_pvals.append(pval)\n",
    "combined_df['delta_corr_pval'] = delta_corr_pvals\n",
    "combined_df['Significant Delta Correlation'] = combined_df['delta_corr_pval'] <= 0.05\n",
    "combined_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
