{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import statsmodels.stats.multitest as ssm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.146442</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>0.095512</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.532265</td>\n",
       "      <td>0.851988</td>\n",
       "      <td>1.691592e-01</td>\n",
       "      <td>0.545156</td>\n",
       "      <td>0.156638</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.098461e-01</td>\n",
       "      <td>1.010117e-01</td>\n",
       "      <td>3.231642e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.104305</td>\n",
       "      <td>7.044145e-01</td>\n",
       "      <td>2.069261e-01</td>\n",
       "      <td>5.672603e-01</td>\n",
       "      <td>0.378694</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>en</td>\n",
       "      <td>0.883318</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.953553</td>\n",
       "      <td>5.628600e-02</td>\n",
       "      <td>0.543085</td>\n",
       "      <td>0.027277</td>\n",
       "      <td>1.438315e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.517563</td>\n",
       "      <td>...</td>\n",
       "      <td>0.769929</td>\n",
       "      <td>8.608802e-01</td>\n",
       "      <td>8.894419e-05</td>\n",
       "      <td>1.330422e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.491506</td>\n",
       "      <td>2.125946e-01</td>\n",
       "      <td>6.985013e-01</td>\n",
       "      <td>4.283120e-01</td>\n",
       "      <td>0.383779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>luad</td>\n",
       "      <td>0.801031</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019206</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.191032</td>\n",
       "      <td>5.371135e-35</td>\n",
       "      <td>0.037629</td>\n",
       "      <td>0.029363</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.594465e-09</td>\n",
       "      <td>7.619459e-19</td>\n",
       "      <td>4.915995e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.641317</td>\n",
       "      <td>7.220021e-10</td>\n",
       "      <td>2.219300e-07</td>\n",
       "      <td>2.915594e-14</td>\n",
       "      <td>0.000043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.998985</td>\n",
       "      <td>0.862365</td>\n",
       "      <td>0.381438</td>\n",
       "      <td>3.371964e-03</td>\n",
       "      <td>0.103999</td>\n",
       "      <td>0.046917</td>\n",
       "      <td>8.900750e-01</td>\n",
       "      <td>0.001904</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.904316e-03</td>\n",
       "      <td>2.797887e-01</td>\n",
       "      <td>7.448440e-03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012482</td>\n",
       "      <td>4.719306e-03</td>\n",
       "      <td>9.340559e-01</td>\n",
       "      <td>2.842001e-01</td>\n",
       "      <td>0.854269</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>lscc</td>\n",
       "      <td>0.183336</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.133999</td>\n",
       "      <td>2.266058e-21</td>\n",
       "      <td>0.202066</td>\n",
       "      <td>0.014338</td>\n",
       "      <td>1.012628e-01</td>\n",
       "      <td>0.522300</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002938</td>\n",
       "      <td>1.262976e-12</td>\n",
       "      <td>2.357531e-49</td>\n",
       "      <td>5.104002e-32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.020118</td>\n",
       "      <td>1.345127e-02</td>\n",
       "      <td>1.587683e-08</td>\n",
       "      <td>6.037113e-06</td>\n",
       "      <td>0.030463</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13670 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Cancer      A1BG      A1CF       A2M         A2ML1    A4GALT      AAAS  \\\n",
       "0  ccrcc  0.146442  0.004995  0.095512           NaN  0.532265  0.851988   \n",
       "1     en  0.883318       NaN  0.953553  5.628600e-02  0.543085  0.027277   \n",
       "2   luad  0.801031       NaN  0.019206           NaN       NaN  0.191032   \n",
       "3  hnscc  0.998985  0.862365  0.381438  3.371964e-03  0.103999  0.046917   \n",
       "4   lscc  0.183336       NaN  0.133999  2.266058e-21  0.202066  0.014338   \n",
       "\n",
       "           AACS     AADAC     AADAT  ...    ZSWIM9          ZW10  \\\n",
       "0  1.691592e-01  0.545156  0.156638  ...       NaN  3.098461e-01   \n",
       "1  1.438315e-07       NaN  0.517563  ...  0.769929  8.608802e-01   \n",
       "2  5.371135e-35  0.037629  0.029363  ...       NaN  2.594465e-09   \n",
       "3  8.900750e-01  0.001904       NaN  ...       NaN  3.904316e-03   \n",
       "4  1.012628e-01  0.522300       NaN  ...  0.002938  1.262976e-12   \n",
       "\n",
       "         ZWILCH         ZWINT  ZXDA      ZXDC        ZYG11B           ZYX  \\\n",
       "0  1.010117e-01  3.231642e-01   NaN  0.104305  7.044145e-01  2.069261e-01   \n",
       "1  8.894419e-05  1.330422e-07   NaN  0.491506  2.125946e-01  6.985013e-01   \n",
       "2  7.619459e-19  4.915995e-05   NaN  0.641317  7.220021e-10  2.219300e-07   \n",
       "3  2.797887e-01  7.448440e-03   NaN  0.012482  4.719306e-03  9.340559e-01   \n",
       "4  2.357531e-49  5.104002e-32   NaN  0.020118  1.345127e-02  1.587683e-08   \n",
       "\n",
       "          ZZEF1      ZZZ3  \n",
       "0  5.672603e-01  0.378694  \n",
       "1  4.283120e-01  0.383779  \n",
       "2  2.915594e-14  0.000043  \n",
       "3  2.842001e-01  0.854269  \n",
       "4  6.037113e-06  0.030463  \n",
       "\n",
       "[5 rows x 13670 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "perm_df = pd.read_csv(\"data/full_10k_permutation_corrected.csv\")\n",
    "perm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>-0.388745</td>\n",
       "      <td>0.146993</td>\n",
       "      <td>-0.277035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.795957</td>\n",
       "      <td>-0.044291</td>\n",
       "      <td>-0.128218</td>\n",
       "      <td>1.196756</td>\n",
       "      <td>-0.174925</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.269728</td>\n",
       "      <td>0.329727</td>\n",
       "      <td>-0.337301</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.337470</td>\n",
       "      <td>0.109414</td>\n",
       "      <td>-0.223934</td>\n",
       "      <td>-0.138352</td>\n",
       "      <td>-0.220727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.061654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.025535</td>\n",
       "      <td>0.402210</td>\n",
       "      <td>0.399061</td>\n",
       "      <td>0.663703</td>\n",
       "      <td>0.593777</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.302879</td>\n",
       "      <td>...</td>\n",
       "      <td>0.177842</td>\n",
       "      <td>0.088061</td>\n",
       "      <td>0.759429</td>\n",
       "      <td>1.006282</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.267078</td>\n",
       "      <td>0.400272</td>\n",
       "      <td>0.147305</td>\n",
       "      <td>0.367791</td>\n",
       "      <td>0.342423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>luad</td>\n",
       "      <td>-0.039149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.097233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.206094</td>\n",
       "      <td>0.683092</td>\n",
       "      <td>-0.119746</td>\n",
       "      <td>0.630155</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.557832</td>\n",
       "      <td>0.904669</td>\n",
       "      <td>0.645076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.087794</td>\n",
       "      <td>0.707991</td>\n",
       "      <td>0.472139</td>\n",
       "      <td>0.418589</td>\n",
       "      <td>0.661323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>-0.002651</td>\n",
       "      <td>0.407295</td>\n",
       "      <td>-0.194186</td>\n",
       "      <td>0.117838</td>\n",
       "      <td>0.555675</td>\n",
       "      <td>0.462864</td>\n",
       "      <td>0.024965</td>\n",
       "      <td>-0.333934</td>\n",
       "      <td>-0.294627</td>\n",
       "      <td>...</td>\n",
       "      <td>1.039965</td>\n",
       "      <td>0.395189</td>\n",
       "      <td>0.119698</td>\n",
       "      <td>1.079145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.591841</td>\n",
       "      <td>-0.453269</td>\n",
       "      <td>-0.010672</td>\n",
       "      <td>0.255505</td>\n",
       "      <td>-0.060708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>lscc</td>\n",
       "      <td>0.196946</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.062630</td>\n",
       "      <td>0.649123</td>\n",
       "      <td>0.246831</td>\n",
       "      <td>0.304418</td>\n",
       "      <td>0.150665</td>\n",
       "      <td>-0.068258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.613891</td>\n",
       "      <td>0.491043</td>\n",
       "      <td>0.646430</td>\n",
       "      <td>0.838981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.332350</td>\n",
       "      <td>0.231167</td>\n",
       "      <td>0.315679</td>\n",
       "      <td>0.268239</td>\n",
       "      <td>0.446553</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13670 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Cancer      A1BG      A1CF       A2M     A2ML1    A4GALT      AAAS  \\\n",
       "0        ccrcc -0.388745  0.146993 -0.277035       NaN -0.795957 -0.044291   \n",
       "1  endometrial  0.061654       NaN -0.025535  0.402210  0.399061  0.663703   \n",
       "2         luad -0.039149       NaN  0.097233       NaN       NaN  0.206094   \n",
       "3        hnscc -0.002651  0.407295 -0.194186  0.117838  0.555675  0.462864   \n",
       "4         lscc  0.196946       NaN -0.062630  0.649123  0.246831  0.304418   \n",
       "\n",
       "       AACS     AADAC     AADAT  ...    ZSWIM9      ZW10    ZWILCH     ZWINT  \\\n",
       "0 -0.128218  1.196756 -0.174925  ...       NaN  0.269728  0.329727 -0.337301   \n",
       "1  0.593777       NaN  0.302879  ...  0.177842  0.088061  0.759429  1.006282   \n",
       "2  0.683092 -0.119746  0.630155  ...       NaN  0.557832  0.904669  0.645076   \n",
       "3  0.024965 -0.333934 -0.294627  ...  1.039965  0.395189  0.119698  1.079145   \n",
       "4  0.150665 -0.068258       NaN  ...  0.613891  0.491043  0.646430  0.838981   \n",
       "\n",
       "   ZXDA      ZXDC    ZYG11B       ZYX     ZZEF1      ZZZ3  \n",
       "0   NaN  0.337470  0.109414 -0.223934 -0.138352 -0.220727  \n",
       "1   NaN -0.267078  0.400272  0.147305  0.367791  0.342423  \n",
       "2   NaN  0.087794  0.707991  0.472139  0.418589  0.661323  \n",
       "3   NaN  0.591841 -0.453269 -0.010672  0.255505 -0.060708  \n",
       "4   NaN  0.332350  0.231167  0.315679  0.268239  0.446553  \n",
       "\n",
       "[5 rows x 13670 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_corr_df = pd.read_csv('data/corr_diff.csv')\n",
    "delta_corr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "perm_df = perm_df.melt(id_vars='Cancer', var_name = 'Gene', value_name = 'pval')\n",
    "delta_corr_df = delta_corr_df.melt(id_vars='Cancer', var_name='Gene', value_name='delta_corr')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Gene</th>\n",
       "      <th>pval</th>\n",
       "      <th>delta_corr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.146442</td>\n",
       "      <td>-0.388745</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>luad</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.801031</td>\n",
       "      <td>-0.039149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.998985</td>\n",
       "      <td>-0.002651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>lscc</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.183336</td>\n",
       "      <td>0.196946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.004995</td>\n",
       "      <td>0.146993</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54671</th>\n",
       "      <td>lscc</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>0.268239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54672</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.378694</td>\n",
       "      <td>-0.220727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54673</th>\n",
       "      <td>luad</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>0.661323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54674</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.854269</td>\n",
       "      <td>-0.060708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54675</th>\n",
       "      <td>lscc</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.030463</td>\n",
       "      <td>0.446553</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43001 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Cancer   Gene      pval  delta_corr\n",
       "0      ccrcc   A1BG  0.146442   -0.388745\n",
       "1       luad   A1BG  0.801031   -0.039149\n",
       "2      hnscc   A1BG  0.998985   -0.002651\n",
       "3       lscc   A1BG  0.183336    0.196946\n",
       "4      ccrcc   A1CF  0.004995    0.146993\n",
       "...      ...    ...       ...         ...\n",
       "54671   lscc  ZZEF1  0.000006    0.268239\n",
       "54672  ccrcc   ZZZ3  0.378694   -0.220727\n",
       "54673   luad   ZZZ3  0.000043    0.661323\n",
       "54674  hnscc   ZZZ3  0.854269   -0.060708\n",
       "54675   lscc   ZZZ3  0.030463    0.446553\n",
       "\n",
       "[43001 rows x 4 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.merge(perm_df, delta_corr_df)\n",
    "df = df.dropna()\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "background_genes = list(pd.unique(df.Gene))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_perm_df = df[df.pval <= 0.05]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>avg_pval</th>\n",
       "      <th>avg_delta_corr</th>\n",
       "      <th>abs_delta_corr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>1.191913e-05</td>\n",
       "      <td>0.513239</td>\n",
       "      <td>0.513239</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>2.187208e-09</td>\n",
       "      <td>0.539893</td>\n",
       "      <td>0.539893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAMP</td>\n",
       "      <td>7.805338e-03</td>\n",
       "      <td>0.422332</td>\n",
       "      <td>0.422332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AARS</td>\n",
       "      <td>1.022777e-02</td>\n",
       "      <td>0.443220</td>\n",
       "      <td>0.443220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>1.766085e-05</td>\n",
       "      <td>0.563538</td>\n",
       "      <td>0.563538</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3044</th>\n",
       "      <td>ZPR1</td>\n",
       "      <td>4.507710e-03</td>\n",
       "      <td>0.552338</td>\n",
       "      <td>0.552338</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3045</th>\n",
       "      <td>ZSCAN18</td>\n",
       "      <td>9.995611e-03</td>\n",
       "      <td>0.461501</td>\n",
       "      <td>0.461501</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3046</th>\n",
       "      <td>ZW10</td>\n",
       "      <td>1.301439e-03</td>\n",
       "      <td>0.481355</td>\n",
       "      <td>0.481355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3047</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>2.499200e-03</td>\n",
       "      <td>0.854401</td>\n",
       "      <td>0.854401</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3048</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>6.056860e-03</td>\n",
       "      <td>0.161963</td>\n",
       "      <td>0.161963</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3049 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene      avg_pval  avg_delta_corr  abs_delta_corr\n",
       "0       AAGAB  1.191913e-05        0.513239        0.513239\n",
       "1        AAK1  2.187208e-09        0.539893        0.539893\n",
       "2        AAMP  7.805338e-03        0.422332        0.422332\n",
       "3        AARS  1.022777e-02        0.443220        0.443220\n",
       "4       AARS2  1.766085e-05        0.563538        0.563538\n",
       "...       ...           ...             ...             ...\n",
       "3044     ZPR1  4.507710e-03        0.552338        0.552338\n",
       "3045  ZSCAN18  9.995611e-03        0.461501        0.461501\n",
       "3046     ZW10  1.301439e-03        0.481355        0.481355\n",
       "3047    ZWINT  2.499200e-03        0.854401        0.854401\n",
       "3048   ZYG11B  6.056860e-03        0.161963        0.161963\n",
       "\n",
       "[3049 rows x 4 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_genes = pd.DataFrame(sig_perm_df.groupby('Gene').size())\n",
    "sig_genes = sig_genes[sig_genes[0] > 2]\n",
    "sig_genes = list(sig_genes.index)\n",
    "avg_pvals = []\n",
    "avg_corrs = []\n",
    "abs_avg_corrs = []\n",
    "for gene in sig_genes:\n",
    "    gene_df = sig_perm_df[sig_perm_df.Gene==gene]\n",
    "    avg_pval = np.average(gene_df.pval)\n",
    "    avg_corr = np.average(gene_df.delta_corr)\n",
    "    avg_corrs.append(avg_corr)\n",
    "    avg_pvals.append(avg_pval)\n",
    "    abs_avg_corrs.append(abs(avg_corr))\n",
    "sig_genes = pd.DataFrame({'Gene': sig_genes, 'avg_pval': avg_pvals, 'avg_delta_corr': avg_corrs, 'abs_delta_corr': abs_avg_corrs})\n",
    "sig_genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "sig_genes = sig_genes.sort_values('avg_pval')\n",
    "pval_ranked_gene_list = list(sig_genes.Gene)\n",
    "sig_genes = sig_genes.sort_values('abs_delta_corr', ascending = False)\n",
    "delta_corr_ranked_gene_list = list(sig_genes.Gene)\n",
    "down_reg = sig_genes[sig_genes.avg_delta_corr < 0]\n",
    "down_reg = down_reg.sort_values('avg_delta_corr')\n",
    "down_reg_ranked_list = list(down_reg.Gene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from gprofiler import GProfiler\n",
    "gp = GProfiler(return_dataframe=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "pval_results_df = gp.profile(organism='hsapiens', query = pval_ranked_gene_list, no_iea=True,\n",
    "                       ordered=True, no_evidences=False, background= background_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "delta_corr_results_df = gp.profile(organism='hsapiens', query = delta_corr_ranked_gene_list, no_iea=True,\n",
    "                       ordered=True, no_evidences=False, background= background_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "down_reg_results_df = gp.profile(organism='hsapiens', query = down_reg_ranked_list, no_iea=True,\n",
    "                       ordered=True, no_evidences=False, background= background_genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "pval_filtered_results = pval_results_df[pval_results_df.term_size <= 500]\n",
    "pval_filtered_results = pval_filtered_results[pval_filtered_results.term_size >=5]\n",
    "delta_corr_filtered_results = delta_corr_results_df[delta_corr_results_df.term_size <= 500]\n",
    "delta_corr_filtered_results = delta_corr_filtered_results[delta_corr_filtered_results.term_size >=5]\n",
    "down_reg_filtered_results = down_reg_results_df[down_reg_results_df.term_size <= 500]\n",
    "down_reg_filtered_results = down_reg_filtered_results[down_reg_filtered_results.term_size >=5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00280</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>0.000101</td>\n",
       "      <td>True</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>48</td>\n",
       "      <td>2397</td>\n",
       "      <td>24</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.010013</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ALDH6A1, OXCT1, ACAT1, IL4I1, ALDH7A1, MUT, D...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01200</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>0.000552</td>\n",
       "      <td>True</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>112</td>\n",
       "      <td>2596</td>\n",
       "      <td>45</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.017334</td>\n",
       "      <td>0.401786</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ALDH6A1, ALDOC, ACAT1, PKM, RPIA, PFKM, RGN, ...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00240</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>0.001212</td>\n",
       "      <td>True</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>51</td>\n",
       "      <td>2738</td>\n",
       "      <td>27</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.009861</td>\n",
       "      <td>0.529412</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[CAD, ENTPD1, TK1, NME1, UPP1, TYMS, CMPK2, CA...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01230</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>0.003258</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>72</td>\n",
       "      <td>2757</td>\n",
       "      <td>32</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011607</td>\n",
       "      <td>0.444444</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ALDOC, PKM, RPIA, PFKM, ASS1, PGAM1, PFKP, EN...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01240</td>\n",
       "      <td>Biosynthesis of cofactors</td>\n",
       "      <td>0.004217</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of cofactors</td>\n",
       "      <td>136</td>\n",
       "      <td>2808</td>\n",
       "      <td>56</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.019943</td>\n",
       "      <td>0.411765</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[FLAD1, TPK1, GCLC, AK4, CAD, ALPL, PKM, NADK2...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00270</td>\n",
       "      <td>Cysteine and methionine metabolism</td>\n",
       "      <td>0.004425</td>\n",
       "      <td>True</td>\n",
       "      <td>Cysteine and methionine metabolism</td>\n",
       "      <td>46</td>\n",
       "      <td>1564</td>\n",
       "      <td>17</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.010870</td>\n",
       "      <td>0.369565</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[GCLC, SMS, MTAP, IL4I1, TST, MPST, DNMT3A, LD...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00920</td>\n",
       "      <td>Sulfur metabolism</td>\n",
       "      <td>0.005339</td>\n",
       "      <td>True</td>\n",
       "      <td>Sulfur metabolism</td>\n",
       "      <td>10</td>\n",
       "      <td>1618</td>\n",
       "      <td>7</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.004326</td>\n",
       "      <td>0.700000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[TST, MPST, PAPSS2, SUOX, SELENBP1, ETHE1, IMP...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>307</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00010</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "      <td>0.018440</td>\n",
       "      <td>True</td>\n",
       "      <td>Glycolysis / Gluconeogenesis</td>\n",
       "      <td>62</td>\n",
       "      <td>565</td>\n",
       "      <td>11</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.019469</td>\n",
       "      <td>0.177419</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ALDOC, PKM, PFKM, ALDH7A1, PGAM1, PFKP, LDHB,...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>316</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03030</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>0.022233</td>\n",
       "      <td>True</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>34</td>\n",
       "      <td>605</td>\n",
       "      <td>8</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.013223</td>\n",
       "      <td>0.235294</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[RFC5, PRIM1, MCM5, MCM6, POLA2, LIG1, POLD2, ...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>317</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00640</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>0.022739</td>\n",
       "      <td>True</td>\n",
       "      <td>Propanoate metabolism</td>\n",
       "      <td>34</td>\n",
       "      <td>2725</td>\n",
       "      <td>17</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.006239</td>\n",
       "      <td>0.500000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[ALDH6A1, ACAT1, LDHB, LDHA, MUT, DBT, ABAT, A...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>318</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00740</td>\n",
       "      <td>Riboflavin metabolism</td>\n",
       "      <td>0.023263</td>\n",
       "      <td>True</td>\n",
       "      <td>Riboflavin metabolism</td>\n",
       "      <td>8</td>\n",
       "      <td>33</td>\n",
       "      <td>2</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.060606</td>\n",
       "      <td>0.250000</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[FLAD1, ACP5]</td>\n",
       "      <td>[[KEGG], [KEGG]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>333</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00250</td>\n",
       "      <td>Alanine, aspartate and glutamate metabolism</td>\n",
       "      <td>0.032579</td>\n",
       "      <td>True</td>\n",
       "      <td>Alanine, aspartate and glutamate metabolism</td>\n",
       "      <td>32</td>\n",
       "      <td>951</td>\n",
       "      <td>10</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.010515</td>\n",
       "      <td>0.312500</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[CAD, PPAT, IL4I1, ASS1, GFPT1, GPT2, ALDH5A1,...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>337</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:04110</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>0.035798</td>\n",
       "      <td>True</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>111</td>\n",
       "      <td>494</td>\n",
       "      <td>14</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.028340</td>\n",
       "      <td>0.126126</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[CHEK2, CCNA2, CDK2, CDK6, MCM5, SMAD4, TP53, ...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    source      native                                         name   p_value  \\\n",
       "219   KEGG  KEGG:00280   Valine, leucine and isoleucine degradation  0.000101   \n",
       "232   KEGG  KEGG:01200                            Carbon metabolism  0.000552   \n",
       "240   KEGG  KEGG:00240                        Pyrimidine metabolism  0.001212   \n",
       "255   KEGG  KEGG:01230                  Biosynthesis of amino acids  0.003258   \n",
       "260   KEGG  KEGG:01240                    Biosynthesis of cofactors  0.004217   \n",
       "261   KEGG  KEGG:00270           Cysteine and methionine metabolism  0.004425   \n",
       "265   KEGG  KEGG:00920                            Sulfur metabolism  0.005339   \n",
       "307   KEGG  KEGG:00010                 Glycolysis / Gluconeogenesis  0.018440   \n",
       "316   KEGG  KEGG:03030                              DNA replication  0.022233   \n",
       "317   KEGG  KEGG:00640                        Propanoate metabolism  0.022739   \n",
       "318   KEGG  KEGG:00740                        Riboflavin metabolism  0.023263   \n",
       "333   KEGG  KEGG:00250  Alanine, aspartate and glutamate metabolism  0.032579   \n",
       "337   KEGG  KEGG:04110                                   Cell cycle  0.035798   \n",
       "\n",
       "     significant                                  description  term_size  \\\n",
       "219         True   Valine, leucine and isoleucine degradation         48   \n",
       "232         True                            Carbon metabolism        112   \n",
       "240         True                        Pyrimidine metabolism         51   \n",
       "255         True                  Biosynthesis of amino acids         72   \n",
       "260         True                    Biosynthesis of cofactors        136   \n",
       "261         True           Cysteine and methionine metabolism         46   \n",
       "265         True                            Sulfur metabolism         10   \n",
       "307         True                 Glycolysis / Gluconeogenesis         62   \n",
       "316         True                              DNA replication         34   \n",
       "317         True                        Propanoate metabolism         34   \n",
       "318         True                        Riboflavin metabolism          8   \n",
       "333         True  Alanine, aspartate and glutamate metabolism         32   \n",
       "337         True                                   Cell cycle        111   \n",
       "\n",
       "     query_size  intersection_size  effective_domain_size  precision  \\\n",
       "219        2397                 24                  12939   0.010013   \n",
       "232        2596                 45                  12939   0.017334   \n",
       "240        2738                 27                  12939   0.009861   \n",
       "255        2757                 32                  12939   0.011607   \n",
       "260        2808                 56                  12939   0.019943   \n",
       "261        1564                 17                  12939   0.010870   \n",
       "265        1618                  7                  12939   0.004326   \n",
       "307         565                 11                  12939   0.019469   \n",
       "316         605                  8                  12939   0.013223   \n",
       "317        2725                 17                  12939   0.006239   \n",
       "318          33                  2                  12939   0.060606   \n",
       "333         951                 10                  12939   0.010515   \n",
       "337         494                 14                  12939   0.028340   \n",
       "\n",
       "       recall    query       parents  \\\n",
       "219  0.500000  query_1  [KEGG:00000]   \n",
       "232  0.401786  query_1  [KEGG:00000]   \n",
       "240  0.529412  query_1  [KEGG:00000]   \n",
       "255  0.444444  query_1  [KEGG:00000]   \n",
       "260  0.411765  query_1  [KEGG:00000]   \n",
       "261  0.369565  query_1  [KEGG:00000]   \n",
       "265  0.700000  query_1  [KEGG:00000]   \n",
       "307  0.177419  query_1  [KEGG:00000]   \n",
       "316  0.235294  query_1  [KEGG:00000]   \n",
       "317  0.500000  query_1  [KEGG:00000]   \n",
       "318  0.250000  query_1  [KEGG:00000]   \n",
       "333  0.312500  query_1  [KEGG:00000]   \n",
       "337  0.126126  query_1  [KEGG:00000]   \n",
       "\n",
       "                                         intersections  \\\n",
       "219  [ALDH6A1, OXCT1, ACAT1, IL4I1, ALDH7A1, MUT, D...   \n",
       "232  [ALDH6A1, ALDOC, ACAT1, PKM, RPIA, PFKM, RGN, ...   \n",
       "240  [CAD, ENTPD1, TK1, NME1, UPP1, TYMS, CMPK2, CA...   \n",
       "255  [ALDOC, PKM, RPIA, PFKM, ASS1, PGAM1, PFKP, EN...   \n",
       "260  [FLAD1, TPK1, GCLC, AK4, CAD, ALPL, PKM, NADK2...   \n",
       "261  [GCLC, SMS, MTAP, IL4I1, TST, MPST, DNMT3A, LD...   \n",
       "265  [TST, MPST, PAPSS2, SUOX, SELENBP1, ETHE1, IMP...   \n",
       "307  [ALDOC, PKM, PFKM, ALDH7A1, PGAM1, PFKP, LDHB,...   \n",
       "316  [RFC5, PRIM1, MCM5, MCM6, POLA2, LIG1, POLD2, ...   \n",
       "317  [ALDH6A1, ACAT1, LDHB, LDHA, MUT, DBT, ABAT, A...   \n",
       "318                                      [FLAD1, ACP5]   \n",
       "333  [CAD, PPAT, IL4I1, ASS1, GFPT1, GPT2, ALDH5A1,...   \n",
       "337  [CHEK2, CCNA2, CDK2, CDK6, MCM5, SMAD4, TP53, ...   \n",
       "\n",
       "                                             evidences  \n",
       "219  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "232  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "240  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "255  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "260  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "261  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "265  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "307  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "316  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "317  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "318                                   [[KEGG], [KEGG]]  \n",
       "333  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "337  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pval_kegg_results = pval_filtered_results[pval_filtered_results.source == 'KEGG']\n",
    "pval_kegg_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03030</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>34</td>\n",
       "      <td>1072</td>\n",
       "      <td>15</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.013993</td>\n",
       "      <td>0.441176</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[RFC5, PRIM1, MCM7, MCM5, MCM2, MCM4, RFC1, PO...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03430</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>True</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>22</td>\n",
       "      <td>1072</td>\n",
       "      <td>12</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011194</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[PMS2, RFC5, RFC1, MSH6, LIG1, RFC4, MSH3, RFC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:04110</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>True</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>111</td>\n",
       "      <td>433</td>\n",
       "      <td>18</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.041570</td>\n",
       "      <td>0.162162</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[CHEK1, CCNA2, PLK1, TP53, CHEK2, YWHAG, CCNB1...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00280</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>0.000340</td>\n",
       "      <td>True</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>48</td>\n",
       "      <td>2964</td>\n",
       "      <td>26</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.541667</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[OXCT1, IL4I1, ALDH7A1, ALDH3A2, ACAD8, HMGCS1...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00240</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>0.001960</td>\n",
       "      <td>True</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>51</td>\n",
       "      <td>2632</td>\n",
       "      <td>26</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.009878</td>\n",
       "      <td>0.509804</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[TYMS, TK1, CAD, DTYMK, DCTPP1, UMPS, UCK2, CT...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01230</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>0.003144</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>72</td>\n",
       "      <td>2890</td>\n",
       "      <td>33</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011419</td>\n",
       "      <td>0.458333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[PGK1, RPIA, CPS1, PGAM1, PSPH, PYCR2, ARG2, P...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00970</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>0.007731</td>\n",
       "      <td>True</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>28</td>\n",
       "      <td>805</td>\n",
       "      <td>11</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.013665</td>\n",
       "      <td>0.392857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[DARS2, TARS2, SEPSECS, HARS2, YARS2, WARS2, N...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01240</td>\n",
       "      <td>Biosynthesis of cofactors</td>\n",
       "      <td>0.011269</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of cofactors</td>\n",
       "      <td>136</td>\n",
       "      <td>2975</td>\n",
       "      <td>57</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.019160</td>\n",
       "      <td>0.419118</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[FLAD1, CAD, NADK2, GCLC, HMBS, UMPS, UROD, UG...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01200</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>0.019601</td>\n",
       "      <td>True</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>112</td>\n",
       "      <td>2975</td>\n",
       "      <td>45</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.015126</td>\n",
       "      <td>0.401786</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[PGK1, RPIA, CPS1, PGAM1, PSPH, GPI, TKFC, PFK...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03013</td>\n",
       "      <td>RNA transport</td>\n",
       "      <td>0.031887</td>\n",
       "      <td>True</td>\n",
       "      <td>RNA transport</td>\n",
       "      <td>152</td>\n",
       "      <td>735</td>\n",
       "      <td>23</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.031293</td>\n",
       "      <td>0.151316</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[STRAP, GEMIN2, EIF5B, XPO5, PAIP1, CLNS1A, EI...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03420</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>0.035977</td>\n",
       "      <td>True</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>44</td>\n",
       "      <td>1072</td>\n",
       "      <td>12</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011194</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[GTF2H1, RFC5, CUL4A, RAD23B, ERCC4, RFC1, LIG...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    source      native                                        name   p_value  \\\n",
       "236   KEGG  KEGG:03030                             DNA replication  0.000005   \n",
       "239   KEGG  KEGG:03430                             Mismatch repair  0.000007   \n",
       "258   KEGG  KEGG:04110                                  Cell cycle  0.000020   \n",
       "303   KEGG  KEGG:00280  Valine, leucine and isoleucine degradation  0.000340   \n",
       "344   KEGG  KEGG:00240                       Pyrimidine metabolism  0.001960   \n",
       "361   KEGG  KEGG:01230                 Biosynthesis of amino acids  0.003144   \n",
       "406   KEGG  KEGG:00970                 Aminoacyl-tRNA biosynthesis  0.007731   \n",
       "425   KEGG  KEGG:01240                   Biosynthesis of cofactors  0.011269   \n",
       "463   KEGG  KEGG:01200                           Carbon metabolism  0.019601   \n",
       "494   KEGG  KEGG:03013                               RNA transport  0.031887   \n",
       "502   KEGG  KEGG:03420                  Nucleotide excision repair  0.035977   \n",
       "\n",
       "     significant                                 description  term_size  \\\n",
       "236         True                             DNA replication         34   \n",
       "239         True                             Mismatch repair         22   \n",
       "258         True                                  Cell cycle        111   \n",
       "303         True  Valine, leucine and isoleucine degradation         48   \n",
       "344         True                       Pyrimidine metabolism         51   \n",
       "361         True                 Biosynthesis of amino acids         72   \n",
       "406         True                 Aminoacyl-tRNA biosynthesis         28   \n",
       "425         True                   Biosynthesis of cofactors        136   \n",
       "463         True                           Carbon metabolism        112   \n",
       "494         True                               RNA transport        152   \n",
       "502         True                  Nucleotide excision repair         44   \n",
       "\n",
       "     query_size  intersection_size  effective_domain_size  precision  \\\n",
       "236        1072                 15                  12939   0.013993   \n",
       "239        1072                 12                  12939   0.011194   \n",
       "258         433                 18                  12939   0.041570   \n",
       "303        2964                 26                  12939   0.008772   \n",
       "344        2632                 26                  12939   0.009878   \n",
       "361        2890                 33                  12939   0.011419   \n",
       "406         805                 11                  12939   0.013665   \n",
       "425        2975                 57                  12939   0.019160   \n",
       "463        2975                 45                  12939   0.015126   \n",
       "494         735                 23                  12939   0.031293   \n",
       "502        1072                 12                  12939   0.011194   \n",
       "\n",
       "       recall    query       parents  \\\n",
       "236  0.441176  query_1  [KEGG:00000]   \n",
       "239  0.545455  query_1  [KEGG:00000]   \n",
       "258  0.162162  query_1  [KEGG:00000]   \n",
       "303  0.541667  query_1  [KEGG:00000]   \n",
       "344  0.509804  query_1  [KEGG:00000]   \n",
       "361  0.458333  query_1  [KEGG:00000]   \n",
       "406  0.392857  query_1  [KEGG:00000]   \n",
       "425  0.419118  query_1  [KEGG:00000]   \n",
       "463  0.401786  query_1  [KEGG:00000]   \n",
       "494  0.151316  query_1  [KEGG:00000]   \n",
       "502  0.272727  query_1  [KEGG:00000]   \n",
       "\n",
       "                                         intersections  \\\n",
       "236  [RFC5, PRIM1, MCM7, MCM5, MCM2, MCM4, RFC1, PO...   \n",
       "239  [PMS2, RFC5, RFC1, MSH6, LIG1, RFC4, MSH3, RFC...   \n",
       "258  [CHEK1, CCNA2, PLK1, TP53, CHEK2, YWHAG, CCNB1...   \n",
       "303  [OXCT1, IL4I1, ALDH7A1, ALDH3A2, ACAD8, HMGCS1...   \n",
       "344  [TYMS, TK1, CAD, DTYMK, DCTPP1, UMPS, UCK2, CT...   \n",
       "361  [PGK1, RPIA, CPS1, PGAM1, PSPH, PYCR2, ARG2, P...   \n",
       "406  [DARS2, TARS2, SEPSECS, HARS2, YARS2, WARS2, N...   \n",
       "425  [FLAD1, CAD, NADK2, GCLC, HMBS, UMPS, UROD, UG...   \n",
       "463  [PGK1, RPIA, CPS1, PGAM1, PSPH, GPI, TKFC, PFK...   \n",
       "494  [STRAP, GEMIN2, EIF5B, XPO5, PAIP1, CLNS1A, EI...   \n",
       "502  [GTF2H1, RFC5, CUL4A, RAD23B, ERCC4, RFC1, LIG...   \n",
       "\n",
       "                                             evidences  \n",
       "236  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "239  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "258  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "303  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "344  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "361  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "406  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "425  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "463  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "494  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "502  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_corr_kegg_results = delta_corr_filtered_results[delta_corr_filtered_results.source == 'KEGG']\n",
    "delta_corr_kegg_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00910</td>\n",
       "      <td>Nitrogen metabolism</td>\n",
       "      <td>0.009448</td>\n",
       "      <td>True</td>\n",
       "      <td>Nitrogen metabolism</td>\n",
       "      <td>14</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.153846</td>\n",
       "      <td>0.142857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[CA4, CA3]</td>\n",
       "      <td>[[KEGG], [KEGG]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00190</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>0.029291</td>\n",
       "      <td>True</td>\n",
       "      <td>Oxidative phosphorylation</td>\n",
       "      <td>116</td>\n",
       "      <td>34</td>\n",
       "      <td>4</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.117647</td>\n",
       "      <td>0.034483</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[NDUFS4, NDUFS1, ATP6V1H, ATP6V1G1]</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00130</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynt...</td>\n",
       "      <td>0.034157</td>\n",
       "      <td>True</td>\n",
       "      <td>Ubiquinone and other terpenoid-quinone biosynt...</td>\n",
       "      <td>9</td>\n",
       "      <td>38</td>\n",
       "      <td>2</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.222222</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[COQ5, COQ6]</td>\n",
       "      <td>[[KEGG], [KEGG]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   source      native                                               name  \\\n",
       "4    KEGG  KEGG:00910                                Nitrogen metabolism   \n",
       "10   KEGG  KEGG:00190                          Oxidative phosphorylation   \n",
       "13   KEGG  KEGG:00130  Ubiquinone and other terpenoid-quinone biosynt...   \n",
       "\n",
       "     p_value  significant                                        description  \\\n",
       "4   0.009448         True                                Nitrogen metabolism   \n",
       "10  0.029291         True                          Oxidative phosphorylation   \n",
       "13  0.034157         True  Ubiquinone and other terpenoid-quinone biosynt...   \n",
       "\n",
       "    term_size  query_size  intersection_size  effective_domain_size  \\\n",
       "4          14          13                  2                  12939   \n",
       "10        116          34                  4                  12939   \n",
       "13          9          38                  2                  12939   \n",
       "\n",
       "    precision    recall    query       parents  \\\n",
       "4    0.153846  0.142857  query_1  [KEGG:00000]   \n",
       "10   0.117647  0.034483  query_1  [KEGG:00000]   \n",
       "13   0.052632  0.222222  query_1  [KEGG:00000]   \n",
       "\n",
       "                          intersections                         evidences  \n",
       "4                            [CA4, CA3]                  [[KEGG], [KEGG]]  \n",
       "10  [NDUFS4, NDUFS1, ATP6V1H, ATP6V1G1]  [[KEGG], [KEGG], [KEGG], [KEGG]]  \n",
       "13                         [COQ5, COQ6]                  [[KEGG], [KEGG]]  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "down_reg_kegg_results = down_reg_filtered_results[down_reg_filtered_results.source == 'KEGG']\n",
    "down_reg_kegg_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make KEGG Mapper scripts\n",
    "https://www.kegg.jp/kegg/tool/map_pathway2.html\n",
    "Search mode: \n",
    "    organism-specific: hsa\n",
    "    outside ID: UniProt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Valine, leucine and isoleucine degradation\n",
      "Q02252 green\n",
      "P55809 green\n",
      "P24752 green\n",
      "Q96RQ9 green\n",
      "P49419 green\n",
      "P22033 green\n",
      "P11182 green\n",
      "P26440 green\n",
      "P80404 green\n",
      "P16219 green\n",
      "Q9UKU7 green\n",
      "Q06278 green\n",
      "P45954 green\n",
      "Q6NVY1 green\n",
      "P35914 green\n",
      "P42765 green\n",
      "P11310 green\n",
      "Q96RQ3 green\n",
      "P30084 green\n",
      "P31937 green\n",
      "P51648 green\n",
      "Q01581 green\n",
      "P05165 green\n",
      "Q99714 green\n",
      "\n",
      "Carbon metabolism\n",
      "Q02252 green\n",
      "P09972 green\n",
      "P24752 green\n",
      "P14618 green\n",
      "P49247 green\n",
      "P08237 green\n",
      "Q15493 green\n",
      "P18669 green\n",
      "Q01813 green\n",
      "P10768 green\n",
      "P13929 green\n",
      "Q8TD30 green\n",
      "P60174 green\n",
      "Q5T6J7 green\n",
      "P78330 green\n",
      "P22033 green\n",
      "P36957 green\n",
      "P11498 green\n",
      "P17174 green\n",
      "P11413 green\n",
      "P16219 green\n",
      "P00558 green\n",
      "Q9P2R7 green\n",
      "P34896 green\n",
      "P48735 green\n",
      "Q6NVY1 green\n",
      "Q9Y617 green\n",
      "P04075 green\n",
      "P07954 green\n",
      "P17858 green\n",
      "Q3LXA3 green\n",
      "P06744 green\n",
      "P08559 green\n",
      "P30084 green\n",
      "O43175 green\n",
      "P00367 green\n",
      "P23378 green\n",
      "P09467 red\n",
      "P53597 red\n",
      "P31327 green\n",
      "P05165 green\n",
      "P19367 green\n",
      "P20132 green\n",
      "O75390 green\n",
      "P06733 green\n",
      "\n",
      "Pyrimidine metabolism\n",
      "P27708 green\n",
      "P49961 green\n",
      "P04183 green\n",
      "P15531 green\n",
      "Q16831 green\n",
      "P04818 green\n",
      "Q5EBM0 green\n",
      "Q8WVQ1 green\n",
      "Q9BZX2 green\n",
      "P49902 green\n",
      "P11172 green\n",
      "Q9H0P0 green\n",
      "P31350 green\n",
      "Q9BXI3 red\n",
      "P00491 green\n",
      "P33316 green\n",
      "Q9Y5B8 green\n",
      "Q96BW1 green\n",
      "P19971 green\n",
      "P32321 green\n",
      "P23919 green\n",
      "Q9NRF8 green\n",
      "Q02127 green\n",
      "P17812 green\n",
      "Q9H773 green\n",
      "P27707 green\n",
      "P30085 green\n",
      "\n",
      "Biosynthesis of amino acids\n",
      "P09972 green\n",
      "P14618 green\n",
      "P49247 green\n",
      "P08237 green\n",
      "P00966 green\n",
      "P18669 green\n",
      "Q01813 green\n",
      "P13929 green\n",
      "Q8TD30 green\n",
      "P60174 green\n",
      "Q03154 green\n",
      "P78330 green\n",
      "P11498 green\n",
      "P17174 green\n",
      "P00558 green\n",
      "Q9NZL9 green\n",
      "P34896 green\n",
      "P48735 green\n",
      "Q9Y617 green\n",
      "P04075 green\n",
      "P17858 green\n",
      "P32322 green\n",
      "P78540 green\n",
      "P54886 green\n",
      "Q96C36 green\n",
      "O43175 green\n",
      "P31327 green\n",
      "P20132 green\n",
      "O75390 green\n",
      "P06733 green\n",
      "P08243 green\n",
      "P05089 red\n",
      "\n",
      "Biosynthesis of cofactors\n",
      "Q8NFF5 green\n",
      "Q9H3S4 green\n",
      "P48506 green\n",
      "P27144 green\n",
      "P27708 green\n",
      "P05186 green\n",
      "P14618 green\n",
      "Q4G0N4 green\n",
      "Q15493 green\n",
      "P15531 green\n",
      "P22830 green\n",
      "Q9Y697 green\n",
      "P00568 green\n",
      "Q5EBM0 green\n",
      "P30520 green\n",
      "P11172 green\n",
      "Q9UIJ7 green\n",
      "P34949 green\n",
      "P08397 green\n",
      "P19224 green\n",
      "Q9HAN9 green\n",
      "P48507 green\n",
      "Q9NZJ6 green\n",
      "Q9NZL9 green\n",
      "P34896 green\n",
      "Q5HYK3 red\n",
      "P13995 green\n",
      "P15559 green\n",
      "Q92820 green\n",
      "Q8N3Y7 green\n",
      "Q9Y617 green\n",
      "P36551 green\n",
      "Q05932 green\n",
      "P30566 green\n",
      "P13716 green\n",
      "P08236 green\n",
      "Q9Y5B8 green\n",
      "Q03393 green\n",
      "Q969G6 green\n",
      "Q9NWU1 green\n",
      "Q6IA69 green\n",
      "P30793 green\n",
      "P51648 green\n",
      "P14902 green\n",
      "P11586 green\n",
      "Q9NRF8 green\n",
      "Q02127 green\n",
      "P17812 green\n",
      "Q96T66 green\n",
      "Q9Y2Z9 red\n",
      "Q9BZ23 green\n",
      "Q9Y3D8 green\n",
      "P30085 green\n",
      "P06132 green\n",
      "Q6XQN6 green\n",
      "Q6UB35 green\n",
      "\n",
      "Cysteine and methionine metabolism\n",
      "P48506 green\n",
      "P52788 green\n",
      "Q13126 green\n",
      "Q96RQ9 green\n",
      "Q16762 green\n",
      "P25325 green\n",
      "Q9Y6K1 green\n",
      "P07195 green\n",
      "P00338 green\n",
      "Q96HN2 green\n",
      "P17174 green\n",
      "P26358 green\n",
      "Q9UHY7 green\n",
      "P48507 green\n",
      "Q9NZL9 green\n",
      "O43865 green\n",
      "Q9Y617 green\n",
      "\n",
      "Sulfur metabolism\n",
      "Q16762 green\n",
      "P25325 green\n",
      "O95340 green\n",
      "P51687 green\n",
      "Q13228 red\n",
      "O95571 green\n",
      "Q9NX62 green\n",
      "\n",
      "Glycolysis / Gluconeogenesis\n",
      "P09972 green\n",
      "P14618 green\n",
      "P08237 green\n",
      "P49419 green\n",
      "P18669 green\n",
      "Q01813 green\n",
      "P07195 green\n",
      "P13929 green\n",
      "P00325 red\n",
      "P00338 green\n",
      "P60174 green\n",
      "\n",
      "DNA replication\n",
      "P40937 green\n",
      "P49642 green\n",
      "P33992 green\n",
      "Q14566 green\n",
      "Q14181 green\n",
      "P18858 green\n",
      "P49005 green\n",
      "P33993 green\n",
      "\n",
      "Propanoate metabolism\n",
      "Q02252 green\n",
      "P24752 green\n",
      "P07195 green\n",
      "P00338 green\n",
      "P22033 green\n",
      "P11182 green\n",
      "P80404 green\n",
      "P16219 green\n",
      "Q9P2R7 green\n",
      "Q6NVY1 green\n",
      "O95822 green\n",
      "P30084 green\n",
      "P53597 red\n",
      "P05165 green\n",
      "Q9H6R3 green\n",
      "Q13085 green\n",
      "P21953 green\n",
      "\n",
      "Riboflavin metabolism\n",
      "Q8NFF5 green\n",
      "P13686 red\n",
      "\n",
      "Alanine, aspartate and glutamate metabolism\n",
      "P27708 green\n",
      "Q06203 green\n",
      "Q96RQ9 green\n",
      "P00966 green\n",
      "Q06210 green\n",
      "Q8TD30 green\n",
      "P51649 green\n",
      "P17174 green\n",
      "P30520 green\n",
      "P80404 green\n",
      "\n",
      "Cell cycle\n",
      "O96017 green\n",
      "P20248 green\n",
      "P24941 green\n",
      "Q00534 green\n",
      "P33992 green\n",
      "Q13485 green\n",
      "P04637 green\n",
      "P61981 green\n",
      "Q14566 green\n",
      "O60566 green\n",
      "Q04917 green\n",
      "P53350 green\n",
      "Q13257 green\n",
      "P14635 green\n"
     ]
    }
   ],
   "source": [
    "for index, row in pval_kegg_results.iterrows():\n",
    "    pathway = row.description\n",
    "    print('\\n' + str(pathway))\n",
    "    old_genes = (row.intersections)\n",
    "    new_genes = list(gp.convert(query = old_genes, organism = 'hsapiens', target_namespace= 'uniprotswissprot_acc').converted)\n",
    "    for gene, ID in zip(old_genes, new_genes):\n",
    "        s = ID\n",
    "        delta_corr = list(sig_genes[sig_genes.Gene == gene].avg_delta_corr)[0]\n",
    "        if delta_corr > 0:\n",
    "            s += ' green'\n",
    "        else:\n",
    "            s+= ' red'\n",
    "        print(s)\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_list = [ccrcc,en,luad,hnscc,lscc]\n",
    "cancer_names = ['ccrcc', 'en', 'luad', 'hnscc', 'lscc']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>source</th>\n",
       "      <th>native</th>\n",
       "      <th>name</th>\n",
       "      <th>p_value</th>\n",
       "      <th>significant</th>\n",
       "      <th>description</th>\n",
       "      <th>term_size</th>\n",
       "      <th>query_size</th>\n",
       "      <th>intersection_size</th>\n",
       "      <th>effective_domain_size</th>\n",
       "      <th>precision</th>\n",
       "      <th>recall</th>\n",
       "      <th>query</th>\n",
       "      <th>parents</th>\n",
       "      <th>intersections</th>\n",
       "      <th>evidences</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03030</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>True</td>\n",
       "      <td>DNA replication</td>\n",
       "      <td>34</td>\n",
       "      <td>1072</td>\n",
       "      <td>15</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.013993</td>\n",
       "      <td>0.441176</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[RFC5, PRIM1, MCM7, MCM5, MCM2, MCM4, RFC1, PO...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03430</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>True</td>\n",
       "      <td>Mismatch repair</td>\n",
       "      <td>22</td>\n",
       "      <td>1072</td>\n",
       "      <td>12</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011194</td>\n",
       "      <td>0.545455</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[PMS2, RFC5, RFC1, MSH6, LIG1, RFC4, MSH3, RFC...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:04110</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>True</td>\n",
       "      <td>Cell cycle</td>\n",
       "      <td>111</td>\n",
       "      <td>433</td>\n",
       "      <td>18</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.041570</td>\n",
       "      <td>0.162162</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[CHEK1, CCNA2, PLK1, TP53, CHEK2, YWHAG, CCNB1...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00280</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>0.000340</td>\n",
       "      <td>True</td>\n",
       "      <td>Valine, leucine and isoleucine degradation</td>\n",
       "      <td>48</td>\n",
       "      <td>2964</td>\n",
       "      <td>26</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.541667</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[OXCT1, IL4I1, ALDH7A1, ALDH3A2, ACAD8, HMGCS1...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>344</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00240</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>0.001960</td>\n",
       "      <td>True</td>\n",
       "      <td>Pyrimidine metabolism</td>\n",
       "      <td>51</td>\n",
       "      <td>2632</td>\n",
       "      <td>26</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.009878</td>\n",
       "      <td>0.509804</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[TYMS, TK1, CAD, DTYMK, DCTPP1, UMPS, UCK2, CT...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>361</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01230</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>0.003144</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of amino acids</td>\n",
       "      <td>72</td>\n",
       "      <td>2890</td>\n",
       "      <td>33</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011419</td>\n",
       "      <td>0.458333</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[PGK1, RPIA, CPS1, PGAM1, PSPH, PYCR2, ARG2, P...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:00970</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>0.007731</td>\n",
       "      <td>True</td>\n",
       "      <td>Aminoacyl-tRNA biosynthesis</td>\n",
       "      <td>28</td>\n",
       "      <td>805</td>\n",
       "      <td>11</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.013665</td>\n",
       "      <td>0.392857</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[DARS2, TARS2, SEPSECS, HARS2, YARS2, WARS2, N...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>425</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01240</td>\n",
       "      <td>Biosynthesis of cofactors</td>\n",
       "      <td>0.011269</td>\n",
       "      <td>True</td>\n",
       "      <td>Biosynthesis of cofactors</td>\n",
       "      <td>136</td>\n",
       "      <td>2975</td>\n",
       "      <td>57</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.019160</td>\n",
       "      <td>0.419118</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[FLAD1, CAD, NADK2, GCLC, HMBS, UMPS, UROD, UG...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>463</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:01200</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>0.019601</td>\n",
       "      <td>True</td>\n",
       "      <td>Carbon metabolism</td>\n",
       "      <td>112</td>\n",
       "      <td>2975</td>\n",
       "      <td>45</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.015126</td>\n",
       "      <td>0.401786</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[PGK1, RPIA, CPS1, PGAM1, PSPH, GPI, TKFC, PFK...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03013</td>\n",
       "      <td>RNA transport</td>\n",
       "      <td>0.031887</td>\n",
       "      <td>True</td>\n",
       "      <td>RNA transport</td>\n",
       "      <td>152</td>\n",
       "      <td>735</td>\n",
       "      <td>23</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.031293</td>\n",
       "      <td>0.151316</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[STRAP, GEMIN2, EIF5B, XPO5, PAIP1, CLNS1A, EI...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>KEGG</td>\n",
       "      <td>KEGG:03420</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>0.035977</td>\n",
       "      <td>True</td>\n",
       "      <td>Nucleotide excision repair</td>\n",
       "      <td>44</td>\n",
       "      <td>1072</td>\n",
       "      <td>12</td>\n",
       "      <td>12939</td>\n",
       "      <td>0.011194</td>\n",
       "      <td>0.272727</td>\n",
       "      <td>query_1</td>\n",
       "      <td>[KEGG:00000]</td>\n",
       "      <td>[GTF2H1, RFC5, CUL4A, RAD23B, ERCC4, RFC1, LIG...</td>\n",
       "      <td>[[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    source      native                                        name   p_value  \\\n",
       "236   KEGG  KEGG:03030                             DNA replication  0.000005   \n",
       "239   KEGG  KEGG:03430                             Mismatch repair  0.000007   \n",
       "258   KEGG  KEGG:04110                                  Cell cycle  0.000020   \n",
       "303   KEGG  KEGG:00280  Valine, leucine and isoleucine degradation  0.000340   \n",
       "344   KEGG  KEGG:00240                       Pyrimidine metabolism  0.001960   \n",
       "361   KEGG  KEGG:01230                 Biosynthesis of amino acids  0.003144   \n",
       "406   KEGG  KEGG:00970                 Aminoacyl-tRNA biosynthesis  0.007731   \n",
       "425   KEGG  KEGG:01240                   Biosynthesis of cofactors  0.011269   \n",
       "463   KEGG  KEGG:01200                           Carbon metabolism  0.019601   \n",
       "494   KEGG  KEGG:03013                               RNA transport  0.031887   \n",
       "502   KEGG  KEGG:03420                  Nucleotide excision repair  0.035977   \n",
       "\n",
       "     significant                                 description  term_size  \\\n",
       "236         True                             DNA replication         34   \n",
       "239         True                             Mismatch repair         22   \n",
       "258         True                                  Cell cycle        111   \n",
       "303         True  Valine, leucine and isoleucine degradation         48   \n",
       "344         True                       Pyrimidine metabolism         51   \n",
       "361         True                 Biosynthesis of amino acids         72   \n",
       "406         True                 Aminoacyl-tRNA biosynthesis         28   \n",
       "425         True                   Biosynthesis of cofactors        136   \n",
       "463         True                           Carbon metabolism        112   \n",
       "494         True                               RNA transport        152   \n",
       "502         True                  Nucleotide excision repair         44   \n",
       "\n",
       "     query_size  intersection_size  effective_domain_size  precision  \\\n",
       "236        1072                 15                  12939   0.013993   \n",
       "239        1072                 12                  12939   0.011194   \n",
       "258         433                 18                  12939   0.041570   \n",
       "303        2964                 26                  12939   0.008772   \n",
       "344        2632                 26                  12939   0.009878   \n",
       "361        2890                 33                  12939   0.011419   \n",
       "406         805                 11                  12939   0.013665   \n",
       "425        2975                 57                  12939   0.019160   \n",
       "463        2975                 45                  12939   0.015126   \n",
       "494         735                 23                  12939   0.031293   \n",
       "502        1072                 12                  12939   0.011194   \n",
       "\n",
       "       recall    query       parents  \\\n",
       "236  0.441176  query_1  [KEGG:00000]   \n",
       "239  0.545455  query_1  [KEGG:00000]   \n",
       "258  0.162162  query_1  [KEGG:00000]   \n",
       "303  0.541667  query_1  [KEGG:00000]   \n",
       "344  0.509804  query_1  [KEGG:00000]   \n",
       "361  0.458333  query_1  [KEGG:00000]   \n",
       "406  0.392857  query_1  [KEGG:00000]   \n",
       "425  0.419118  query_1  [KEGG:00000]   \n",
       "463  0.401786  query_1  [KEGG:00000]   \n",
       "494  0.151316  query_1  [KEGG:00000]   \n",
       "502  0.272727  query_1  [KEGG:00000]   \n",
       "\n",
       "                                         intersections  \\\n",
       "236  [RFC5, PRIM1, MCM7, MCM5, MCM2, MCM4, RFC1, PO...   \n",
       "239  [PMS2, RFC5, RFC1, MSH6, LIG1, RFC4, MSH3, RFC...   \n",
       "258  [CHEK1, CCNA2, PLK1, TP53, CHEK2, YWHAG, CCNB1...   \n",
       "303  [OXCT1, IL4I1, ALDH7A1, ALDH3A2, ACAD8, HMGCS1...   \n",
       "344  [TYMS, TK1, CAD, DTYMK, DCTPP1, UMPS, UCK2, CT...   \n",
       "361  [PGK1, RPIA, CPS1, PGAM1, PSPH, PYCR2, ARG2, P...   \n",
       "406  [DARS2, TARS2, SEPSECS, HARS2, YARS2, WARS2, N...   \n",
       "425  [FLAD1, CAD, NADK2, GCLC, HMBS, UMPS, UROD, UG...   \n",
       "463  [PGK1, RPIA, CPS1, PGAM1, PSPH, GPI, TKFC, PFK...   \n",
       "494  [STRAP, GEMIN2, EIF5B, XPO5, PAIP1, CLNS1A, EI...   \n",
       "502  [GTF2H1, RFC5, CUL4A, RAD23B, ERCC4, RFC1, LIG...   \n",
       "\n",
       "                                             evidences  \n",
       "236  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "239  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "258  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "303  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "344  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "361  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "406  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "425  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "463  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "494  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  \n",
       "502  [[KEGG], [KEGG], [KEGG], [KEGG], [KEGG], [KEGG...  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_corr_kegg_results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['RFC5', 'PRIM1', 'MCM7', 'MCM5', 'MCM2', 'MCM4', 'RFC1', 'POLA2', 'LIG1', 'RFC4', 'MCM6', 'RFC3', 'SSBP1', 'POLD2', 'POLD3']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ccrcc\n",
      "RFC5\n",
      "PRIM1\n",
      "MCM7\n",
      "MCM5\n",
      "MCM2\n",
      "MCM4\n",
      "RFC1\n",
      "POLA2\n",
      "LIG1\n",
      "RFC4\n",
      "MCM6\n",
      "RFC3\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "en\n",
      "RFC5\n",
      "PRIM1\n",
      "MCM7\n",
      "MCM5\n",
      "MCM2\n",
      "MCM4\n",
      "RFC1\n",
      "POLA2\n",
      "LIG1\n",
      "RFC4\n",
      "MCM6\n",
      "RFC3\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "luad\n",
      "RFC5\n",
      "PRIM1\n",
      "MCM7\n",
      "MCM5\n",
      "MCM2\n",
      "MCM4\n",
      "RFC1\n",
      "POLA2\n",
      "LIG1\n",
      "RFC4\n",
      "MCM6\n",
      "RFC3\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "hnscc\n",
      "RFC5\n",
      "PRIM1\n",
      "MCM7\n",
      "MCM5\n",
      "MCM2\n",
      "MCM4\n",
      "RFC1\n",
      "POLA2\n",
      "LIG1\n",
      "RFC4\n",
      "MCM6\n",
      "RFC3\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "lscc\n",
      "RFC5\n",
      "PRIM1\n",
      "MCM7\n",
      "MCM5\n",
      "MCM2\n",
      "MCM4\n",
      "RFC1\n",
      "POLA2\n",
      "LIG1\n",
      "RFC4\n",
      "MCM6\n",
      "RFC3\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "\n",
      "DNA replication\n",
      "     gene cancer  interaction_coeff  condition_coeff  transcript_coeff  \\\n",
      "0    RFC5  ccrcc          -0.043390         0.446613          0.061585   \n",
      "1   PRIM1  ccrcc           0.036482         0.411458         -0.001128   \n",
      "2    MCM7  ccrcc          -0.015694         1.019508          0.045208   \n",
      "3    MCM5  ccrcc          -0.045800         0.894918          0.090340   \n",
      "4    MCM2  ccrcc          -0.039351         1.055137          0.088322   \n",
      "..    ...    ...                ...              ...               ...   \n",
      "10   MCM6   lscc           2.012947       -33.505771          0.126389   \n",
      "11   RFC3   lscc           1.007726       -15.742462         -0.229781   \n",
      "12  SSBP1   lscc           2.526943       -43.768370         -0.779431   \n",
      "13  POLD2   lscc           0.673479       -11.811183          0.277750   \n",
      "14  POLD3   lscc           1.286471       -20.427045          0.346120   \n",
      "\n",
      "    intercept  interaction_pval  condition_pval  transcript_pval  \\\n",
      "0   -0.433473          0.049659    6.715109e-04         0.003201   \n",
      "1   -0.382691          0.640052    1.334458e-02         0.980581   \n",
      "2   -1.165363          0.662331    1.578294e-04         0.098216   \n",
      "3   -1.175187          0.311647    8.036796e-05         0.017467   \n",
      "4   -1.052501          0.566878    4.216188e-10         0.072586   \n",
      "..        ...               ...             ...              ...   \n",
      "10  -5.851948          0.000004    9.633485e-06         0.836973   \n",
      "11   2.725751          0.000137    2.543813e-04         0.836973   \n",
      "12  12.667787          0.001257    1.292928e-03         0.836973   \n",
      "13  -5.554126          0.036907    3.700180e-02         0.836973   \n",
      "14  -7.505188          0.002169    4.131294e-03         0.836973   \n",
      "\n",
      "    intercept_pval  \n",
      "0     3.095094e-04  \n",
      "1     4.900774e-03  \n",
      "2     3.368693e-06  \n",
      "3     2.801189e-08  \n",
      "4     8.166463e-12  \n",
      "..             ...  \n",
      "10    9.516768e-01  \n",
      "11    9.516768e-01  \n",
      "12    9.516768e-01  \n",
      "13    9.516768e-01  \n",
      "14    9.516768e-01  \n",
      "\n",
      "[75 rows x 10 columns]\n",
      "['PMS2', 'RFC5', 'RFC1', 'MSH6', 'LIG1', 'RFC4', 'MSH3', 'RFC3', 'MSH2', 'SSBP1', 'POLD2', 'POLD3']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ccrcc\n",
      "PMS2\n",
      "RFC5\n",
      "RFC1\n",
      "MSH6\n",
      "LIG1\n",
      "RFC4\n",
      "MSH3\n",
      "RFC3\n",
      "MSH2\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "en\n",
      "PMS2\n",
      "RFC5\n",
      "RFC1\n",
      "MSH6\n",
      "LIG1\n",
      "RFC4\n",
      "MSH3\n",
      "RFC3\n",
      "MSH2\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "luad\n",
      "PMS2\n",
      "RFC5\n",
      "RFC1\n",
      "MSH6\n",
      "LIG1\n",
      "RFC4\n",
      "MSH3\n",
      "RFC3\n",
      "MSH2\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "hnscc\n",
      "PMS2\n",
      "RFC5\n",
      "RFC1\n",
      "MSH6\n",
      "LIG1\n",
      "RFC4\n",
      "MSH3\n",
      "RFC3\n",
      "MSH2\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "lscc\n",
      "PMS2\n",
      "RFC5\n",
      "RFC1\n",
      "MSH6\n",
      "LIG1\n",
      "RFC4\n",
      "MSH3\n",
      "RFC3\n",
      "MSH2\n",
      "SSBP1\n",
      "POLD2\n",
      "POLD3\n",
      "\n",
      "Mismatch repair\n",
      "     gene cancer  interaction_coeff  condition_coeff  transcript_coeff  \\\n",
      "0    PMS2  ccrcc           0.000794        -0.121777          0.024171   \n",
      "1    RFC5  ccrcc          -0.043390         0.446613          0.061585   \n",
      "2    RFC1  ccrcc           0.001244        -0.048388          0.013927   \n",
      "3    MSH6  ccrcc          -0.061074         0.777337          0.106847   \n",
      "4    LIG1  ccrcc          -0.129574         0.831057          0.181179   \n",
      "5    RFC4  ccrcc          -0.068957         0.455625          0.087354   \n",
      "6    MSH3  ccrcc           0.032595        -0.400533         -0.001073   \n",
      "7    RFC3  ccrcc          -0.083975         0.322909          0.118123   \n",
      "8    MSH2  ccrcc          -0.037585         0.455130          0.072196   \n",
      "9   SSBP1  ccrcc           0.005566        -0.333080          0.007627   \n",
      "10  POLD2  ccrcc           0.022474        -0.112340         -0.012361   \n",
      "11  POLD3  ccrcc           0.038263        -0.047117         -0.004188   \n",
      "0    PMS2     en           0.872686        -9.234108         -0.383943   \n",
      "1    RFC5     en           0.443203        -4.015793          0.155960   \n",
      "2    RFC1     en           1.287258       -15.391127         -0.723350   \n",
      "3    MSH6     en           0.517577        -5.796562          0.414309   \n",
      "4    LIG1     en           0.336736        -3.126533          0.486529   \n",
      "5    RFC4     en           0.225774        -2.025932          0.232771   \n",
      "6    MSH3     en           1.285798       -13.691078         -0.405567   \n",
      "7    RFC3     en           0.266031        -2.670189          0.045891   \n",
      "8    MSH2     en           0.263409        -2.963589          0.191138   \n",
      "9   SSBP1     en           1.068610       -11.162575         -0.352263   \n",
      "10  POLD2     en           0.528843        -5.747714         -0.069256   \n",
      "11  POLD3     en           0.837322        -8.419227         -0.031975   \n",
      "0    PMS2   luad           1.939518        -5.191905         -0.685986   \n",
      "1    RFC5   luad           0.944283        -1.716023          0.027406   \n",
      "2    RFC1   luad           2.069780        -8.351884         -0.595889   \n",
      "3    MSH6   luad           1.994466        -6.237861          0.851810   \n",
      "4    LIG1   luad           2.020491        -3.800752         -0.167983   \n",
      "5    RFC4   luad           0.939635        -1.871392          0.093261   \n",
      "6    MSH3   luad           0.546894        -1.288694          0.972935   \n",
      "7    RFC3   luad           1.001154        -1.521844         -0.298821   \n",
      "8    MSH2   luad           1.091965        -2.860851          0.343063   \n",
      "9   SSBP1   luad           1.656539        -7.258542          0.120784   \n",
      "10  POLD2   luad           0.297820        -1.193882          0.390609   \n",
      "11  POLD3   luad           0.722106        -0.819130          1.271962   \n",
      "0    PMS2  hnscc           0.452257        -4.538379          0.050540   \n",
      "1    RFC5  hnscc           0.330784        -2.893216          0.363030   \n",
      "2    RFC1  hnscc           0.480796        -5.294891          0.168600   \n",
      "3    MSH6  hnscc           0.347140        -4.030369          0.552248   \n",
      "4    LIG1  hnscc           0.323613        -3.172715          0.526319   \n",
      "5    RFC4  hnscc           0.160687        -1.471921          0.308567   \n",
      "6    MSH3  hnscc           0.536001        -5.490779          0.048170   \n",
      "7    RFC3  hnscc           0.047185        -0.175032          0.250557   \n",
      "8    MSH2  hnscc           0.055138        -0.581519          0.826461   \n",
      "9   SSBP1  hnscc           0.399337        -4.480687          0.180822   \n",
      "10  POLD2  hnscc           0.012553         0.060443          0.295053   \n",
      "11  POLD3  hnscc          -0.036063         0.461518          0.672543   \n",
      "0    PMS2   lscc           1.288360       -21.280315          0.055852   \n",
      "1    RFC5   lscc           1.377588       -22.047313         -0.215177   \n",
      "2    RFC1   lscc           1.549166       -26.884683         -0.147326   \n",
      "3    MSH6   lscc           2.074899       -35.578757          0.416725   \n",
      "4    LIG1   lscc           1.643490       -26.165922          0.376402   \n",
      "5    RFC4   lscc           1.326649       -21.334068         -0.111382   \n",
      "6    MSH3   lscc           1.185864       -20.174085          0.638687   \n",
      "7    RFC3   lscc           1.007726       -15.742462         -0.229781   \n",
      "8    MSH2   lscc           1.699999       -29.073605          0.266386   \n",
      "9   SSBP1   lscc           2.526943       -43.768370         -0.779431   \n",
      "10  POLD2   lscc           0.673479       -11.811183          0.277750   \n",
      "11  POLD3   lscc           1.286471       -20.427045          0.346120   \n",
      "\n",
      "    intercept  interaction_pval  condition_pval  transcript_pval  \\\n",
      "0   -0.186612      9.256399e-01    3.537386e-01         0.004013   \n",
      "1   -0.433473      3.089866e-02    7.162783e-04         0.001707   \n",
      "2   -0.228216      9.256399e-01    7.323099e-01         0.050811   \n",
      "3   -1.042109      1.251055e-02    2.247397e-04         0.000020   \n",
      "4   -0.980491      7.362990e-03    1.233497e-05         0.000066   \n",
      "5   -0.441664      7.362990e-03    7.692467e-05         0.000307   \n",
      "6    0.039116      1.670231e-10    1.233497e-05         0.714135   \n",
      "7   -0.338143      7.362990e-03    8.980367e-04         0.000066   \n",
      "8   -0.589023      3.089866e-02    5.480659e-04         0.000044   \n",
      "9   -0.182530      6.971420e-01    3.537386e-01         0.480188   \n",
      "10   0.054000      7.362990e-03    4.831425e-01         0.130003   \n",
      "11  -0.044611      7.362990e-03    7.323099e-01         0.432499   \n",
      "0    3.891207      3.646426e-01    3.314274e-01         0.912166   \n",
      "1   -1.580390      3.646426e-01    3.314274e-01         0.912166   \n",
      "2    8.799077      3.816566e-02    4.679531e-02         0.504663   \n",
      "3   -5.143202      2.589982e-01    2.677785e-01         0.504663   \n",
      "4   -5.148540      2.630799e-01    3.074182e-01         0.322483   \n",
      "5   -2.137314      4.412701e-01    4.403979e-01         0.912166   \n",
      "6    4.270799      2.630799e-01    2.677785e-01         0.912166   \n",
      "7   -0.425401      3.646426e-01    3.314274e-01         0.912166   \n",
      "8   -2.166149      3.646426e-01    3.314274e-01         0.912166   \n",
      "9    3.481099      3.646426e-01    3.314274e-01         0.912166   \n",
      "10   0.757119      2.630799e-01    2.677785e-01         0.912166   \n",
      "11   0.119467      3.816566e-02    4.679531e-02         0.912166   \n",
      "0    1.358231      5.589876e-10    6.319327e-11         0.027188   \n",
      "1   -0.859125      3.747335e-03    3.178153e-02         0.925818   \n",
      "2    2.151703      5.004393e-06    3.871015e-05         0.235182   \n",
      "3   -3.997834      1.576887e-06    1.147794e-05         0.060456   \n",
      "4   -0.949113      5.589876e-10    7.238236e-07         0.674994   \n",
      "5   -1.163235      5.554588e-07    3.871015e-05         0.674994   \n",
      "6   -3.629583      9.750358e-02    2.648023e-01         0.006024   \n",
      "7    0.037095      1.632553e-03    2.744014e-02         0.429120   \n",
      "8   -1.808245      5.554588e-07    8.502214e-06         0.141898   \n",
      "9   -1.323843      1.143959e-02    2.131983e-02         0.913129   \n",
      "10  -1.665708      3.587160e-01    3.522669e-01         0.347691   \n",
      "11  -4.906081      1.497643e-01    5.797324e-01         0.027188   \n",
      "0   23.630559      6.696682e-02    6.593497e-02         0.782003   \n",
      "1   21.009244      6.696682e-02    6.593497e-02         0.010013   \n",
      "2   22.452823      7.715172e-02    9.591519e-02         0.549619   \n",
      "3   18.082628      7.715172e-02    8.038631e-02         0.003703   \n",
      "4   19.382069      3.031066e-02    3.100944e-02         0.000009   \n",
      "5   21.823156      2.064248e-01    1.792969e-01         0.006867   \n",
      "6   23.860652      1.308943e-02    1.590519e-02         0.782003   \n",
      "7   21.944801      9.220753e-01    9.717877e-01         0.139470   \n",
      "8   16.448529      9.220753e-01    9.641471e-01         0.000912   \n",
      "9   25.381103      6.805231e-02    6.593497e-02         0.384671   \n",
      "10  20.800181      9.358535e-01    9.717877e-01         0.079283   \n",
      "11  16.438949      9.220753e-01    9.641471e-01         0.000912   \n",
      "0   -1.257432      2.382649e-05    2.026917e-05         0.822193   \n",
      "1    1.959067      1.109526e-07    1.359557e-07         0.512524   \n",
      "2    1.808265      3.609655e-05    9.216223e-05         0.699793   \n",
      "3   -9.503581      2.382649e-05    2.026917e-05         0.512524   \n",
      "4   -7.945779      3.405009e-06    6.168946e-06         0.512524   \n",
      "5   -0.377724      2.063331e-07    1.359557e-07         0.699793   \n",
      "6  -11.356742      1.173818e-04    1.623826e-04         0.196247   \n",
      "7    2.725751      1.218101e-04    2.261167e-04         0.512524   \n",
      "8   -6.467966      1.535752e-08    1.489505e-08         0.512524   \n",
      "9   12.667787      1.106325e-03    1.137777e-03         0.512524   \n",
      "10  -5.554126      3.690718e-02    3.700180e-02         0.512524   \n",
      "11  -7.505188      2.208686e-03    4.206408e-03         0.512524   \n",
      "\n",
      "    intercept_pval  \n",
      "0     1.101366e-01  \n",
      "1     3.714113e-04  \n",
      "2     1.171569e-01  \n",
      "3     1.230869e-07  \n",
      "4     6.874942e-08  \n",
      "5     3.922125e-05  \n",
      "6     5.789357e-01  \n",
      "7     1.033603e-04  \n",
      "8     8.030731e-07  \n",
      "9     5.789357e-01  \n",
      "10    6.745153e-01  \n",
      "11    5.789357e-01  \n",
      "0     9.308004e-01  \n",
      "1     9.308004e-01  \n",
      "2     5.028817e-01  \n",
      "3     5.028817e-01  \n",
      "4     2.038891e-01  \n",
      "5     9.308004e-01  \n",
      "6     9.308004e-01  \n",
      "7     9.308004e-01  \n",
      "8     9.308004e-01  \n",
      "9     9.308004e-01  \n",
      "10    9.308004e-01  \n",
      "11    9.683897e-01  \n",
      "0     4.528189e-02  \n",
      "1     2.638409e-01  \n",
      "2     2.638409e-01  \n",
      "3     3.363088e-03  \n",
      "4     1.987525e-01  \n",
      "5     3.667996e-03  \n",
      "6     1.340874e-03  \n",
      "7     9.495098e-01  \n",
      "8     1.838184e-03  \n",
      "9     6.760138e-01  \n",
      "10    1.987525e-01  \n",
      "11    1.838184e-03  \n",
      "0     1.090736e-24  \n",
      "1     1.481648e-37  \n",
      "2     1.658202e-14  \n",
      "3     5.768899e-17  \n",
      "4     4.401968e-42  \n",
      "5     7.212082e-54  \n",
      "6     2.348157e-31  \n",
      "7     3.695008e-33  \n",
      "8     2.975223e-11  \n",
      "9     4.965379e-28  \n",
      "10    7.293297e-26  \n",
      "11    1.624202e-16  \n",
      "0     8.240608e-01  \n",
      "1     7.757577e-01  \n",
      "2     8.240608e-01  \n",
      "3     5.222401e-01  \n",
      "4     4.967015e-01  \n",
      "5     9.130074e-01  \n",
      "6     1.684457e-01  \n",
      "7     7.229331e-01  \n",
      "8     4.967015e-01  \n",
      "9     5.222401e-01  \n",
      "10    5.222401e-01  \n",
      "11    5.222401e-01  \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CHEK1', 'CCNA2', 'PLK1', 'TP53', 'CHEK2', 'YWHAG', 'CCNB1', 'BUB1B', 'TFDP1', 'MCM7', 'MAD2L1', 'MCM5', 'MCM2', 'CDKN2A', 'MCM4', 'YWHAE', 'CDK1', 'EP300']\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n",
      "cptac warning: You tried to flatten a column index that didn't have multiple levels, so we didn't actually change anything. (C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\cptac\\dataset.py, line 700)\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Length mismatch: Expected axis has 3 elements, new values have 2 elements",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-23-24deeb6284a3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      6\u001b[0m     \u001b[0mold_genes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mrow\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mintersections\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mold_genes\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 8\u001b[1;33m     \u001b[0mslopes_df\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mrc\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcalculate_regression\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcancer_list\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mold_genes\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcancer_names\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      9\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'\\n'\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mpathway\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m     \u001b[0mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mslopes_df\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Regression_calculator.py\u001b[0m in \u001b[0;36mcalculate_regression\u001b[1;34m(cancer_list, gene_list, cancer_names)\u001b[0m\n\u001b[0;32m     48\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mnormal\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m<\u001b[0m \u001b[1;36m2\u001b[0m \u001b[1;32mor\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtumor\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m<\u001b[0m \u001b[1;36m2\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     49\u001b[0m                 \u001b[1;32mcontinue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 50\u001b[1;33m             \u001b[0mtumor\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m'proteomics'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'transcriptomics'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     51\u001b[0m             \u001b[0mnormal\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m'proteomics'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'transcriptomics'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     52\u001b[0m             \u001b[0mgroups\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m'tumor'\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;33m*\u001b[0m \u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mtumor\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m__setattr__\u001b[1;34m(self, name, value)\u001b[0m\n\u001b[0;32m   5285\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5286\u001b[0m             \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5287\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   5288\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5289\u001b[0m             \u001b[1;32mpass\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\properties.pyx\u001b[0m in \u001b[0;36mpandas._libs.properties.AxisProperty.__set__\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m_set_axis\u001b[1;34m(self, axis, labels)\u001b[0m\n\u001b[0;32m    659\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    660\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_set_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 661\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    662\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_clear_item_cache\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    663\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m~\\anaconda3\\lib\\site-packages\\pandas\\core\\internals\\managers.py\u001b[0m in \u001b[0;36mset_axis\u001b[1;34m(self, axis, new_labels)\u001b[0m\n\u001b[0;32m    175\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    176\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mnew_len\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mold_len\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 177\u001b[1;33m             raise ValueError(\n\u001b[0m\u001b[0;32m    178\u001b[0m                 \u001b[1;34mf\"Length mismatch: Expected axis has {old_len} elements, new \"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    179\u001b[0m                 \u001b[1;34mf\"values have {new_len} elements\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Length mismatch: Expected axis has 3 elements, new values have 2 elements"
     ]
    }
   ],
   "source": [
    "\n",
    "gene_ids = []\n",
    "txt = ''\n",
    "for index, row in delta_corr_kegg_results.iterrows():\n",
    "    pathway = row.description\n",
    "    \n",
    "    old_genes = (row.intersections)\n",
    "    print(old_genes)\n",
    "    slopes_df = rc.calculate_regression(cancer_list, old_genes, cancer_names)\n",
    "    print('\\n' + str(pathway))\n",
    "    print(slopes_df)\n",
    "    new_genes = list(gp.convert(query = old_genes, organism = 'hsapiens', target_namespace= 'uniprotswissprot_acc').converted)\n",
    "    for gene, ID in zip(old_genes, new_genes):\n",
    "        s = ID\n",
    "        gene_df = slopes_df[slopes_df.gene == gene]\n",
    "        avg_slope = np.mean(gene_df.interaction_coeff)\n",
    "        if avg_slope > 0:\n",
    "            s += ' green'\n",
    "        else:\n",
    "            s+= ' red'\n",
    "        txt += s + '\\n'\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_ids = []\n",
    "for index, row in down_reg_kegg_results.iterrows():\n",
    "    pathway = row.description\n",
    "    print('\\n' + str(pathway))\n",
    "    old_genes = (row.intersections)\n",
    "    new_genes = list(gp.convert(query = old_genes, organism = 'hsapiens', target_namespace= 'uniprotswissprot_acc').converted)\n",
    "    for gene, ID in zip(old_genes, new_genes):\n",
    "        s = ID\n",
    "        delta_corr = list(sig_genes[sig_genes.Gene == gene].avg_delta_corr)[0]\n",
    "        if delta_corr > 0:\n",
    "            s += ' green'\n",
    "        else:\n",
    "            s+= ' red'\n",
    "        print(s)\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pval_results_df = gp.profile(organism='hsapiens', query = pval_ranked_gene_list, no_iea=True,\n",
    "                       ordered=False, no_evidences=False, background= background_genes)\n",
    "pval_filtered_results = pval_results_df[pval_results_df.term_size <= 500]\n",
    "pval_filtered_results = pval_filtered_results[pval_filtered_results.term_size >=5]\n",
    "pval_kegg_results = pval_filtered_results[pval_filtered_results.source == 'KEGG']\n",
    "for index, row in pval_kegg_results.iterrows():\n",
    "    pathway = row.description\n",
    "    print('\\n' + str(pathway))\n",
    "    old_genes = (row.intersections)\n",
    "    new_genes = list(gp.convert(query = old_genes, organism = 'hsapiens', target_namespace= 'uniprotswissprot_acc').converted)\n",
    "    for gene, ID in zip(old_genes, new_genes):\n",
    "        s = ID\n",
    "        delta_corr = list(sig_genes[sig_genes.Gene == gene].avg_delta_corr)[0]\n",
    "        if delta_corr > 0:\n",
    "            s += ' green'\n",
    "        else:\n",
    "            s+= ' red'\n",
    "        print(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pval_results_df = gp.profile(organism='hsapiens', query = pval_ranked_gene_list, no_iea=True,\n",
    "                       ordered=False, no_evidences=False)\n",
    "pval_filtered_results = pval_results_df[pval_results_df.term_size <= 500]\n",
    "pval_filtered_results = pval_filtered_results[pval_filtered_results.term_size >=5]\n",
    "pval_kegg_results = pval_filtered_results[pval_filtered_results.source == 'KEGG']\n",
    "for index, row in pval_kegg_results.iterrows():\n",
    "    pathway = row.description\n",
    "    print('\\n' + str(pathway))\n",
    "    old_genes = (row.intersections)\n",
    "    new_genes = list(gp.convert(query = old_genes, organism = 'hsapiens', target_namespace= 'uniprotswissprot_acc').converted)\n",
    "    for gene, ID in zip(old_genes, new_genes):\n",
    "        s = ID\n",
    "        delta_corr = list(sig_genes[sig_genes.Gene == gene].avg_delta_corr)[0]\n",
    "        if delta_corr > 0:\n",
    "            s += ' green'\n",
    "        else:\n",
    "            s+= ' red'\n",
    "        print(s)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
