{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import pandas as pd\n",
    "import cptac.utils as ut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that hnscc index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-5f7029bc4773>, line 2)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "luad = cptac.Luad()\n",
    "lscc = cptac.Lscc()\n",
    "hnscc = cptac.Hnscc()\n",
    "en = cptac.Endometrial()\n",
    "ccrcc = cptac.Ccrcc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_list = [luad, lscc, hnscc, en, ccrcc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_dict = {lscc: 'LSCC', luad: 'LUAD', hnscc: 'HNSCC', en: 'Endometrial', ccrcc: 'CCRCC'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Unique_Samples_Mut</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9409</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.536364</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2622</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>0.345455</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5580</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.336364</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9680</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7886</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5011</th>\n",
       "      <td>LRP1B</td>\n",
       "      <td>0.309091</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4711</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10204</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>0.236364</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3045</th>\n",
       "      <td>FAT3</td>\n",
       "      <td>0.209091</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1880</th>\n",
       "      <td>COL11A1</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9925</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.654867</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2122</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>0.451327</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10494</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>0.353982</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8097</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.353982</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5079</th>\n",
       "      <td>LRP1B</td>\n",
       "      <td>0.300885</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9121</th>\n",
       "      <td>SYNE1</td>\n",
       "      <td>0.292035</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5686</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.292035</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4763</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>0.247788</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2495</th>\n",
       "      <td>DNAH5</td>\n",
       "      <td>0.230088</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10351</th>\n",
       "      <td>XIRP2</td>\n",
       "      <td>0.212389</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8412</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.396396</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1391</th>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>0.261261</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2709</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>0.234234</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5174</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>0.234234</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4137</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>0.171171</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4863</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.171171</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4868</th>\n",
       "      <td>MUC4</td>\n",
       "      <td>0.126126</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6016</th>\n",
       "      <td>PLEC</td>\n",
       "      <td>0.117117</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2139</th>\n",
       "      <td>DMD</td>\n",
       "      <td>0.081081</td>\n",
       "      <td>HNSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10096</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>0.789474</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9415</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>0.494737</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>836</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>0.452632</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9419</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.389474</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6514</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0.326316</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2935</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>0.305263</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2917</th>\n",
       "      <td>CTCF</td>\n",
       "      <td>0.284211</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6494</th>\n",
       "      <td>KMT2B</td>\n",
       "      <td>0.242105</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14231</th>\n",
       "      <td>ZFHX3</td>\n",
       "      <td>0.221053</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13126</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.221053</td>\n",
       "      <td>Endometrial</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2997</th>\n",
       "      <td>PBRM1</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3721</th>\n",
       "      <td>SETD2</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4439</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Name      Gene  Unique_Samples_Mut       Cancer\n",
       "9409      TP53            0.536364         LUAD\n",
       "2622      EGFR            0.345455         LUAD\n",
       "5580     MUC16            0.336364         LUAD\n",
       "9680       TTN            0.318182         LUAD\n",
       "7886      RYR2            0.318182         LUAD\n",
       "5011     LRP1B            0.309091         LUAD\n",
       "4711      KRAS            0.300000         LUAD\n",
       "10204    ZFHX4            0.236364         LUAD\n",
       "3045      FAT3            0.209091         LUAD\n",
       "1880   COL11A1            0.200000         LUAD\n",
       "9925       TTN            0.654867         LSCC\n",
       "2122     CSMD3            0.451327         LSCC\n",
       "10494    ZFHX4            0.353982         LSCC\n",
       "8097      RYR2            0.353982         LSCC\n",
       "5079     LRP1B            0.300885         LSCC\n",
       "9121     SYNE1            0.292035         LSCC\n",
       "5686     MUC16            0.292035         LSCC\n",
       "4763     KMT2D            0.247788         LSCC\n",
       "2495     DNAH5            0.230088         LSCC\n",
       "10351    XIRP2            0.212389         LSCC\n",
       "8412       TTN            0.396396        HNSCC\n",
       "1391    CDKN2A            0.261261        HNSCC\n",
       "2709      FAT1            0.234234        HNSCC\n",
       "5174    NOTCH1            0.234234        HNSCC\n",
       "4137     KMT2D            0.171171        HNSCC\n",
       "4863     MUC16            0.171171        HNSCC\n",
       "4868      MUC4            0.126126        HNSCC\n",
       "6016      PLEC            0.117117        HNSCC\n",
       "2139       DMD            0.081081        HNSCC\n",
       "10096     PTEN            0.789474  Endometrial\n",
       "9415    PIK3CA            0.494737  Endometrial\n",
       "836     ARID1A            0.452632  Endometrial\n",
       "9419    PIK3R1            0.389474  Endometrial\n",
       "6514      KRAS            0.326316  Endometrial\n",
       "2935    CTNNB1            0.305263  Endometrial\n",
       "2917      CTCF            0.284211  Endometrial\n",
       "6494     KMT2B            0.242105  Endometrial\n",
       "14231    ZFHX3            0.221053  Endometrial\n",
       "13126     TP53            0.221053  Endometrial\n",
       "2997     PBRM1            0.400000        CCRCC\n",
       "3721     SETD2            0.136364        CCRCC\n",
       "4439       TTN            0.118182        CCRCC"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_frequency_df = []\n",
    "for cancer in cancer_list:\n",
    "    mutation_df = cancer.get_somatic_mutation()\n",
    "    mutation_df = mutation_df[mutation_df.Mutation != 'Silent']\n",
    "    mutation_df = mutation_df[mutation_df.Mutation != 'RNA']\n",
    "    mutation_df = mutation_df[mutation_df.Mutation != 'synonymous SNV']\n",
    "    #look only at gene with both transciptomic and proteomic data\n",
    "    gene_df = cancer.get_transcriptomics('tumor')\n",
    "    if isinstance(gene_df.columns, pd.MultiIndex):\n",
    "        gene_df = gene_df.droplevel('Database_ID', axis = 1)\n",
    "    prot_df = cancer.get_proteomics('tumor')\n",
    "    if isinstance(prot_df.columns, pd.MultiIndex):\n",
    "        prot_df = prot_df.droplevel('Database_ID', axis = 1)\n",
    "    gene_df['Patient_ID'] = gene_df.index\n",
    "    prot_df['Patient_ID'] = prot_df.index\n",
    "    gene_df = gene_df.melt(id_vars = 'Patient_ID',\n",
    "                           var_name ='Gene',\n",
    "                           value_name = 'Transcriptomics')\n",
    "    prot_df = prot_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomics')\n",
    "    omics_df = pd.merge(gene_df, prot_df, how = 'outer')\n",
    "    omics_df = omics_df.dropna()\n",
    "    mutation_df = mutation_df.reset_index()\n",
    "    df = pd.merge(omics_df, mutation_df, how = 'outer')\n",
    "    df = df.drop(columns = 'Location')\n",
    "    df['Mutation'] = df.Mutation.notnull()\n",
    "    df = df.dropna()\n",
    "    #filter to only include genes with a minimum number of samples\n",
    "    genes = df.groupby(['Mutation', 'Gene'])\n",
    "    sample_count = genes.size()\n",
    "    cutoff = 15\n",
    "    if cancer == en:\n",
    "        cutoff = 10\n",
    "    sample_count = sample_count[sample_count>=cutoff]\n",
    "    sample_count\n",
    "    mutated_genes = set(sample_count[sample_count.index.get_level_values(0) == False].index.get_level_values('Gene'))\n",
    "    wt_genes = set(sample_count[sample_count.index.get_level_values(0) == True].index.get_level_values('Gene'))\n",
    "    genes = wt_genes & mutated_genes\n",
    "    #get mutation frequency \n",
    "    df = ut.get_frequently_mutated(cancer, 0)\n",
    "    df = df[df.Gene.isin(genes)]\n",
    "    df = df.sort_values('Unique_Samples_Mut', ascending = False)\n",
    "    df=df.drop(columns = ['Missense_Mut', 'Truncation_Mut'])\n",
    "    if len(df) > 10:\n",
    "        df = df[0:10]\n",
    "    df['Cancer'] = [cancer_dict[cancer]] * len(df)\n",
    "    mutation_frequency_df.append(df)\n",
    "mutation_frequency_df = pd.concat(mutation_frequency_df)\n",
    "mutation_frequency_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write bash command for finding permutation p-val for each gene/cancer pair\n",
    "header = '#!/bin/bash\\n#SBATCH --time=168:00:00   # walltime\\n#SBATCH --ntasks=2   # number of processor cores (i.e. tasks)\\n#SBATCH --mem-per-cpu=8192M   # memory per CPU core\\n#SBATCH --mail-user=nanelbarton@gmail.com   # email address\\n#SBATCH --mail-type=BEGIN\\n#SBATCH --mail-type=END\\n# Set the max number of threads to use for programs using OpenMP. Should be <= ppn. Does nothing if the program doesn\\'t use OpenMP.\\nexport OMP_NUM_THREADS=$SLURM_CPUS_ON_NODE\\n'\n",
    "for index, row in mutation_frequency_df.iterrows():\n",
    "    file_name = 'trans_' + row.Cancer + '_' + row.Gene + '.sh'\n",
    "    with open(file_name, 'w+') as file:\n",
    "        file.write(header)\n",
    "        s = 'python3 transmutation_effects.py ' + row.Cancer + ' ' + row.Gene + ' 10000\\n'\n",
    "        file.write(s)\n",
    "        file.close()\n",
    "    with open ('run_transmutation_scripts.sh', 'a+') as file:\n",
    "        s = 'sbatch ' + file_name +'\\n'\n",
    "        file.write(s)\n",
    "        file.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
