{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "11e22838",
   "metadata": {},
   "source": [
    "## Getting the genes that have a significant P-value for each cancer's delta correlation\n",
    "### Do we want to use all of the significant genes (approx. 5000 genes for each cancer) or just the most significant (i.e. lowest 50 p-values)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "10cf2246",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pcprutils as ut\n",
    "import scipy.odr\n",
    "import altair as alt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9ae8d797",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Delta_Correlation</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.268533</td>\n",
       "      <td>5.703182e-02</td>\n",
       "      <td>1.320375e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.192038</td>\n",
       "      <td>1.063340e-04</td>\n",
       "      <td>6.401858e-04</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.191619</td>\n",
       "      <td>1.277644e-01</td>\n",
       "      <td>2.439276e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.019654</td>\n",
       "      <td>8.963138e-01</td>\n",
       "      <td>9.409267e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS</td>\n",
       "      <td>-0.169937</td>\n",
       "      <td>6.007042e-02</td>\n",
       "      <td>1.375402e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50303</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.073854</td>\n",
       "      <td>6.395306e-01</td>\n",
       "      <td>6.962162e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50304</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.734569</td>\n",
       "      <td>1.583743e-10</td>\n",
       "      <td>1.504331e-09</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50305</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.440861</td>\n",
       "      <td>2.253307e-06</td>\n",
       "      <td>9.499585e-06</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50306</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.422001</td>\n",
       "      <td>3.731185e-11</td>\n",
       "      <td>3.889196e-10</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50307</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.717501</td>\n",
       "      <td>2.955080e-06</td>\n",
       "      <td>1.215796e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50308 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Delta_Correlation       P_Value           FDR Cancer\n",
       "0        A1BG          -0.268533  5.703182e-02  1.320375e-01  CCRCC\n",
       "1        A1CF           0.192038  1.063340e-04  6.401858e-04  CCRCC\n",
       "2         A2M          -0.191619  1.277644e-01  2.439276e-01  CCRCC\n",
       "3        AAAS           0.019654  8.963138e-01  9.409267e-01  CCRCC\n",
       "4        AACS          -0.169937  6.007042e-02  1.375402e-01  CCRCC\n",
       "...       ...                ...           ...           ...    ...\n",
       "50303    ZXDC           0.073854  6.395306e-01  6.962162e-01   LUAD\n",
       "50304  ZYG11B           0.734569  1.583743e-10  1.504331e-09   LUAD\n",
       "50305     ZYX           0.440861  2.253307e-06  9.499585e-06   LUAD\n",
       "50306   ZZEF1           0.422001  3.731185e-11  3.889196e-10   LUAD\n",
       "50307    ZZZ3           0.717501  2.955080e-06  1.215796e-05   LUAD\n",
       "\n",
       "[50308 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_correlation_df = pd.read_csv('../data/delta_correlation_df.csv')\n",
    "delta_correlation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c9335365",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Delta_Correlation</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.192038</td>\n",
       "      <td>1.063340e-04</td>\n",
       "      <td>6.401858e-04</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>-0.263372</td>\n",
       "      <td>1.531939e-02</td>\n",
       "      <td>4.572896e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.364999</td>\n",
       "      <td>7.633656e-04</td>\n",
       "      <td>3.646252e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.356932</td>\n",
       "      <td>6.347464e-14</td>\n",
       "      <td>1.385608e-12</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.415815</td>\n",
       "      <td>6.031667e-03</td>\n",
       "      <td>2.119475e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50302</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>0.837582</td>\n",
       "      <td>2.770355e-08</td>\n",
       "      <td>1.722312e-07</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50304</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.734569</td>\n",
       "      <td>1.583743e-10</td>\n",
       "      <td>1.504331e-09</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50305</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.440861</td>\n",
       "      <td>2.253307e-06</td>\n",
       "      <td>9.499585e-06</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50306</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.422001</td>\n",
       "      <td>3.731185e-11</td>\n",
       "      <td>3.889196e-10</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50307</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.717501</td>\n",
       "      <td>2.955080e-06</td>\n",
       "      <td>1.215796e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21749 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Delta_Correlation       P_Value           FDR Cancer\n",
       "1        A1CF           0.192038  1.063340e-04  6.401858e-04  CCRCC\n",
       "5       AADAT          -0.263372  1.531939e-02  4.572896e-02  CCRCC\n",
       "7       AAGAB           0.364999  7.633656e-04  3.646252e-03  CCRCC\n",
       "8        AAK1           0.356932  6.347464e-14  1.385608e-12  CCRCC\n",
       "11       AARS           0.415815  6.031667e-03  2.119475e-02  CCRCC\n",
       "...       ...                ...           ...           ...    ...\n",
       "50302   ZWINT           0.837582  2.770355e-08  1.722312e-07   LUAD\n",
       "50304  ZYG11B           0.734569  1.583743e-10  1.504331e-09   LUAD\n",
       "50305     ZYX           0.440861  2.253307e-06  9.499585e-06   LUAD\n",
       "50306   ZZEF1           0.422001  3.731185e-11  3.889196e-10   LUAD\n",
       "50307    ZZZ3           0.717501  2.955080e-06  1.215796e-05   LUAD\n",
       "\n",
       "[21749 rows x 5 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get all significant delta correlations\n",
    "\n",
    "delta_corr_sig = delta_correlation_df.loc[delta_correlation_df['FDR'] < 0.05]\n",
    "delta_corr_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4be676d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_freq = delta_corr_sig[\"Gene\"].\\\n",
    "value_counts().\\\n",
    "to_frame().\\\n",
    "reset_index(drop=False).\\\n",
    "rename(columns={\"Gene\": \"count\", \"index\": \"Gene\"})\n",
    "\n",
    "delta_corr_sig_with_counts = delta_corr_sig.merge(gene_freq, on=\"Gene\", how=\"inner\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "185ae351",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Delta_Correlation</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.364999</td>\n",
       "      <td>7.633656e-04</td>\n",
       "      <td>3.646252e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.534630</td>\n",
       "      <td>9.313146e-27</td>\n",
       "      <td>2.721592e-25</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.541641</td>\n",
       "      <td>9.576553e-08</td>\n",
       "      <td>5.369596e-07</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.626523</td>\n",
       "      <td>1.265797e-04</td>\n",
       "      <td>1.448009e-03</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.356932</td>\n",
       "      <td>6.347464e-14</td>\n",
       "      <td>1.385608e-12</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.614117</td>\n",
       "      <td>1.000902e-11</td>\n",
       "      <td>3.887029e-10</td>\n",
       "      <td>HNSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.655869</td>\n",
       "      <td>1.061364e-08</td>\n",
       "      <td>5.576651e-08</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.711044</td>\n",
       "      <td>4.736550e-04</td>\n",
       "      <td>4.325435e-03</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.415815</td>\n",
       "      <td>6.031667e-03</td>\n",
       "      <td>2.119475e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.485189</td>\n",
       "      <td>2.635135e-22</td>\n",
       "      <td>5.364523e-21</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.641203</td>\n",
       "      <td>4.377718e-40</td>\n",
       "      <td>1.258156e-37</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.730099</td>\n",
       "      <td>2.820537e-05</td>\n",
       "      <td>4.122256e-04</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9654</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>0.483164</td>\n",
       "      <td>2.912272e-05</td>\n",
       "      <td>8.598400e-05</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9653</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>0.565982</td>\n",
       "      <td>3.124882e-05</td>\n",
       "      <td>3.164216e-04</td>\n",
       "      <td>HNSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9655</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>0.649032</td>\n",
       "      <td>3.683187e-07</td>\n",
       "      <td>1.817918e-06</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9652</th>\n",
       "      <td>AARS2</td>\n",
       "      <td>0.733763</td>\n",
       "      <td>2.162371e-03</td>\n",
       "      <td>1.498109e-02</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>0.450818</td>\n",
       "      <td>7.373637e-03</td>\n",
       "      <td>2.865766e-02</td>\n",
       "      <td>HNSCC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>0.464768</td>\n",
       "      <td>4.388407e-04</td>\n",
       "      <td>2.263637e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>0.505070</td>\n",
       "      <td>8.178253e-06</td>\n",
       "      <td>2.699022e-05</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>0.645277</td>\n",
       "      <td>3.941371e-18</td>\n",
       "      <td>1.040584e-16</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AARSD1</td>\n",
       "      <td>0.815017</td>\n",
       "      <td>4.670956e-03</td>\n",
       "      <td>2.752783e-02</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>AASS</td>\n",
       "      <td>-0.278696</td>\n",
       "      <td>1.150360e-06</td>\n",
       "      <td>1.030519e-05</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>AASS</td>\n",
       "      <td>0.446930</td>\n",
       "      <td>1.591147e-05</td>\n",
       "      <td>5.572996e-05</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>AASS</td>\n",
       "      <td>0.483940</td>\n",
       "      <td>1.686322e-08</td>\n",
       "      <td>5.986765e-07</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>AASS</td>\n",
       "      <td>0.526886</td>\n",
       "      <td>9.033256e-24</td>\n",
       "      <td>1.090013e-21</td>\n",
       "      <td>HNSCC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>AASS</td>\n",
       "      <td>0.664868</td>\n",
       "      <td>6.183710e-09</td>\n",
       "      <td>3.365928e-08</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9660</th>\n",
       "      <td>ABCF2</td>\n",
       "      <td>0.465571</td>\n",
       "      <td>2.684193e-07</td>\n",
       "      <td>4.588023e-06</td>\n",
       "      <td>HNSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9659</th>\n",
       "      <td>ABCF2</td>\n",
       "      <td>0.769383</td>\n",
       "      <td>2.429924e-04</td>\n",
       "      <td>2.463422e-03</td>\n",
       "      <td>Endometrial</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9662</th>\n",
       "      <td>ABCF2</td>\n",
       "      <td>0.814284</td>\n",
       "      <td>1.237177e-21</td>\n",
       "      <td>4.823550e-20</td>\n",
       "      <td>LUAD</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9661</th>\n",
       "      <td>ABCF2</td>\n",
       "      <td>0.821995</td>\n",
       "      <td>3.610732e-27</td>\n",
       "      <td>1.111962e-25</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene  Delta_Correlation       P_Value           FDR       Cancer  \\\n",
       "3      AAGAB           0.364999  7.633656e-04  3.646252e-03        CCRCC   \n",
       "5      AAGAB           0.534630  9.313146e-27  2.721592e-25         LSCC   \n",
       "6      AAGAB           0.541641  9.576553e-08  5.369596e-07         LUAD   \n",
       "4      AAGAB           0.626523  1.265797e-04  1.448009e-03  Endometrial   \n",
       "7       AAK1           0.356932  6.347464e-14  1.385608e-12        CCRCC   \n",
       "9       AAK1           0.614117  1.000902e-11  3.887029e-10        HNSCC   \n",
       "10      AAK1           0.655869  1.061364e-08  5.576651e-08         LSCC   \n",
       "8       AAK1           0.711044  4.736550e-04  4.325435e-03  Endometrial   \n",
       "11      AARS           0.415815  6.031667e-03  2.119475e-02        CCRCC   \n",
       "13      AARS           0.485189  2.635135e-22  5.364523e-21         LSCC   \n",
       "14      AARS           0.641203  4.377718e-40  1.258156e-37         LUAD   \n",
       "12      AARS           0.730099  2.820537e-05  4.122256e-04  Endometrial   \n",
       "9654   AARS2           0.483164  2.912272e-05  8.598400e-05         LSCC   \n",
       "9653   AARS2           0.565982  3.124882e-05  3.164216e-04        HNSCC   \n",
       "9655   AARS2           0.649032  3.683187e-07  1.817918e-06         LUAD   \n",
       "9652   AARS2           0.733763  2.162371e-03  1.498109e-02  Endometrial   \n",
       "17    AARSD1           0.450818  7.373637e-03  2.865766e-02        HNSCC   \n",
       "15    AARSD1           0.464768  4.388407e-04  2.263637e-03        CCRCC   \n",
       "18    AARSD1           0.505070  8.178253e-06  2.699022e-05         LSCC   \n",
       "19    AARSD1           0.645277  3.941371e-18  1.040584e-16         LUAD   \n",
       "16    AARSD1           0.815017  4.670956e-03  2.752783e-02  Endometrial   \n",
       "20      AASS          -0.278696  1.150360e-06  1.030519e-05        CCRCC   \n",
       "24      AASS           0.446930  1.591147e-05  5.572996e-05         LUAD   \n",
       "21      AASS           0.483940  1.686322e-08  5.986765e-07  Endometrial   \n",
       "22      AASS           0.526886  9.033256e-24  1.090013e-21        HNSCC   \n",
       "23      AASS           0.664868  6.183710e-09  3.365928e-08         LSCC   \n",
       "9660   ABCF2           0.465571  2.684193e-07  4.588023e-06        HNSCC   \n",
       "9659   ABCF2           0.769383  2.429924e-04  2.463422e-03  Endometrial   \n",
       "9662   ABCF2           0.814284  1.237177e-21  4.823550e-20         LUAD   \n",
       "9661   ABCF2           0.821995  3.610732e-27  1.111962e-25         LSCC   \n",
       "\n",
       "      count  \n",
       "3         4  \n",
       "5         4  \n",
       "6         4  \n",
       "4         4  \n",
       "7         4  \n",
       "9         4  \n",
       "10        4  \n",
       "8         4  \n",
       "11        4  \n",
       "13        4  \n",
       "14        4  \n",
       "12        4  \n",
       "9654      4  \n",
       "9653      4  \n",
       "9655      4  \n",
       "9652      4  \n",
       "17        5  \n",
       "15        5  \n",
       "18        5  \n",
       "19        5  \n",
       "16        5  \n",
       "20        5  \n",
       "24        5  \n",
       "21        5  \n",
       "22        5  \n",
       "23        5  \n",
       "9660      4  \n",
       "9659      4  \n",
       "9662      4  \n",
       "9661      4  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_corr_sig_with_counts[delta_corr_sig_with_counts[\"count\"] > 3].sort_values(by=['Gene','Delta_Correlation']).head(30)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "72ff79e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all significant genes for each specific cancer\n",
    "\n",
    "ccrcc_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'CCRCC']\n",
    "hnscc_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'HNSCC']\n",
    "endo_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'Endometrial']\n",
    "lscc_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'LSCC']\n",
    "luad_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'LUAD']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ea880e1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "sort_hnscc = hnscc_delta.sort_values(by=['FDR'])\n",
    "hnscc_genes = sort_hnscc['Gene'].to_list()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0e6b2db",
   "metadata": {},
   "source": [
    "## Calculate regression line for all patients for each tissue type for each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a38bc205",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 41)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 163)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 41)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 41)\n"
     ]
    }
   ],
   "source": [
    "prot_trans = ut.load_prot_trans([\n",
    "    \"ccrcc\",\n",
    "    \"endometrial\",\n",
    "    \"hnscc\",\n",
    "    \"lscc\",\n",
    "    \"luad\",\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7408728f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_regression(corr_df, prot_trans_df):\n",
    "    \n",
    "    sig_prot_trans = prot_trans_df[prot_trans_df[\"Gene\"].isin(corr_df[\"Gene\"])]\n",
    "    \n",
    "    def get_odr(df):\n",
    "        \n",
    "        def f(B, x):\n",
    "            return B[0]*x + B[1]\n",
    "        \n",
    "        linear = scipy.odr.Model(f)\n",
    "        data = scipy.odr.Data(df[\"Transcriptomics\"], df[\"Proteomics\"])\n",
    "        odr = scipy.odr.ODR(data, linear, beta0=[1, 1])\n",
    "        output = odr.run()\n",
    "        \n",
    "        return output.beta\n",
    "    \n",
    "    results = sig_prot_trans.\\\n",
    "    groupby([\"Tissue\", \"Gene\"]).\\\n",
    "    apply(get_odr).\\\n",
    "    reset_index(drop=False).\\\n",
    "    rename(columns={0: \"beta\"})\n",
    "    \n",
    "    mb = pd.DataFrame(results[\"beta\"].tolist(), columns=[\"m\", \"b\"])\n",
    "    results = pd.concat([results, mb], axis=1).drop(columns=[\"beta\"])\n",
    "    \n",
    "    return results\n",
    "\n",
    "hnscc_results = get_regression(hnscc_delta, prot_trans[\"hnscc\"])\n",
    "luad_results = get_regression(luad_delta, prot_trans[\"luad\"])\n",
    "lscc_results = get_regression(lscc_delta, prot_trans[\"lscc\"])\n",
    "ccrcc_results = get_regression(ccrcc_delta, prot_trans[\"ccrcc\"])\n",
    "endo_results = get_regression(endo_delta, prot_trans[\"endometrial\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "93292e84",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_chart(df, m, b):\n",
    "\n",
    "    scatter = alt.Chart(df).mark_circle().encode(\n",
    "        x=\"Transcriptomics\",\n",
    "        y=\"Proteomics\",\n",
    "        color=\"Tissue\",\n",
    "        tooltip=[\"Patient_ID\"]\n",
    "    )\n",
    "    \n",
    "    x = np.arange(14)\n",
    "    y = m * x + b\n",
    "    \n",
    "    df = pd.DataFrame({\"x\":x, \"y\": y})\n",
    "    \n",
    "    line = alt.Chart(df).mark_line().encode(\n",
    "        x='x',\n",
    "        y='y'\n",
    "    )\n",
    "    \n",
    "    return scatter+line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4a5236da",
   "metadata": {},
   "outputs": [],
   "source": [
    "#for gene in hnscc_results['Gene']:\n",
    "gene = 'AADAC'\n",
    "tissue_type = ['Tumor', 'Normal']\n",
    "\n",
    "chart = []\n",
    "\n",
    "for tissue in tissue_type:\n",
    "    pr_tr_cancer = prot_trans['hnscc']\n",
    "    gene_df = pr_tr_cancer.loc[(pr_tr_cancer['Gene'] == gene) & (pr_tr_cancer['Tissue'] == tissue)]\n",
    "    gene_slope = hnscc_results.loc[(hnscc_results['Gene'] == gene) & (hnscc_results['Tissue'] == tissue)]\n",
    "    m = gene_slope['m'].values[0]\n",
    "    b = gene_slope['b'].values[0]\n",
    "    \n",
    "    chart.append(make_chart(gene_df, m, b))\n",
    "\n",
    "a = alt.vconcat(*chart)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "35108bef",
   "metadata": {},
   "outputs": [],
   "source": [
    "hnscc_results.to_csv('hnscc_regression.tsv', sep='\\t', index=False)\n",
    "ccrcc_results.to_csv('ccrcc_regression.tsv', sep='\\t', index=False)\n",
    "lscc_results.to_csv('lscc_regression.tsv', sep='\\t', index=False)\n",
    "luad_results.to_csv('luad_regression.tsv', sep='\\t', index=False)\n",
    "endo_results.to_csv('endometrial_regression.tsv', sep='\\t', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
