{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "11e22838",
   "metadata": {},
   "source": [
    "## Getting the genes that have a significant P-value for each cancer's delta correlation\n",
    "### Do we want to use all of the significant genes (approx. 5000 genes for each cancer) or just the most significant (i.e. lowest 50 p-values)?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "10cf2246",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import pcprutils as ut\n",
    "import scipy.odr\n",
    "import altair as alt\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "9ae8d797",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Delta_Correlation</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.268533</td>\n",
       "      <td>5.703182e-02</td>\n",
       "      <td>1.320375e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.192038</td>\n",
       "      <td>1.063340e-04</td>\n",
       "      <td>6.401858e-04</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.191619</td>\n",
       "      <td>1.277644e-01</td>\n",
       "      <td>2.439276e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.019654</td>\n",
       "      <td>8.963138e-01</td>\n",
       "      <td>9.409267e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AACS</td>\n",
       "      <td>-0.169937</td>\n",
       "      <td>6.007042e-02</td>\n",
       "      <td>1.375402e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50303</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>0.073854</td>\n",
       "      <td>6.395306e-01</td>\n",
       "      <td>6.962162e-01</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50304</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.734569</td>\n",
       "      <td>1.583743e-10</td>\n",
       "      <td>1.504331e-09</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50305</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.440861</td>\n",
       "      <td>2.253307e-06</td>\n",
       "      <td>9.499585e-06</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50306</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.422001</td>\n",
       "      <td>3.731185e-11</td>\n",
       "      <td>3.889196e-10</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50307</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.717501</td>\n",
       "      <td>2.955080e-06</td>\n",
       "      <td>1.215796e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50308 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Delta_Correlation       P_Value           FDR Cancer\n",
       "0        A1BG          -0.268533  5.703182e-02  1.320375e-01  CCRCC\n",
       "1        A1CF           0.192038  1.063340e-04  6.401858e-04  CCRCC\n",
       "2         A2M          -0.191619  1.277644e-01  2.439276e-01  CCRCC\n",
       "3        AAAS           0.019654  8.963138e-01  9.409267e-01  CCRCC\n",
       "4        AACS          -0.169937  6.007042e-02  1.375402e-01  CCRCC\n",
       "...       ...                ...           ...           ...    ...\n",
       "50303    ZXDC           0.073854  6.395306e-01  6.962162e-01   LUAD\n",
       "50304  ZYG11B           0.734569  1.583743e-10  1.504331e-09   LUAD\n",
       "50305     ZYX           0.440861  2.253307e-06  9.499585e-06   LUAD\n",
       "50306   ZZEF1           0.422001  3.731185e-11  3.889196e-10   LUAD\n",
       "50307    ZZZ3           0.717501  2.955080e-06  1.215796e-05   LUAD\n",
       "\n",
       "[50308 rows x 5 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "delta_correlation_df = pd.read_csv('../data/delta_correlation_df.csv')\n",
    "delta_correlation_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "c9335365",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Delta_Correlation</th>\n",
       "      <th>P_Value</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.192038</td>\n",
       "      <td>1.063340e-04</td>\n",
       "      <td>6.401858e-04</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AADAT</td>\n",
       "      <td>-0.263372</td>\n",
       "      <td>1.531939e-02</td>\n",
       "      <td>4.572896e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AAGAB</td>\n",
       "      <td>0.364999</td>\n",
       "      <td>7.633656e-04</td>\n",
       "      <td>3.646252e-03</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AAK1</td>\n",
       "      <td>0.356932</td>\n",
       "      <td>6.347464e-14</td>\n",
       "      <td>1.385608e-12</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AARS</td>\n",
       "      <td>0.415815</td>\n",
       "      <td>6.031667e-03</td>\n",
       "      <td>2.119475e-02</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50302</th>\n",
       "      <td>ZWINT</td>\n",
       "      <td>0.837582</td>\n",
       "      <td>2.770355e-08</td>\n",
       "      <td>1.722312e-07</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50304</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.734569</td>\n",
       "      <td>1.583743e-10</td>\n",
       "      <td>1.504331e-09</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50305</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.440861</td>\n",
       "      <td>2.253307e-06</td>\n",
       "      <td>9.499585e-06</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50306</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.422001</td>\n",
       "      <td>3.731185e-11</td>\n",
       "      <td>3.889196e-10</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50307</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.717501</td>\n",
       "      <td>2.955080e-06</td>\n",
       "      <td>1.215796e-05</td>\n",
       "      <td>LUAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21749 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Delta_Correlation       P_Value           FDR Cancer\n",
       "1        A1CF           0.192038  1.063340e-04  6.401858e-04  CCRCC\n",
       "5       AADAT          -0.263372  1.531939e-02  4.572896e-02  CCRCC\n",
       "7       AAGAB           0.364999  7.633656e-04  3.646252e-03  CCRCC\n",
       "8        AAK1           0.356932  6.347464e-14  1.385608e-12  CCRCC\n",
       "11       AARS           0.415815  6.031667e-03  2.119475e-02  CCRCC\n",
       "...       ...                ...           ...           ...    ...\n",
       "50302   ZWINT           0.837582  2.770355e-08  1.722312e-07   LUAD\n",
       "50304  ZYG11B           0.734569  1.583743e-10  1.504331e-09   LUAD\n",
       "50305     ZYX           0.440861  2.253307e-06  9.499585e-06   LUAD\n",
       "50306   ZZEF1           0.422001  3.731185e-11  3.889196e-10   LUAD\n",
       "50307    ZZZ3           0.717501  2.955080e-06  1.215796e-05   LUAD\n",
       "\n",
       "[21749 rows x 5 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Get all significant delta correlations\n",
    "\n",
    "delta_corr_sig = delta_correlation_df.loc[delta_correlation_df['FDR'] < 0.05]\n",
    "delta_corr_sig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "72ff79e8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Get all significant genes for each specific cancer\n",
    "\n",
    "ccrcc_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'CCRCC']\n",
    "hnscc_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'HNSCC']\n",
    "endo_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'Endometrial']\n",
    "lscc_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'LSCC']\n",
    "luad_delta = delta_corr_sig.loc[delta_corr_sig['Cancer'] == 'LUAD']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a0e6b2db",
   "metadata": {},
   "source": [
    "## Calculate regression line for all patients for each tissue type for each gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "a38bc205",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                          \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 41)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 163)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 41)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 27)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (/Library/Frameworks/Python.framework/Versions/3.9/lib/python3.9/site-packages/pcprutils/__init__.py, line 41)\n"
     ]
    }
   ],
   "source": [
    "prot_trans = ut.load_prot_trans([\n",
    "    \"ccrcc\",\n",
    "    \"endometrial\",\n",
    "    \"hnscc\",\n",
    "    \"lscc\",\n",
    "    \"luad\",\n",
    "])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "7408728f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_regression(corr_df, prot_trans_df):\n",
    "    \n",
    "    sig_prot_trans = prot_trans_df[prot_trans_df[\"Gene\"].isin(corr_df[\"Gene\"])]\n",
    "    \n",
    "    def get_odr(df):\n",
    "        \n",
    "        def f(B, x):\n",
    "            return B[0]*x + B[1]\n",
    "        \n",
    "        linear = scipy.odr.Model(f)\n",
    "        data = scipy.odr.Data(df[\"Transcriptomics\"], df[\"Proteomics\"])\n",
    "        odr = scipy.odr.ODR(data, linear, beta0=[1, 1])\n",
    "        output = odr.run()\n",
    "        \n",
    "        return output.beta\n",
    "    \n",
    "    results = sig_prot_trans.\\\n",
    "    groupby([\"Tissue\", \"Gene\"]).\\\n",
    "    apply(get_odr).\\\n",
    "    reset_index(drop=False).\\\n",
    "    rename(columns={0: \"beta\"})\n",
    "    \n",
    "    mb = pd.DataFrame(results[\"beta\"].tolist(), columns=[\"m\", \"b\"])\n",
    "    results = pd.concat([results, mb], axis=1).drop(columns=[\"beta\"])\n",
    "    \n",
    "    return results\n",
    "\n",
    "hnscc_results = get_regression(hnscc_delta, prot_trans[\"hnscc\"])\n",
    "luad_results = get_regression(luad_delta, prot_trans[\"luad\"])\n",
    "lscc_results = get_regression(lscc_delta, prot_trans[\"lscc\"])\n",
    "ccrcc_results = get_regression(ccrcc_delta, prot_trans[\"ccrcc\"])\n",
    "endo_results = get_regression(endo_delta, prot_trans[\"endometrial\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "93292e84",
   "metadata": {},
   "outputs": [],
   "source": [
    "def make_chart(df, m, b):\n",
    "\n",
    "    scatter = alt.Chart(df).mark_circle().encode(\n",
    "        x=\"Transcriptomics\",\n",
    "        y=\"Proteomics\",\n",
    "        color=\"Tissue\",\n",
    "        tooltip=[\"Patient_ID\"]\n",
    "    )\n",
    "    \n",
    "    x = np.arange(14)\n",
    "    y = m * x + b\n",
    "    \n",
    "    df = pd.DataFrame({\"x\":x, \"y\": y})\n",
    "    \n",
    "    line = alt.Chart(df).mark_line().encode(\n",
    "        x='x',\n",
    "        y='y'\n",
    "    )\n",
    "    \n",
    "    return scatter+line"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4a5236da",
   "metadata": {},
   "outputs": [],
   "source": [
    "#for gene in hnscc_results['Gene']:\n",
    "gene = 'AADAC'\n",
    "tissue_type = ['Tumor', 'Normal']\n",
    "\n",
    "chart = []\n",
    "\n",
    "for tissue in tissue_type:\n",
    "    pr_tr_cancer = prot_trans['hnscc']\n",
    "    gene_df = pr_tr_cancer.loc[(pr_tr_cancer['Gene'] == gene) & (pr_tr_cancer['Tissue'] == tissue)]\n",
    "    gene_slope = hnscc_results.loc[(hnscc_results['Gene'] == gene) & (hnscc_results['Tissue'] == tissue)]\n",
    "    m = gene_slope['m'].values[0]\n",
    "    b = gene_slope['b'].values[0]\n",
    "    \n",
    "    chart.append(make_chart(gene_df, m, b))\n",
    "\n",
    "a = alt.vconcat(*chart)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "35108bef",
   "metadata": {},
   "outputs": [],
   "source": [
    "hnscc_results.to_csv('hnscc_regression.tsv', sep='\\t', index=False)\n",
    "ccrcc_results.to_csv('ccrcc_regression.tsv', sep='\\t', index=False)\n",
    "lscc_results.to_csv('lscc_regression.tsv', sep='\\t', index=False)\n",
    "luad_results.to_csv('luad_regression.tsv', sep='\\t', index=False)\n",
    "endo_results.to_csv('endometrial_regression.tsv', sep='\\t', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
