{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import pandas as pd\n",
    "import scipy.stats as stats\n",
    "import cptac.utils as ut\n",
    "import scipy\n",
    "import statsmodels.stats.multitest as ssm\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that pdac index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-891d3f5549c9>, line 5)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The pdac data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-891d3f5549c9>, line 6)\n"
     ]
    }
   ],
   "source": [
    "ccrcc = cptac.Ccrcc()\n",
    "en = cptac.Endometrial()\n",
    "luad = cptac.Luad()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()\n",
    "pdac = cptac.Pdac()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [ccrcc, en, luad, hnscc, lscc, pdac]\n",
    "cancer_names = ['CCRCC', 'Endometrial', 'LUAD', 'HNSCC', 'LSCC', 'PDAC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def log2_fold_change(tumor, normal):\n",
    "    tumor = np.mean(tumor)\n",
    "    normal = np.mean(normal)\n",
    "    if normal == 0 or tumor == 0:\n",
    "        return(float('Nan'))\n",
    "    fold_change = tumor / normal\n",
    "    fold_change = abs(fold_change)\n",
    "    log2_change = np.log2(fold_change)\n",
    "    return log2_change    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-8-a6967ad66291>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-8-a6967ad66291>, line 11)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-8-a6967ad66291>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-8-a6967ad66291>, line 11)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-8-a6967ad66291>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-8-a6967ad66291>, line 11)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>statistic</th>\n",
       "      <th>pval</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Log2_fold_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>4.062020</td>\n",
       "      <td>4.864990e-05</td>\n",
       "      <td>7.977389e-05</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.846625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>-6.998855</td>\n",
       "      <td>2.580621e-12</td>\n",
       "      <td>6.376757e-12</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-0.218309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>6.557556</td>\n",
       "      <td>5.469684e-11</td>\n",
       "      <td>1.261222e-10</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-1.256118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>0.979796</td>\n",
       "      <td>3.271869e-01</td>\n",
       "      <td>3.660405e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>3.045874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>9.324065</td>\n",
       "      <td>1.119664e-20</td>\n",
       "      <td>4.340440e-20</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>4.527650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11626</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>2.500000</td>\n",
       "      <td>1.241933e-02</td>\n",
       "      <td>2.168582e-02</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>0.068020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11627</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>1.778682</td>\n",
       "      <td>7.529194e-02</td>\n",
       "      <td>1.102090e-01</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>0.005597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11628</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>9.327386</td>\n",
       "      <td>1.085139e-20</td>\n",
       "      <td>2.743751e-19</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>0.036329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11629</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-4.140393</td>\n",
       "      <td>3.467108e-05</td>\n",
       "      <td>8.841467e-05</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>-0.008661</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11630</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-4.816231</td>\n",
       "      <td>1.462954e-06</td>\n",
       "      <td>4.469559e-06</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>-0.028503</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>66996 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  statistic          pval           FDR Cancer  Log2_fold_change\n",
       "0        A1BG   4.062020  4.864990e-05  7.977389e-05  CCRCC          0.846625\n",
       "1        A1CF  -6.998855  2.580621e-12  6.376757e-12  CCRCC         -0.218309\n",
       "2         A2M   6.557556  5.469684e-11  1.261222e-10  CCRCC         -1.256118\n",
       "3      A4GALT   0.979796  3.271869e-01  3.660405e-01  CCRCC          3.045874\n",
       "4        AAAS   9.324065  1.119664e-20  4.340440e-20  CCRCC          4.527650\n",
       "...       ...        ...           ...           ...    ...               ...\n",
       "11626    ZXDC   2.500000  1.241933e-02  2.168582e-02   PDAC          0.068020\n",
       "11627  ZYG11B   1.778682  7.529194e-02  1.102090e-01   PDAC          0.005597\n",
       "11628     ZYX   9.327386  1.085139e-20  2.743751e-19   PDAC          0.036329\n",
       "11629   ZZEF1  -4.140393  3.467108e-05  8.841467e-05   PDAC         -0.008661\n",
       "11630    ZZZ3  -4.816231  1.462954e-06  4.469559e-06   PDAC         -0.028503\n",
       "\n",
       "[66996 rows x 6 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diff_expression_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_normal_df = cancer.get_proteomics('normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID')\n",
    "    prot_normal_df.reset_index(inplace = True)\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    prot_normal_df['Tissue'] = ['normal'] * len(prot_normal_df)\n",
    "    prot_tumor_df = cancer.get_proteomics('tumor')\n",
    "    if(isinstance(prot_tumor_df.columns, pd.MultiIndex)):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID')\n",
    "    prot_tumor_df.reset_index(inplace = True)\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    prot_tumor_df['Tissue'] = ['tumor'] * len(prot_tumor_df)\n",
    "    prot_df = pd.merge(prot_tumor_df, prot_normal_df, how = 'outer')\n",
    "    prot_df = prot_df.dropna()\n",
    "    fold_change = prot_df.groupby('Gene').apply(\n",
    "        lambda df: log2_fold_change(df[df['Tissue']=='tumor'].Proteomic,\n",
    "                                    df[df['Tissue']=='normal'].Proteomic))\n",
    "    fold_change = pd.DataFrame(fold_change, columns = ['Log2_fold_change'])\n",
    "    fold_change.reset_index(inplace = True)\n",
    "    ranksums = prot_df.groupby('Gene').apply(lambda df: stats.ranksums(\n",
    "        df[df['Tissue']=='tumor'].Proteomic,\n",
    "        df[df['Tissue']=='normal'].Proteomic))\n",
    "    ranksums = pd.DataFrame.from_records(ranksums, index = ranksums.index, columns = ['statistic', 'pval'])\n",
    "    ranksums.reset_index(inplace = True)\n",
    "    ranksums = ranksums.dropna()\n",
    "    ranksums['FDR'] = ssm.fdrcorrection(ranksums['pval'])[1]\n",
    "    ranksums['Cancer'] = [cancer_name] * len(ranksums)\n",
    "    diff_expression_df = pd.merge(ranksums, fold_change)\n",
    "    diff_expression_dfs.append(diff_expression_df)\n",
    "diff_expression_df = pd.concat(diff_expression_dfs)\n",
    "diff_expression_df.to_csv('data/Proteomics_differential_expression_df.csv', index = False)\n",
    "diff_expression_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>statistic</th>\n",
       "      <th>pval</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Log2_fold_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>6.602420</td>\n",
       "      <td>4.044989e-11</td>\n",
       "      <td>9.028157e-11</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.881967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>-5.201398</td>\n",
       "      <td>1.977946e-07</td>\n",
       "      <td>3.574768e-07</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-0.641470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>5.970422</td>\n",
       "      <td>2.366402e-09</td>\n",
       "      <td>4.783680e-09</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.514646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ML1</td>\n",
       "      <td>-4.798710</td>\n",
       "      <td>1.596912e-06</td>\n",
       "      <td>2.739452e-06</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-0.304295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A3GALT2</td>\n",
       "      <td>1.037483</td>\n",
       "      <td>2.995107e-01</td>\n",
       "      <td>3.425340e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.314083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28052</th>\n",
       "      <td>ZYG11A</td>\n",
       "      <td>-0.120468</td>\n",
       "      <td>9.041126e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>-0.001721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28053</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>-0.983820</td>\n",
       "      <td>3.252042e-01</td>\n",
       "      <td>4.961530e-01</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>-0.008388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28054</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>2.730602</td>\n",
       "      <td>6.321883e-03</td>\n",
       "      <td>1.768955e-02</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>0.043200</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28055</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-6.806426</td>\n",
       "      <td>1.000530e-11</td>\n",
       "      <td>1.711700e-09</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>-0.087991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28056</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.005019</td>\n",
       "      <td>9.959950e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>PDAC</td>\n",
       "      <td>-0.000978</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>153736 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Gene  statistic          pval           FDR Cancer  Log2_fold_change\n",
       "0         A1BG   6.602420  4.044989e-11  9.028157e-11  CCRCC          0.881967\n",
       "1         A1CF  -5.201398  1.977946e-07  3.574768e-07  CCRCC         -0.641470\n",
       "2          A2M   5.970422  2.366402e-09  4.783680e-09  CCRCC          0.514646\n",
       "3        A2ML1  -4.798710  1.596912e-06  2.739452e-06  CCRCC         -0.304295\n",
       "4      A3GALT2   1.037483  2.995107e-01  3.425340e-01  CCRCC          0.314083\n",
       "...        ...        ...           ...           ...    ...               ...\n",
       "28052   ZYG11A  -0.120468  9.041126e-01  1.000000e+00   PDAC         -0.001721\n",
       "28053   ZYG11B  -0.983820  3.252042e-01  4.961530e-01   PDAC         -0.008388\n",
       "28054      ZYX   2.730602  6.321883e-03  1.768955e-02   PDAC          0.043200\n",
       "28055    ZZEF1  -6.806426  1.000530e-11  1.711700e-09   PDAC         -0.087991\n",
       "28056     ZZZ3  -0.005019  9.959950e-01  1.000000e+00   PDAC         -0.000978\n",
       "\n",
       "[153736 rows x 6 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diff_expression_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_normal_df = cancer.get_transcriptomics('normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID')\n",
    "    prot_normal_df.reset_index(inplace = True)\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    prot_normal_df['Tissue'] = ['normal'] * len(prot_normal_df)\n",
    "    prot_tumor_df = cancer.get_transcriptomics('tumor')\n",
    "    if(isinstance(prot_tumor_df.columns, pd.MultiIndex)):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID')\n",
    "    prot_tumor_df.reset_index(inplace = True)\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    prot_tumor_df['Tissue'] = ['tumor'] * len(prot_tumor_df)\n",
    "    prot_df = pd.merge(prot_tumor_df, prot_normal_df, how = 'outer')\n",
    "    prot_df = prot_df.dropna()\n",
    "    fold_change = prot_df.groupby('Gene').apply(\n",
    "        lambda df: log2_fold_change(df[df['Tissue']=='tumor'].Transcriptomics,\n",
    "                                    df[df['Tissue']=='normal'].Transcriptomics))\n",
    "    fold_change = pd.DataFrame(fold_change, columns = ['Log2_fold_change'])\n",
    "    fold_change.reset_index(inplace = True)\n",
    "    ranksums = prot_df.groupby('Gene').apply(lambda df: stats.ranksums(\n",
    "        df[df['Tissue']=='tumor'].Transcriptomics,\n",
    "        df[df['Tissue']=='normal'].Transcriptomics))\n",
    "    ranksums = pd.DataFrame.from_records(ranksums, index = ranksums.index, columns = ['statistic', 'pval'])\n",
    "    ranksums.reset_index(inplace = True)\n",
    "    ranksums = ranksums.dropna()\n",
    "    ranksums['FDR'] = ssm.fdrcorrection(ranksums['pval'])[1]\n",
    "    ranksums['Cancer'] = [cancer_name] * len(ranksums)\n",
    "    diff_expression_df = pd.merge(ranksums, fold_change)\n",
    "    diff_expression_dfs.append(diff_expression_df)\n",
    "diff_expression_df = pd.concat(diff_expression_dfs)\n",
    "diff_expression_df.to_csv('Transcriptomics_differential_expression_df.csv', index = False)\n",
    "diff_expression_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
