{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Your version of cptac (0.8.8) is out-of-date. Latest is 0.9.0. Please run 'pip install --upgrade cptac' to update it. (<ipython-input-1-8bc4673ec60a>, line 1)\n"
     ]
    }
   ],
   "source": [
    "import cptac\n",
    "import cptac.utils as ut\n",
    "import scipy\n",
    "import seaborn as sns; sns.set()\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy\n",
    "import math\n",
    "import pandas as pd\n",
    "import statsmodels.stats.multitest as ssm\n",
    "sns.set(style = 'white')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-9e8f8515bbc2>, line 10)\n"
     ]
    }
   ],
   "source": [
    "cptac.download(\"ccrcc\")\n",
    "cptac.download(\"endometrial\")\n",
    "cptac.download(\"luad\")\n",
    "cptac.download(\"hnscc\")\n",
    "cptac.download(\"lscc\")\n",
    "ccrcc = cptac.Ccrcc()\n",
    "en = cptac.Endometrial()\n",
    "luad = cptac.Luad()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [ccrcc, en, luad, hnscc, lscc]\n",
    "cancer_names = ['CCRCC', 'Endometrial', 'LUAD', 'HNSCC', 'LSCC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-4-065c8696527d>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-4-065c8696527d>, line 62)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-4-065c8696527d>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-4-065c8696527d>, line 62)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-4-065c8696527d>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-4-065c8696527d>, line 62)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Proteomic</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.859821</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.291127</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.824992</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.579658</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.673249</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.096200</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.043751</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.209196</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.801038</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.395915</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4598431</th>\n",
       "      <td>C3L-02646.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.402600</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4598432</th>\n",
       "      <td>C3N-03072.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>1.336400</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4598433</th>\n",
       "      <td>C3N-03662.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.726400</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4598434</th>\n",
       "      <td>C3N-03886.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.359300</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4598435</th>\n",
       "      <td>C3N-04155.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.809200</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>22344914 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID  Gene  Transcriptomics   Stage  Proteomic Cancer\n",
       "0        C3L-00004.N  A1BG         0.859821  Normal   0.291127  CCRCC\n",
       "1        C3L-00010.N  A1BG         0.824992  Normal  -0.579658  CCRCC\n",
       "2        C3L-00011.N  A1BG         0.673249  Normal  -0.096200  CCRCC\n",
       "3        C3L-00026.N  A1BG         3.043751  Normal  -0.209196  CCRCC\n",
       "4        C3L-00079.N  A1BG         0.801038  Normal  -0.395915  CCRCC\n",
       "...              ...   ...              ...     ...        ...    ...\n",
       "4598431  C3L-02646.N  ZZZ3              NaN  Normal   0.402600   LSCC\n",
       "4598432  C3N-03072.N  ZZZ3              NaN  Normal   1.336400   LSCC\n",
       "4598433  C3N-03662.N  ZZZ3              NaN  Normal   0.726400   LSCC\n",
       "4598434  C3N-03886.N  ZZZ3              NaN  Normal   0.359300   LSCC\n",
       "4598435  C3N-04155.N  ZZZ3              NaN  Normal   0.809200   LSCC\n",
       "\n",
       "[22344914 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_normal_df = cancer.get_proteomics('normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID')\n",
    "    trans_normal_df = cancer.get_transcriptomics('normal')\n",
    "    if isinstance(trans_normal_df.columns, pd.MultiIndex):\n",
    "        trans_normal_df = ut.reduce_multiindex(df = trans_normal_df, levels_to_drop='Database_ID')\n",
    "    prot_normal_df['Patient_ID'] = prot_normal_df.index\n",
    "    trans_normal_df['Patient_ID'] = trans_normal_df.index\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    trans_normal_df = trans_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    prot_normal_df['Stage'] = ['Normal'] * len(prot_normal_df)\n",
    "    trans_normal_df['Stage'] = ['Normal'] * len(trans_normal_df)\n",
    "    if cancer == ccrcc:\n",
    "        stage = cancer.get_clinical('tumor')[['tumor_stage_pathological']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'tumor_stage_pathological': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "    elif cancer == luad or cancer == lscc:\n",
    "        stage = cancer.get_clinical('tumor')[['Stage']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['1A', '1B', '1', 1, 'I', 'IA', 'IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['2A', '2B', 'II', 'IIA', 'IIB']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['3A', '3', 3, 'III', 'IIIA', 'IIIB']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    elif cancer == en:\n",
    "        stage = cancer.get_clinical('tumor')[['FIGO_stage']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'FIGO_stage': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['IA', 'IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['II']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['IIIA', 'IIIB', 'IIIC2', 'IIIC1']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['IVB', 'IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    elif cancer == hnscc:\n",
    "        stage = cancer.get_clinical('tumor')[['patho_staging_curated']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'patho_staging_curated': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "    prot_tumor_df = cancer.get_proteomics('tumor')\n",
    "    if isinstance(prot_tumor_df.columns, pd.MultiIndex):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID')\n",
    "    trans_tumor_df = cancer.get_transcriptomics('tumor')\n",
    "    if isinstance(trans_tumor_df.columns, pd.MultiIndex):\n",
    "        trans_tumor_df = ut.reduce_multiindex(df = trans_tumor_df, levels_to_drop='Database_ID')\n",
    "    prot_tumor_df['Patient_ID'] = prot_tumor_df.index\n",
    "    trans_tumor_df['Patient_ID'] = trans_tumor_df.index\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    trans_tumor_df = trans_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    trans_tumor_df = pd.merge(trans_tumor_df, stage, how = 'outer')\n",
    "    prot_tumor_df = pd.merge(prot_tumor_df, stage, how = 'outer')\n",
    "    prot_df = pd.merge(prot_tumor_df, prot_normal_df, how = 'outer')\n",
    "    trans_df = pd.merge(trans_normal_df, trans_tumor_df, how = 'outer')\n",
    "    cancer_df = pd.merge(trans_df, prot_df,how = 'outer')\n",
    "    cancer_df['Cancer'] = [cancer_name] * len(cancer_df)\n",
    "    cancer_dfs.append(cancer_df)\n",
    "prot_trans_stages_df = pd.concat(cancer_dfs)\n",
    "prot_trans_stages_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Proteomic</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.859821</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.291127</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.824992</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.579658</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.673249</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.096200</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.043751</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.209196</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.801038</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.395915</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491060</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>17.376300</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>0.450600</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491062</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>17.324700</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>-0.583000</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491063</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>18.029000</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>-2.542300</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491064</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>17.693600</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>-0.589400</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491065</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>18.028500</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>0.916700</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7952124 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID    Gene  Transcriptomics     Stage  Proteomic Cancer\n",
       "0        C3L-00004.N    A1BG         0.859821    Normal   0.291127  CCRCC\n",
       "1        C3L-00010.N    A1BG         0.824992    Normal  -0.579658  CCRCC\n",
       "2        C3L-00011.N    A1BG         0.673249    Normal  -0.096200  CCRCC\n",
       "3        C3L-00026.N    A1BG         3.043751    Normal  -0.209196  CCRCC\n",
       "4        C3L-00079.N    A1BG         0.801038    Normal  -0.395915  CCRCC\n",
       "...              ...     ...              ...       ...        ...    ...\n",
       "4491060    C3N-04162    ZXDC        17.376300  Stage II   0.450600   LSCC\n",
       "4491062    C3N-04162  ZYG11B        17.324700  Stage II  -0.583000   LSCC\n",
       "4491063    C3N-04162     ZYX        18.029000  Stage II  -2.542300   LSCC\n",
       "4491064    C3N-04162   ZZEF1        17.693600  Stage II  -0.589400   LSCC\n",
       "4491065    C3N-04162    ZZZ3        18.028500  Stage II   0.916700   LSCC\n",
       "\n",
       "[7952124 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_trans_stages_df = prot_trans_stages_df.dropna()\n",
    "prot_trans_stages_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Proteomic</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.859821</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.291127</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.824992</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.579658</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.673249</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.096200</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.043751</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.209196</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.801038</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.395915</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491060</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZXDC</td>\n",
       "      <td>17.376300</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>0.450600</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491062</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>17.324700</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>-0.583000</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491063</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>18.029000</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>-2.542300</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491064</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>17.693600</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>-0.589400</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4491065</th>\n",
       "      <td>C3N-04162</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>18.028500</td>\n",
       "      <td>Stage II</td>\n",
       "      <td>0.916700</td>\n",
       "      <td>LSCC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7952124 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID    Gene  Transcriptomics     Stage  Proteomic Cancer\n",
       "0        C3L-00004.N    A1BG         0.859821    Normal   0.291127  CCRCC\n",
       "1        C3L-00010.N    A1BG         0.824992    Normal  -0.579658  CCRCC\n",
       "2        C3L-00011.N    A1BG         0.673249    Normal  -0.096200  CCRCC\n",
       "3        C3L-00026.N    A1BG         3.043751    Normal  -0.209196  CCRCC\n",
       "4        C3L-00079.N    A1BG         0.801038    Normal  -0.395915  CCRCC\n",
       "...              ...     ...              ...       ...        ...    ...\n",
       "4491060    C3N-04162    ZXDC        17.376300  Stage II   0.450600   LSCC\n",
       "4491062    C3N-04162  ZYG11B        17.324700  Stage II  -0.583000   LSCC\n",
       "4491063    C3N-04162     ZYX        18.029000  Stage II  -2.542300   LSCC\n",
       "4491064    C3N-04162   ZZEF1        17.693600  Stage II  -0.589400   LSCC\n",
       "4491065    C3N-04162    ZZZ3        18.028500  Stage II   0.916700   LSCC\n",
       "\n",
       "[7952124 rows x 6 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_trans_stages_df.groupby(['Cancer', 'Stage', 'Gene'])\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
