{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac.pancan as pc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: cptac in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (1.1.2)\n",
      "Requirement already satisfied: requests>=2.21.0 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (2.27.1)\n",
      "Requirement already satisfied: xlrd>=2.0.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (2.0.1)\n",
      "Requirement already satisfied: numpy>=1.16.3 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (1.21.5)\n",
      "Requirement already satisfied: pandas>=1.2.0 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (1.4.2)\n",
      "Requirement already satisfied: packaging>=19.2 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (21.3)\n",
      "Requirement already satisfied: statsmodels>=0.10.0 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (0.13.2)\n",
      "Requirement already satisfied: scipy>=1.2.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (1.7.3)\n",
      "Requirement already satisfied: beautifulsoup4>=4.7.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (4.11.1)\n",
      "Requirement already satisfied: gtfparse>=1.2.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (1.2.1)\n",
      "Requirement already satisfied: flask>=1.1.0 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (1.1.2)\n",
      "Requirement already satisfied: urllib3>=1.24.2 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (1.26.9)\n",
      "Requirement already satisfied: mygene>=3.2.2 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (3.2.2)\n",
      "Requirement already satisfied: openpyxl>=2.6.0 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from cptac) (3.0.9)\n",
      "Requirement already satisfied: soupsieve>1.2 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from beautifulsoup4>=4.7.1->cptac) (2.3.1)\n",
      "Requirement already satisfied: click>=5.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from flask>=1.1.0->cptac) (8.0.4)\n",
      "Requirement already satisfied: Jinja2>=2.10.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from flask>=1.1.0->cptac) (2.11.3)\n",
      "Requirement already satisfied: Werkzeug>=0.15 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from flask>=1.1.0->cptac) (2.0.3)\n",
      "Requirement already satisfied: itsdangerous>=0.24 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from flask>=1.1.0->cptac) (2.0.1)\n",
      "Requirement already satisfied: MarkupSafe>=0.23 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from Jinja2>=2.10.1->flask>=1.1.0->cptac) (2.0.1)\n",
      "Requirement already satisfied: biothings-client>=0.2.6 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from mygene>=3.2.2->cptac) (0.2.6)\n",
      "Requirement already satisfied: et-xmlfile in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from openpyxl>=2.6.0->cptac) (1.1.0)\n",
      "Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from packaging>=19.2->cptac) (3.0.4)\n",
      "Requirement already satisfied: python-dateutil>=2.8.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from pandas>=1.2.0->cptac) (2.8.2)\n",
      "Requirement already satisfied: pytz>=2020.1 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from pandas>=1.2.0->cptac) (2021.3)\n",
      "Requirement already satisfied: six>=1.5 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from python-dateutil>=2.8.1->pandas>=1.2.0->cptac) (1.16.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from requests>=2.21.0->cptac) (3.3)\n",
      "Requirement already satisfied: charset-normalizer~=2.0.0 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from requests>=2.21.0->cptac) (2.0.4)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from requests>=2.21.0->cptac) (2021.10.8)\n",
      "Requirement already satisfied: patsy>=0.5.2 in /Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages (from statsmodels>=0.10.0->cptac) (0.5.2)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'1.1.2'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "!pip install --upgrade cptac\n",
    "\n",
    "import cptac\n",
    "cptac.version()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import cptac\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import copy\n",
    "import cptac.utils as ut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac.utils"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cptac.utils.reduce_multiindex()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading broadluad v1.0...                     \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/gtfparse/read_gtf.py:82: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.\n",
      "\n",
      "\n",
      "  chunk_iterator = pd.read_csv(\n",
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/gtfparse/read_gtf.py:82: FutureWarning: The warn_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.\n",
      "\n",
      "\n",
      "  chunk_iterator = pd.read_csv(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading washuluad v1.0.........                  \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/cptac/pancan/washuluad.py:136: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  all_df = tumor.append(normal)\n",
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/cptac/pancan/washuluad.py:136: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  all_df = tumor.append(normal)\n",
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/cptac/pancan/washuluad.py:136: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  all_df = tumor.append(normal)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading washuluad v1.0..........\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/gtfparse/read_gtf.py:82: FutureWarning: The error_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.\n",
      "\n",
      "\n",
      "  chunk_iterator = pd.read_csv(\n",
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/gtfparse/read_gtf.py:82: FutureWarning: The warn_bad_lines argument has been deprecated and will be removed in a future version. Use on_bad_lines in the future.\n",
      "\n",
      "\n",
      "  chunk_iterator = pd.read_csv(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading washuluad v1.0...........\r",
      "Loading washuluad v1.0............\r",
      "Loading washuluad v1.0.............\r",
      "Loading washuluad v1.0..............\r",
      "Loading washuluad v1.0...............\r",
      "Loading washuluad v1.0................\r",
      "Loading washuluad v1.0.................\r",
      "Loading washuluad v1.0..................\r",
      "                                        \r",
      "Formatting washuluad dataframes...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/humbertogiraldez/opt/anaconda3/lib/python3.9/site-packages/cptac/pancan/washuluad.py:215: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.\n",
      "  rna_combined = rna_tumor.append(rna_normal)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                               \r"
     ]
    }
   ],
   "source": [
    "# ccrcc = pc.PancanCcrcc()\n",
    "# hnscc = pc.PancanHnscc()\n",
    "luad = pc.PancanLuad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Data type</th>\n",
       "      <th>Available sources</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CNV</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cancer_diagnosis</td>\n",
       "      <td>mssm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>circular_RNA</td>\n",
       "      <td>bcm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>clinical</td>\n",
       "      <td>mssm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>deconvolution_cibersort</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>deconvolution_xcell</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>demographic</td>\n",
       "      <td>mssm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>followup</td>\n",
       "      <td>mssm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>mature_miRNA</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>medical_conditions</td>\n",
       "      <td>mssm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>phosphoproteomics</td>\n",
       "      <td>umich</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>precursor_miRNA</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>previous_cancer</td>\n",
       "      <td>mssm</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>proteomics</td>\n",
       "      <td>umich</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>somatic_mutation</td>\n",
       "      <td>harmonized, washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>total_miRNA</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>transcriptomics</td>\n",
       "      <td>bcm, broad, washu</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>tumor_purity</td>\n",
       "      <td>washu</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  Data type  Available sources\n",
       "0                       CNV              washu\n",
       "1          cancer_diagnosis               mssm\n",
       "2              circular_RNA                bcm\n",
       "3                  clinical               mssm\n",
       "4   deconvolution_cibersort              washu\n",
       "5       deconvolution_xcell              washu\n",
       "6               demographic               mssm\n",
       "7                  followup               mssm\n",
       "8              mature_miRNA              washu\n",
       "9        medical_conditions               mssm\n",
       "10        phosphoproteomics              umich\n",
       "11          precursor_miRNA              washu\n",
       "12          previous_cancer               mssm\n",
       "13               proteomics              umich\n",
       "14         somatic_mutation  harmonized, washu\n",
       "15              total_miRNA              washu\n",
       "16          transcriptomics  bcm, broad, washu\n",
       "17             tumor_purity              washu"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "luad.list_data_sources()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th>Name</th>\n",
       "      <th colspan=\"10\" halign=\"left\">5S_rRNA</th>\n",
       "      <th>...</th>\n",
       "      <th colspan=\"9\" halign=\"left\">uc_338</th>\n",
       "      <th>yR211F11.2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Database_ID</th>\n",
       "      <th>ENSG00000201285.1</th>\n",
       "      <th>ENSG00000212595.1</th>\n",
       "      <th>ENSG00000252830.2</th>\n",
       "      <th>ENSG00000271924.1</th>\n",
       "      <th>ENSG00000272253.1</th>\n",
       "      <th>ENSG00000272351.1</th>\n",
       "      <th>ENSG00000272435.1</th>\n",
       "      <th>ENSG00000274059.1</th>\n",
       "      <th>ENSG00000274097.1</th>\n",
       "      <th>ENSG00000274164.1</th>\n",
       "      <th>...</th>\n",
       "      <th>ENSG00000278040.1</th>\n",
       "      <th>ENSG00000278113.1</th>\n",
       "      <th>ENSG00000278218.1</th>\n",
       "      <th>ENSG00000278236.1</th>\n",
       "      <th>ENSG00000278333.1</th>\n",
       "      <th>ENSG00000278413.1</th>\n",
       "      <th>ENSG00000278498.1</th>\n",
       "      <th>ENSG00000278587.1</th>\n",
       "      <th>ENSG00000278643.1</th>\n",
       "      <th>ENSG00000213076.3</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.710629</td>\n",
       "      <td>1.554085</td>\n",
       "      <td>0.465790</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.824837</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.894017</td>\n",
       "      <td>4.350535</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.040333</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.149003</td>\n",
       "      <td>0.095567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.391454</td>\n",
       "      <td>0.428039</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.184678</td>\n",
       "      <td>...</td>\n",
       "      <td>1.090479</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.492475</td>\n",
       "      <td>6.390716</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.826386</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.246237</td>\n",
       "      <td>0.276380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.557894</td>\n",
       "      <td>1.703492</td>\n",
       "      <td>1.094079</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.789902</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.679941</td>\n",
       "      <td>7.872282</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.792470</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.089790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.564504</td>\n",
       "      <td>2.443884</td>\n",
       "      <td>0.219744</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.488777</td>\n",
       "      <td>0.421767</td>\n",
       "      <td>...</td>\n",
       "      <td>2.179130</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.843534</td>\n",
       "      <td>4.713003</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.566828</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.140589</td>\n",
       "      <td>0.225427</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.206354</td>\n",
       "      <td>1.608370</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.195109</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.462623</td>\n",
       "      <td>6.003336</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.639504</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.154208</td>\n",
       "      <td>0.049453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.043907</td>\n",
       "      <td>0.957825</td>\n",
       "      <td>0.287079</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.813391</td>\n",
       "      <td>0.203348</td>\n",
       "      <td>2.571366</td>\n",
       "      <td>4.568233</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.724612</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.294504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.570147</td>\n",
       "      <td>1.870295</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>2.580933</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.896602</td>\n",
       "      <td>6.011403</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.887842</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.613883</td>\n",
       "      <td>0.632568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.704438</td>\n",
       "      <td>1.797306</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.256758</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.981182</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.329344</td>\n",
       "      <td>7.347457</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.865293</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.443115</td>\n",
       "      <td>0.426307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.584157</td>\n",
       "      <td>3.117976</td>\n",
       "      <td>0.623013</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.661951</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2.590863</td>\n",
       "      <td>3.448304</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.458052</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.447388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.204842</td>\n",
       "      <td>2.015876</td>\n",
       "      <td>0.604198</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>1.283921</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.901966</td>\n",
       "      <td>3.344166</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.940677</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.128852</td>\n",
       "      <td>0.371894</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75 rows × 60483 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name                  5S_rRNA                                      \\\n",
       "Database_ID ENSG00000201285.1 ENSG00000212595.1 ENSG00000252830.2   \n",
       "Patient_ID                                                          \n",
       "C3L-00004.N               0.0               0.0               0.0   \n",
       "C3L-00010.N               0.0               0.0               0.0   \n",
       "C3L-00011.N               0.0               0.0               0.0   \n",
       "C3L-00026.N               0.0               0.0               0.0   \n",
       "C3L-00079.N               0.0               0.0               0.0   \n",
       "...                       ...               ...               ...   \n",
       "C3N-01646.N               0.0               0.0               0.0   \n",
       "C3N-01648.N               0.0               0.0               0.0   \n",
       "C3N-01649.N               0.0               0.0               0.0   \n",
       "C3N-01651.N               0.0               0.0               0.0   \n",
       "C3N-01808.N               0.0               0.0               0.0   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000271924.1 ENSG00000272253.1 ENSG00000272351.1   \n",
       "Patient_ID                                                          \n",
       "C3L-00004.N          0.710629          1.554085          0.465790   \n",
       "C3L-00010.N          0.391454          0.428039          0.000000   \n",
       "C3L-00011.N          1.557894          1.703492          1.094079   \n",
       "C3L-00026.N          1.564504          2.443884          0.219744   \n",
       "C3L-00079.N          2.206354          1.608370          0.000000   \n",
       "...                       ...               ...               ...   \n",
       "C3N-01646.N          2.043907          0.957825          0.287079   \n",
       "C3N-01648.N          0.570147          1.870295          0.000000   \n",
       "C3N-01649.N          0.704438          1.797306          0.000000   \n",
       "C3N-01651.N          1.584157          3.117976          0.623013   \n",
       "C3N-01808.N          0.204842          2.015876          0.604198   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000272435.1 ENSG00000274059.1 ENSG00000274097.1   \n",
       "Patient_ID                                                          \n",
       "C3L-00004.N               0.0               0.0          0.000000   \n",
       "C3L-00010.N               0.0               0.0          0.000000   \n",
       "C3L-00011.N               0.0               0.0          0.000000   \n",
       "C3L-00026.N               0.0               0.0          0.488777   \n",
       "C3L-00079.N               0.0               0.0          0.000000   \n",
       "...                       ...               ...               ...   \n",
       "C3N-01646.N               0.0               0.0          0.000000   \n",
       "C3N-01648.N               0.0               0.0          0.000000   \n",
       "C3N-01649.N               0.0               0.0          0.256758   \n",
       "C3N-01651.N               0.0               0.0          0.000000   \n",
       "C3N-01808.N               0.0               0.0          0.000000   \n",
       "\n",
       "Name                           ...            uc_338                    \\\n",
       "Database_ID ENSG00000274164.1  ... ENSG00000278040.1 ENSG00000278113.1   \n",
       "Patient_ID                     ...                                       \n",
       "C3L-00004.N          0.000000  ...          0.824837          0.000000   \n",
       "C3L-00010.N          0.184678  ...          1.090479          0.000000   \n",
       "C3L-00011.N          0.000000  ...          2.789902          0.000000   \n",
       "C3L-00026.N          0.421767  ...          2.179130          0.000000   \n",
       "C3L-00079.N          0.000000  ...          1.195109          0.000000   \n",
       "...                       ...  ...               ...               ...   \n",
       "C3N-01646.N          0.000000  ...          0.813391          0.203348   \n",
       "C3N-01648.N          0.000000  ...          2.580933          0.000000   \n",
       "C3N-01649.N          0.000000  ...          0.981182          0.000000   \n",
       "C3N-01651.N          0.000000  ...          0.661951          0.000000   \n",
       "C3N-01808.N          0.000000  ...          1.283921          0.000000   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000278218.1 ENSG00000278236.1 ENSG00000278333.1   \n",
       "Patient_ID                                                          \n",
       "C3L-00004.N          0.894017          4.350535               0.0   \n",
       "C3L-00010.N          0.492475          6.390716               0.0   \n",
       "C3L-00011.N          1.679941          7.872282               0.0   \n",
       "C3L-00026.N          0.843534          4.713003               0.0   \n",
       "C3L-00079.N          0.462623          6.003336               0.0   \n",
       "...                       ...               ...               ...   \n",
       "C3N-01646.N          2.571366          4.568233               0.0   \n",
       "C3N-01648.N          0.896602          6.011403               0.0   \n",
       "C3N-01649.N          1.329344          7.347457               0.0   \n",
       "C3N-01651.N          2.590863          3.448304               0.0   \n",
       "C3N-01808.N          0.901966          3.344166               0.0   \n",
       "\n",
       "Name                                                               \\\n",
       "Database_ID ENSG00000278413.1 ENSG00000278498.1 ENSG00000278587.1   \n",
       "Patient_ID                                                          \n",
       "C3L-00004.N               0.0          2.040333               0.0   \n",
       "C3L-00010.N               0.0          1.826386               0.0   \n",
       "C3L-00011.N               0.0          4.792470               0.0   \n",
       "C3L-00026.N               0.0          2.566828               0.0   \n",
       "C3L-00079.N               0.0          2.639504               0.0   \n",
       "...                       ...               ...               ...   \n",
       "C3N-01646.N               0.0          2.724612               0.0   \n",
       "C3N-01648.N               0.0          3.887842               0.0   \n",
       "C3N-01649.N               0.0          2.865293               0.0   \n",
       "C3N-01651.N               0.0          5.458052               0.0   \n",
       "C3N-01808.N               0.0          2.940677               0.0   \n",
       "\n",
       "Name                                 yR211F11.2  \n",
       "Database_ID ENSG00000278643.1 ENSG00000213076.3  \n",
       "Patient_ID                                       \n",
       "C3L-00004.N          0.149003          0.095567  \n",
       "C3L-00010.N          0.246237          0.276380  \n",
       "C3L-00011.N          0.000000          0.089790  \n",
       "C3L-00026.N          0.140589          0.225427  \n",
       "C3L-00079.N          0.154208          0.049453  \n",
       "...                       ...               ...  \n",
       "C3N-01646.N          0.000000          0.294504  \n",
       "C3N-01648.N          1.613883          0.632568  \n",
       "C3N-01649.N          0.443115          0.426307  \n",
       "C3N-01651.N          0.000000          0.447388  \n",
       "C3N-01808.N          0.128852          0.371894  \n",
       "\n",
       "[75 rows x 60483 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ccrcc.get_transcriptomics(\"washu\", \"normal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_prot_trans_df(cancer):\n",
    "\n",
    "    prot_normal_df = cancer.get_proteomics(\"umich\", 'normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID', quiet=True)\n",
    "\n",
    "    trans_normal_df = cancer.get_transcriptomics(\"washu\", 'normal')\n",
    "    if isinstance(trans_normal_df.columns, pd.MultiIndex):\n",
    "        trans_normal_df = ut.reduce_multiindex(df = trans_normal_df, levels_to_drop='Database_ID', quiet=True)\n",
    "\n",
    "    prot_normal_df['Name'] = prot_normal_df.index\n",
    "    trans_normal_df['Name'] = trans_normal_df.index\n",
    "\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Proteomics')\n",
    "    trans_normal_df = trans_normal_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "\n",
    "    prot_tumor_df = cancer.get_proteomics(\"umich\", 'tumor')\n",
    "    if isinstance(prot_tumor_df.columns, pd.MultiIndex):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID', quiet=True)\n",
    "\n",
    "    trans_tumor_df = cancer.get_transcriptomics(\"washu\", 'tumor')\n",
    "    if isinstance(trans_tumor_df.columns, pd.MultiIndex):\n",
    "        trans_tumor_df = ut.reduce_multiindex(df = trans_tumor_df, levels_to_drop='Database_ID', quiet=True)\n",
    "\n",
    "    prot_tumor_df['Name'] = prot_tumor_df.index\n",
    "    trans_tumor_df['Name'] = trans_tumor_df.index\n",
    "\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Proteomics')\n",
    "    trans_tumor_df = trans_tumor_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "\n",
    "    prot_tumor_df['Tissue'] = ['Tumor'] * len(prot_tumor_df)\n",
    "    prot_normal_df['Tissue'] = ['Normal'] * len(prot_normal_df)\n",
    "    trans_tumor_df['Tissue'] = ['Tumor'] * len(trans_tumor_df)\n",
    "    trans_normal_df['Tissue'] = ['Normal'] * len(trans_normal_df)\n",
    "\n",
    "    prot_df = pd.concat([prot_tumor_df, prot_normal_df])\n",
    "    trans_df = pd.concat([trans_tumor_df, trans_normal_df])\n",
    "\n",
    "    return(pd.merge(prot_df, trans_df).dropna())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer = luad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "prot_normal_df = cancer.get_proteomics(\"umich\", 'normal')\n",
    "if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "    prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID', quiet=True)\n",
    "\n",
    "trans_normal_df = cancer.get_transcriptomics(\"washu\", 'normal')\n",
    "if isinstance(trans_normal_df.columns, pd.MultiIndex):\n",
    "    trans_normal_df = ut.reduce_multiindex(df = trans_normal_df, levels_to_drop='Database_ID', quiet=True)\n",
    "\n",
    "prot_normal_df['Name'] = prot_normal_df.index\n",
    "trans_normal_df['Name'] = trans_normal_df.index\n",
    "\n",
    "prot_normal_df = prot_normal_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Proteomics')\n",
    "trans_normal_df = trans_normal_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "\n",
    "prot_tumor_df = cancer.get_proteomics(\"umich\", 'tumor')\n",
    "if isinstance(prot_tumor_df.columns, pd.MultiIndex):\n",
    "    prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID', quiet=True)\n",
    "\n",
    "trans_tumor_df = cancer.get_transcriptomics(\"washu\", 'tumor')\n",
    "if isinstance(trans_tumor_df.columns, pd.MultiIndex):\n",
    "    trans_tumor_df = ut.reduce_multiindex(df = trans_tumor_df, levels_to_drop='Database_ID', quiet=True)\n",
    "\n",
    "prot_tumor_df['Name'] = prot_tumor_df.index\n",
    "trans_tumor_df['Name'] = trans_tumor_df.index\n",
    "\n",
    "prot_tumor_df = prot_tumor_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Proteomics')\n",
    "# trans_tumor_df = trans_tumor_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Transcriptomics')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Proteomics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11LU013</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.522000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11LU016</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.416894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11LU022</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.114286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11LU035</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.138902</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00001</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.779792</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476517</th>\n",
       "      <td>C3N-02582</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-0.292263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476518</th>\n",
       "      <td>C3N-02586</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-0.164234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476519</th>\n",
       "      <td>C3N-02587</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.098843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476520</th>\n",
       "      <td>C3N-02588</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-0.224152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1476521</th>\n",
       "      <td>C3N-02729</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-0.255202</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1476522 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Name   Gene  Proteomics\n",
       "0          11LU013   A1BG   -0.522000\n",
       "1          11LU016   A1BG   -0.416894\n",
       "2          11LU022   A1BG   -0.114286\n",
       "3          11LU035   A1BG   -0.138902\n",
       "4        C3L-00001   A1BG   -0.779792\n",
       "...            ...    ...         ...\n",
       "1476517  C3N-02582  ZZEF1   -0.292263\n",
       "1476518  C3N-02586  ZZEF1   -0.164234\n",
       "1476519  C3N-02587  ZZEF1    0.098843\n",
       "1476520  C3N-02588  ZZEF1   -0.224152\n",
       "1476521  C3N-02729  ZZEF1   -0.255202\n",
       "\n",
       "[1476522 rows x 3 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_tumor_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df[df.Gene == 'ARID1B']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/16/01vxw5fs66d9b01nlwf3w5yc0000gn/T/ipykernel_88445/2063825305.py:7: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.\n",
      "  normal_corr = df[df[column] == label2].corr(method = 'spearman',min_periods = cutoff ).iloc[0][1]\n",
      "/var/folders/16/01vxw5fs66d9b01nlwf3w5yc0000gn/T/ipykernel_88445/2063825305.py:8: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.\n",
      "  tumor_corr = df[df[column] == label1].corr(method = 'spearman',min_periods = cutoff).iloc[0][1]\n"
     ]
    }
   ],
   "source": [
    "# df = get_prot_trans_df(luad)\n",
    "# delta_correlation(get_prot_trans_df(luad))\n",
    "# column = 'Tissue'\n",
    "# label1 = 'Tumor'\n",
    "# label2 ='Normal'\n",
    "# cutoff = 15\n",
    "normal_corr = df[df[column] == label2].corr(method = 'spearman',min_periods = cutoff ).iloc[0][1]\n",
    "tumor_corr = df[df[column] == label1].corr(method = 'spearman',min_periods = cutoff).iloc[0][1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.02675596971461852\n",
      "0.5646011315736086\n"
     ]
    }
   ],
   "source": [
    "print(normal_corr)\n",
    "print(tumor_corr)\n",
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.02675596971461852\n",
      "0.5646011315736086\n"
     ]
    }
   ],
   "source": [
    "print(normal_corr)\n",
    "print(tumor_corr)\n",
    "# "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def delta_correlation(df, column = 'Tissue', label1 = 'Tumor', label2 ='Normal', cutoff = 15):\n",
    "    normal_corr = df[df[column] == label2].corr(method = 'spearman',min_periods = cutoff ).iloc[0][1]\n",
    "    tumor_corr = df[df[column] == label1].corr(method = 'spearman',min_periods = cutoff).iloc[0][1]\n",
    "    delta_corr = tumor_corr - normal_corr\n",
    "    return delta_corr\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Please login to Box on the webpage that was just opened and grant access for cptac to download files through your account. If you accidentally closed the browser window, press Ctrl+C and call the download function again.\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'pDsRMupmWmnAwVcP1UOuWwQnTbmnaQaB'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import cptac.pancan as pc\n",
    "pc.get_box_token()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                 \r"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'input_permutation_number' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-40a3db8e56dd>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     95\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m \u001b[0mprot_trans_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_prot_trans_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcancer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 97\u001b[0;31m \u001b[0mperm_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprot_trans_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Gene'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mpermutate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcutoff\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcutoff\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnum_permutations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minput_permutation_number\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     98\u001b[0m \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_records\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mperm_results\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'Delta_Correlation'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'P_Value'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mperm_results\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/groupby.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, *args, **kwargs)\u001b[0m\n\u001b[1;32m   1412\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0moption_context\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"mode.chained_assignment\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1413\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1414\u001b[0;31m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_python_apply_general\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_selected_obj\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1415\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mTypeError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1416\u001b[0m                 \u001b[0;31m# gh-20949\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/groupby.py\u001b[0m in \u001b[0;36m_python_apply_general\u001b[0;34m(self, f, data, not_indexed_same)\u001b[0m\n\u001b[1;32m   1453\u001b[0m             \u001b[0mdata\u001b[0m \u001b[0mafter\u001b[0m \u001b[0mapplying\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1454\u001b[0m         \"\"\"\n\u001b[0;32m-> 1455\u001b[0;31m         \u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmutated\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgrouper\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1456\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1457\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mnot_indexed_same\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/ops.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, f, data, axis)\u001b[0m\n\u001b[1;32m    759\u001b[0m             \u001b[0;31m# group might be modified\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    760\u001b[0m             \u001b[0mgroup_axes\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgroup\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxes\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 761\u001b[0;31m             \u001b[0mres\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgroup\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    762\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mmutated\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0m_is_indexed_like\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mres\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgroup_axes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    763\u001b[0m                 \u001b[0mmutated\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-4-40a3db8e56dd>\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m     95\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     96\u001b[0m \u001b[0mprot_trans_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_prot_trans_df\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcancer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 97\u001b[0;31m \u001b[0mperm_results\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprot_trans_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgroupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Gene'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mpermutate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcutoff\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcutoff\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnum_permutations\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minput_permutation_number\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     98\u001b[0m \u001b[0mdf\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_records\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mperm_results\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m'Delta_Correlation'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'P_Value'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     99\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mperm_results\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'input_permutation_number' is not defined"
     ]
    }
   ],
   "source": [
    "import cptac.pancan as pc\n",
    "import scipy.stats as stats\n",
    "import numpy as np\n",
    "import copy\n",
    "import cptac.utils as ut\n",
    "import statsmodels.stats.multitest as ssm\n",
    "import pandas as pd\n",
    "import warnings\n",
    "import sys, os\n",
    "\n",
    "def get_prot_trans_df(cancer):\n",
    "\n",
    "    prot_normal_df = cancer.get_proteomics(\"pdc\", 'normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID', quiet=True)\n",
    "\n",
    "    trans_normal_df = cancer.get_transcriptomics(\"washu\", 'normal')\n",
    "    if isinstance(trans_normal_df.columns, pd.MultiIndex):\n",
    "        trans_normal_df = ut.reduce_multiindex(df = trans_normal_df, levels_to_drop='Database_ID', quiet=True)\n",
    "\n",
    "    prot_normal_df['Name'] = prot_normal_df.index\n",
    "    trans_normal_df['Name'] = trans_normal_df.index\n",
    "\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Proteomics')\n",
    "    trans_normal_df = trans_normal_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "\n",
    "    prot_tumor_df = cancer.get_proteomics(\"pdc\", 'tumor')\n",
    "    if isinstance(prot_tumor_df.columns, pd.MultiIndex):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID', quiet=True)\n",
    "\n",
    "    trans_tumor_df = cancer.get_transcriptomics(\"washu\", 'tumor')\n",
    "    if isinstance(trans_tumor_df.columns, pd.MultiIndex):\n",
    "        trans_tumor_df = ut.reduce_multiindex(df = trans_tumor_df, levels_to_drop='Database_ID', quiet=True)\n",
    "\n",
    "    prot_tumor_df['Name'] = prot_tumor_df.index\n",
    "    trans_tumor_df['Name'] = trans_tumor_df.index\n",
    "\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Proteomics')\n",
    "    trans_tumor_df = trans_tumor_df.melt(id_vars = 'Name', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "\n",
    "    prot_tumor_df['Tissue'] = ['Tumor'] * len(prot_tumor_df)\n",
    "    prot_normal_df['Tissue'] = ['Normal'] * len(prot_normal_df)\n",
    "    trans_tumor_df['Tissue'] = ['Tumor'] * len(trans_tumor_df)\n",
    "    trans_normal_df['Tissue'] = ['Normal'] * len(trans_normal_df)\n",
    "\n",
    "    prot_df = pd.concat([prot_tumor_df, prot_normal_df])\n",
    "    trans_df = pd.concat([trans_tumor_df, trans_normal_df])\n",
    "\n",
    "    return(pd.merge(prot_df, trans_df).dropna())\n",
    "\n",
    "def permutate(df, column = 'Tissue', label1 = 'Tumor', label2 ='Normal', cutoff = 15, num_permutations = 10000, return_perm_list = False):\n",
    "    delta_corr = delta_correlation(df, column = column, label1 = label1, label2 = label2, cutoff= cutoff)\n",
    "    perm_delta_corrs = []\n",
    "    for i in range(0, num_permutations):\n",
    "        df[column] = np.random.permutation(df[column])\n",
    "        perm_delta_corr = delta_correlation(df, column = column, label1 = label1, label2 = label2, cutoff = cutoff)\n",
    "        perm_delta_corrs.append(perm_delta_corr)\n",
    "    if return_perm_list:\n",
    "        return perm_delta_corrs\n",
    "    z_score = (delta_corr - np.mean(perm_delta_corrs)) / np.std(perm_delta_corrs)\n",
    "    p_val = stats.norm.sf(abs(z_score)) * 2  \n",
    "    return(delta_corr, p_val)\n",
    "\n",
    "\n",
    "warnings.filterwarnings('ignore')\n",
    "# currentdir = os.path.dirname(os.path.realpath('Make_Cancer_Delta_Corr_and_P_Value_Dataframe'))\n",
    "# parentdir = os.path.dirname(currentdir)\n",
    "# parentdir = os.path.dirname(parentdir)\n",
    "# sys.path.append(parentdir)\n",
    "\n",
    "\n",
    "# input_cancer_type = sys.argv[1]\n",
    "# input_permutation_number = int(sys.argv[2])\n",
    "# token = sys.argv[3]\n",
    "cutoff = 15\n",
    "input_cancer_type = \"CCRCC\"\n",
    "token =\"pDsRMupmWmnAwVcP1UOuWwQnTbmnaQaB\"\n",
    "if input_cancer_type == \"CCRCC\":\n",
    "    pc.download(\"pancanccrcc\", box_token=token)\n",
    "    cancer = pc.PancanCcrcc()\n",
    "elif input_cancer_type == \"Endometrial\":\n",
    "    pc.download(\"pancanucec\", box_token=token)\n",
    "    cancer = pc.PancanUcec()\n",
    "    cutoff = 10\n",
    "elif input_cancer_type == \"LUAD\":\n",
    "    pc.download(\"pancanluad\", box_token=token)\n",
    "    cancer = pc.PancanLuad()\n",
    "elif input_cancer_type == \"HNSCC\":\n",
    "    pc.download(\"pancanhnscc\", box_token=token)\n",
    "    cancer = pc.PancanHnscc()\n",
    "elif input_cancer_type == \"LSCC\":\n",
    "    pc.download(\"pancanlscc\", box_token=token)\n",
    "    cancer = pc.PancanLscc()\n",
    "\n",
    "\n",
    "prot_trans_df = get_prot_trans_df(cancer)\n",
    "perm_results = prot_trans_df.groupby('Gene').apply(lambda x: permutate(df = x, cutoff = cutoff, num_permutations = input_permutation_number))\n",
    "df = pd.DataFrame.from_records(perm_results, columns = ['Delta_Correlation', 'P_Value'])\n",
    "df.index = perm_results.index\n",
    "df = df.dropna()\n",
    "df.reset_index(inplace = True)\n",
    "df['FDR'] = ssm.fdrcorrection(df.P_Value)[1]\n",
    "df['Cancer'] = [input_cancer_type] * len(df)\n",
    "file_name = input_cancer_type + '_delta_corr_pval_df.csv'\n",
    "df.to_csv(file_name, index = False)\n",
    "  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "def permutate(df, column = 'Tissue', label1 = 'Tumor', label2 ='Normal', cutoff = 15, num_permutations = 10000, return_perm_list = False):\n",
    "    delta_corr = delta_correlation(df, column = column, label1 = label1, label2 = label2, cutoff= cutoff)\n",
    "    perm_delta_corrs = []\n",
    "    for i in range(0, num_permutations):\n",
    "        df[column] = np.random.permutation(df[column])\n",
    "        perm_delta_corr = delta_correlation(df, column = column, label1 = label1, label2 = label2, cutoff = cutoff)\n",
    "        perm_delta_corrs.append(perm_delta_corr)\n",
    "    if return_perm_list:\n",
    "        return perm_delta_corrs\n",
    "    z_score = (delta_corr - np.mean(perm_delta_corrs)) / np.std(perm_delta_corrs)\n",
    "    p_val = stats.norm.sf(abs(z_score)) * 2  \n",
    "    return(delta_corr, p_val)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'delta_correlation' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Input \u001b[0;32mIn [31]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mpermutate\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "Input \u001b[0;32mIn [29]\u001b[0m, in \u001b[0;36mpermutate\u001b[0;34m(df, column, label1, label2, cutoff, num_permutations, return_perm_list)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mpermutate\u001b[39m(df, column \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTissue\u001b[39m\u001b[38;5;124m'\u001b[39m, label1 \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTumor\u001b[39m\u001b[38;5;124m'\u001b[39m, label2 \u001b[38;5;241m=\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mNormal\u001b[39m\u001b[38;5;124m'\u001b[39m, cutoff \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m15\u001b[39m, num_permutations \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m10000\u001b[39m, return_perm_list \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m):\n\u001b[0;32m----> 2\u001b[0m     delta_corr \u001b[38;5;241m=\u001b[39m \u001b[43mdelta_correlation\u001b[49m(df, column \u001b[38;5;241m=\u001b[39m column, label1 \u001b[38;5;241m=\u001b[39m label1, label2 \u001b[38;5;241m=\u001b[39m label2, cutoff\u001b[38;5;241m=\u001b[39m cutoff)\n\u001b[1;32m      3\u001b[0m     perm_delta_corrs \u001b[38;5;241m=\u001b[39m []\n\u001b[1;32m      4\u001b[0m     \u001b[38;5;28;01mfor\u001b[39;00m i \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mrange\u001b[39m(\u001b[38;5;241m0\u001b[39m, num_permutations):\n",
      "\u001b[0;31mNameError\u001b[0m: name 'delta_correlation' is not defined"
     ]
    }
   ],
   "source": [
    "permutate(0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
