{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac \n",
    "import pandas as pd\n",
    "import os, sys\n",
    "currentdir = os.path.dirname(os.path.realpath('notebook_steps'))\n",
    "parentdir = os.path.dirname(currentdir)\n",
    "sys.path.append(parentdir)\n",
    "import Delta_Correlation as dc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    }
   ],
   "source": [
    "cancer = cptac.Luad()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 15)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 25)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Transcriptomics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00001</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-2.5347</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2.2545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00009</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.5627</td>\n",
       "      <td>tumor</td>\n",
       "      <td>1.4770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00080</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-1.9422</td>\n",
       "      <td>tumor</td>\n",
       "      <td>1.5103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00083</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>2.1636</td>\n",
       "      <td>tumor</td>\n",
       "      <td>3.0398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00093</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-1.0022</td>\n",
       "      <td>tumor</td>\n",
       "      <td>1.7528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2205156</th>\n",
       "      <td>C3N-02582.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.9497</td>\n",
       "      <td>normal</td>\n",
       "      <td>4.2045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2205157</th>\n",
       "      <td>C3N-02586.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>1.1807</td>\n",
       "      <td>normal</td>\n",
       "      <td>4.2214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2205158</th>\n",
       "      <td>C3N-02587.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>1.1825</td>\n",
       "      <td>normal</td>\n",
       "      <td>4.0836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2205159</th>\n",
       "      <td>C3N-02588.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>1.2729</td>\n",
       "      <td>normal</td>\n",
       "      <td>4.3164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2205160</th>\n",
       "      <td>C3N-02729.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>0.8390</td>\n",
       "      <td>normal</td>\n",
       "      <td>4.2474</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2027299 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID  Gene  Proteomics  Tissue  Transcriptomics\n",
       "0          C3L-00001  A1BG     -2.5347   tumor           2.2545\n",
       "1          C3L-00009  A1BG     -0.5627   tumor           1.4770\n",
       "2          C3L-00080  A1BG     -1.9422   tumor           1.5103\n",
       "3          C3L-00083  A1BG      2.1636   tumor           3.0398\n",
       "4          C3L-00093  A1BG     -1.0022   tumor           1.7528\n",
       "...              ...   ...         ...     ...              ...\n",
       "2205156  C3N-02582.N  ZZZ3      0.9497  normal           4.2045\n",
       "2205157  C3N-02586.N  ZZZ3      1.1807  normal           4.2214\n",
       "2205158  C3N-02587.N  ZZZ3      1.1825  normal           4.0836\n",
       "2205159  C3N-02588.N  ZZZ3      1.2729  normal           4.3164\n",
       "2205160  C3N-02729.N  ZZZ3      0.8390  normal           4.2474\n",
       "\n",
       "[2027299 rows x 5 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = dc.get_prot_trans_df(cancer)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_genes = pd.unique(df.Gene)[0:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Transcriptomics</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00001</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-2.5347</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2.2545</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00009</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.5627</td>\n",
       "      <td>tumor</td>\n",
       "      <td>1.4770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00080</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-1.9422</td>\n",
       "      <td>tumor</td>\n",
       "      <td>1.5103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00083</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>2.1636</td>\n",
       "      <td>tumor</td>\n",
       "      <td>3.0398</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00093</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-1.0022</td>\n",
       "      <td>tumor</td>\n",
       "      <td>1.7528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150110</th>\n",
       "      <td>C3N-02582.N</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>0.6852</td>\n",
       "      <td>normal</td>\n",
       "      <td>2.1092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150111</th>\n",
       "      <td>C3N-02586.N</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>1.4085</td>\n",
       "      <td>normal</td>\n",
       "      <td>1.0860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150112</th>\n",
       "      <td>C3N-02587.N</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>4.8264</td>\n",
       "      <td>normal</td>\n",
       "      <td>3.3245</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150113</th>\n",
       "      <td>C3N-02588.N</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>4.5153</td>\n",
       "      <td>normal</td>\n",
       "      <td>3.5570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1150114</th>\n",
       "      <td>C3N-02729.N</td>\n",
       "      <td>AADAC</td>\n",
       "      <td>4.1405</td>\n",
       "      <td>normal</td>\n",
       "      <td>2.8387</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1052 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID   Gene  Proteomics  Tissue  Transcriptomics\n",
       "0          C3L-00001   A1BG     -2.5347   tumor           2.2545\n",
       "1          C3L-00009   A1BG     -0.5627   tumor           1.4770\n",
       "2          C3L-00080   A1BG     -1.9422   tumor           1.5103\n",
       "3          C3L-00083   A1BG      2.1636   tumor           3.0398\n",
       "4          C3L-00093   A1BG     -1.0022   tumor           1.7528\n",
       "...              ...    ...         ...     ...              ...\n",
       "1150110  C3N-02582.N  AADAC      0.6852  normal           2.1092\n",
       "1150111  C3N-02586.N  AADAC      1.4085  normal           1.0860\n",
       "1150112  C3N-02587.N  AADAC      4.8264  normal           3.3245\n",
       "1150113  C3N-02588.N  AADAC      4.5153  normal           3.5570\n",
       "1150114  C3N-02729.N  AADAC      4.1405  normal           2.8387\n",
       "\n",
       "[1052 rows x 5 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test_df = df[df.Gene.isin(test_genes)]\n",
    "test_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "13.386112451553345\n"
     ]
    }
   ],
   "source": [
    "start = time.time()\n",
    "results = test_df.groupby('Gene').apply(dc.regression)\n",
    "print(time.time() - start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene\n",
       "A1BG     {'interaction_coeff': -0.005621377742532035, '...\n",
       "A2M      {'interaction_coeff': -0.7324729815685269, 'co...\n",
       "AAAS     {'interaction_coeff': 0.3098589828045592, 'con...\n",
       "AACS     {'interaction_coeff': 1.7600473881731062, 'con...\n",
       "AADAC    {'interaction_coeff': -0.7163769385780298, 'co...\n",
       "dtype: object"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>interaction_coeff</th>\n",
       "      <th>condition_coeff</th>\n",
       "      <th>transcript_coeff</th>\n",
       "      <th>intercept</th>\n",
       "      <th>interaction_pval</th>\n",
       "      <th>condition_pval</th>\n",
       "      <th>transcript_pval</th>\n",
       "      <th>intercept_pval</th>\n",
       "      <th>interaction_FDR</th>\n",
       "      <th>condition_FDR</th>\n",
       "      <th>intercept_FDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.005621</td>\n",
       "      <td>-3.138192</td>\n",
       "      <td>-0.069847</td>\n",
       "      <td>1.982755</td>\n",
       "      <td>0.983767</td>\n",
       "      <td>6.692320e-14</td>\n",
       "      <td>7.854612e-01</td>\n",
       "      <td>6.670211e-08</td>\n",
       "      <td>0.983767</td>\n",
       "      <td>3.346160e-13</td>\n",
       "      <td>1.667553e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.732473</td>\n",
       "      <td>4.320107</td>\n",
       "      <td>1.371296</td>\n",
       "      <td>-11.647115</td>\n",
       "      <td>0.128614</td>\n",
       "      <td>3.732341e-01</td>\n",
       "      <td>2.917195e-03</td>\n",
       "      <td>1.347250e-02</td>\n",
       "      <td>0.214357</td>\n",
       "      <td>3.779691e-01</td>\n",
       "      <td>1.684063e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.309859</td>\n",
       "      <td>-1.068018</td>\n",
       "      <td>0.077440</td>\n",
       "      <td>-0.282065</td>\n",
       "      <td>0.268019</td>\n",
       "      <td>3.044971e-01</td>\n",
       "      <td>7.513657e-01</td>\n",
       "      <td>7.524439e-01</td>\n",
       "      <td>0.335023</td>\n",
       "      <td>3.779691e-01</td>\n",
       "      <td>7.524439e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AACS</td>\n",
       "      <td>1.760047</td>\n",
       "      <td>-3.663687</td>\n",
       "      <td>0.273041</td>\n",
       "      <td>-2.546285</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>1.523813e-04</td>\n",
       "      <td>3.992679e-01</td>\n",
       "      <td>2.589706e-03</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>3.809533e-04</td>\n",
       "      <td>4.316177e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AADAC</td>\n",
       "      <td>-0.716377</td>\n",
       "      <td>0.332234</td>\n",
       "      <td>1.616517</td>\n",
       "      <td>-1.877289</td>\n",
       "      <td>0.000211</td>\n",
       "      <td>3.779691e-01</td>\n",
       "      <td>9.359110e-18</td>\n",
       "      <td>3.903905e-08</td>\n",
       "      <td>0.000527</td>\n",
       "      <td>3.779691e-01</td>\n",
       "      <td>1.667553e-07</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene  interaction_coeff  condition_coeff  transcript_coeff  intercept  \\\n",
       "0   A1BG          -0.005621        -3.138192         -0.069847   1.982755   \n",
       "1    A2M          -0.732473         4.320107          1.371296 -11.647115   \n",
       "2   AAAS           0.309859        -1.068018          0.077440  -0.282065   \n",
       "3   AACS           1.760047        -3.663687          0.273041  -2.546285   \n",
       "4  AADAC          -0.716377         0.332234          1.616517  -1.877289   \n",
       "\n",
       "   interaction_pval  condition_pval  transcript_pval  intercept_pval  \\\n",
       "0          0.983767    6.692320e-14     7.854612e-01    6.670211e-08   \n",
       "1          0.128614    3.732341e-01     2.917195e-03    1.347250e-02   \n",
       "2          0.268019    3.044971e-01     7.513657e-01    7.524439e-01   \n",
       "3          0.000001    1.523813e-04     3.992679e-01    2.589706e-03   \n",
       "4          0.000211    3.779691e-01     9.359110e-18    3.903905e-08   \n",
       "\n",
       "   interaction_FDR  condition_FDR  intercept_FDR  \n",
       "0         0.983767   3.346160e-13   1.667553e-07  \n",
       "1         0.214357   3.779691e-01   1.684063e-02  \n",
       "2         0.335023   3.779691e-01   7.524439e-01  \n",
       "3         0.000005   3.809533e-04   4.316177e-03  \n",
       "4         0.000527   3.779691e-01   1.667553e-07  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reg_df = pd.DataFrame(list(results))\n",
    "reg_df.index = results.index\n",
    "reg_df.reset_index(inplace = True)\n",
    "reg_df = reg_df.dropna()\n",
    "reg_df['interaction_FDR'] = ssm.fdrcorrection(reg_df['interaction_pval'])[1]\n",
    "reg_df['condition_FDR'] = ssm.fdrcorrection(reg_df['condition_pval'])[1]\n",
    "reg_df['intercept_FDR'] = ssm.fdrcorrection(reg_df['intercept_pval'])[1]\n",
    "reg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>interaction_coeff</th>\n",
       "      <th>condition_coeff</th>\n",
       "      <th>transcript_coeff</th>\n",
       "      <th>intercept</th>\n",
       "      <th>interaction_pval</th>\n",
       "      <th>condition_pval</th>\n",
       "      <th>transcript_pval</th>\n",
       "      <th>intercept_pval</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>-0.005621</td>\n",
       "      <td>-3.138192</td>\n",
       "      <td>-0.069847</td>\n",
       "      <td>1.982755</td>\n",
       "      <td>0.983767</td>\n",
       "      <td>6.692320e-14</td>\n",
       "      <td>7.854612e-01</td>\n",
       "      <td>6.670211e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>-0.732473</td>\n",
       "      <td>4.320107</td>\n",
       "      <td>1.371296</td>\n",
       "      <td>-11.647115</td>\n",
       "      <td>0.128614</td>\n",
       "      <td>3.732341e-01</td>\n",
       "      <td>2.917195e-03</td>\n",
       "      <td>1.347250e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>0.309859</td>\n",
       "      <td>-1.068018</td>\n",
       "      <td>0.077440</td>\n",
       "      <td>-0.282065</td>\n",
       "      <td>0.268019</td>\n",
       "      <td>3.044971e-01</td>\n",
       "      <td>7.513657e-01</td>\n",
       "      <td>7.524439e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>1.760047</td>\n",
       "      <td>-3.663687</td>\n",
       "      <td>0.273041</td>\n",
       "      <td>-2.546285</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>1.523813e-04</td>\n",
       "      <td>3.992679e-01</td>\n",
       "      <td>2.589706e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AADAC</th>\n",
       "      <td>-0.716377</td>\n",
       "      <td>0.332234</td>\n",
       "      <td>1.616517</td>\n",
       "      <td>-1.877289</td>\n",
       "      <td>0.000211</td>\n",
       "      <td>3.779691e-01</td>\n",
       "      <td>9.359110e-18</td>\n",
       "      <td>3.903905e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       interaction_coeff  condition_coeff  transcript_coeff  intercept  \\\n",
       "Gene                                                                     \n",
       "A1BG           -0.005621        -3.138192         -0.069847   1.982755   \n",
       "A2M            -0.732473         4.320107          1.371296 -11.647115   \n",
       "AAAS            0.309859        -1.068018          0.077440  -0.282065   \n",
       "AACS            1.760047        -3.663687          0.273041  -2.546285   \n",
       "AADAC          -0.716377         0.332234          1.616517  -1.877289   \n",
       "\n",
       "       interaction_pval  condition_pval  transcript_pval  intercept_pval  \n",
       "Gene                                                                      \n",
       "A1BG           0.983767    6.692320e-14     7.854612e-01    6.670211e-08  \n",
       "A2M            0.128614    3.732341e-01     2.917195e-03    1.347250e-02  \n",
       "AAAS           0.268019    3.044971e-01     7.513657e-01    7.524439e-01  \n",
       "AACS           0.000001    1.523813e-04     3.992679e-01    2.589706e-03  \n",
       "AADAC          0.000211    3.779691e-01     9.359110e-18    3.903905e-08  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reg_df['interaction_FDR'] = ssm.fdrcorrection(reg_df['interaction_pval'])\n",
    "reg_df['condition_FDR'] = ssm.fdrcorrection(reg_df['condition_pval'])\n",
    "reg_df['intercept_FDR'] = ssm.fdrcorrection(reg_df['intercept_pval'])\n",
    "reg_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'interaction_coeff': -0.005621377742532035,\n",
       "  'condition_coeff': -3.138191751761857,\n",
       "  'transcript_coeff': -0.0698474956997143,\n",
       "  'intercept': 1.9827545815127643,\n",
       "  'interaction_pval': 0.9837666980841135,\n",
       "  'condition_pval': 6.692320347219538e-14,\n",
       "  'transcript_pval': 0.7854612291907325,\n",
       "  'intercept_pval': 6.67021134035188e-08},\n",
       " {'interaction_coeff': -0.7324729815685269,\n",
       "  'condition_coeff': 4.320106808375722,\n",
       "  'transcript_coeff': 1.3712963041007216,\n",
       "  'intercept': -11.64711542661222,\n",
       "  'interaction_pval': 0.12861448532769265,\n",
       "  'condition_pval': 0.3732341271332117,\n",
       "  'transcript_pval': 0.0029171947230322115,\n",
       "  'intercept_pval': 0.013472500900094433},\n",
       " {'interaction_coeff': 0.3098589828045592,\n",
       "  'condition_coeff': -1.0680182194239456,\n",
       "  'transcript_coeff': 0.07744015339345421,\n",
       "  'intercept': -0.282064610526283,\n",
       "  'interaction_pval': 0.2680186000391487,\n",
       "  'condition_pval': 0.304497050482342,\n",
       "  'transcript_pval': 0.7513657495382859,\n",
       "  'intercept_pval': 0.752443948475882},\n",
       " {'interaction_coeff': 1.7600473881731062,\n",
       "  'condition_coeff': -3.6636866643914567,\n",
       "  'transcript_coeff': 0.27304140345966843,\n",
       "  'intercept': -2.546285178868433,\n",
       "  'interaction_pval': 1.097822037186981e-06,\n",
       "  'condition_pval': 0.0001523813075010661,\n",
       "  'transcript_pval': 0.39926792463185257,\n",
       "  'intercept_pval': 0.0025897063867259345},\n",
       " {'interaction_coeff': -0.7163769385780298,\n",
       "  'condition_coeff': 0.33223364875033967,\n",
       "  'transcript_coeff': 1.6165168402172554,\n",
       "  'intercept': -1.8772893621111963,\n",
       "  'interaction_pval': 0.0002109454012993184,\n",
       "  'condition_pval': 0.3779690954878847,\n",
       "  'transcript_pval': 9.359109764496483e-18,\n",
       "  'intercept_pval': 3.903904537400446e-08}]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(results)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "import statsmodels.stats.multitest as ssm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
