{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import cptac.utils as ut\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that pdac index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-891d3f5549c9>, line 5)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The pdac data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-891d3f5549c9>, line 6)\n"
     ]
    }
   ],
   "source": [
    "ccrcc = cptac.Ccrcc()\n",
    "en = cptac.Endometrial()\n",
    "luad = cptac.Luad()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()\n",
    "pdac = cptac.Pdac()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [ccrcc, en, luad, hnscc, lscc, pdac]\n",
    "cancer_names = ['CCRCC', 'Endometrial', 'LUAD', 'HNSCC', 'LSCC', 'PDAC']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-4-81639f6741fb>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (<ipython-input-4-81639f6741fb>, line 83)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-4-81639f6741fb>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (<ipython-input-4-81639f6741fb>, line 83)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-4-81639f6741fb>, line 5)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (<ipython-input-4-81639f6741fb>, line 83)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Proteomic</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.859821</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.291127</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.824992</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.579658</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.673249</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.096200</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.043751</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.209196</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.801038</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.395915</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148696</th>\n",
       "      <td>C3N-03839.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148697</th>\n",
       "      <td>C3N-03840.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148698</th>\n",
       "      <td>C3N-03884.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148699</th>\n",
       "      <td>C3N-04119.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5148700</th>\n",
       "      <td>C3N-04282.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Normal</td>\n",
       "      <td>20.068122</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27493621 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID  Gene  Transcriptomics   Stage  Proteomic Cancer\n",
       "0        C3L-00004.N  A1BG         0.859821  Normal   0.291127  CCRCC\n",
       "1        C3L-00010.N  A1BG         0.824992  Normal  -0.579658  CCRCC\n",
       "2        C3L-00011.N  A1BG         0.673249  Normal  -0.096200  CCRCC\n",
       "3        C3L-00026.N  A1BG         3.043751  Normal  -0.209196  CCRCC\n",
       "4        C3L-00079.N  A1BG         0.801038  Normal  -0.395915  CCRCC\n",
       "...              ...   ...              ...     ...        ...    ...\n",
       "5148696  C3N-03839.N  ZZZ3              NaN  Normal        NaN   PDAC\n",
       "5148697  C3N-03840.N  ZZZ3              NaN  Normal        NaN   PDAC\n",
       "5148698  C3N-03884.N  ZZZ3              NaN  Normal        NaN   PDAC\n",
       "5148699  C3N-04119.N  ZZZ3              NaN  Normal        NaN   PDAC\n",
       "5148700  C3N-04282.N  ZZZ3              NaN  Normal  20.068122   PDAC\n",
       "\n",
       "[27493621 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_normal_df = cancer.get_proteomics('normal')\n",
    "    if isinstance(prot_normal_df.columns, pd.MultiIndex):\n",
    "        prot_normal_df = ut.reduce_multiindex(df= prot_normal_df, levels_to_drop = 'Database_ID')\n",
    "    trans_normal_df = cancer.get_transcriptomics('normal')\n",
    "    if isinstance(trans_normal_df.columns, pd.MultiIndex):\n",
    "        trans_normal_df = ut.reduce_multiindex(df = trans_normal_df, levels_to_drop='Database_ID')\n",
    "    prot_normal_df['Patient_ID'] = prot_normal_df.index\n",
    "    trans_normal_df['Patient_ID'] = trans_normal_df.index\n",
    "    prot_normal_df = prot_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    trans_normal_df = trans_normal_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    prot_normal_df['Stage'] = ['Normal'] * len(prot_normal_df)\n",
    "    trans_normal_df['Stage'] = ['Normal'] * len(trans_normal_df)\n",
    "    if cancer == ccrcc:\n",
    "        stage = cancer.get_clinical('tumor')[['tumor_stage_pathological']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'tumor_stage_pathological': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "    elif cancer == luad or cancer == lscc:\n",
    "        stage = cancer.get_clinical('tumor')[['Stage']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['1A', '1B', '1', 1, 'I', 'IA', 'IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['2A', '2B', 'II', 'IIA', 'IIB']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['3A', '3', 3, 'III', 'IIIA', 'IIIB']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    elif cancer == en:\n",
    "        stage = cancer.get_clinical('tumor')[['FIGO_stage']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'FIGO_stage': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['IA', 'IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['II']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['IIIA', 'IIIB', 'IIIC2', 'IIIC1']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['IVB', 'IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    elif cancer == hnscc:\n",
    "        stage = cancer.get_clinical('tumor')[['patho_staging_curated']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'patho_staging_curated': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        \n",
    "    elif cancer == pdac:\n",
    "        stage = cancer.get_clinical('tumor')[['tumor_stage_pathological']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'tumor_stage_pathological': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['Stage IA', 'Stage IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['Stage IIA', 'Stage IIB']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['Stage III']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['Stage IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "        \n",
    "        \n",
    "    prot_tumor_df = cancer.get_proteomics('tumor')\n",
    "    if isinstance(prot_tumor_df.columns, pd.MultiIndex):\n",
    "        prot_tumor_df = ut.reduce_multiindex(df= prot_tumor_df, levels_to_drop = 'Database_ID')\n",
    "    trans_tumor_df = cancer.get_transcriptomics('tumor')\n",
    "    if isinstance(trans_tumor_df.columns, pd.MultiIndex):\n",
    "        trans_tumor_df = ut.reduce_multiindex(df = trans_tumor_df, levels_to_drop='Database_ID')\n",
    "    prot_tumor_df['Patient_ID'] = prot_tumor_df.index\n",
    "    trans_tumor_df['Patient_ID'] = trans_tumor_df.index\n",
    "    prot_tumor_df = prot_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Proteomic')\n",
    "    trans_tumor_df = trans_tumor_df.melt(id_vars = 'Patient_ID', var_name = 'Gene', value_name = 'Transcriptomics')\n",
    "    trans_tumor_df = pd.merge(trans_tumor_df, stage, how = 'outer')\n",
    "    prot_tumor_df = pd.merge(prot_tumor_df, stage, how = 'outer')\n",
    "    prot_df = pd.merge(prot_tumor_df, prot_normal_df, how = 'outer')\n",
    "    trans_df = pd.merge(trans_normal_df, trans_tumor_df, how = 'outer')\n",
    "    cancer_df = pd.merge(trans_df, prot_df,how = 'outer')\n",
    "    cancer_df['Cancer'] = [cancer_name] * len(cancer_df)\n",
    "    cancer_dfs.append(cancer_df)\n",
    "prot_trans_stages_df = pd.concat(cancer_dfs)\n",
    "prot_trans_stages_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Proteomic</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.859821</td>\n",
       "      <td>Normal</td>\n",
       "      <td>0.291127</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00010.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.824992</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.579658</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00011.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.673249</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.096200</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00026.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.043751</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.209196</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00079.N</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.801038</td>\n",
       "      <td>Normal</td>\n",
       "      <td>-0.395915</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4517166</th>\n",
       "      <td>C3N-04284</td>\n",
       "      <td>ZW10</td>\n",
       "      <td>10.161812</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>24.909713</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4517167</th>\n",
       "      <td>C3N-04284</td>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>8.849157</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>20.684593</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4517173</th>\n",
       "      <td>C3N-04284</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>11.665104</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>20.666625</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4517174</th>\n",
       "      <td>C3N-04284</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>10.118784</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>27.897518</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4517175</th>\n",
       "      <td>C3N-04284</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>12.709588</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>24.613373</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9311569 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID    Gene  Transcriptomics      Stage  Proteomic Cancer\n",
       "0        C3L-00004.N    A1BG         0.859821     Normal   0.291127  CCRCC\n",
       "1        C3L-00010.N    A1BG         0.824992     Normal  -0.579658  CCRCC\n",
       "2        C3L-00011.N    A1BG         0.673249     Normal  -0.096200  CCRCC\n",
       "3        C3L-00026.N    A1BG         3.043751     Normal  -0.209196  CCRCC\n",
       "4        C3L-00079.N    A1BG         0.801038     Normal  -0.395915  CCRCC\n",
       "...              ...     ...              ...        ...        ...    ...\n",
       "4517166    C3N-04284    ZW10        10.161812  Stage III  24.909713   PDAC\n",
       "4517167    C3N-04284  ZWILCH         8.849157  Stage III  20.684593   PDAC\n",
       "4517173    C3N-04284  ZYG11B        11.665104  Stage III  20.666625   PDAC\n",
       "4517174    C3N-04284     ZYX        10.118784  Stage III  27.897518   PDAC\n",
       "4517175    C3N-04284   ZZEF1        12.709588  Stage III  24.613373   PDAC\n",
       "\n",
       "[9311569 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "prot_trans_stages_df = prot_trans_stages_df.dropna()\n",
    "prot_trans_stages_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Gene</th>\n",
       "      <th>PearsonR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.328964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.745763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>A2M</td>\n",
       "      <td>0.287323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.156649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>AACS</td>\n",
       "      <td>0.707274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611575</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>0.385813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611583</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.171495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611585</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.480944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611587</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.097533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>611589</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.090946</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>235819 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Cancer      Stage    Gene  PearsonR\n",
       "1       CCRCC     Normal    A1BG  0.328964\n",
       "3       CCRCC     Normal    A1CF  0.745763\n",
       "5       CCRCC     Normal     A2M  0.287323\n",
       "9       CCRCC     Normal    AAAS  0.156649\n",
       "11      CCRCC     Normal    AACS  0.707274\n",
       "...       ...        ...     ...       ...\n",
       "611575   PDAC  Stage III  ZWILCH  0.385813\n",
       "611583   PDAC  Stage III  ZYG11B  0.171495\n",
       "611585   PDAC  Stage III     ZYX  0.480944\n",
       "611587   PDAC  Stage III   ZZEF1  0.097533\n",
       "611589   PDAC  Stage III    ZZZ3 -0.090946\n",
       "\n",
       "[235819 rows x 4 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = prot_trans_stages_df\n",
    "df = df.drop(columns = 'Patient_ID')\n",
    "grouped_stages = df.groupby(['Cancer', 'Stage', 'Gene'])\n",
    "corr_df = grouped_stages.apply(lambda x: x.corr(method = 'pearson', min_periods = 10))\n",
    "corr_df = corr_df.drop(columns = 'Proteomic')\n",
    "corr_df.reset_index(inplace = True)\n",
    "corr_df = corr_df[corr_df.level_3 == 'Proteomic']\n",
    "corr_df = corr_df.drop(columns = 'level_3')\n",
    "corr_df = corr_df.rename(columns = {'Transcriptomics': 'PearsonR'})\n",
    "corr_df = corr_df.dropna()\n",
    "corr_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df.to_csv('data/Cancer_stages_correlations.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
