{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import scipy\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import math\n",
    "import pandas as pd\n",
    "import statistics\n",
    "import parse_correlations_dataframe as get_corr\n",
    "import copy\n",
    "import get_correlations\n",
    "import cptac.utils as ut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "brca = cptac.Brca()\n",
    "ccrcc = cptac.Ccrcc()\n",
    "colon = cptac.Colon()\n",
    "en = cptac.Endometrial()\n",
    "gbm = cptac.Gbm()\n",
    "luad = cptac.Luad()\n",
    "ovarian = cptac.Ovarian()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [brca, ccrcc, colon, en, gbm, luad, ovarian, hnscc, lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers2 = [ccrcc, en, gbm, luad, hnscc, lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_gene_list(tissues):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    whole_gene_list = set()\n",
    "    for tissue in tissues:\n",
    "        tissue_list = get_gene_list(tissue)\n",
    "        for gene in tissue_list:\n",
    "            whole_gene_list.add(gene)\n",
    "    return sorted(list(whole_gene_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_list(tissue):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    gene_list = []\n",
    "    prot = tissue.get_proteomics()\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = prot.columns.get_level_values(0)\n",
    "    trans = tissue.get_transcriptomics()\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = trans.columns.get_level_values(0)\n",
    "    for i in prot:\n",
    "        if i in trans and i not in gene_list:\n",
    "            gene_list.append(i)\n",
    "    return gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = get_all_gene_list(cancers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "type_dict = {brca:\"brca\",ccrcc:\"ccrcc\",colon:\"colon\",en:\"endometrial\",gbm:\"gbm\",luad:\"luad\",\n",
    "                  ovarian:\"ovarian\",hnscc:\"hnscc\",lscc:\"lscc\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://link.springer.com/article/10.3758/s13428-012-0289-7\n",
    "def compare_correlations(r1, r2, n1, n2):\n",
    "    rp1 = np.arctanh(r1)\n",
    "    rp2 = np.arctanh(r2)\n",
    "    \n",
    "    if n1 < 4 or n2 < 4:\n",
    "        return(np.nan)\n",
    "    Sr12 = math.sqrt((1/(n1-3))+(1/(n2-3)))\n",
    "    z = (rp1-rp2) / Sr12\n",
    "    p = scipy.stats.norm.sf(abs(z))*2\n",
    "    return (p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def permute(df,original_correlation, label_1, label_2, column_one, column_two, permutation_times):\n",
    "    permutation_list = []\n",
    "    permu_df = copy.deepcopy(df)\n",
    "\n",
    "    for i in range(permutation_times):\n",
    "        permu_df[\"Type\"] = np.random.permutation(permu_df[\"Type\"])\n",
    "        permu_is_label_1 = permu_df[\"Type\"] == label_1\n",
    "        permu_is_label_2 = permu_df[\"Type\"] == label_2\n",
    "        label_1_correlation,label_1_pval = scipy.stats.pearsonr(permu_df[permu_is_label_1][column_one], permu_df[permu_is_label_1][column_two])\n",
    "        label_2_correlation,label_2_pval = scipy.stats.pearsonr(permu_df[permu_is_label_2][column_one], permu_df[permu_is_label_2][column_two])\n",
    "        delta = label_1_correlation - label_2_correlation\n",
    "        permutation_list.append(delta)\n",
    "        \n",
    "    z_score = (original_correlation - np.mean(permutation_list)) / np.std(permutation_list)\n",
    "    p_val = scipy.stats.norm.sf(abs(z_score))*2\n",
    "    return p_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "tot_diff_list = []\n",
    "tot_pval_list = []\n",
    "tot_perm_list = []\n",
    "for cancer in cancers2:\n",
    "    cancer_diff_list = [type_dict[cancer]]\n",
    "    cancer_pval_list = [type_dict[cancer]]\n",
    "    cancer_perm_list = [type_dict[cancer]]\n",
    "    \n",
    "    tumor_cancer_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"tumor\",quiet=True)\n",
    "    if isinstance(tumor_cancer_df.columns, pd.MultiIndex):\n",
    "        tumor_cancer_df = ut.reduce_multiindex(df = tumor_cancer_df, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "            \n",
    "    normal_cancer_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"normal\",quiet=True)\n",
    "    if isinstance(normal_cancer_df.columns, pd.MultiIndex):\n",
    "        normal_cancer_df = ut.reduce_multiindex(df = normal_cancer_df, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "    \n",
    "    for gene in gene_list:\n",
    "        gene_trans = gene + \"_transcriptomics\"\n",
    "        gene_prot = gene + \"_proteomics\"\n",
    "        gene_in_tumor = gene_trans in tumor_cancer_df.columns and gene_prot in tumor_cancer_df.columns\n",
    "        gene_in_normal = gene_trans in normal_cancer_df.columns and gene_prot in normal_cancer_df.columns\n",
    "        \n",
    "        if not(gene_in_tumor and gene_in_normal):\n",
    "            cancer_diff_list.append(np.nan)\n",
    "            cancer_pval_list.append(np.nan)\n",
    "            cancer_perm_list.append(np.nan)\n",
    "            continue\n",
    "\n",
    "        tumor_df = tumor_cancer_df[[gene_trans,gene_prot]]\n",
    "        #The following line takes care of the problem that arises when reducing the multi-index like we did earlier.\n",
    "        #There are sometimes multiple columns of proteomics and sometimes multiple of transcriptomics, this \n",
    "        #takes the first one.\n",
    "        if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame) or isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "            trans_col = tumor_df[gene_trans]\n",
    "            if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame):\n",
    "                trans_col = trans_col.iloc[:,0]\n",
    "            prot_col = tumor_df[gene_prot]\n",
    "            if isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame):\n",
    "                prot_col = prot_col.iloc[:,0]\n",
    "            frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "            tumor_df = pd.DataFrame(frame)\n",
    "        tumor_df = tumor_df.dropna()\n",
    "        num_tumor = len(tumor_df)\n",
    "        tumor_corr = tumor_df.corr().iloc[0][1]\n",
    "        \n",
    "        normal_df = normal_cancer_df[[gene_trans,gene_prot]]\n",
    "        if isinstance(normal_df[gene_trans], pd.core.frame.DataFrame) or isinstance(normal_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "            trans_col = normal_df[gene_trans]\n",
    "            if isinstance(normal_df[gene_trans], pd.core.frame.DataFrame):\n",
    "                trans_col = trans_col.iloc[:,0]\n",
    "            prot_col = normal_df[gene_prot]\n",
    "            if isinstance(normal_df[gene_prot], pd.core.frame.DataFrame):\n",
    "                prot_col = prot_col.iloc[:,0]\n",
    "            frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "            normal_df = pd.DataFrame(frame)\n",
    "        normal_df = normal_df.dropna()\n",
    "        num_normal = len(normal_df)\n",
    "        normal_corr = normal_df.corr().iloc[0][1]\n",
    "        \n",
    "        corr_diff = tumor_corr - normal_corr\n",
    "        cancer_diff_list.append(corr_diff)\n",
    "        \n",
    "        gene_pval = compare_correlations(tumor_corr, normal_corr, num_tumor, num_normal)\n",
    "        cancer_pval_list.append(gene_pval)\n",
    "        \n",
    "        #Permutations\n",
    "        if num_tumor < 4 or num_normal < 4 or gene_pval > .01:\n",
    "            cancer_perm_list.append(np.nan)\n",
    "            continue\n",
    "        tumor_label_list = ['tumor'] * len(tumor_df)\n",
    "        tumor_df[\"Type\"] = tumor_label_list\n",
    "        \n",
    "        normal_label_list = ['normal'] * len(normal_df)\n",
    "        normal_df[\"Type\"] = normal_label_list\n",
    "        \n",
    "        perm_list = [tumor_df,normal_df]\n",
    "        perm_df = pd.concat(perm_list)\n",
    "        \n",
    "        column_one = perm_df.columns[0]\n",
    "        column_two = perm_df.columns[1]\n",
    "        perm_val = permute(perm_df,corr_diff,\"tumor\",\"normal\",column_one,column_two,1000)\n",
    "        cancer_perm_list.append(perm_val)\n",
    "        \n",
    "    tot_diff_list.append(cancer_diff_list)\n",
    "    tot_pval_list.append(cancer_pval_list)\n",
    "    tot_perm_list.append(cancer_perm_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = [\"Cancer\"] \n",
    "labels.extend(gene_list)\n",
    "df = pd.DataFrame.from_records(tot_diff_list,columns=labels)\n",
    "df2 = pd.DataFrame.from_records(tot_pval_list,columns=labels)\n",
    "df3 = pd.DataFrame.from_records(tot_perm_list,columns=labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.047054</td>\n",
       "      <td>0.001344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.939375e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.134020e-06</td>\n",
       "      <td>2.205229e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gbm</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.947565e-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>luad</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.362011e-40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.016913</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.288210e-10</td>\n",
       "      <td>1.115662e-20</td>\n",
       "      <td>1.769466e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.566817e-10</td>\n",
       "      <td>4.873755e-08</td>\n",
       "      <td>1.684699e-15</td>\n",
       "      <td>0.000017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000226</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.368489e-04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.733422e-03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003677</td>\n",
       "      <td>6.991469e-04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>lscc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.333383e-22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.001093</td>\n",
       "      <td>4.714786e-13</td>\n",
       "      <td>6.781454e-52</td>\n",
       "      <td>9.657885e-34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011712</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.821957e-09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.019456</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 14637 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Cancer      A1BG      A1CF  A2M         A2ML1  A4GALT  AAAS  \\\n",
       "0        ccrcc  0.047054  0.001344  NaN           NaN     NaN   NaN   \n",
       "1  endometrial       NaN       NaN  NaN           NaN     NaN   NaN   \n",
       "2          gbm       NaN       NaN  NaN           NaN     NaN   NaN   \n",
       "3         luad       NaN       NaN  NaN           NaN     NaN   NaN   \n",
       "4        hnscc       NaN       NaN  NaN           NaN     NaN   NaN   \n",
       "5         lscc       NaN       NaN  NaN  3.333383e-22     NaN   NaN   \n",
       "\n",
       "           AACS     AADAC     AADAT  ...    ZSWIM9          ZW10  \\\n",
       "0           NaN       NaN       NaN  ...       NaN           NaN   \n",
       "1  6.939375e-08       NaN       NaN  ...       NaN           NaN   \n",
       "2           NaN       NaN       NaN  ...       NaN           NaN   \n",
       "3  2.362011e-40       NaN  0.016913  ...       NaN  1.288210e-10   \n",
       "4           NaN  0.000226       NaN  ...       NaN  6.368489e-04   \n",
       "5           NaN       NaN       NaN  ...  0.001093  4.714786e-13   \n",
       "\n",
       "         ZWILCH         ZWINT  ZXDA      ZXDC        ZYG11B           ZYX  \\\n",
       "0           NaN           NaN   NaN       NaN           NaN           NaN   \n",
       "1  5.134020e-06  2.205229e-09   NaN       NaN           NaN           NaN   \n",
       "2  3.947565e-08           NaN   NaN       NaN           NaN           NaN   \n",
       "3  1.115662e-20  1.769466e-05   NaN       NaN  2.566817e-10  4.873755e-08   \n",
       "4           NaN  1.733422e-03   NaN  0.003677  6.991469e-04           NaN   \n",
       "5  6.781454e-52  9.657885e-34   NaN  0.011712           NaN  6.821957e-09   \n",
       "\n",
       "          ZZEF1      ZZZ3  \n",
       "0           NaN       NaN  \n",
       "1           NaN       NaN  \n",
       "2           NaN       NaN  \n",
       "3  1.684699e-15  0.000017  \n",
       "4           NaN       NaN  \n",
       "5           NaN  0.019456  \n",
       "\n",
       "[6 rows x 14637 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.to_csv(\"corr_diff.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df2.to_csv(\"p_val.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df3.to_csv(\"permutation_pval01.csv\",index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
