{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import scipy\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import math\n",
    "import pandas as pd\n",
    "import statistics\n",
    "import parse_correlations_dataframe as get_corr\n",
    "import copy\n",
    "import get_correlations\n",
    "import cptac.utils as ut\n",
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "brca = cptac.Brca()\n",
    "ccrcc = cptac.Ccrcc()\n",
    "colon = cptac.Colon()\n",
    "en = cptac.Endometrial()\n",
    "gbm = cptac.Gbm()\n",
    "luad = cptac.Luad()\n",
    "ovarian = cptac.Ovarian()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [ccrcc, en, luad, hnscc, lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_gene_list(tissues):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    whole_gene_list = set()\n",
    "    for tissue in tissues:\n",
    "        tissue_list = get_gene_list(tissue)\n",
    "        for gene in tissue_list:\n",
    "            whole_gene_list.add(gene)\n",
    "    return sorted(list(whole_gene_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_list(tissue):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    gene_list = []\n",
    "    prot = tissue.get_proteomics()\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = prot.columns.get_level_values(0)\n",
    "    trans = tissue.get_transcriptomics()\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = trans.columns.get_level_values(0)\n",
    "    for i in prot:\n",
    "        if i in trans and i not in gene_list:\n",
    "            gene_list.append(i)\n",
    "    return gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# gene_list = get_all_gene_list(cancers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('gene_list.csv', newline='') as in_file:\n",
    "    reader = csv.reader(in_file)\n",
    "    gene_list = list(reader)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "type_dict = {brca:\"brca\",ccrcc:\"ccrcc\",colon:\"colon\",en:\"endometrial\",gbm:\"gbm\",luad:\"luad\",\n",
    "                  ovarian:\"ovarian\",hnscc:\"hnscc\",lscc:\"lscc\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://link.springer.com/article/10.3758/s13428-012-0289-7\n",
    "def compare_correlations(r1, r2, n1, n2):\n",
    "    rp1 = np.arctanh(r1)\n",
    "    rp2 = np.arctanh(r2)\n",
    "    \n",
    "    if n1 < 4 or n2 < 4:\n",
    "        return(np.nan)\n",
    "    Sr12 = math.sqrt((1/(n1-3))+(1/(n2-3)))\n",
    "    z = (rp1-rp2) / Sr12\n",
    "    p = scipy.stats.norm.sf(abs(z))*2\n",
    "    return (p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def permute(df,original_correlation, label_1, label_2, column_one, column_two, permutation_times):\n",
    "    permutation_list = []\n",
    "    permu_df = copy.deepcopy(df)\n",
    "\n",
    "    for i in range(permutation_times):\n",
    "        permu_df[\"Type\"] = np.random.permutation(permu_df[\"Type\"])\n",
    "        permu_is_label_1 = permu_df[\"Type\"] == label_1\n",
    "        permu_is_label_2 = permu_df[\"Type\"] == label_2\n",
    "        label_1_correlation,label_1_pval = scipy.stats.pearsonr(permu_df[permu_is_label_1][column_one], permu_df[permu_is_label_1][column_two])\n",
    "        label_2_correlation,label_2_pval = scipy.stats.pearsonr(permu_df[permu_is_label_2][column_one], permu_df[permu_is_label_2][column_two])\n",
    "        delta = label_1_correlation - label_2_correlation\n",
    "        permutation_list.append(delta)\n",
    "        \n",
    "    z_score = (original_correlation - np.mean(permutation_list)) / np.std(permutation_list)\n",
    "    p_val = scipy.stats.norm.sf(abs(z_score))*2\n",
    "    return p_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "tot_diff_list = []\n",
    "tot_pval_list = []\n",
    "tot_perm_list = []\n",
    "for cancer in cancers:\n",
    "    cancer_diff_list = [type_dict[cancer]]\n",
    "    cancer_pval_list = [type_dict[cancer]]\n",
    "    cancer_perm_list = [type_dict[cancer]]\n",
    "    \n",
    "    tumor_cancer_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"tumor\",quiet=True)\n",
    "    if isinstance(tumor_cancer_df.columns, pd.MultiIndex):\n",
    "        tumor_cancer_df = ut.reduce_multiindex(df = tumor_cancer_df, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "            \n",
    "    normal_cancer_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"normal\",quiet=True)\n",
    "    if isinstance(normal_cancer_df.columns, pd.MultiIndex):\n",
    "        normal_cancer_df = ut.reduce_multiindex(df = normal_cancer_df, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "    \n",
    "    for gene in gene_list:\n",
    "        gene_trans = gene + \"_transcriptomics\"\n",
    "        gene_prot = gene + \"_proteomics\"\n",
    "        gene_in_tumor = gene_trans in tumor_cancer_df.columns and gene_prot in tumor_cancer_df.columns\n",
    "        gene_in_normal = gene_trans in normal_cancer_df.columns and gene_prot in normal_cancer_df.columns\n",
    "        \n",
    "        if not(gene_in_tumor and gene_in_normal):\n",
    "            cancer_diff_list.append(np.nan)\n",
    "            cancer_pval_list.append(np.nan)\n",
    "            cancer_perm_list.append(np.nan)\n",
    "            continue\n",
    "\n",
    "        tumor_df = tumor_cancer_df[[gene_trans,gene_prot]]\n",
    "        #The following line takes care of the problem that arises when reducing the multi-index like we did earlier.\n",
    "        #There are sometimes multiple columns of proteomics and sometimes multiple of transcriptomics, this \n",
    "        #takes the first one.\n",
    "        if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame) or isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "            trans_col = tumor_df[gene_trans]\n",
    "            if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame):\n",
    "                trans_col = trans_col.iloc[:,0]\n",
    "            prot_col = tumor_df[gene_prot]\n",
    "            if isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame):\n",
    "                prot_col = prot_col.iloc[:,0]\n",
    "            frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "            tumor_df = pd.DataFrame(frame)\n",
    "        tumor_df = tumor_df.dropna()\n",
    "        num_tumor = len(tumor_df)\n",
    "        tumor_corr = tumor_df.corr().iloc[0][1]\n",
    "        \n",
    "        normal_df = normal_cancer_df[[gene_trans,gene_prot]]\n",
    "        if isinstance(normal_df[gene_trans], pd.core.frame.DataFrame) or isinstance(normal_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "            trans_col = normal_df[gene_trans]\n",
    "            if isinstance(normal_df[gene_trans], pd.core.frame.DataFrame):\n",
    "                trans_col = trans_col.iloc[:,0]\n",
    "            prot_col = normal_df[gene_prot]\n",
    "            if isinstance(normal_df[gene_prot], pd.core.frame.DataFrame):\n",
    "                prot_col = prot_col.iloc[:,0]\n",
    "            frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "            normal_df = pd.DataFrame(frame)\n",
    "        normal_df = normal_df.dropna()\n",
    "        num_normal = len(normal_df)\n",
    "        normal_corr = normal_df.corr().iloc[0][1]\n",
    "        \n",
    "        corr_diff = tumor_corr - normal_corr\n",
    "        cancer_diff_list.append(corr_diff)\n",
    "        \n",
    "        gene_pval = compare_correlations(tumor_corr, normal_corr, num_tumor, num_normal)\n",
    "        cancer_pval_list.append(gene_pval)\n",
    "        cancer_perm_list.append(np.nan)\n",
    "        \n",
    "        #Permutations\n",
    "#         if num_tumor < 4 or num_normal < 4 or gene_pval > .2:\n",
    "#             cancer_perm_list.append(np.nan)\n",
    "#             continue\n",
    "#         tumor_label_list = ['tumor'] * len(tumor_df)\n",
    "#         tumor_df[\"Type\"] = tumor_label_list\n",
    "        \n",
    "#         normal_label_list = ['normal'] * len(normal_df)\n",
    "#         normal_df[\"Type\"] = normal_label_list\n",
    "        \n",
    "#         perm_list = [tumor_df,normal_df]\n",
    "#         perm_df = pd.concat(perm_list)\n",
    "        \n",
    "#         column_one = perm_df.columns[0]\n",
    "#         column_two = perm_df.columns[1]\n",
    "#         perm_val = permute(perm_df,corr_diff,\"tumor\",\"normal\",column_one,column_two,1000)\n",
    "#         cancer_perm_list.append(perm_val)\n",
    "        \n",
    "    tot_diff_list.append(cancer_diff_list)\n",
    "    tot_pval_list.append(cancer_pval_list)\n",
    "    tot_perm_list.append(cancer_perm_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = [\"Cancer\"] \n",
    "labels.extend(gene_list)\n",
    "df = pd.DataFrame.from_records(tot_diff_list,columns=labels)\n",
    "df2 = pd.DataFrame.from_records(tot_pval_list,columns=labels)\n",
    "df3 = pd.DataFrame.from_records(tot_perm_list,columns=labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>-0.388745</td>\n",
       "      <td>0.146993</td>\n",
       "      <td>-0.277035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.795957</td>\n",
       "      <td>-0.044291</td>\n",
       "      <td>-0.128218</td>\n",
       "      <td>1.196756</td>\n",
       "      <td>-0.174925</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.269728</td>\n",
       "      <td>0.329727</td>\n",
       "      <td>-0.337301</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.337470</td>\n",
       "      <td>0.109414</td>\n",
       "      <td>-0.223934</td>\n",
       "      <td>-0.138352</td>\n",
       "      <td>-0.220727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.061654</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.025535</td>\n",
       "      <td>0.402210</td>\n",
       "      <td>0.399061</td>\n",
       "      <td>0.663703</td>\n",
       "      <td>0.593777</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.302879</td>\n",
       "      <td>...</td>\n",
       "      <td>0.177842</td>\n",
       "      <td>0.088061</td>\n",
       "      <td>0.759429</td>\n",
       "      <td>1.006282</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.267078</td>\n",
       "      <td>0.400272</td>\n",
       "      <td>0.147305</td>\n",
       "      <td>0.367791</td>\n",
       "      <td>0.342423</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>luad</td>\n",
       "      <td>-0.039149</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.097233</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.206094</td>\n",
       "      <td>0.683092</td>\n",
       "      <td>-0.119746</td>\n",
       "      <td>0.630155</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.557832</td>\n",
       "      <td>0.904669</td>\n",
       "      <td>0.645076</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.087794</td>\n",
       "      <td>0.707991</td>\n",
       "      <td>0.472139</td>\n",
       "      <td>0.418589</td>\n",
       "      <td>0.661323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>-0.002651</td>\n",
       "      <td>0.407295</td>\n",
       "      <td>-0.194186</td>\n",
       "      <td>0.117838</td>\n",
       "      <td>0.555675</td>\n",
       "      <td>0.462864</td>\n",
       "      <td>0.024965</td>\n",
       "      <td>-0.333934</td>\n",
       "      <td>-0.294627</td>\n",
       "      <td>...</td>\n",
       "      <td>1.039965</td>\n",
       "      <td>0.395189</td>\n",
       "      <td>0.119698</td>\n",
       "      <td>1.079145</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.591841</td>\n",
       "      <td>-0.453269</td>\n",
       "      <td>-0.010672</td>\n",
       "      <td>0.255505</td>\n",
       "      <td>-0.060708</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>lscc</td>\n",
       "      <td>0.196946</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.062630</td>\n",
       "      <td>0.649123</td>\n",
       "      <td>0.246831</td>\n",
       "      <td>0.304418</td>\n",
       "      <td>0.150665</td>\n",
       "      <td>-0.068258</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.613891</td>\n",
       "      <td>0.491043</td>\n",
       "      <td>0.646430</td>\n",
       "      <td>0.838981</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.332350</td>\n",
       "      <td>0.231167</td>\n",
       "      <td>0.315679</td>\n",
       "      <td>0.268239</td>\n",
       "      <td>0.446553</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 13670 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Cancer      A1BG      A1CF       A2M     A2ML1    A4GALT      AAAS  \\\n",
       "0        ccrcc -0.388745  0.146993 -0.277035       NaN -0.795957 -0.044291   \n",
       "1  endometrial  0.061654       NaN -0.025535  0.402210  0.399061  0.663703   \n",
       "2         luad -0.039149       NaN  0.097233       NaN       NaN  0.206094   \n",
       "3        hnscc -0.002651  0.407295 -0.194186  0.117838  0.555675  0.462864   \n",
       "4         lscc  0.196946       NaN -0.062630  0.649123  0.246831  0.304418   \n",
       "\n",
       "       AACS     AADAC     AADAT  ...    ZSWIM9      ZW10    ZWILCH     ZWINT  \\\n",
       "0 -0.128218  1.196756 -0.174925  ...       NaN  0.269728  0.329727 -0.337301   \n",
       "1  0.593777       NaN  0.302879  ...  0.177842  0.088061  0.759429  1.006282   \n",
       "2  0.683092 -0.119746  0.630155  ...       NaN  0.557832  0.904669  0.645076   \n",
       "3  0.024965 -0.333934 -0.294627  ...  1.039965  0.395189  0.119698  1.079145   \n",
       "4  0.150665 -0.068258       NaN  ...  0.613891  0.491043  0.646430  0.838981   \n",
       "\n",
       "   ZXDA      ZXDC    ZYG11B       ZYX     ZZEF1      ZZZ3  \n",
       "0   NaN  0.337470  0.109414 -0.223934 -0.138352 -0.220727  \n",
       "1   NaN -0.267078  0.400272  0.147305  0.367791  0.342423  \n",
       "2   NaN  0.087794  0.707991  0.472139  0.418589  0.661323  \n",
       "3   NaN  0.591841 -0.453269 -0.010672  0.255505 -0.060708  \n",
       "4   NaN  0.332350  0.231167  0.315679  0.268239  0.446553  \n",
       "\n",
       "[5 rows x 13670 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"corr_diff.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2.to_csv(\"p_val.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  df3.to_csv(\"nathaniel_list_permutation_pval01.csv\",index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
