{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import pandas as pd\n",
    "import cptac.utils as ut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that hnscc index is up-to-date...\r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (<ipython-input-2-5f7029bc4773>, line 2)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "luad = cptac.Luad()\n",
    "lscc = cptac.Lscc()\n",
    "hnscc = cptac.Hnscc()\n",
    "en = cptac.Endometrial()\n",
    "ccrcc = cptac.Ccrcc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_list = [luad, lscc, hnscc, en, ccrcc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancer_dict = {lscc: 'lscc', luad: 'luad', hnscc: 'hnscc', en: 'en', ccrcc: 'ccrcc'}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Mutation_Frequency</th>\n",
       "      <th>cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.536364</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>871</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>0.345455</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1020</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.336364</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.318182</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>153</th>\n",
       "      <td>LRP1B</td>\n",
       "      <td>0.309091</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1010</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1221</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>USH2A</td>\n",
       "      <td>0.254545</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>281</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>0.236364</td>\n",
       "      <td>luad</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.953704</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>386</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.685185</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>632</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>0.472222</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>835</th>\n",
       "      <td>USH2A</td>\n",
       "      <td>0.398148</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>842</th>\n",
       "      <td>ZFHX4</td>\n",
       "      <td>0.370370</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.370370</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>327</th>\n",
       "      <td>LRP1B</td>\n",
       "      <td>0.314815</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>SYNE1</td>\n",
       "      <td>0.305556</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>332</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.305556</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>294</th>\n",
       "      <td>FAM135B</td>\n",
       "      <td>0.296296</td>\n",
       "      <td>lscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.880734</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.403670</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>343</th>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>0.266055</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1828</th>\n",
       "      <td>FAT1</td>\n",
       "      <td>0.238532</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>0.238532</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>0.238532</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DNAH5</td>\n",
       "      <td>0.192661</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1346</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>892</th>\n",
       "      <td>RYR2</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>726</th>\n",
       "      <td>CTNNA2</td>\n",
       "      <td>0.174312</td>\n",
       "      <td>hnscc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>0.789474</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>PIK3CA</td>\n",
       "      <td>0.494737</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>0.452632</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1117</th>\n",
       "      <td>PIK3R1</td>\n",
       "      <td>0.389474</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>883</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>0.326316</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1181</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>0.305263</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160</th>\n",
       "      <td>CTCF</td>\n",
       "      <td>0.284211</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>353</th>\n",
       "      <td>KMT2B</td>\n",
       "      <td>0.242105</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>TP53</td>\n",
       "      <td>0.221053</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1668</th>\n",
       "      <td>ZFHX3</td>\n",
       "      <td>0.221053</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>VHL</td>\n",
       "      <td>0.745455</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>PBRM1</td>\n",
       "      <td>0.400000</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>KDM5C</td>\n",
       "      <td>0.172727</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>BAP1</td>\n",
       "      <td>0.154545</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>SETD2</td>\n",
       "      <td>0.136364</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>805</th>\n",
       "      <td>TTN</td>\n",
       "      <td>0.118182</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>565</th>\n",
       "      <td>CSMD3</td>\n",
       "      <td>0.090909</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>USH2A</td>\n",
       "      <td>0.081818</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>409</th>\n",
       "      <td>MUC16</td>\n",
       "      <td>0.072727</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>BIRC6</td>\n",
       "      <td>0.063636</td>\n",
       "      <td>ccrcc</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  Mutation_Frequency cancer\n",
       "263      TP53            0.536364   luad\n",
       "871      EGFR            0.345455   luad\n",
       "1020    MUC16            0.336364   luad\n",
       "225      RYR2            0.318182   luad\n",
       "40        TTN            0.318182   luad\n",
       "153     LRP1B            0.309091   luad\n",
       "1010     KRAS            0.300000   luad\n",
       "1221    CSMD3            0.300000   luad\n",
       "273     USH2A            0.254545   luad\n",
       "281     ZFHX4            0.236364   luad\n",
       "237      TP53            0.953704   lscc\n",
       "386       TTN            0.685185   lscc\n",
       "632     CSMD3            0.472222   lscc\n",
       "835     USH2A            0.398148   lscc\n",
       "842     ZFHX4            0.370370   lscc\n",
       "204      RYR2            0.370370   lscc\n",
       "327     LRP1B            0.314815   lscc\n",
       "971     SYNE1            0.305556   lscc\n",
       "332     MUC16            0.305556   lscc\n",
       "294   FAM135B            0.296296   lscc\n",
       "73       TP53            0.880734  hnscc\n",
       "423       TTN            0.403670  hnscc\n",
       "343    CDKN2A            0.266055  hnscc\n",
       "1828     FAT1            0.238532  hnscc\n",
       "293    NOTCH1            0.238532  hnscc\n",
       "13      CSMD3            0.238532  hnscc\n",
       "14      DNAH5            0.192661  hnscc\n",
       "1346    MUC16            0.174312  hnscc\n",
       "892      RYR2            0.174312  hnscc\n",
       "726    CTNNA2            0.174312  hnscc\n",
       "532      PTEN            0.789474     en\n",
       "497    PIK3CA            0.494737     en\n",
       "54     ARID1A            0.452632     en\n",
       "1117   PIK3R1            0.389474     en\n",
       "883      KRAS            0.326316     en\n",
       "1181   CTNNB1            0.305263     en\n",
       "160      CTCF            0.284211     en\n",
       "353     KMT2B            0.242105     en\n",
       "676      TP53            0.221053     en\n",
       "1668    ZFHX3            0.221053     en\n",
       "70        VHL            0.745455  ccrcc\n",
       "49      PBRM1            0.400000  ccrcc\n",
       "80      KDM5C            0.172727  ccrcc\n",
       "93       BAP1            0.154545  ccrcc\n",
       "56      SETD2            0.136364  ccrcc\n",
       "805       TTN            0.118182  ccrcc\n",
       "565     CSMD3            0.090909  ccrcc\n",
       "718     USH2A            0.081818  ccrcc\n",
       "409     MUC16            0.072727  ccrcc\n",
       "976     BIRC6            0.063636  ccrcc"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutation_frequency_dfs = []\n",
    "for cancer in cancer_list:\n",
    "    mutation_df = cancer.get_somatic_mutation()\n",
    "    mutation_df = mutation_df[mutation_df.Mutation != 'Silent']\n",
    "    mutation_df = mutation_df[mutation_df.Mutation != 'RNA']\n",
    "    gene_df = cancer.get_transcriptomics('tumor')\n",
    "    if isinstance(gene_df.columns, pd.MultiIndex):\n",
    "        gene_df = gene_df.droplevel('Database_ID', axis = 1)\n",
    "    mutation_frequency = []\n",
    "    genes = list(pd.unique(mutation_df.Gene))\n",
    "    for gene in genes:\n",
    "        num_mutated = len(pd.unique(mutation_df[mutation_df.Gene == gene].index))\n",
    "        total = len(pd.unique(gene_df.index))\n",
    "        frequency = num_mutated / total \n",
    "        mutation_frequency.append(frequency)\n",
    "    df = pd.DataFrame({'Gene': genes, 'Mutation_Frequency': mutation_frequency})\n",
    "    df = df.sort_values(by= 'Mutation_Frequency', ascending = False)\n",
    "    if len(df) > 10:\n",
    "        df = df[0:10]\n",
    "    df['cancer'] = [cancer_dict[cancer]] * len(df)\n",
    "    mutation_frequency_dfs.append(df)\n",
    "mutation_frequency_df = pd.concat(mutation_frequency_dfs)\n",
    "mutation_frequency_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# write bash command for finding permutation p-val for each gene/cancer pair\n",
    "with open('transmutation_effects_permutations.sh', 'w') as file:\n",
    "    for index, row in mutation_frequency_df.iterrows():\n",
    "        s = 'python supercomputer_transmutation_effects.py ' + row.cancer + ' ' + row.Gene + ' 10000\\n'\n",
    "        file.write(s)\n",
    "    file.close()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
