{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import cptac\n",
    "import scipy\n",
    "from scipy import stats\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import math\n",
    "import pandas as pd\n",
    "import statistics\n",
    "import parse_correlations_dataframe as get_corr\n",
    "import copy\n",
    "import get_correlations\n",
    "import cptac.utils as ut"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    }
   ],
   "source": [
    "brca = cptac.Brca()\n",
    "ccrcc = cptac.Ccrcc()\n",
    "colon = cptac.Colon()\n",
    "en = cptac.Endometrial()\n",
    "gbm = cptac.Gbm()\n",
    "luad = cptac.Luad()\n",
    "ovarian = cptac.Ovarian()\n",
    "hnscc  = cptac.Hnscc()\n",
    "lscc = cptac.Lscc()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers = [brca, ccrcc, colon, en, gbm, luad, ovarian, hnscc, lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "cancers2 = [ccrcc, en, gbm, luad, hnscc, lscc]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "hi = set()\n",
    "hi.add(\"yo\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'yo'}"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_gene_list(tissues):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    whole_gene_list = set()\n",
    "    for tissue in tissues:\n",
    "        tissue_list = get_gene_list(tissue)\n",
    "        for gene in tissue_list:\n",
    "            whole_gene_list.add(gene)\n",
    "    return sorted(list(whole_gene_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_all_gene_list_old(tissues):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    whole_gene_list = set()\n",
    "    for tissue in tissues:\n",
    "        gene_list = []\n",
    "        prot = tissue.get_proteomics()\n",
    "        if isinstance(prot.columns, pd.MultiIndex):\n",
    "            prot = ut.reduce_multiindex(df = prot, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "        trans = tissue.get_transcriptomics()\n",
    "        if isinstance(trans.columns, pd.MultiIndex):\n",
    "            trans = ut.reduce_multiindex(df = trans, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "        for gene in prot:\n",
    "            whole_gene_list.add(gene)\n",
    "        for gene in trans:\n",
    "            whole_gene_list.add(gene)\n",
    "    return sorted(list(whole_gene_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_list(tissue):\n",
    "    #Returns a list of the genes that are in both proteomics and transcriptomics\n",
    "    gene_list = []\n",
    "    prot = tissue.get_proteomics()\n",
    "    if isinstance(prot.columns, pd.MultiIndex):\n",
    "        prot = prot.columns.get_level_values(0)\n",
    "    trans = tissue.get_transcriptomics()\n",
    "    if isinstance(trans.columns, pd.MultiIndex):\n",
    "        trans = trans.columns.get_level_values(0)\n",
    "    for i in prot:\n",
    "        if i in trans and i not in gene_list:\n",
    "            gene_list.append(i)\n",
    "    return gene_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_list = get_all_gene_list(cancers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "type_dict = {brca:\"brca\",ccrcc:\"ccrcc\",colon:\"colon\",en:\"endometrial\",gbm:\"gbm\",luad:\"luad\",\n",
    "                  ovarian:\"ovarian\",hnscc:\"hnscc\",lscc:\"lscc\"}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "# https://link.springer.com/article/10.3758/s13428-012-0289-7\n",
    "def compare_correlations(r1, r2, n1, n2):\n",
    "    rp1 = np.arctanh(r1)\n",
    "    rp2 = np.arctanh(r2)\n",
    "    \n",
    "    if n1 < 4 or n2 < 4:\n",
    "        return(np.nan)\n",
    "    Sr12 = math.sqrt((1/(n1-3))+(1/(n2-3)))\n",
    "    z = (rp1-rp2) / Sr12\n",
    "    p = scipy.stats.norm.sf(abs(z))*2\n",
    "    return (p)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def find_mut_tumor(cancer_type, gene):\n",
    "    try:\n",
    "        gene_multi = cancer_type.multi_join({'proteomics': gene, 'transcriptomics': gene, 'somatic_mutation': gene}, tissue_type = 'tumor', flatten = True)\n",
    "        normal = cancer_type.multi_join({'proteomics': gene, 'transcriptomics': gene}, tissue_type= 'normal', flatten = True)\n",
    "        normal.columns = ['proteomics', 'transcriptomics']\n",
    "        gene_multi.columns = ['proteomics', 'transcriptomics', 'mutation', 'location', 'mutation_status']\n",
    "        trans = list(gene_multi[\"transcriptomics\"])\n",
    "        prot = list(gene_multi['proteomics'])\n",
    "        group = []\n",
    "        for i in gene_multi['mutation_status']:\n",
    "            if type(i) == str:\n",
    "                group.append(\"mutation\")\n",
    "            else:\n",
    "                group.append(\"wt\") \n",
    "        if group.count('wt') < 2 or len(normal) < 2:\n",
    "            return float(\"NaN\")\n",
    "        group.extend(['normal']*len(normal))\n",
    "        prot.extend(list(normal['proteomics']))\n",
    "        trans.extend(list(normal['transcriptomics']))\n",
    "        gene_df = pd.DataFrame({'Type': group, 'Proteomics': prot, 'Transcriptomics': trans})\n",
    "        gene_df = gene_df.dropna()\n",
    "        return gene_df\n",
    "    except:\n",
    "        return float(\"NaN\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_df_with_type(df,input_type):\n",
    "    return(df.loc[df['Type']==input_type])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "def permute(df,original_correlation, label_1, label_2, column_one, column_two, permutation_times):\n",
    "    permutation_list = []\n",
    "    permu_df = copy.deepcopy(df)\n",
    "\n",
    "    for i in range(permutation_times):\n",
    "        permu_df[\"Type\"] = np.random.permutation(permu_df[\"Type\"])\n",
    "        permu_is_label_1 = permu_df[\"Type\"] == label_1\n",
    "        permu_is_label_2 = permu_df[\"Type\"] == label_2\n",
    "        label_1_correlation,label_1_pval = scipy.stats.pearsonr(permu_df[permu_is_label_1][column_one], permu_df[permu_is_label_1][column_two])\n",
    "        label_2_correlation,label_2_pval = scipy.stats.pearsonr(permu_df[permu_is_label_2][column_one], permu_df[permu_is_label_2][column_two])\n",
    "        delta = label_1_correlation - label_2_correlation\n",
    "        permutation_list.append(delta)\n",
    "        \n",
    "    z_score = (original_correlation - np.mean(permutation_list)) / np.std(permutation_list)\n",
    "    p_val = scipy.stats.norm.sf(abs(z_score))*2\n",
    "    return p_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "from mlxtend.evaluate import permutation_test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 227,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "x and y must have length at least 2.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-227-58322045fb9e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     80\u001b[0m         \u001b[0mcolumn_one\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mperm_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     81\u001b[0m         \u001b[0mcolumn_two\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mperm_df\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 82\u001b[0;31m         \u001b[0mperm_val\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpermute\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mperm_df\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcorr_diff\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"tumor\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"normal\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcolumn_one\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     83\u001b[0m         \u001b[0mcancer_perm_list\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mperm_val\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     84\u001b[0m \u001b[0;31m#         cancer_perm_list.append(np.nan)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-16-d8ddddf354f8>\u001b[0m in \u001b[0;36mpermute\u001b[0;34m(df, original_correlation, label_1, label_2, column_one, column_two, permutation_times)\u001b[0m\n\u001b[1;32m      8\u001b[0m         \u001b[0mpermu_is_label_2\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Type\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mlabel_2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m         \u001b[0mlabel_1_correlation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlabel_1_pval\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpearsonr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_one\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m         \u001b[0mlabel_2_correlation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlabel_2_pval\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpearsonr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_one\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m         \u001b[0mdelta\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlabel_1_correlation\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mlabel_2_correlation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m         \u001b[0mpermutation_list\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdelta\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/scipy/stats/stats.py\u001b[0m in \u001b[0;36mpearsonr\u001b[0;34m(x, y)\u001b[0m\n\u001b[1;32m   3499\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3500\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mn\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m2\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3501\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'x and y must have length at least 2.'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3502\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3503\u001b[0m     \u001b[0mx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: x and y must have length at least 2."
     ]
    }
   ],
   "source": [
    "# first_pass = dict()\n",
    "tot_diff_list = []\n",
    "tot_pval_list = []\n",
    "tot_perm_list = []\n",
    "for cancer in cancers2:\n",
    "    cancer_diff_list = [type_dict[cancer]]\n",
    "    cancer_pval_list = [type_dict[cancer]]\n",
    "    cancer_perm_list = [type_dict[cancer]]\n",
    "    tumor_cancer_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"tumor\",quiet=True)\n",
    "    if isinstance(tumor_cancer_df.columns, pd.MultiIndex):\n",
    "        tumor_cancer_df = ut.reduce_multiindex(df = tumor_cancer_df, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "            \n",
    "    normal_cancer_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"normal\",quiet=True)\n",
    "    if isinstance(normal_cancer_df.columns, pd.MultiIndex):\n",
    "        normal_cancer_df = ut.reduce_multiindex(df = normal_cancer_df, levels_to_drop=\"Database_ID\",quiet=True)\n",
    "    \n",
    "    for gene in gene_list:\n",
    "        gene_trans = gene + \"_transcriptomics\"\n",
    "        gene_prot = gene + \"_proteomics\"\n",
    "        gene_in_tumor = gene_trans in tumor_cancer_df.columns and gene_prot in tumor_cancer_df.columns\n",
    "        gene_in_normal = gene_trans in normal_cancer_df.columns and gene_prot in normal_cancer_df.columns\n",
    "        \n",
    "        if not(gene_in_tumor and gene_in_normal):\n",
    "            cancer_diff_list.append(np.nan)\n",
    "            cancer_pval_list.append(np.nan)\n",
    "            cancer_perm_list.append(np.nan)\n",
    "            continue\n",
    "\n",
    "        tumor_df = tumor_cancer_df[[gene_trans,gene_prot]]\n",
    "#         print(tumor_df[gene_trans])\n",
    "#         print(tumor_df[gene_prot])\n",
    "        if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame) or isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "            trans_col = tumor_df[gene_trans]\n",
    "            if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame):\n",
    "                trans_col = trans_col.iloc[:,0]\n",
    "            prot_col = tumor_df[gene_prot]\n",
    "            if isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame):\n",
    "                prot_col = prot_col.iloc[:,0]\n",
    "            frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "            tumor_df = pd.DataFrame(frame)\n",
    "        num_tumor = len(tumor_df)\n",
    "        tumor_df = tumor_df.dropna()\n",
    "        tumor_corr = tumor_df.corr().iloc[0][1]\n",
    "        \n",
    "#         print(normal_df[gene_trans])\n",
    "#         print(normal_df[gene_prot])\n",
    "        normal_df = normal_cancer_df[[gene_trans,gene_prot]]\n",
    "        if isinstance(normal_df[gene_trans], pd.core.frame.DataFrame) or isinstance(normal_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "            trans_col = normal_df[gene_trans]\n",
    "            if isinstance(normal_df[gene_trans], pd.core.frame.DataFrame):\n",
    "                trans_col = trans_col.iloc[:,0]\n",
    "            prot_col = normal_df[gene_prot]\n",
    "            if isinstance(normal_df[gene_prot], pd.core.frame.DataFrame):\n",
    "                prot_col = prot_col.iloc[:,0]\n",
    "            frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "            normal_df = pd.DataFrame(frame)\n",
    "        num_normal = len(normal_df)\n",
    "        normal_df = normal_df.dropna()\n",
    "        normal_corr = normal_df.corr().iloc[0][1]\n",
    "        \n",
    "        corr_diff = tumor_corr - normal_corr\n",
    "        cancer_diff_list.append(corr_diff)\n",
    "        \n",
    "        gene_pval = compare_correlations(tumor_corr, normal_corr, num_tumor, num_normal)\n",
    "        cancer_pval_list.append(gene_pval)\n",
    "        \n",
    "        #Here we do permutations. Cut down number?\n",
    "        if num_tumor < 4 or num_normal < 4 or gene_pval > .01:\n",
    "            cancer_perm_list.append(np.nan)\n",
    "            continue\n",
    "        tumor_label_list = ['tumor'] * len(tumor_df)\n",
    "        tumor_df[\"Type\"] = tumor_label_list\n",
    "        \n",
    "        normal_label_list = ['normal'] * len(normal_df)\n",
    "        normal_df[\"Type\"] = normal_label_list\n",
    "        \n",
    "        perm_list = [tumor_df,normal_df]\n",
    "        perm_df = pd.concat(perm_list)\n",
    "        \n",
    "        column_one = perm_df.columns[0]\n",
    "        column_two = perm_df.columns[1]\n",
    "        perm_val = permute(perm_df,corr_diff,\"tumor\",\"normal\",column_one,column_two,1000)\n",
    "        cancer_perm_list.append(perm_val)\n",
    "#         cancer_perm_list.append(np.nan)\n",
    "#17 minutes and 10 seconds for 100 genes permutation, p val < .05 and 1000 permutations per\n",
    "        \n",
    "    tot_diff_list.append(cancer_diff_list)\n",
    "    tot_pval_list.append(cancer_pval_list)\n",
    "    tot_perm_list.append(cancer_perm_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [],
   "source": [
    "tumor_df = tumor_cancer_df[[gene_trans,gene_prot]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ABI1_transcriptomics</th>\n",
       "      <th>ABI1_proteomics</th>\n",
       "      <th>ABI1_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>26.692315</td>\n",
       "      <td>-0.130282</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>24.618292</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>47.983112</td>\n",
       "      <td>0.200461</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>21.324011</td>\n",
       "      <td>0.582155</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>25.178665</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646</th>\n",
       "      <td>23.429352</td>\n",
       "      <td>0.046889</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648</th>\n",
       "      <td>39.167637</td>\n",
       "      <td>0.094193</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649</th>\n",
       "      <td>23.722191</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651</th>\n",
       "      <td>23.938233</td>\n",
       "      <td>0.118175</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808</th>\n",
       "      <td>18.867214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        ABI1_transcriptomics  ABI1_proteomics  ABI1_proteomics\n",
       "Patient_ID                                                        \n",
       "C3L-00004              26.692315        -0.130282              NaN\n",
       "C3L-00010              24.618292              NaN              NaN\n",
       "C3L-00011              47.983112         0.200461              NaN\n",
       "C3L-00026              21.324011         0.582155              NaN\n",
       "C3L-00079              25.178665              NaN              NaN\n",
       "...                          ...              ...              ...\n",
       "C3N-01646              23.429352         0.046889              NaN\n",
       "C3N-01648              39.167637         0.094193              NaN\n",
       "C3N-01649              23.722191              NaN              NaN\n",
       "C3N-01651              23.938233         0.118175              NaN\n",
       "C3N-01808              18.867214              NaN              NaN\n",
       "\n",
       "[110 rows x 3 columns]"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hi\n"
     ]
    }
   ],
   "source": [
    "if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame) or isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "    print(\"hi\")\n",
    "    trans_col = tumor_df[gene_trans]\n",
    "    if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame):\n",
    "        trans_col = trans_col.iloc[:,0]\n",
    "    prot_col = tumor_df[gene_prot]\n",
    "    if isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame):\n",
    "        prot_col = prot_col.iloc[:,0]\n",
    "    frame = {gene_trans : trans_col, gene_prot : prot_col}\n",
    "    tumor_df = pd.DataFrame(frame)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ABI1_transcriptomics</th>\n",
       "      <th>ABI1_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>26.692315</td>\n",
       "      <td>-0.130282</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>24.618292</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>47.983112</td>\n",
       "      <td>0.200461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>21.324011</td>\n",
       "      <td>0.582155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>25.178665</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646</th>\n",
       "      <td>23.429352</td>\n",
       "      <td>0.046889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648</th>\n",
       "      <td>39.167637</td>\n",
       "      <td>0.094193</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649</th>\n",
       "      <td>23.722191</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651</th>\n",
       "      <td>23.938233</td>\n",
       "      <td>0.118175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808</th>\n",
       "      <td>18.867214</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            ABI1_transcriptomics  ABI1_proteomics\n",
       "Patient_ID                                       \n",
       "C3L-00004              26.692315        -0.130282\n",
       "C3L-00010              24.618292              NaN\n",
       "C3L-00011              47.983112         0.200461\n",
       "C3L-00026              21.324011         0.582155\n",
       "C3L-00079              25.178665              NaN\n",
       "...                          ...              ...\n",
       "C3N-01646              23.429352         0.046889\n",
       "C3N-01648              39.167637         0.094193\n",
       "C3N-01649              23.722191              NaN\n",
       "C3N-01651              23.938233         0.118175\n",
       "C3N-01808              18.867214              NaN\n",
       "\n",
       "[110 rows x 2 columns]"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [],
   "source": [
    "if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame) or isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame):\n",
    "    print(\"hi\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ABHD6_transcriptomics</th>\n",
       "      <th>ABHD6_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004.N</th>\n",
       "      <td>20.690153</td>\n",
       "      <td>0.041242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010.N</th>\n",
       "      <td>20.026105</td>\n",
       "      <td>0.177484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011.N</th>\n",
       "      <td>28.403306</td>\n",
       "      <td>0.442436</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026.N</th>\n",
       "      <td>25.670387</td>\n",
       "      <td>0.043691</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079.N</th>\n",
       "      <td>17.425588</td>\n",
       "      <td>-0.395671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646.N</th>\n",
       "      <td>30.047970</td>\n",
       "      <td>0.522017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648.N</th>\n",
       "      <td>19.367721</td>\n",
       "      <td>0.402142</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649.N</th>\n",
       "      <td>23.215246</td>\n",
       "      <td>0.602206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651.N</th>\n",
       "      <td>22.201853</td>\n",
       "      <td>0.501867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808.N</th>\n",
       "      <td>37.099596</td>\n",
       "      <td>0.755867</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>75 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name         ABHD6_transcriptomics  ABHD6_proteomics\n",
       "Patient_ID                                          \n",
       "C3L-00004.N              20.690153          0.041242\n",
       "C3L-00010.N              20.026105          0.177484\n",
       "C3L-00011.N              28.403306          0.442436\n",
       "C3L-00026.N              25.670387          0.043691\n",
       "C3L-00079.N              17.425588         -0.395671\n",
       "...                            ...               ...\n",
       "C3N-01646.N              30.047970          0.522017\n",
       "C3N-01648.N              19.367721          0.402142\n",
       "C3N-01649.N              23.215246          0.602206\n",
       "C3N-01651.N              22.201853          0.501867\n",
       "C3N-01808.N              37.099596          0.755867\n",
       "\n",
       "[75 rows x 2 columns]"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'A1BG_proteomics'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2645\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2646\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2647\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'A1BG_proteomics'",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-183-762a3b5cac1d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mnormal_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mgene_prot\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   2798\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2799\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2800\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2801\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2802\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   2646\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2647\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 2648\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_maybe_cast_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   2649\u001b[0m         \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtolerance\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtolerance\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   2650\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndim\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msize\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'A1BG_proteomics'"
     ]
    }
   ],
   "source": [
    "normal_df[gene_prot]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Patient_ID\n",
       "ABI1_proteomics    Empty DataFrame\n",
       "Columns: [ABI1_proteomics]\n",
       "Ind...\n",
       "Name: ABI1_transcriptomics, dtype: object"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKAP17A_transcriptomics</th>\n",
       "      <th>AKAP17A_transcriptomics</th>\n",
       "      <th>AKAP17A_proteomics</th>\n",
       "      <th>Type</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>224925.013065</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.544187</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>251674.374330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.203106</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>205830.598754</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020474</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>268763.653430</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.139892</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>180929.603827</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.107471</td>\n",
       "      <td>tumor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PT-RN5K.N</th>\n",
       "      <td>121224.465429</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.727790</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PT-RU72.N</th>\n",
       "      <td>115447.303360</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.708098</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PT-UTHO.N</th>\n",
       "      <td>162974.757645</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.349061</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PT-WVLH.N</th>\n",
       "      <td>117977.283349</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.602661</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PT-Y8DK.N</th>\n",
       "      <td>106738.735377</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.644847</td>\n",
       "      <td>normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>108 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKAP17A_transcriptomics  AKAP17A_transcriptomics  \\\n",
       "Patient_ID                                                     \n",
       "C3L-00104             224925.013065                      0.0   \n",
       "C3L-00365             251674.374330                      0.0   \n",
       "C3L-00674             205830.598754                      0.0   \n",
       "C3L-00677             268763.653430                      0.0   \n",
       "C3L-01040             180929.603827                      0.0   \n",
       "...                             ...                      ...   \n",
       "PT-RN5K.N             121224.465429                      0.0   \n",
       "PT-RU72.N             115447.303360                      0.0   \n",
       "PT-UTHO.N             162974.757645                      0.0   \n",
       "PT-WVLH.N             117977.283349                      0.0   \n",
       "PT-Y8DK.N             106738.735377                      0.0   \n",
       "\n",
       "Name        AKAP17A_proteomics    Type  \n",
       "Patient_ID                              \n",
       "C3L-00104             0.544187   tumor  \n",
       "C3L-00365             0.203106   tumor  \n",
       "C3L-00674             0.020474   tumor  \n",
       "C3L-00677             0.139892   tumor  \n",
       "C3L-01040             0.107471   tumor  \n",
       "...                        ...     ...  \n",
       "PT-RN5K.N            -0.727790  normal  \n",
       "PT-RU72.N            -0.708098  normal  \n",
       "PT-UTHO.N            -0.349061  normal  \n",
       "PT-WVLH.N            -0.602661  normal  \n",
       "PT-Y8DK.N            -0.644847  normal  \n",
       "\n",
       "[108 rows x 4 columns]"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "perm_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name         ACLY_transcriptomics  ACLY_proteomics  ACLY_proteomics    Type\n",
      "Patient_ID                                                                 \n",
      "C3L-00004              127.431921         0.640347         1.058305   tumor\n",
      "C3L-00010              157.060848         0.887528         1.331261   tumor\n",
      "C3L-00011               80.023745         0.349048        -0.115492   tumor\n",
      "C3L-00026              158.636481         0.907927         0.459999   tumor\n",
      "C3L-00079               90.420426         0.064131         0.073501   tumor\n",
      "...                           ...              ...              ...     ...\n",
      "C3N-01646.N             43.665836        -1.010941        -0.607228  normal\n",
      "C3N-01648.N             48.895973        -1.718317        -0.776481  normal\n",
      "C3N-01649.N             47.823339        -1.359644        -0.846564  normal\n",
      "C3N-01651.N             65.531812        -1.144428        -0.500871  normal\n",
      "C3N-01808.N             64.300822        -0.980743        -0.606716  normal\n",
      "\n",
      "[185 rows x 4 columns]\n",
      "tumor\n",
      "normal\n",
      "ACLY_transcriptomics\n",
      "ACLY_proteomics\n",
      "Here:\n",
      "Name         ACLY_proteomics  ACLY_proteomics\n",
      "Patient_ID                                   \n",
      "C3L-00004           0.640347         1.058305\n",
      "C3L-00011           0.349048        -0.115492\n",
      "C3L-00026           0.907927         0.459999\n",
      "C3L-00079           0.064131         0.073501\n",
      "C3L-00088           1.232395         1.178499\n",
      "...                      ...              ...\n",
      "C3N-01261.N        -1.306552        -0.940408\n",
      "C3N-01361.N        -0.822686        -0.125430\n",
      "C3N-01524.N        -0.864498        -0.714217\n",
      "C3N-01646.N        -1.010941        -0.607228\n",
      "C3N-01649.N        -1.359644        -0.846564\n",
      "\n",
      "[110 rows x 2 columns]\n"
     ]
    },
    {
     "ename": "TypeError",
     "evalue": "No loop matching the specified signature and casting was found for ufunc add",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-113-bf4985d2846f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpermute2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mperm_df\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcorr_diff\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"tumor\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"normal\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcolumn_one\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m1000\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-112-fe373f988780>\u001b[0m in \u001b[0;36mpermute2\u001b[0;34m(df, original_correlation, label_1, label_2, column_one, column_two, permutation_times)\u001b[0m\n\u001b[1;32m     13\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Here:\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     14\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 15\u001b[0;31m         \u001b[0mlabel_1_correlation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlabel_1_pval\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpearsonr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_one\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     16\u001b[0m         \u001b[0mlabel_2_correlation\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mlabel_2_pval\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mscipy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstats\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpearsonr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_one\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpermu_df\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mpermu_is_label_2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcolumn_two\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     17\u001b[0m         \u001b[0mdelta\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlabel_1_correlation\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mlabel_2_correlation\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/scipy/stats/stats.py\u001b[0m in \u001b[0;36mpearsonr\u001b[0;34m(x, y)\u001b[0m\n\u001b[1;32m   3517\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msign\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msign\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0my\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1.0\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3518\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3519\u001b[0;31m     \u001b[0mxmean\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3520\u001b[0m     \u001b[0mymean\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmean\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3521\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/envs/cptac_dev/lib/python3.7/site-packages/numpy/core/_methods.py\u001b[0m in \u001b[0;36m_mean\u001b[0;34m(a, axis, dtype, out, keepdims)\u001b[0m\n\u001b[1;32m    149\u001b[0m             \u001b[0mis_float16_result\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mTrue\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    150\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 151\u001b[0;31m     \u001b[0mret\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mumr_sum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0marr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkeepdims\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    152\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mret\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmu\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mndarray\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    153\u001b[0m         ret = um.true_divide(\n",
      "\u001b[0;31mTypeError\u001b[0m: No loop matching the specified signature and casting was found for ufunc add"
     ]
    }
   ],
   "source": [
    "permute2(perm_df,corr_diff,\"tumor\",\"normal\",column_one,column_two,1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "hi = ut.reduce_multiindex(df = ccrcc.get_proteomics(), levels_to_drop=\"Database_ID\",quiet=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\"ACLY_proteomics\" in hi.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name        ACLY_proteomics  ACLY_proteomics\n",
      "Patient_ID                                  \n",
      "C3L-00004          0.640347         1.058305\n",
      "C3L-00010          0.887528         1.331261\n",
      "C3L-00011          0.349048        -0.115492\n",
      "C3L-00026          0.907927         0.459999\n",
      "C3L-00079          0.064131         0.073501\n",
      "...                     ...              ...\n",
      "C3N-01646          0.525380         0.629745\n",
      "C3N-01648         -0.089212        -0.351126\n",
      "C3N-01649          0.034884        -0.285154\n",
      "C3N-01651          0.442074         0.025897\n",
      "C3N-01808          0.625828         0.697103\n",
      "\n",
      "[110 rows x 2 columns]\n"
     ]
    }
   ],
   "source": [
    "print(tumor_df[\"ACLY_proteomics\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "isinstance(tumor_df[\"ACLY_transcriptomics\"], pd.core.frame.DataFrame)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Here we are only currently taking the first column of multi_indices\n",
    "dftry = tumor_df.iloc[:, 0:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>ACLY_transcriptomics</th>\n",
       "      <th>ACLY_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00004</th>\n",
       "      <td>127.431921</td>\n",
       "      <td>0.640347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00010</th>\n",
       "      <td>157.060848</td>\n",
       "      <td>0.887528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00011</th>\n",
       "      <td>80.023745</td>\n",
       "      <td>0.349048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00026</th>\n",
       "      <td>158.636481</td>\n",
       "      <td>0.907927</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00079</th>\n",
       "      <td>90.420426</td>\n",
       "      <td>0.064131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01646</th>\n",
       "      <td>159.803693</td>\n",
       "      <td>0.525380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01648</th>\n",
       "      <td>25.479118</td>\n",
       "      <td>-0.089212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01649</th>\n",
       "      <td>116.451276</td>\n",
       "      <td>0.034884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01651</th>\n",
       "      <td>87.199126</td>\n",
       "      <td>0.442074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-01808</th>\n",
       "      <td>156.924589</td>\n",
       "      <td>0.625828</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>110 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        ACLY_transcriptomics  ACLY_proteomics\n",
       "Patient_ID                                       \n",
       "C3L-00004             127.431921         0.640347\n",
       "C3L-00010             157.060848         0.887528\n",
       "C3L-00011              80.023745         0.349048\n",
       "C3L-00026             158.636481         0.907927\n",
       "C3L-00079              90.420426         0.064131\n",
       "...                          ...              ...\n",
       "C3N-01646             159.803693         0.525380\n",
       "C3N-01648              25.479118        -0.089212\n",
       "C3N-01649             116.451276         0.034884\n",
       "C3N-01651              87.199126         0.442074\n",
       "C3N-01808             156.924589         0.625828\n",
       "\n",
       "[110 rows x 2 columns]"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dftry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "def permute2(df,original_correlation, label_1, label_2, column_one, column_two, permutation_times):\n",
    "    permutation_list = []\n",
    "    permu_df = copy.deepcopy(df)\n",
    "    print(permu_df)\n",
    "    print(label_1)\n",
    "    print(label_2)\n",
    "    print(column_one)\n",
    "    print(column_two)\n",
    "    for i in range(permutation_times):\n",
    "        permu_df[\"Type\"] = np.random.permutation(permu_df[\"Type\"])\n",
    "        permu_is_label_1 = permu_df[\"Type\"] == label_1\n",
    "        permu_is_label_2 = permu_df[\"Type\"] == label_2\n",
    "        print(\"Here:\")\n",
    "        print(permu_df[permu_is_label_1][column_two])\n",
    "        label_1_correlation,label_1_pval = scipy.stats.pearsonr(permu_df[permu_is_label_1][column_one], permu_df[permu_is_label_1][column_two])\n",
    "        label_2_correlation,label_2_pval = scipy.stats.pearsonr(permu_df[permu_is_label_2][column_one], permu_df[permu_is_label_2][column_two])\n",
    "        delta = label_1_correlation - label_2_correlation\n",
    "        permutation_list.append(delta)\n",
    "        print(\"done\")\n",
    "        \n",
    "    z_score = (original_correlation - np.mean(permutation_list)) / np.std(permutation_list)\n",
    "    p_val = scipy.stats.norm.sf(abs(z_score))*2\n",
    "    return p_val"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = [\"Cancer\"] \n",
    "labels.extend(gene_list)\n",
    "df = pd.DataFrame.from_records(tot_diff_list,columns=labels)\n",
    "df2 = pd.DataFrame.from_records(tot_pval_list,columns=labels)\n",
    "df3 = pd.DataFrame.from_records(tot_perm_list,columns=labels)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gbm</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>luad</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>lscc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 14637 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Cancer  A1BG  A1CF  A2M  A2ML1  A4GALT  AAAS  AACS  AADAC  AADAT  ...  \\\n",
       "0        ccrcc   NaN   NaN  NaN    NaN     NaN   NaN   NaN    NaN    NaN  ...   \n",
       "1  endometrial   NaN   NaN  NaN    NaN     NaN   NaN   NaN    NaN    NaN  ...   \n",
       "2          gbm   NaN   NaN  NaN    NaN     NaN   NaN   NaN    NaN    NaN  ...   \n",
       "3         luad   NaN   NaN  NaN    NaN     NaN   NaN   NaN    NaN    NaN  ...   \n",
       "4        hnscc   NaN   NaN  NaN    NaN     NaN   NaN   NaN    NaN    NaN  ...   \n",
       "5         lscc   NaN   NaN  NaN    NaN     NaN   NaN   NaN    NaN    NaN  ...   \n",
       "\n",
       "   ZSWIM9  ZW10  ZWILCH  ZWINT  ZXDA  ZXDC  ZYG11B  ZYX  ZZEF1  ZZZ3  \n",
       "0     NaN   NaN     NaN    NaN   NaN   NaN     NaN  NaN    NaN   NaN  \n",
       "1     NaN   NaN     NaN    NaN   NaN   NaN     NaN  NaN    NaN   NaN  \n",
       "2     NaN   NaN     NaN    NaN   NaN   NaN     NaN  NaN    NaN   NaN  \n",
       "3     NaN   NaN     NaN    NaN   NaN   NaN     NaN  NaN    NaN   NaN  \n",
       "4     NaN   NaN     NaN    NaN   NaN   NaN     NaN  NaN    NaN   NaN  \n",
       "5     NaN   NaN     NaN    NaN   NaN   NaN     NaN  NaN    NaN   NaN  \n",
       "\n",
       "[6 rows x 14637 columns]"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df.to_csv(\"corr_diff.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [],
   "source": [
    "# df2.to_csv(\"p_val.csv\",index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "      <th>AAAS</th>\n",
       "      <th>AACS</th>\n",
       "      <th>AADAC</th>\n",
       "      <th>AADAT</th>\n",
       "      <th>...</th>\n",
       "      <th>ZSWIM9</th>\n",
       "      <th>ZW10</th>\n",
       "      <th>ZWILCH</th>\n",
       "      <th>ZWINT</th>\n",
       "      <th>ZXDA</th>\n",
       "      <th>ZXDC</th>\n",
       "      <th>ZYG11B</th>\n",
       "      <th>ZYX</th>\n",
       "      <th>ZZEF1</th>\n",
       "      <th>ZZZ3</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.008435</td>\n",
       "      <td>0.001958</td>\n",
       "      <td>0.061198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.347296</td>\n",
       "      <td>0.767248</td>\n",
       "      <td>1.477092e-01</td>\n",
       "      <td>0.081185</td>\n",
       "      <td>0.370717</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.069946</td>\n",
       "      <td>1.166079e-01</td>\n",
       "      <td>2.703798e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.065181</td>\n",
       "      <td>4.565249e-01</td>\n",
       "      <td>0.106062</td>\n",
       "      <td>0.327473</td>\n",
       "      <td>0.191946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.824775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.936116</td>\n",
       "      <td>3.499895e-02</td>\n",
       "      <td>0.264571</td>\n",
       "      <td>0.030133</td>\n",
       "      <td>3.175433e-03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.287843</td>\n",
       "      <td>...</td>\n",
       "      <td>0.654079</td>\n",
       "      <td>0.781930</td>\n",
       "      <td>7.278204e-03</td>\n",
       "      <td>2.185463e-03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.359389</td>\n",
       "      <td>2.029671e-01</td>\n",
       "      <td>0.556780</td>\n",
       "      <td>0.236825</td>\n",
       "      <td>0.261292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gbm</td>\n",
       "      <td>0.475921</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.953751</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.944833</td>\n",
       "      <td>2.692001e-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.995065</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.428331</td>\n",
       "      <td>4.868691e-04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.752768</td>\n",
       "      <td>4.370716e-01</td>\n",
       "      <td>0.036739</td>\n",
       "      <td>0.512177</td>\n",
       "      <td>0.185044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>luad</td>\n",
       "      <td>0.778949</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.433724</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.131262</td>\n",
       "      <td>7.607273e-12</td>\n",
       "      <td>0.065583</td>\n",
       "      <td>0.003088</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>5.299375e-16</td>\n",
       "      <td>2.143507e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.574093</td>\n",
       "      <td>2.386150e-08</td>\n",
       "      <td>0.000360</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.000002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.988560</td>\n",
       "      <td>0.446635</td>\n",
       "      <td>0.252807</td>\n",
       "      <td>6.755440e-02</td>\n",
       "      <td>0.118225</td>\n",
       "      <td>0.011762</td>\n",
       "      <td>7.665152e-01</td>\n",
       "      <td>0.000098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005881</td>\n",
       "      <td>3.638297e-01</td>\n",
       "      <td>8.984708e-06</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001435</td>\n",
       "      <td>1.751749e-04</td>\n",
       "      <td>0.934953</td>\n",
       "      <td>0.090658</td>\n",
       "      <td>0.795908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>lscc</td>\n",
       "      <td>0.169387</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.610199</td>\n",
       "      <td>2.377650e-12</td>\n",
       "      <td>0.156933</td>\n",
       "      <td>0.030961</td>\n",
       "      <td>1.035001e-01</td>\n",
       "      <td>0.300931</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000682</td>\n",
       "      <td>0.000139</td>\n",
       "      <td>6.967096e-08</td>\n",
       "      <td>6.784408e-12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005178</td>\n",
       "      <td>8.362345e-02</td>\n",
       "      <td>0.008578</td>\n",
       "      <td>0.011934</td>\n",
       "      <td>0.000571</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6 rows × 14637 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Cancer      A1BG      A1CF       A2M         A2ML1    A4GALT  \\\n",
       "0        ccrcc  0.008435  0.001958  0.061198           NaN  0.347296   \n",
       "1  endometrial  0.824775       NaN  0.936116  3.499895e-02  0.264571   \n",
       "2          gbm  0.475921       NaN  0.953751           NaN       NaN   \n",
       "3         luad  0.778949       NaN  0.433724           NaN       NaN   \n",
       "4        hnscc  0.988560  0.446635  0.252807  6.755440e-02  0.118225   \n",
       "5         lscc  0.169387       NaN  0.610199  2.377650e-12  0.156933   \n",
       "\n",
       "       AAAS          AACS     AADAC     AADAT  ...    ZSWIM9      ZW10  \\\n",
       "0  0.767248  1.477092e-01  0.081185  0.370717  ...       NaN  0.069946   \n",
       "1  0.030133  3.175433e-03       NaN  0.287843  ...  0.654079  0.781930   \n",
       "2  0.944833  2.692001e-01       NaN  0.995065  ...       NaN  0.428331   \n",
       "3  0.131262  7.607273e-12  0.065583  0.003088  ...       NaN  0.000012   \n",
       "4  0.011762  7.665152e-01  0.000098       NaN  ...       NaN  0.005881   \n",
       "5  0.030961  1.035001e-01  0.300931       NaN  ...  0.000682  0.000139   \n",
       "\n",
       "         ZWILCH         ZWINT  ZXDA      ZXDC        ZYG11B       ZYX  \\\n",
       "0  1.166079e-01  2.703798e-01   NaN  0.065181  4.565249e-01  0.106062   \n",
       "1  7.278204e-03  2.185463e-03   NaN  0.359389  2.029671e-01  0.556780   \n",
       "2  4.868691e-04           NaN   NaN  0.752768  4.370716e-01  0.036739   \n",
       "3  5.299375e-16  2.143507e-05   NaN  0.574093  2.386150e-08  0.000360   \n",
       "4  3.638297e-01  8.984708e-06   NaN  0.001435  1.751749e-04  0.934953   \n",
       "5  6.967096e-08  6.784408e-12   NaN  0.005178  8.362345e-02  0.008578   \n",
       "\n",
       "      ZZEF1      ZZZ3  \n",
       "0  0.327473  0.191946  \n",
       "1  0.236825  0.261292  \n",
       "2  0.512177  0.185044  \n",
       "3  0.000060  0.000002  \n",
       "4  0.090658  0.795908  \n",
       "5  0.011934  0.000571  \n",
       "\n",
       "[6 rows x 14637 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "normal_df = cancer.join_omics_to_omics(\"transcriptomics\",\"proteomics\",tissue_type=\"normal\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# normal_df[[\"A1BG_transcriptomics\",\"A1CF_transcriptomics\"]]\n",
    "# normal_df\n",
    "\"A1BG_transcriptomics\" in normal_df.columns and \"A1CF_transcriptomics\" in normal_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'yoyip'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hi = \"yo\"\n",
    "hi2 = \"yip\"\n",
    "hi + hi2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>A1BG</th>\n",
       "      <th>A1CF</th>\n",
       "      <th>A2M</th>\n",
       "      <th>A2ML1</th>\n",
       "      <th>A4GALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>brca</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ccrcc</td>\n",
       "      <td>0.049344</td>\n",
       "      <td>0.000831</td>\n",
       "      <td>0.019891</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>colon</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>endometrial</td>\n",
       "      <td>0.775861</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.912147</td>\n",
       "      <td>1.393949e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>gbm</td>\n",
       "      <td>0.380497</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.720140</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>luad</td>\n",
       "      <td>0.771030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009256</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ovarian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>hnscc</td>\n",
       "      <td>0.991894</td>\n",
       "      <td>0.788576</td>\n",
       "      <td>0.211574</td>\n",
       "      <td>4.487003e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>lscc</td>\n",
       "      <td>0.140028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.098622</td>\n",
       "      <td>1.278508e-22</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          A1BG      A1CF       A2M     A2ML1        A4GALT\n",
       "0         brca       NaN       NaN       NaN           NaN\n",
       "1        ccrcc  0.049344  0.000831  0.019891           NaN\n",
       "2        colon       NaN       NaN       NaN           NaN\n",
       "3  endometrial  0.775861       NaN  0.912147  1.393949e-02\n",
       "4          gbm  0.380497       NaN  0.720140           NaN\n",
       "5         luad  0.771030       NaN  0.009256           NaN\n",
       "6      ovarian       NaN       NaN       NaN           NaN\n",
       "7        hnscc  0.991894  0.788576  0.211574  4.487003e-04\n",
       "8         lscc  0.140028       NaN  0.098622  1.278508e-22"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "        \n",
    "        #join omics to omics, proteomics to transcriptomics. One call for tumor, one to normal\n",
    "        #join one column from proteomics, one from transcriptomics. Make one normal, one tumor. each will have 2 columns\n",
    "        #here we call Humberto's function as well, giving us p-value - make dataframe with multijoin that has label for \n",
    "        #whether tumor or normal (3 columns in total, transcrip, prot, and tumor/normal. delta correlation as corr1 - corr2)\n",
    "        #just a stats call to get corr numbers\n",
    "        #If Nathaniel's p-value looks bad, then don't give it to Humberto. Otherwise, pass it in\n",
    "#         corr_1,num_samples_1 = get_single_gene_correlations(gene, [cancer],input_tissue_type = \"tumor\")\n",
    "#         corr_2,num_samples_2 = get_single_gene_correlations(gene,[cancer],input_tissue_type = \"normal\")\n",
    "#             check mut vs wt tumor:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create 3 giant dataframes - difference in correlation, Nathaniel's p-value from z score, and Humberto's permutation based p-value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "[\"hi\"].extend(['yo','wuz'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['hi', 'yo', 'wuz']"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "one = [\"hi\"]\n",
    "one.extend(['yo','wuz'])\n",
    "one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "second_df = copy.deepcopy(tumor_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "second_df\n",
    "second_df = second_df.iloc[:, 0:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKAP17A_transcriptomics</th>\n",
       "      <th>AKAP17A_transcriptomics</th>\n",
       "      <th>AKAP17A_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>224925.013065</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.544187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>251674.374330</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.203106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>205830.598754</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.020474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>268763.653430</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.139892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>180929.603827</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.107471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>258341.911092</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.290588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>166729.683250</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.660597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>181008.038925</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-0.255908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>222308.197768</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.028869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>297698.055560</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.540620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKAP17A_transcriptomics  AKAP17A_transcriptomics  \\\n",
       "Patient_ID                                                     \n",
       "C3L-00104             224925.013065                      0.0   \n",
       "C3L-00365             251674.374330                      0.0   \n",
       "C3L-00674             205830.598754                      0.0   \n",
       "C3L-00677             268763.653430                      0.0   \n",
       "C3L-01040             180929.603827                      0.0   \n",
       "...                             ...                      ...   \n",
       "C3N-03183             258341.911092                      0.0   \n",
       "C3N-03184             166729.683250                      0.0   \n",
       "C3N-03186             181008.038925                      0.0   \n",
       "C3N-03188             222308.197768                      0.0   \n",
       "C3N-03473             297698.055560                      0.0   \n",
       "\n",
       "Name        AKAP17A_proteomics  \n",
       "Patient_ID                      \n",
       "C3L-00104             0.544187  \n",
       "C3L-00365             0.203106  \n",
       "C3L-00674             0.020474  \n",
       "C3L-00677             0.139892  \n",
       "C3L-01040             0.107471  \n",
       "...                        ...  \n",
       "C3N-03183             0.290588  \n",
       "C3N-03184             0.660597  \n",
       "C3N-03186            -0.255908  \n",
       "C3N-03188             0.028869  \n",
       "C3N-03473             0.540620  \n",
       "\n",
       "[99 rows x 3 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "second_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "third_df = second_df.iloc[:,0:1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [],
   "source": [
    "add_col = second_df['AKAP17A_proteomics']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "third_df['AKAP17A_proteomics'] = add_col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKAP17A_transcriptomics</th>\n",
       "      <th>AKAP17A_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>224925.013065</td>\n",
       "      <td>0.544187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>251674.374330</td>\n",
       "      <td>0.203106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>205830.598754</td>\n",
       "      <td>0.020474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>268763.653430</td>\n",
       "      <td>0.139892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>180929.603827</td>\n",
       "      <td>0.107471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>258341.911092</td>\n",
       "      <td>0.290588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>166729.683250</td>\n",
       "      <td>0.660597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>181008.038925</td>\n",
       "      <td>-0.255908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>222308.197768</td>\n",
       "      <td>0.028869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>297698.055560</td>\n",
       "      <td>0.540620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKAP17A_transcriptomics  AKAP17A_proteomics\n",
       "Patient_ID                                             \n",
       "C3L-00104             224925.013065            0.544187\n",
       "C3L-00365             251674.374330            0.203106\n",
       "C3L-00674             205830.598754            0.020474\n",
       "C3L-00677             268763.653430            0.139892\n",
       "C3L-01040             180929.603827            0.107471\n",
       "...                             ...                 ...\n",
       "C3N-03183             258341.911092            0.290588\n",
       "C3N-03184             166729.683250            0.660597\n",
       "C3N-03186             181008.038925           -0.255908\n",
       "C3N-03188             222308.197768            0.028869\n",
       "C3N-03473             297698.055560            0.540620\n",
       "\n",
       "[99 rows x 2 columns]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "third_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [],
   "source": [
    "if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame) or isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame): #This is to take first column of multi-index\n",
    "    trans_col = tumor_df[gene_trans]\n",
    "    if isinstance(tumor_df[gene_trans], pd.core.frame.DataFrame):\n",
    "        trans_col = trans_col.iloc[:,0:1]\n",
    "    prot_col = tumor_df[gene_prot]\n",
    "    if isinstance(tumor_df[gene_prot], pd.core.frame.DataFrame):\n",
    "        prot_col = prot_col.iloc[:,0:1]\n",
    "    trans_col[gene_prot] = prot_col\n",
    "    tumor_df = trans_col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Name</th>\n",
       "      <th>AKAP17A_transcriptomics</th>\n",
       "      <th>AKAP17A_proteomics</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Patient_ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>C3L-00104</th>\n",
       "      <td>224925.013065</td>\n",
       "      <td>0.544187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00365</th>\n",
       "      <td>251674.374330</td>\n",
       "      <td>0.203106</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00674</th>\n",
       "      <td>205830.598754</td>\n",
       "      <td>0.020474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-00677</th>\n",
       "      <td>268763.653430</td>\n",
       "      <td>0.139892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3L-01040</th>\n",
       "      <td>180929.603827</td>\n",
       "      <td>0.107471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03183</th>\n",
       "      <td>258341.911092</td>\n",
       "      <td>0.290588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03184</th>\n",
       "      <td>166729.683250</td>\n",
       "      <td>0.660597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03186</th>\n",
       "      <td>181008.038925</td>\n",
       "      <td>-0.255908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03188</th>\n",
       "      <td>222308.197768</td>\n",
       "      <td>0.028869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C3N-03473</th>\n",
       "      <td>297698.055560</td>\n",
       "      <td>0.540620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>99 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Name        AKAP17A_transcriptomics  AKAP17A_proteomics\n",
       "Patient_ID                                             \n",
       "C3L-00104             224925.013065            0.544187\n",
       "C3L-00365             251674.374330            0.203106\n",
       "C3L-00674             205830.598754            0.020474\n",
       "C3L-00677             268763.653430            0.139892\n",
       "C3L-01040             180929.603827            0.107471\n",
       "...                             ...                 ...\n",
       "C3N-03183             258341.911092            0.290588\n",
       "C3N-03184             166729.683250            0.660597\n",
       "C3N-03186             181008.038925           -0.255908\n",
       "C3N-03188             222308.197768            0.028869\n",
       "C3N-03473             297698.055560            0.540620\n",
       "\n",
       "[99 rows x 2 columns]"
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tumor_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
