{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import scipy \n",
    "import dataframe_image as dfi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option('display.max_rows',70)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('data/Positive_delta_corr_individual_cancer_pathways.csv')\n",
    "df = df[df.source.isin(['GO:BP', 'KEGG'])]\n",
    "df = df[['source', 'name', 'p_value', 'term_size', 'intersection_size', 'Cancer']]\n",
    "df = df.rename(columns = {'source': 'Database', 'name': 'Pathway', 'term_size' : 'Pathway size', 'intersection_size': '# upregulated genes'})\n",
    "df = df.groupby(['Database', 'Pathway', 'Cancer', 'p_value', 'Pathway size', '# upregulated genes'])#.apply(lambda x: x['# upregulated genes'] / x['Pathway size'])\n",
    "\n",
    "df = pd.DataFrame(df)\n",
    "# df = df.droplevel(None)\n",
    "# #df['Cancer'] = df.index.get_level_values('Cancer')\n",
    "# df['Pathway size'] = df.index.get_level_values('Pathway size')\n",
    "# df['# upregulated genes'] = df.index.get_level_values('# upregulated genes')\n",
    "# df['p_value'] = df.index.get_level_values('p_value')\n",
    "# df = df.droplevel([ '# upregulated genes', 'Pathway size', 'p_value'])\n",
    "#df = df.drop(columns = ['percent_pathway'])\n",
    "# df = df.style.hide_columns(['percent_pathway']).bar(subset=['p_value'], color = 'red')\n",
    "#df = df.style.bar(subset=['p_value'], color = 'red')\n",
    "# df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'Requested level (Cancer) does not match index name (None)'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "Input \u001b[0;32mIn [120]\u001b[0m, in \u001b[0;36m<cell line: 1>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mdf\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mindex\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_level_values\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mCancer\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py:2008\u001b[0m, in \u001b[0;36mIndex._get_level_values\u001b[0;34m(self, level)\u001b[0m\n\u001b[1;32m   1972\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_get_level_values\u001b[39m(\u001b[38;5;28mself\u001b[39m, level) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m Index:\n\u001b[1;32m   1973\u001b[0m     \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m   1974\u001b[0m \u001b[38;5;124;03m    Return an Index of values for requested level.\u001b[39;00m\n\u001b[1;32m   1975\u001b[0m \n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   2006\u001b[0m \u001b[38;5;124;03m    Index(['a', 'b', 'c'], dtype='object')\u001b[39;00m\n\u001b[1;32m   2007\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[0;32m-> 2008\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate_index_level\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlevel\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   2009\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\n",
      "File \u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/pandas/core/indexes/base.py:1931\u001b[0m, in \u001b[0;36mIndex._validate_index_level\u001b[0;34m(self, level)\u001b[0m\n\u001b[1;32m   1927\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mIndexError\u001b[39;00m(\n\u001b[1;32m   1928\u001b[0m             \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mToo many levels: Index has only 1 level, not \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlevel \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1929\u001b[0m         )\n\u001b[1;32m   1930\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m level \u001b[38;5;241m!=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mname:\n\u001b[0;32m-> 1931\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mKeyError\u001b[39;00m(\n\u001b[1;32m   1932\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mRequested level (\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mlevel\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m) does not match index name (\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mname\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m)\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   1933\u001b[0m     )\n",
      "\u001b[0;31mKeyError\u001b[0m: 'Requested level (Cancer) does not match index name (None)'"
     ]
    }
   ],
   "source": [
    "df.index.get_level_values('Cancer')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "# dfi.export(df, 'Figures/Positive_delta_corr_pathways.png', max_cols= -1, max_rows=-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# g = sns.FacetGrid(data = df, col = 'Cancer', row = 'source', sharex = True, sharey=False)\n",
    "# g.map_dataframe(sns.barplot,y = 'name', x ='intersection_size', hue = 'p_value')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>(GO:BP, DNA strand elongation involved in DNA ...</td>\n",
       "      <td>Database                                   ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>(GO:BP, DNA strand elongation involved in DNA ...</td>\n",
       "      <td>Database                                  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>(GO:BP, cell cycle, LUAD, 0.0044683395228127, ...</td>\n",
       "      <td>Database     Pathway   p_value  Pathway si...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>(GO:BP, cell cycle DNA replication, LSCC, 0.01...</td>\n",
       "      <td>Database                     Pathway   p_v...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>(GO:BP, cell cycle DNA replication, LUAD, 0.00...</td>\n",
       "      <td>Database                     Pathway  p_va...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>(GO:BP, cell cycle process, LUAD, 0.0182447136...</td>\n",
       "      <td>Database             Pathway   p_value  Pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>(GO:BP, cellular response to type I interferon...</td>\n",
       "      <td>Database                                 Pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>(GO:BP, chromosome condensation, LUAD, 0.04548...</td>\n",
       "      <td>Database                  Pathway   p_valu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>(GO:BP, chromosome segregation, LSCC, 0.000908...</td>\n",
       "      <td>Database                 Pathway   p_value ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>(GO:BP, chromosome segregation, LUAD, 0.001316...</td>\n",
       "      <td>Database                 Pathway   p_value...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>(GO:BP, cornification, LSCC, 0.032485050755549...</td>\n",
       "      <td>Database        Pathway   p_value  Pathway...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>(GO:BP, cytokinetic process, LSCC, 0.000624200...</td>\n",
       "      <td>Database              Pathway   p_value  Pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>(GO:BP, double-strand break repair via break-i...</td>\n",
       "      <td>Database                                  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>(GO:BP, membrane assembly, LSCC, 0.00971231718...</td>\n",
       "      <td>Database            Pathway   p_value  Pat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>(GO:BP, membrane biogenesis, LSCC, 0.030091860...</td>\n",
       "      <td>Database              Pathway   p_value  P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>(GO:BP, microtubule cytoskeleton organization ...</td>\n",
       "      <td>Database                                   ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>(GO:BP, mitotic cell cycle, LSCC, 0.0063367798...</td>\n",
       "      <td>Database             Pathway   p_value  Pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>(GO:BP, mitotic cell cycle, LUAD, 0.0005315150...</td>\n",
       "      <td>Database             Pathway   p_value  Pa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>(GO:BP, mitotic cell cycle process, LSCC, 0.00...</td>\n",
       "      <td>Database                     Pathway   p_va...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>(GO:BP, mitotic cell cycle process, LUAD, 0.00...</td>\n",
       "      <td>Database                     Pathway   p_v...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>(GO:BP, mitotic cytokinesis, LSCC, 0.005653310...</td>\n",
       "      <td>Database              Pathway   p_value  P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>(GO:BP, mitotic metaphase plate congression, L...</td>\n",
       "      <td>Database                              Path...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>(GO:BP, mitotic nuclear division, LSCC, 0.0081...</td>\n",
       "      <td>Database                   Pathway   p_val...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>(GO:BP, mitotic nuclear division, LUAD, 4.5490...</td>\n",
       "      <td>Database                   Pathway   p_val...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>(GO:BP, mitotic nuclear membrane organization,...</td>\n",
       "      <td>Database                                Pat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>(GO:BP, mitotic nuclear membrane reassembly, L...</td>\n",
       "      <td>Database                              Pathw...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>(GO:BP, mitotic sister chromatid segregation, ...</td>\n",
       "      <td>Database                               Path...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>(GO:BP, mitotic sister chromatid segregation, ...</td>\n",
       "      <td>Database                               Pat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>(GO:BP, mitotic spindle organization, LSCC, 0....</td>\n",
       "      <td>Database                       Pathway   p_...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>(GO:BP, nuclear DNA replication, LSCC, 0.02219...</td>\n",
       "      <td>Database                  Pathway   p_valu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>(GO:BP, nuclear DNA replication, LUAD, 0.00556...</td>\n",
       "      <td>Database                  Pathway   p_valu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>(GO:BP, nuclear chromosome segregation, LSCC, ...</td>\n",
       "      <td>Database                         Pathway  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>(GO:BP, nuclear chromosome segregation, LUAD, ...</td>\n",
       "      <td>Database                         Pathway  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>(GO:BP, nuclear division, LUAD, 1.489325682026...</td>\n",
       "      <td>Database           Pathway   p_value  Path...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>(GO:BP, nuclear export, LSCC, 0.01137637001516...</td>\n",
       "      <td>Database         Pathway   p_value  Pathwa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>(GO:BP, nuclear membrane organization, LSCC, 0...</td>\n",
       "      <td>Database                        Pathway   ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>(GO:BP, nuclear membrane reassembly, LSCC, 6.3...</td>\n",
       "      <td>Database                      Pathway   p_v...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>(GO:BP, organelle fission, LUAD, 3.22175772331...</td>\n",
       "      <td>Database            Pathway   p_value  Pat...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>(GO:BP, pre-replicative complex assembly invol...</td>\n",
       "      <td>Database                                  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>(GO:BP, pre-replicative complex assembly invol...</td>\n",
       "      <td>Database                                  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>(GO:BP, response to type I interferon, HNSCC, ...</td>\n",
       "      <td>Database                        Pathway   p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>(GO:BP, ribonucleoprotein complex export from ...</td>\n",
       "      <td>Database                                  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>(GO:BP, ribonucleoprotein complex localization...</td>\n",
       "      <td>Database                                 P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>(GO:BP, sister chromatid segregation, LSCC, 0....</td>\n",
       "      <td>Database                       Pathway   p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>(GO:BP, sister chromatid segregation, LUAD, 4....</td>\n",
       "      <td>Database                       Pathway   p...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>(GO:BP, spindle organization, LSCC, 0.00332009...</td>\n",
       "      <td>Database               Pathway  p_value  P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>(GO:BP, type I interferon signaling pathway, H...</td>\n",
       "      <td>Database                              Pathw...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>(KEGG, Biosynthesis of amino acids, Endometria...</td>\n",
       "      <td>Database                      Pathway   p_va...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>(KEGG, Cell cycle, LSCC, 0.0017038538134792, 9...</td>\n",
       "      <td>Database     Pathway   p_value  Pathway siz...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>(KEGG, Cell cycle, LUAD, 0.0005631656266224, 8...</td>\n",
       "      <td>Database     Pathway   p_value  Pathway si...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>(KEGG, DNA replication, LSCC, 8.90451489694245...</td>\n",
       "      <td>Database          Pathway   p_value  Pathwa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>(KEGG, DNA replication, LUAD, 1.20435905223152...</td>\n",
       "      <td>Database          Pathway   p_value  Pathw...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>(KEGG, Fluid shear stress and atherosclerosis,...</td>\n",
       "      <td>Database                                 P...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>(KEGG, Mismatch repair, LSCC, 0.00476448109165...</td>\n",
       "      <td>Database          Pathway   p_value  Pathw...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>(KEGG, Mismatch repair, LUAD, 0.01534040365560...</td>\n",
       "      <td>Database          Pathway  p_value  Pathwa...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>(KEGG, Prostate cancer, LSCC, 0.03730622902957...</td>\n",
       "      <td>Database          Pathway   p_value  Pathw...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>(KEGG, Pyrimidine metabolism, LUAD, 0.00110846...</td>\n",
       "      <td>Database                Pathway   p_value ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>(KEGG, RNA transport, LSCC, 0.0110484471133605...</td>\n",
       "      <td>Database        Pathway   p_value  Pathway...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>(KEGG, Small cell lung cancer, HNSCC, 0.033020...</td>\n",
       "      <td>Database                 Pathway  p_value  ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>(KEGG, TNF signaling pathway, LSCC, 0.02732864...</td>\n",
       "      <td>Database                Pathway   p_value ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>(KEGG, Valine, leucine and isoleucine degradat...</td>\n",
       "      <td>Database                                    ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    0  \\\n",
       "0   (GO:BP, DNA strand elongation involved in DNA ...   \n",
       "1   (GO:BP, DNA strand elongation involved in DNA ...   \n",
       "2   (GO:BP, cell cycle, LUAD, 0.0044683395228127, ...   \n",
       "3   (GO:BP, cell cycle DNA replication, LSCC, 0.01...   \n",
       "4   (GO:BP, cell cycle DNA replication, LUAD, 0.00...   \n",
       "5   (GO:BP, cell cycle process, LUAD, 0.0182447136...   \n",
       "6   (GO:BP, cellular response to type I interferon...   \n",
       "7   (GO:BP, chromosome condensation, LUAD, 0.04548...   \n",
       "8   (GO:BP, chromosome segregation, LSCC, 0.000908...   \n",
       "9   (GO:BP, chromosome segregation, LUAD, 0.001316...   \n",
       "10  (GO:BP, cornification, LSCC, 0.032485050755549...   \n",
       "11  (GO:BP, cytokinetic process, LSCC, 0.000624200...   \n",
       "12  (GO:BP, double-strand break repair via break-i...   \n",
       "13  (GO:BP, membrane assembly, LSCC, 0.00971231718...   \n",
       "14  (GO:BP, membrane biogenesis, LSCC, 0.030091860...   \n",
       "15  (GO:BP, microtubule cytoskeleton organization ...   \n",
       "16  (GO:BP, mitotic cell cycle, LSCC, 0.0063367798...   \n",
       "17  (GO:BP, mitotic cell cycle, LUAD, 0.0005315150...   \n",
       "18  (GO:BP, mitotic cell cycle process, LSCC, 0.00...   \n",
       "19  (GO:BP, mitotic cell cycle process, LUAD, 0.00...   \n",
       "20  (GO:BP, mitotic cytokinesis, LSCC, 0.005653310...   \n",
       "21  (GO:BP, mitotic metaphase plate congression, L...   \n",
       "22  (GO:BP, mitotic nuclear division, LSCC, 0.0081...   \n",
       "23  (GO:BP, mitotic nuclear division, LUAD, 4.5490...   \n",
       "24  (GO:BP, mitotic nuclear membrane organization,...   \n",
       "25  (GO:BP, mitotic nuclear membrane reassembly, L...   \n",
       "26  (GO:BP, mitotic sister chromatid segregation, ...   \n",
       "27  (GO:BP, mitotic sister chromatid segregation, ...   \n",
       "28  (GO:BP, mitotic spindle organization, LSCC, 0....   \n",
       "29  (GO:BP, nuclear DNA replication, LSCC, 0.02219...   \n",
       "30  (GO:BP, nuclear DNA replication, LUAD, 0.00556...   \n",
       "31  (GO:BP, nuclear chromosome segregation, LSCC, ...   \n",
       "32  (GO:BP, nuclear chromosome segregation, LUAD, ...   \n",
       "33  (GO:BP, nuclear division, LUAD, 1.489325682026...   \n",
       "34  (GO:BP, nuclear export, LSCC, 0.01137637001516...   \n",
       "35  (GO:BP, nuclear membrane organization, LSCC, 0...   \n",
       "36  (GO:BP, nuclear membrane reassembly, LSCC, 6.3...   \n",
       "37  (GO:BP, organelle fission, LUAD, 3.22175772331...   \n",
       "38  (GO:BP, pre-replicative complex assembly invol...   \n",
       "39  (GO:BP, pre-replicative complex assembly invol...   \n",
       "40  (GO:BP, response to type I interferon, HNSCC, ...   \n",
       "41  (GO:BP, ribonucleoprotein complex export from ...   \n",
       "42  (GO:BP, ribonucleoprotein complex localization...   \n",
       "43  (GO:BP, sister chromatid segregation, LSCC, 0....   \n",
       "44  (GO:BP, sister chromatid segregation, LUAD, 4....   \n",
       "45  (GO:BP, spindle organization, LSCC, 0.00332009...   \n",
       "46  (GO:BP, type I interferon signaling pathway, H...   \n",
       "47  (KEGG, Biosynthesis of amino acids, Endometria...   \n",
       "48  (KEGG, Cell cycle, LSCC, 0.0017038538134792, 9...   \n",
       "49  (KEGG, Cell cycle, LUAD, 0.0005631656266224, 8...   \n",
       "50  (KEGG, DNA replication, LSCC, 8.90451489694245...   \n",
       "51  (KEGG, DNA replication, LUAD, 1.20435905223152...   \n",
       "52  (KEGG, Fluid shear stress and atherosclerosis,...   \n",
       "53  (KEGG, Mismatch repair, LSCC, 0.00476448109165...   \n",
       "54  (KEGG, Mismatch repair, LUAD, 0.01534040365560...   \n",
       "55  (KEGG, Prostate cancer, LSCC, 0.03730622902957...   \n",
       "56  (KEGG, Pyrimidine metabolism, LUAD, 0.00110846...   \n",
       "57  (KEGG, RNA transport, LSCC, 0.0110484471133605...   \n",
       "58  (KEGG, Small cell lung cancer, HNSCC, 0.033020...   \n",
       "59  (KEGG, TNF signaling pathway, LSCC, 0.02732864...   \n",
       "60  (KEGG, Valine, leucine and isoleucine degradat...   \n",
       "\n",
       "                                                    1  \n",
       "0      Database                                   ...  \n",
       "1       Database                                  ...  \n",
       "2       Database     Pathway   p_value  Pathway si...  \n",
       "3       Database                     Pathway   p_v...  \n",
       "4       Database                     Pathway  p_va...  \n",
       "5       Database             Pathway   p_value  Pa...  \n",
       "6      Database                                 Pa...  \n",
       "7       Database                  Pathway   p_valu...  \n",
       "8      Database                 Pathway   p_value ...  \n",
       "9       Database                 Pathway   p_value...  \n",
       "10      Database        Pathway   p_value  Pathway...  \n",
       "11     Database              Pathway   p_value  Pa...  \n",
       "12      Database                                  ...  \n",
       "13      Database            Pathway   p_value  Pat...  \n",
       "14      Database              Pathway   p_value  P...  \n",
       "15     Database                                   ...  \n",
       "16      Database             Pathway   p_value  Pa...  \n",
       "17      Database             Pathway   p_value  Pa...  \n",
       "18     Database                     Pathway   p_va...  \n",
       "19      Database                     Pathway   p_v...  \n",
       "20      Database              Pathway   p_value  P...  \n",
       "21      Database                              Path...  \n",
       "22      Database                   Pathway   p_val...  \n",
       "23      Database                   Pathway   p_val...  \n",
       "24     Database                                Pat...  \n",
       "25     Database                              Pathw...  \n",
       "26     Database                               Path...  \n",
       "27      Database                               Pat...  \n",
       "28     Database                       Pathway   p_...  \n",
       "29      Database                  Pathway   p_valu...  \n",
       "30      Database                  Pathway   p_valu...  \n",
       "31      Database                         Pathway  ...  \n",
       "32      Database                         Pathway  ...  \n",
       "33      Database           Pathway   p_value  Path...  \n",
       "34      Database         Pathway   p_value  Pathwa...  \n",
       "35      Database                        Pathway   ...  \n",
       "36     Database                      Pathway   p_v...  \n",
       "37      Database            Pathway   p_value  Pat...  \n",
       "38      Database                                  ...  \n",
       "39      Database                                  ...  \n",
       "40     Database                        Pathway   p...  \n",
       "41      Database                                  ...  \n",
       "42      Database                                 P...  \n",
       "43      Database                       Pathway   p...  \n",
       "44      Database                       Pathway   p...  \n",
       "45      Database               Pathway  p_value  P...  \n",
       "46     Database                              Pathw...  \n",
       "47    Database                      Pathway   p_va...  \n",
       "48     Database     Pathway   p_value  Pathway siz...  \n",
       "49      Database     Pathway   p_value  Pathway si...  \n",
       "50     Database          Pathway   p_value  Pathwa...  \n",
       "51      Database          Pathway   p_value  Pathw...  \n",
       "52      Database                                 P...  \n",
       "53      Database          Pathway   p_value  Pathw...  \n",
       "54      Database          Pathway  p_value  Pathwa...  \n",
       "55      Database          Pathway   p_value  Pathw...  \n",
       "56      Database                Pathway   p_value ...  \n",
       "57      Database        Pathway   p_value  Pathway...  \n",
       "58     Database                 Pathway  p_value  ...  \n",
       "59      Database                Pathway   p_value ...  \n",
       "60    Database                                    ...  "
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['percent_pathway'], dtype='object')"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Could not interpret input 'p_value'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "Input \u001b[0;32mIn [98]\u001b[0m, in \u001b[0;36m<cell line: 2>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m plt\u001b[38;5;241m.\u001b[39mfigure(figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m9\u001b[39m,\u001b[38;5;241m13\u001b[39m))\n\u001b[0;32m----> 2\u001b[0m g \u001b[38;5;241m=\u001b[39m \u001b[43msns\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mbarplot\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mdf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mx\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mp_value\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43my\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mDatabase\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhue\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mCancer Hallmark\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      3\u001b[0m g\u001b[38;5;241m.\u001b[39mlegend(bbox_to_anchor\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m1.01\u001b[39m, \u001b[38;5;241m.5\u001b[39m), ncol\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, title \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mCancer Hallmark\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m      4\u001b[0m plt\u001b[38;5;241m.\u001b[39mylabel(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGO:BP Pathway\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/seaborn/_decorators.py:46\u001b[0m, in \u001b[0;36m_deprecate_positional_args.<locals>.inner_f\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m     36\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[1;32m     37\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPass the following variable\u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m as \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124mkeyword arg\u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m: \u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m     38\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFrom version 0.12, the only valid positional argument \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     43\u001b[0m         \u001b[38;5;167;01mFutureWarning\u001b[39;00m\n\u001b[1;32m     44\u001b[0m     )\n\u001b[1;32m     45\u001b[0m kwargs\u001b[38;5;241m.\u001b[39mupdate({k: arg \u001b[38;5;28;01mfor\u001b[39;00m k, arg \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(sig\u001b[38;5;241m.\u001b[39mparameters, args)})\n\u001b[0;32m---> 46\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mf\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/seaborn/categorical.py:3182\u001b[0m, in \u001b[0;36mbarplot\u001b[0;34m(x, y, hue, data, order, hue_order, estimator, ci, n_boot, units, seed, orient, color, palette, saturation, errcolor, errwidth, capsize, dodge, ax, **kwargs)\u001b[0m\n\u001b[1;32m   3169\u001b[0m \u001b[38;5;129m@_deprecate_positional_args\u001b[39m\n\u001b[1;32m   3170\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mbarplot\u001b[39m(\n\u001b[1;32m   3171\u001b[0m     \u001b[38;5;241m*\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m   3179\u001b[0m     \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs,\n\u001b[1;32m   3180\u001b[0m ):\n\u001b[0;32m-> 3182\u001b[0m     plotter \u001b[38;5;241m=\u001b[39m \u001b[43m_BarPlotter\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43my\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43morder\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhue_order\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3183\u001b[0m \u001b[43m                          \u001b[49m\u001b[43mestimator\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mci\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mn_boot\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43munits\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mseed\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3184\u001b[0m \u001b[43m                          \u001b[49m\u001b[43morient\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcolor\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpalette\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msaturation\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3185\u001b[0m \u001b[43m                          \u001b[49m\u001b[43merrcolor\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43merrwidth\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcapsize\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdodge\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3187\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ax \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   3188\u001b[0m         ax \u001b[38;5;241m=\u001b[39m plt\u001b[38;5;241m.\u001b[39mgca()\n",
      "File \u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/seaborn/categorical.py:1584\u001b[0m, in \u001b[0;36m_BarPlotter.__init__\u001b[0;34m(self, x, y, hue, data, order, hue_order, estimator, ci, n_boot, units, seed, orient, color, palette, saturation, errcolor, errwidth, capsize, dodge)\u001b[0m\n\u001b[1;32m   1579\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m, x, y, hue, data, order, hue_order,\n\u001b[1;32m   1580\u001b[0m              estimator, ci, n_boot, units, seed,\n\u001b[1;32m   1581\u001b[0m              orient, color, palette, saturation, errcolor,\n\u001b[1;32m   1582\u001b[0m              errwidth, capsize, dodge):\n\u001b[1;32m   1583\u001b[0m     \u001b[38;5;124;03m\"\"\"Initialize the plotter.\"\"\"\u001b[39;00m\n\u001b[0;32m-> 1584\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mestablish_variables\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43my\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhue\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdata\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43morient\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   1585\u001b[0m \u001b[43m                             \u001b[49m\u001b[43morder\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mhue_order\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43munits\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1586\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mestablish_colors(color, palette, saturation)\n\u001b[1;32m   1587\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mestimate_statistic(estimator, ci, n_boot, seed)\n",
      "File \u001b[0;32m~/opt/anaconda3/lib/python3.9/site-packages/seaborn/categorical.py:153\u001b[0m, in \u001b[0;36m_CategoricalPlotter.establish_variables\u001b[0;34m(self, x, y, hue, data, orient, order, hue_order, units)\u001b[0m\n\u001b[1;32m    151\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(var, \u001b[38;5;28mstr\u001b[39m):\n\u001b[1;32m    152\u001b[0m         err \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCould not interpret input \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{}\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;241m.\u001b[39mformat(var)\n\u001b[0;32m--> 153\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(err)\n\u001b[1;32m    155\u001b[0m \u001b[38;5;66;03m# Figure out the plotting orientation\u001b[39;00m\n\u001b[1;32m    156\u001b[0m orient \u001b[38;5;241m=\u001b[39m infer_orient(\n\u001b[1;32m    157\u001b[0m     x, y, orient, require_numeric\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mrequire_numeric\n\u001b[1;32m    158\u001b[0m )\n",
      "\u001b[0;31mValueError\u001b[0m: Could not interpret input 'p_value'"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 648x936 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(9,13))\n",
    "g = sns.barplot(data = df, x = 'p_value', y = 'Database', hue = 'Cancer Hallmark')\n",
    "g.legend(bbox_to_anchor=(1.01, .5), ncol=1, title = 'Cancer Hallmark')\n",
    "plt.ylabel('GO:BP Pathway')\n",
    "#plt.tight_layout()\n",
    "# plt.savefig('Figures/Figure 5) Cancer Hallmarks Pathways.png', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
