{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import sys, os\n",
    "currentdir = os.path.dirname(os.path.realpath('make_regression_df.py'))\n",
    "parentdir = os.path.dirname(currentdir)\n",
    "sys.path.append(parentdir)\n",
    "import Delta_Correlation as dc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Checking that pdac index is up-to-date...       \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 14)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                         \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The pdac data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 15)\n"
     ]
    }
   ],
   "source": [
    "cancers, cancer_names = dc.load_cancers()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 33)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 33)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 23)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 33)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Patient_ID</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Proteomics</th>\n",
       "      <th>Tissue</th>\n",
       "      <th>Transcriptomics</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Cancer</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>-0.304302</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>0.995336</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.641447</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>16.677828</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>A2M</td>\n",
       "      <td>-0.000025</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>353.263362</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.207831</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>15.831130</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>C3L-00004</td>\n",
       "      <td>AACS</td>\n",
       "      <td>-0.364128</td>\n",
       "      <td>Tumor</td>\n",
       "      <td>2.938550</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>CCRCC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413169</th>\n",
       "      <td>C3N-03780.N</td>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>22.013502</td>\n",
       "      <td>Normal</td>\n",
       "      <td>8.752524</td>\n",
       "      <td>Normal</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413170</th>\n",
       "      <td>C3N-03780.N</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>20.983310</td>\n",
       "      <td>Normal</td>\n",
       "      <td>11.899037</td>\n",
       "      <td>Normal</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413171</th>\n",
       "      <td>C3N-03780.N</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>28.040295</td>\n",
       "      <td>Normal</td>\n",
       "      <td>10.204664</td>\n",
       "      <td>Normal</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413172</th>\n",
       "      <td>C3N-03780.N</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>24.455977</td>\n",
       "      <td>Normal</td>\n",
       "      <td>12.287269</td>\n",
       "      <td>Normal</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1413173</th>\n",
       "      <td>C3N-03780.N</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>19.395490</td>\n",
       "      <td>Normal</td>\n",
       "      <td>11.376311</td>\n",
       "      <td>Normal</td>\n",
       "      <td>PDAC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9748307 rows Ã— 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Patient_ID    Gene  Proteomics  Tissue  Transcriptomics      Stage  \\\n",
       "0          C3L-00004    A1BG   -0.304302   Tumor         0.995336  Stage III   \n",
       "1          C3L-00004    A1CF    0.641447   Tumor        16.677828  Stage III   \n",
       "2          C3L-00004     A2M   -0.000025   Tumor       353.263362  Stage III   \n",
       "3          C3L-00004    AAAS    0.207831   Tumor        15.831130  Stage III   \n",
       "4          C3L-00004    AACS   -0.364128   Tumor         2.938550  Stage III   \n",
       "...              ...     ...         ...     ...              ...        ...   \n",
       "1413169  C3N-03780.N  ZWILCH   22.013502  Normal         8.752524     Normal   \n",
       "1413170  C3N-03780.N  ZYG11B   20.983310  Normal        11.899037     Normal   \n",
       "1413171  C3N-03780.N     ZYX   28.040295  Normal        10.204664     Normal   \n",
       "1413172  C3N-03780.N   ZZEF1   24.455977  Normal        12.287269     Normal   \n",
       "1413173  C3N-03780.N    ZZZ3   19.395490  Normal        11.376311     Normal   \n",
       "\n",
       "        Cancer  \n",
       "0        CCRCC  \n",
       "1        CCRCC  \n",
       "2        CCRCC  \n",
       "3        CCRCC  \n",
       "4        CCRCC  \n",
       "...        ...  \n",
       "1413169   PDAC  \n",
       "1413170   PDAC  \n",
       "1413171   PDAC  \n",
       "1413172   PDAC  \n",
       "1413173   PDAC  \n",
       "\n",
       "[9748307 rows x 7 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):    \n",
    "    if cancer_name == 'CCRCC':\n",
    "        stage = cancer.get_clinical('tumor')[['tumor_stage_pathological']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'tumor_stage_pathological': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "    elif cancer_name == 'LUAD' or cancer_name == 'LSCC':\n",
    "        stage = cancer.get_clinical('tumor')[['Stage']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['1A', '1B', '1', 1, 'I', 'IA', 'IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['2A', '2B', 'II', 'IIA', 'IIB']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['3A', '3', 3, 'III', 'IIIA', 'IIIB']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    elif cancer_name == 'Endometrial':\n",
    "        stage = cancer.get_clinical('tumor')[['FIGO_stage']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'FIGO_stage': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['IA', 'IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['II']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['IIIA', 'IIIB', 'IIIC2', 'IIIC1']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['IVB', 'IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    elif cancer_name == 'HNSCC':\n",
    "        stage = cancer.get_clinical('tumor')[['patho_staging_curated']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'patho_staging_curated': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        \n",
    "    elif cancer_name == 'PDAC':\n",
    "        stage = cancer.get_clinical('tumor')[['tumor_stage_pathological']]\n",
    "        stage['Patient_ID'] = stage.index\n",
    "        stage = stage.rename(columns = {'tumor_stage_pathological': 'Stage'})\n",
    "        stage = stage.reset_index(level = 0, drop = True)\n",
    "        stages = []\n",
    "        for s in stage['Stage']:\n",
    "            if s in ['Stage IA', 'Stage IB']:\n",
    "                stages.append('Stage I')\n",
    "            elif s in ['Stage IIA', 'Stage IIB']:\n",
    "                stages.append('Stage II')\n",
    "            elif s in ['Stage III']:\n",
    "                stages.append('Stage III')\n",
    "            elif s in ['Stage IV']:\n",
    "                stages.append('Stage IV')\n",
    "            else:\n",
    "                stages.append(float('NaN'))\n",
    "        stage['Stage'] = stages\n",
    "    stage = stage.dropna()\n",
    "    cancer_df = dc.get_prot_trans_df(cancer)\n",
    "    cancer_df = cancer_df.dropna()\n",
    "    cancer_df = pd.merge(cancer_df, stage, how = 'outer')\n",
    "    cancer_df[['Stage']] = cancer_df[['Stage']].fillna(value='Normal')\n",
    "    cancer_df = cancer_df.dropna()\n",
    "    cancer_df['Cancer'] = [cancer_name] * len(cancer_df)\n",
    "    cancer_dfs.append(cancer_df)\n",
    "prot_trans_stages_df = pd.concat(cancer_dfs)\n",
    "prot_trans_stages_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Stage</th>\n",
       "      <th>Gene</th>\n",
       "      <th>PearsonR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>A1BG</td>\n",
       "      <td>0.328964</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>A1CF</td>\n",
       "      <td>0.745763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>A2M</td>\n",
       "      <td>0.287323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>AAAS</td>\n",
       "      <td>0.156649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CCRCC</td>\n",
       "      <td>Normal</td>\n",
       "      <td>AACS</td>\n",
       "      <td>0.707274</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305886</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZWILCH</td>\n",
       "      <td>0.385813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305890</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>0.171495</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305891</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZYX</td>\n",
       "      <td>0.480944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305892</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>0.097533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>305893</th>\n",
       "      <td>PDAC</td>\n",
       "      <td>Stage III</td>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-0.090946</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>236317 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Cancer      Stage    Gene  PearsonR\n",
       "0       CCRCC     Normal    A1BG  0.328964\n",
       "1       CCRCC     Normal    A1CF  0.745763\n",
       "2       CCRCC     Normal     A2M  0.287323\n",
       "4       CCRCC     Normal    AAAS  0.156649\n",
       "5       CCRCC     Normal    AACS  0.707274\n",
       "...       ...        ...     ...       ...\n",
       "305886   PDAC  Stage III  ZWILCH  0.385813\n",
       "305890   PDAC  Stage III  ZYG11B  0.171495\n",
       "305891   PDAC  Stage III     ZYX  0.480944\n",
       "305892   PDAC  Stage III   ZZEF1  0.097533\n",
       "305893   PDAC  Stage III    ZZZ3 -0.090946\n",
       "\n",
       "[236317 rows x 4 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = prot_trans_stages_df\n",
    "df = df.drop(columns = 'Patient_ID')\n",
    "grouped_stages = df.groupby(['Cancer', 'Stage', 'Gene'])\n",
    "corr_df = grouped_stages.apply(lambda x: x.corr(method = 'pearson', min_periods = 10).iloc[0][1])\n",
    "corr_df = pd.DataFrame(corr_df, columns = ['PearsonR'])\n",
    "corr_df.reset_index(inplace = True)\n",
    "corr_df = corr_df.dropna()\n",
    "corr_df\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "corr_df.to_csv('Cancer_stages_correlations.csv', index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "    "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
