{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import scipy.stats as stats\n",
    "import cptac.utils as ut\n",
    "import scipy\n",
    "import statsmodels.stats.multitest as ssm\n",
    "import numpy as np\n",
    "import sys, os\n",
    "currentdir = os.path.dirname(os.path.realpath('make_regression_df.py'))\n",
    "parentdir = os.path.dirname(currentdir)\n",
    "sys.path.append(parentdir)\n",
    "import Delta_Correlation as dc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                                \r"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: The LSCC data is currently strictly reserved for CPTAC investigators. Otherwise, you are not authorized to access these data. Additionally, even after these data become publicly available, they will be subject to a publication embargo (see https://proteomics.cancer.gov/data-portal/about/data-use-agreement or enter cptac.embargo() to open the webpage for more details). (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 14)\n"
     ]
    }
   ],
   "source": [
    "cancers, cancer_names = dc.load_cancers()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def log2_fold_change(tumor, normal):\n",
    "    tumor = np.mean(tumor)\n",
    "    normal = np.mean(normal)\n",
    "    if normal == 0 or tumor == 0:\n",
    "        return(float('Nan'))\n",
    "    fold_change = tumor / normal\n",
    "    fold_change = abs(fold_change)\n",
    "    log2_change = np.log2(fold_change)\n",
    "    return log2_change    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 36)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 36)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 36)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>statistic</th>\n",
       "      <th>pval</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Log2_fold_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>3.652165</td>\n",
       "      <td>2.600390e-04</td>\n",
       "      <td>4.129259e-04</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>1.095344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>-6.297607</td>\n",
       "      <td>3.022755e-10</td>\n",
       "      <td>6.980369e-10</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-0.001629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>6.104652</td>\n",
       "      <td>1.030249e-09</td>\n",
       "      <td>2.307942e-09</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-1.195447</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>0.979796</td>\n",
       "      <td>3.271869e-01</td>\n",
       "      <td>3.669641e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>3.045874</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>8.984996</td>\n",
       "      <td>2.587430e-19</td>\n",
       "      <td>1.009474e-18</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>4.400288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10884</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>9.450203</td>\n",
       "      <td>3.381747e-21</td>\n",
       "      <td>1.368916e-20</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-2.931537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10885</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>-9.184078</td>\n",
       "      <td>4.150661e-20</td>\n",
       "      <td>1.563894e-19</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-1.555629</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10886</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>-11.703305</td>\n",
       "      <td>1.225837e-31</td>\n",
       "      <td>1.960080e-30</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-0.718442</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10887</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-10.267539</td>\n",
       "      <td>9.869002e-25</td>\n",
       "      <td>5.397467e-24</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-1.789191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10888</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>-2.233280</td>\n",
       "      <td>2.553047e-02</td>\n",
       "      <td>3.141968e-02</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-1.126394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54444 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  statistic          pval           FDR Cancer  Log2_fold_change\n",
       "0        A1BG   3.652165  2.600390e-04  4.129259e-04  CCRCC          1.095344\n",
       "1        A1CF  -6.297607  3.022755e-10  6.980369e-10  CCRCC         -0.001629\n",
       "2         A2M   6.104652  1.030249e-09  2.307942e-09  CCRCC         -1.195447\n",
       "3      A4GALT   0.979796  3.271869e-01  3.669641e-01  CCRCC          3.045874\n",
       "4        AAAS   8.984996  2.587430e-19  1.009474e-18  CCRCC          4.400288\n",
       "...       ...        ...           ...           ...    ...               ...\n",
       "10884    ZXDC   9.450203  3.381747e-21  1.368916e-20   LSCC         -2.931537\n",
       "10885  ZYG11B  -9.184078  4.150661e-20  1.563894e-19   LSCC         -1.555629\n",
       "10886     ZYX -11.703305  1.225837e-31  1.960080e-30   LSCC         -0.718442\n",
       "10887   ZZEF1 -10.267539  9.869002e-25  5.397467e-24   LSCC         -1.789191\n",
       "10888    ZZZ3  -2.233280  2.553047e-02  3.141968e-02   LSCC         -1.126394\n",
       "\n",
       "[54444 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diff_expression_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_df = dc.get_prot_trans_df(cancer)\n",
    "    prot_df = prot_df.drop(columns = 'Transcriptomics')\n",
    "    fold_change = prot_df.groupby('Gene').apply(\n",
    "        lambda df: log2_fold_change(df[df['Tissue']=='Tumor'].Proteomics,\n",
    "                                    df[df['Tissue']=='Normal'].Proteomics))\n",
    "    fold_change = pd.DataFrame(fold_change, columns = ['Log2_fold_change'])\n",
    "    fold_change.reset_index(inplace = True)\n",
    "    ranksums = prot_df.groupby('Gene').apply(lambda df: stats.ranksums(\n",
    "        df[df['Tissue']=='Tumor'].Proteomics,\n",
    "        df[df['Tissue']=='Normal'].Proteomics))\n",
    "    ranksums = pd.DataFrame.from_records(ranksums, index = ranksums.index, columns = ['statistic', 'pval'])\n",
    "    ranksums.reset_index(inplace = True)\n",
    "    ranksums = ranksums.dropna()\n",
    "    ranksums['FDR'] = ssm.fdrcorrection(ranksums['pval'])[1]\n",
    "    ranksums['Cancer'] = [cancer_name] * len(ranksums)\n",
    "    diff_expression_df = pd.merge(ranksums, fold_change)\n",
    "    diff_expression_dfs.append(diff_expression_df)\n",
    "diff_expression_df = pd.concat(diff_expression_dfs)\n",
    "diff_expression_df.to_csv('Proteomics_differential_expression_df.csv', index = False)\n",
    "diff_expression_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 907 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 36)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 745 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 36)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\stats.py:7114: RuntimeWarning: invalid value encountered in double_scalars\n",
      "  z = (s - expected) / np.sqrt(n1*n2*(n1+n2+1)/12.0)\n",
      "C:\\Users\\nanel\\anaconda3\\lib\\site-packages\\scipy\\stats\\_distn_infrastructure.py:1932: RuntimeWarning: invalid value encountered in less_equal\n",
      "  cond2 = cond0 & (x <= _a)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 26)\n",
      "cptac warning: Due to dropping the specified levels, dataframe now has 858 duplicated column headers. (C:\\Users\\nanel\\Documents\\GitHub\\pancancerProteinMRNA\\notebook_steps\\Delta_Correlation.py, line 36)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>statistic</th>\n",
       "      <th>pval</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Cancer</th>\n",
       "      <th>Log2_fold_change</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1BG</td>\n",
       "      <td>6.602420</td>\n",
       "      <td>4.044989e-11</td>\n",
       "      <td>9.864646e-11</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.881967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>-5.201398</td>\n",
       "      <td>1.977946e-07</td>\n",
       "      <td>3.785302e-07</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>-0.641470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2M</td>\n",
       "      <td>5.970422</td>\n",
       "      <td>2.366402e-09</td>\n",
       "      <td>5.132955e-09</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.514646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A4GALT</td>\n",
       "      <td>0.244949</td>\n",
       "      <td>8.064959e-01</td>\n",
       "      <td>8.241685e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.096223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAAS</td>\n",
       "      <td>1.596773</td>\n",
       "      <td>1.103162e-01</td>\n",
       "      <td>1.296881e-01</td>\n",
       "      <td>CCRCC</td>\n",
       "      <td>0.078019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10884</th>\n",
       "      <td>ZXDC</td>\n",
       "      <td>-1.037788</td>\n",
       "      <td>2.993687e-01</td>\n",
       "      <td>3.241350e-01</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>0.000374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10885</th>\n",
       "      <td>ZYG11B</td>\n",
       "      <td>-9.816298</td>\n",
       "      <td>9.579687e-23</td>\n",
       "      <td>4.007423e-22</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-0.040386</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10886</th>\n",
       "      <td>ZYX</td>\n",
       "      <td>-9.708917</td>\n",
       "      <td>2.762557e-22</td>\n",
       "      <td>1.116610e-21</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-0.068806</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10887</th>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>-9.406079</td>\n",
       "      <td>5.149893e-21</td>\n",
       "      <td>1.891943e-20</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>-0.039705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10888</th>\n",
       "      <td>ZZZ3</td>\n",
       "      <td>4.149244</td>\n",
       "      <td>3.335750e-05</td>\n",
       "      <td>4.889351e-05</td>\n",
       "      <td>LSCC</td>\n",
       "      <td>0.014515</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54444 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene  statistic          pval           FDR Cancer  Log2_fold_change\n",
       "0        A1BG   6.602420  4.044989e-11  9.864646e-11  CCRCC          0.881967\n",
       "1        A1CF  -5.201398  1.977946e-07  3.785302e-07  CCRCC         -0.641470\n",
       "2         A2M   5.970422  2.366402e-09  5.132955e-09  CCRCC          0.514646\n",
       "3      A4GALT   0.244949  8.064959e-01  8.241685e-01  CCRCC          0.096223\n",
       "4        AAAS   1.596773  1.103162e-01  1.296881e-01  CCRCC          0.078019\n",
       "...       ...        ...           ...           ...    ...               ...\n",
       "10884    ZXDC  -1.037788  2.993687e-01  3.241350e-01   LSCC          0.000374\n",
       "10885  ZYG11B  -9.816298  9.579687e-23  4.007423e-22   LSCC         -0.040386\n",
       "10886     ZYX  -9.708917  2.762557e-22  1.116610e-21   LSCC         -0.068806\n",
       "10887   ZZEF1  -9.406079  5.149893e-21  1.891943e-20   LSCC         -0.039705\n",
       "10888    ZZZ3   4.149244  3.335750e-05  4.889351e-05   LSCC          0.014515\n",
       "\n",
       "[54444 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "diff_expression_dfs = []\n",
    "for cancer, cancer_name in zip(cancers, cancer_names):\n",
    "    prot_df = dc.get_prot_trans_df(cancer)\n",
    "    prot_df = prot_df.drop(columns = 'Proteomics')\n",
    "    fold_change = prot_df.groupby('Gene').apply(\n",
    "        lambda df: log2_fold_change(df[df['Tissue']=='Tumor'].Transcriptomics,\n",
    "                                    df[df['Tissue']=='Normal'].Transcriptomics))\n",
    "    fold_change = pd.DataFrame(fold_change, columns = ['Log2_fold_change'])\n",
    "    fold_change.reset_index(inplace = True)\n",
    "    ranksums = prot_df.groupby('Gene').apply(lambda df: stats.ranksums(\n",
    "        df[df['Tissue']=='Tumor'].Transcriptomics,\n",
    "        df[df['Tissue']=='Normal'].Transcriptomics))\n",
    "    ranksums = pd.DataFrame.from_records(ranksums, index = ranksums.index, columns = ['statistic', 'pval'])\n",
    "    ranksums.reset_index(inplace = True)\n",
    "    ranksums = ranksums.dropna()\n",
    "    ranksums['FDR'] = ssm.fdrcorrection(ranksums['pval'])[1]\n",
    "    ranksums['Cancer'] = [cancer_name] * len(ranksums)\n",
    "    diff_expression_df = pd.merge(ranksums, fold_change)\n",
    "    diff_expression_dfs.append(diff_expression_df)\n",
    "diff_expression_df = pd.concat(diff_expression_dfs)\n",
    "diff_expression_df.to_csv('Transcriptomics_differential_expression_df.csv', index = False)\n",
    "diff_expression_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
